#' IGVF Project API
#'
#' No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
#'
#' The version of the OpenAPI document: 0.1.0
#' Generated by: https://openapi-generator.tech
#'
#' @docType class
#' @title Igvf operations
#' @description IgvfApi
#' @format An \code{R6Class} generator object
#' @field api_client Handles the client-server communication.
#'
#' @section Methods:
#' \describe{
#' \strong{ AccessKeys } \emph{ List items in the AccessKey collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AccessKey items. Supports filtering on fields within AccessKey items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } access_key_id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } secret_access_key_hash list( character )
#' \item \emph{ @param } status Enum < [current, deleted] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } user list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{AccessKeyResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AccessKeyResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ AlignmentFiles } \emph{ List items in the AlignmentFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AlignmentFile items. Supports filtering on fields within AlignmentFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } anvil_url list( character )
#' \item \emph{ @param } assembly Enum < [GRCh38, GRCm39] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_summary list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } controlled_access list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [bam] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } filtered list( character )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } redacted list( character )
#' \item \emph{ @param } reference_files list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } transcriptome_annotation Enum < [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34] >
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{AlignmentFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AlignmentFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ AnalysisSets } \emph{ List items in the AnalysisSet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AnalysisSet items. Supports filtering on fields within AnalysisSet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } assay_titles list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_contact_pi_id list( character )
#' \item \emph{ @param } award_contact_pi_title list( character )
#' \item \emph{ @param } award_title list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors_id list( character )
#' \item \emph{ @param } donors_accession list( character )
#' \item \emph{ @param } donors_aliases list( character )
#' \item \emph{ @param } donors_sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } donors_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } donors_taxa list( character )
#' \item \emph{ @param } file_set_type Enum < [intermediate analysis, principal analysis] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_creation_timestamp list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_file_size list( integer )
#' \item \emph{ @param } files_href list( character )
#' \item \emph{ @param } files_s3_uri list( character )
#' \item \emph{ @param } files_sequencing_platform list( character )
#' \item \emph{ @param } files_submitted_file_name list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } input_file_sets_id list( character )
#' \item \emph{ @param } input_file_sets_accession list( character )
#' \item \emph{ @param } input_file_sets_aliases list( character )
#' \item \emph{ @param } input_file_sets_file_set_type Enum < [pathogenicity, functional effect, protein stability, activity level, binding effect] >
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } samples_id list( character )
#' \item \emph{ @param } samples_accession list( character )
#' \item \emph{ @param } samples_aliases list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments list( character )
#' \item \emph{ @param } samples_classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } samples_construct_library_sets list( character )
#' \item \emph{ @param } samples_disease_terms_id list( character )
#' \item \emph{ @param } samples_disease_terms_term_name list( character )
#' \item \emph{ @param } samples_modifications list( character )
#' \item \emph{ @param } samples_sample_terms_id list( character )
#' \item \emph{ @param } samples_sample_terms_aliases list( character )
#' \item \emph{ @param } samples_sample_terms_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } samples_sample_terms_summary list( character )
#' \item \emph{ @param } samples_sample_terms_term_name list( character )
#' \item \emph{ @param } samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } samples_summary list( character )
#' \item \emph{ @param } samples_targeted_sample_term_id list( character )
#' \item \emph{ @param } samples_targeted_sample_term_term_name list( character )
#' \item \emph{ @param } samples_taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } samples_treatments list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{AnalysisSetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AnalysisSetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ AnalysisStepVersions } \emph{ List items in the AnalysisStepVersion collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AnalysisStepVersion items. Supports filtering on fields within AnalysisStepVersion items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } analysis_step_id list( character )
#' \item \emph{ @param } analysis_step_name list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } software_versions_id list( character )
#' \item \emph{ @param } software_versions_name list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{AnalysisStepVersionResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AnalysisStepVersionResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ AnalysisSteps } \emph{ List items in the AnalysisStep collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AnalysisStep items. Supports filtering on fields within AnalysisStep items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } analysis_step_types Enum < [alignment, file format conversion, signal generation] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } input_content_types list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } output_content_types list( character )
#' \item \emph{ @param } parents_id list( character )
#' \item \emph{ @param } parents_title list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } step_label list( character )
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } workflow_id list( character )
#' \item \emph{ @param } workflow_accession list( character )
#' \item \emph{ @returnType } \link{AnalysisStepResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AnalysisStepResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ AssayTerms } \emph{ List items in the AssayTerm collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AssayTerm items. Supports filtering on fields within AssayTerm items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } ancestors list( character )
#' \item \emph{ @param } assay_slims list( character )
#' \item \emph{ @param } category_slims list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } deprecated_ntr_terms list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } is_a list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } objective_slims list( character )
#' \item \emph{ @param } ontology list( character )
#' \item \emph{ @param } preferred_assay_titles Enum < [10x multiome, 10x multiome with MULTI-seq, AAV-MPRA, ATAC-seq, CERES-seq, Cell painting, CRISPR FlowFISH, DOGMA-seq, Histone ChIP-seq, Hi-C, lentiMPRA, MERFISH, MIAA, mN2H, MPRA, MPRA (scQer), MULTI-seq, Parse SPLiT-seq, Perturb-seq, RNA-seq, SGE, scATAC-seq, scNT-seq, scNT-seq2, scRNA-seq, semi-qY2H, SHARE-seq, smFISH, snATAC-seq, snmC-Seq2, snMCT-seq, snM3C-seq, snRNA-seq, SUPERSTARR, TAP-seq, TF ChIP-seq, VAMP-seq, Variant FlowFISH, Variant painting, Y2H, yN2H] >
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } synonyms list( character )
#' \item \emph{ @param } term_id list( character )
#' \item \emph{ @param } term_name list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{AssayTermResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AssayTermResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ AuxiliarySets } \emph{ List items in the AuxiliarySet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate AuxiliarySet items. Supports filtering on fields within AuxiliarySet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_contact_pi_id list( character )
#' \item \emph{ @param } award_contact_pi_title list( character )
#' \item \emph{ @param } award_title list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors_id list( character )
#' \item \emph{ @param } donors_accession list( character )
#' \item \emph{ @param } donors_aliases list( character )
#' \item \emph{ @param } donors_sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } donors_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } donors_taxa list( character )
#' \item \emph{ @param } file_set_type Enum < [cell hashing, cell sorting, circularized RNA barcode detection, gRNA sequencing, oligo-conjugated lipids, quantification DNA barcode sequencing, variant sequencing] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_creation_timestamp list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_file_size list( integer )
#' \item \emph{ @param } files_href list( character )
#' \item \emph{ @param } files_s3_uri list( character )
#' \item \emph{ @param } files_sequencing_platform list( character )
#' \item \emph{ @param } files_submitted_file_name list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } library_construction_platform list( character )
#' \item \emph{ @param } measurement_sets_id list( character )
#' \item \emph{ @param } measurement_sets_accession list( character )
#' \item \emph{ @param } measurement_sets_aliases list( character )
#' \item \emph{ @param } measurement_sets_preferred_assay_title Enum < [10x multiome, 10x multiome with MULTI-seq, AAV-MPRA, ATAC-seq, CERES-seq, Cell painting, CRISPR FlowFISH, DOGMA-seq, Histone ChIP-seq, Hi-C, lentiMPRA, MERFISH, MIAA, mN2H, MPRA, MPRA (scQer), MULTI-seq, Parse SPLiT-seq, Perturb-seq, RNA-seq, SGE, scATAC-seq, scNT-seq, scNT-seq2, scRNA-seq, semi-qY2H, SHARE-seq, smFISH, snATAC-seq, snmC-Seq2, snMCT-seq, snM3C-seq, snRNA-seq, SUPERSTARR, TAP-seq, TF ChIP-seq, VAMP-seq, Variant FlowFISH, Variant painting, Y2H, yN2H] >
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } samples_id list( character )
#' \item \emph{ @param } samples_accession list( character )
#' \item \emph{ @param } samples_aliases list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments list( character )
#' \item \emph{ @param } samples_classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } samples_construct_library_sets list( character )
#' \item \emph{ @param } samples_disease_terms_id list( character )
#' \item \emph{ @param } samples_disease_terms_term_name list( character )
#' \item \emph{ @param } samples_modifications list( character )
#' \item \emph{ @param } samples_sample_terms_id list( character )
#' \item \emph{ @param } samples_sample_terms_aliases list( character )
#' \item \emph{ @param } samples_sample_terms_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } samples_sample_terms_summary list( character )
#' \item \emph{ @param } samples_sample_terms_term_name list( character )
#' \item \emph{ @param } samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } samples_summary list( character )
#' \item \emph{ @param } samples_targeted_sample_term_id list( character )
#' \item \emph{ @param } samples_targeted_sample_term_term_name list( character )
#' \item \emph{ @param } samples_taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } samples_treatments list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{AuxiliarySetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AuxiliarySetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Awards } \emph{ List items in the Award collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Award items. Supports filtering on fields within Award items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } contact_pi list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } end_date list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } pis list( character )
#' \item \emph{ @param } project Enum < [community, ENCODE, IGVF] >
#' \item \emph{ @param } start_date list( character )
#' \item \emph{ @param } status Enum < [current, deleted, disabled] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } viewing_group Enum < [community, IGVF] >
#' \item \emph{ @returnType } \link{AwardResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : AwardResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ BatchDownload } \emph{ List files to download based on search query. All results are returned. }
#' Generates TSV of files contained in FileSets in search results.
#'
#' \itemize{
#' \item \emph{ @param } type list( character )
#' \item \emph{ @param } query character
#' \item \emph{ @param } field_filters \link{object}
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : character
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Biomarkers } \emph{ List items in the Biomarker collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Biomarker items. Supports filtering on fields within Biomarker items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_name list( character )
#' \item \emph{ @param } biomarker_for list( character )
#' \item \emph{ @param } classification Enum < [cell surface protein, marker gene] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } gene list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } name_quantification list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } quantification Enum < [negative, positive, low, intermediate, high] >
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } synonyms list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{BiomarkerResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : BiomarkerResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ ConfigurationFiles } \emph{ List items in the ConfigurationFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate ConfigurationFile items. Supports filtering on fields within ConfigurationFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [yaml] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } seqspec_of list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{ConfigurationFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ConfigurationFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ ConstructLibrarySets } \emph{ List items in the ConstructLibrarySet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate ConstructLibrarySet items. Supports filtering on fields within ConstructLibrarySet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } applied_to_samples_id list( character )
#' \item \emph{ @param } applied_to_samples_accession list( character )
#' \item \emph{ @param } applied_to_samples_aliases list( character )
#' \item \emph{ @param } applied_to_samples_disease_terms_id list( character )
#' \item \emph{ @param } applied_to_samples_disease_terms_term_name list( character )
#' \item \emph{ @param } applied_to_samples_sample_terms_id list( character )
#' \item \emph{ @param } applied_to_samples_sample_terms_term_name list( character )
#' \item \emph{ @param } applied_to_samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } applied_to_samples_summary list( character )
#' \item \emph{ @param } associated_phenotypes_id list( character )
#' \item \emph{ @param } associated_phenotypes_term_id list( character )
#' \item \emph{ @param } associated_phenotypes_term_name list( character )
#' \item \emph{ @param } average_guide_coverage list( numeric )
#' \item \emph{ @param } average_insert_size list( numeric )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } exon list( character )
#' \item \emph{ @param } file_set_type Enum < [guide library, reporter library, expression vector library, editing template library] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } guide_type Enum < [sgRNA, pgRNA] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } integrated_content_files list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } large_scale_gene_list_id list( character )
#' \item \emph{ @param } large_scale_gene_list_accession list( character )
#' \item \emph{ @param } large_scale_gene_list_aliases list( character )
#' \item \emph{ @param } large_scale_loci_list_id list( character )
#' \item \emph{ @param } large_scale_loci_list_accession list( character )
#' \item \emph{ @param } large_scale_loci_list_aliases list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } lower_bound_guide_coverage list( integer )
#' \item \emph{ @param } lower_bound_insert_size list( integer )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } orf_list_id list( character )
#' \item \emph{ @param } orf_list_aliases list( character )
#' \item \emph{ @param } orf_list_gene list( character )
#' \item \emph{ @param } orf_list_orf_id list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } scope Enum < [tile, exon, genes, loci, genome-wide, interactors, alleles, targeton] >
#' \item \emph{ @param } selection_criteria Enum < [accessible genome regions, candidate cis-regulatory elements, chromatin states, phenotype-associated variants, DNase hypersensitive sites, genes, histone modifications, protein interactors, sequence variants, synthetic elements, transcription start sites, TF binding sites] >
#' \item \emph{ @param } small_scale_gene_list_id list( character )
#' \item \emph{ @param } small_scale_gene_list_geneid list( character )
#' \item \emph{ @param } small_scale_gene_list_name list( character )
#' \item \emph{ @param } small_scale_gene_list_symbol list( character )
#' \item \emph{ @param } small_scale_gene_list_synonyms list( character )
#' \item \emph{ @param } small_scale_loci_list list( \link{Locus} )
#' \item \emph{ @param } sources list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } targeton list( character )
#' \item \emph{ @param } tiling_modality Enum < [peak tiling, full tiling, sparse peaks] >
#' \item \emph{ @param } upper_bound_guide_coverage list( integer )
#' \item \emph{ @param } upper_bound_insert_size list( integer )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{ConstructLibrarySetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ConstructLibrarySetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ CrisprModifications } \emph{ List items in the CrisprModification collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate CrisprModification items. Supports filtering on fields within CrisprModification items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } activated list( character )
#' \item \emph{ @param } activating_agent_term_id list( character )
#' \item \emph{ @param } activating_agent_term_name list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biosamples_modified list( character )
#' \item \emph{ @param } cas Enum < [Cas9, Cas12a, Cas13, dCas9, nCas9, SpG, SpRY] >
#' \item \emph{ @param } cas_species Enum < [Streptococcus pyogenes (Sp), Staphylococcus aureus (Sa), Campylobacter jejuni (Cj), Neisseria meningitidis (Nm)] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } fused_domain Enum < [2xVP64, 3xVP64, ABE8e, ABE8.20, ANTI-FLAG, BE4, BE4max, eA3A, eA3A-T31A, eA3A-T44D-S45A, KOX1-KRAB, M-MLV RT (PE2), p300, TdCBE, TdCGBE, TdDE, VPH, VP64, VP64-p65-Rta (VPR), ZIM3-KRAB] >
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } modality Enum < [activation, base editing, cutting, interference, knockout, localizing, prime editing] >
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } sources list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } tagged_protein list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{CrisprModificationResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : CrisprModificationResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ CuratedSets } \emph{ List items in the CuratedSet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate CuratedSet items. Supports filtering on fields within CuratedSet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } assemblies list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_contact_pi_id list( character )
#' \item \emph{ @param } award_contact_pi_title list( character )
#' \item \emph{ @param } award_title list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors_id list( character )
#' \item \emph{ @param } donors_accession list( character )
#' \item \emph{ @param } donors_aliases list( character )
#' \item \emph{ @param } donors_sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } donors_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } donors_taxa list( character )
#' \item \emph{ @param } file_set_type Enum < [barcodes, editing templates, elements, external data for catalog, genome, genes, guide RNAs, transcriptome, variants] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_creation_timestamp list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_file_size list( integer )
#' \item \emph{ @param } files_href list( character )
#' \item \emph{ @param } files_s3_uri list( character )
#' \item \emph{ @param } files_sequencing_platform list( character )
#' \item \emph{ @param } files_submitted_file_name list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } samples_id list( character )
#' \item \emph{ @param } samples_accession list( character )
#' \item \emph{ @param } samples_aliases list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments list( character )
#' \item \emph{ @param } samples_classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } samples_construct_library_sets list( character )
#' \item \emph{ @param } samples_disease_terms_id list( character )
#' \item \emph{ @param } samples_disease_terms_term_name list( character )
#' \item \emph{ @param } samples_modifications list( character )
#' \item \emph{ @param } samples_sample_terms_id list( character )
#' \item \emph{ @param } samples_sample_terms_aliases list( character )
#' \item \emph{ @param } samples_sample_terms_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } samples_sample_terms_summary list( character )
#' \item \emph{ @param } samples_sample_terms_term_name list( character )
#' \item \emph{ @param } samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } samples_summary list( character )
#' \item \emph{ @param } samples_targeted_sample_term_id list( character )
#' \item \emph{ @param } samples_targeted_sample_term_term_name list( character )
#' \item \emph{ @param } samples_taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } samples_treatments list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } transcriptome_annotations list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{CuratedSetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : CuratedSetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ DegronModifications } \emph{ List items in the DegronModification collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate DegronModification items. Supports filtering on fields within DegronModification items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } activated list( character )
#' \item \emph{ @param } activating_agent_term_id list( character )
#' \item \emph{ @param } activating_agent_term_name list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biosamples_modified list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } degron_system Enum < [AID, AlissAid, ssAID] >
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } modality Enum < [degradation] >
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } sources list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } tagged_proteins list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{DegronModificationResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : DegronModificationResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Documents } \emph{ List items in the Document collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Document items. Supports filtering on fields within Document items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } characterization_method Enum < [FACS, immunoblot, immunofluorescence, immunoprecipitation, mass spectrometry, PCR, restriction digest, RT-qPCR, sequencing] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } document_type Enum < [cell fate change protocol, characterization, computational protocol, experimental protocol, file format specification, image, model source data, plate map, plasmid map, plasmid sequence, standards] >
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } urls list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{DocumentResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : DocumentResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Download } \emph{ Download file. }
#' Returns underlying file associated with file metadata
#'
#' \itemize{
#' \item \emph{ @param } file_id character
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : data.frame
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | Item not found
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Genes } \emph{ List items in the Gene collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Gene items. Supports filtering on fields within Gene items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } geneid list( character )
#' \item \emph{ @param } geneid_with_version list( character )
#' \item \emph{ @param } locations list( \link{GeneLocation} )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } symbol list( character )
#' \item \emph{ @param } synonyms list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } transcriptome_annotation Enum < [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } version_number list( character )
#' \item \emph{ @returnType } \link{GeneResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : GeneResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ GenomeBrowserAnnotationFiles } \emph{ List items in the GenomeBrowserAnnotationFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate GenomeBrowserAnnotationFile items. Supports filtering on fields within GenomeBrowserAnnotationFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } assembly Enum < [GRCh38, GRCm39] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [bigBed, tabix] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_format_type Enum < [bed12, bed3, bed3+, bed5, bed6, bed6+, bed9, bed9+, mpra_starr] >
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } transcriptome_annotation Enum < [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34] >
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{GenomeBrowserAnnotationFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : GenomeBrowserAnnotationFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ GetById } \emph{ Get item information }
#' Retrieve detailed information about a specific item using its @id or uuid.
#'
#' \itemize{
#' \item \emph{ @param } resource_id character
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @returnType } \link{Item} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : Item
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | Object not found
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ HumanDonors } \emph{ List items in the HumanDonor collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate HumanDonor items. Supports filtering on fields within HumanDonor items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } ethnicities Enum < [African American, African Caribbean, Arab, Asian, Black, Black African, Chinese, Colombian, Dai Chinese, Esan, Eskimo, European, Gambian, Han Chinese, Hispanic, Indian, Japanese, Kinh Vietnamese, Luhya, Maasai, Mende, Native Hawaiian, Pacific Islander, Puerto Rican, Yoruba] >
#' \item \emph{ @param } human_donor_identifiers list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } phenotypic_features_id list( character )
#' \item \emph{ @param } phenotypic_features_feature_id list( character )
#' \item \emph{ @param } phenotypic_features_feature_term_id list( character )
#' \item \emph{ @param } phenotypic_features_feature_term_name list( character )
#' \item \emph{ @param } phenotypic_features_observation_date list( character )
#' \item \emph{ @param } phenotypic_features_quantity_units Enum < [meter, micromole, nanogram, microgram, milligram, gram, kilogram, milli-International Unit per milliliter, picogram per milliliter, nanogram per milliliter, milligram per deciliter] >
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } related_donors_donor_id list( character )
#' \item \emph{ @param } related_donors_donor_accession list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens] >
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{HumanDonorResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : HumanDonorResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ ImageFiles } \emph{ List items in the ImageFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate ImageFile items. Supports filtering on fields within ImageFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [jpg, png] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{ImageFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ImageFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Images } \emph{ List items in the Image collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Image items. Supports filtering on fields within Image items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } caption list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } download_url list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } thumb_nail list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{ImageResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ImageResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ InVitroSystems } \emph{ List items in the InVitroSystem collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate InVitroSystem items. Supports filtering on fields within InVitroSystem items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } age list( character )
#' \item \emph{ @param } age_units Enum < [minute, hour, day, week, month, year] >
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biomarkers list( character )
#' \item \emph{ @param } cell_fate_change_protocol list( character )
#' \item \emph{ @param } cell_fate_change_treatments_id list( character )
#' \item \emph{ @param } cell_fate_change_treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } cell_fate_change_treatments_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } cell_fate_change_treatments_summary list( character )
#' \item \emph{ @param } cell_fate_change_treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } cellular_sub_pool list( character )
#' \item \emph{ @param } classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } construct_library_sets list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_obtained list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } demultiplexed_from list( character )
#' \item \emph{ @param } demultiplexed_to list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } disease_terms_id list( character )
#' \item \emph{ @param } disease_terms_term_name list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors list( character )
#' \item \emph{ @param } embryonic list( character )
#' \item \emph{ @param } file_sets_id list( character )
#' \item \emph{ @param } file_sets_accession list( character )
#' \item \emph{ @param } file_sets_aliases list( character )
#' \item \emph{ @param } file_sets_assay_term_term_name list( character )
#' \item \emph{ @param } file_sets_lab_title list( character )
#' \item \emph{ @param } file_sets_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } file_sets_summary list( character )
#' \item \emph{ @param } growth_medium Enum < [DMEM with serum, DMEM without serum, SMBM with serum, SMBM without serum] >
#' \item \emph{ @param } institutional_certificates_id list( character )
#' \item \emph{ @param } institutional_certificates_certificate_identifier list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } lower_bound_age list( numeric )
#' \item \emph{ @param } lower_bound_age_in_hours list( numeric )
#' \item \emph{ @param } modifications_id list( character )
#' \item \emph{ @param } modifications_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } modifications_summary list( character )
#' \item \emph{ @param } moi list( numeric )
#' \item \emph{ @param } multiplexed_in_id list( character )
#' \item \emph{ @param } multiplexed_in_accession list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } nucleic_acid_delivery Enum < [transfection, adenoviral transduction, lentiviral transduction] >
#' \item \emph{ @param } origin_of list( character )
#' \item \emph{ @param } originated_from_id list( character )
#' \item \emph{ @param } originated_from_accession list( character )
#' \item \emph{ @param } part_of list( character )
#' \item \emph{ @param } parts list( character )
#' \item \emph{ @param } passage_number list( integer )
#' \item \emph{ @param } pooled_from list( character )
#' \item \emph{ @param } pooled_in list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sample_terms_id list( character )
#' \item \emph{ @param } sample_terms_term_name list( character )
#' \item \emph{ @param } sex Enum < [female, male, mixed, unspecified] >
#' \item \emph{ @param } sorted_fractions list( character )
#' \item \emph{ @param } sorted_from_id list( character )
#' \item \emph{ @param } sorted_from_accession list( character )
#' \item \emph{ @param } sorted_from_detail list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } starting_amount list( numeric )
#' \item \emph{ @param } starting_amount_units Enum < [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } targeted_sample_term list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } time_post_change list( numeric )
#' \item \emph{ @param } time_post_change_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } time_post_library_delivery list( numeric )
#' \item \emph{ @param } time_post_library_delivery_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } treatments_id list( character )
#' \item \emph{ @param } treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } treatments_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } treatments_summary list( character )
#' \item \emph{ @param } treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } upper_bound_age list( numeric )
#' \item \emph{ @param } upper_bound_age_in_hours list( numeric )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{InVitroSystemResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : InVitroSystemResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ InstitutionalCertificates } \emph{ List items in the InstitutionalCertificate collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate InstitutionalCertificate items. Supports filtering on fields within InstitutionalCertificate items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } certificate_identifier list( character )
#' \item \emph{ @param } controlled_access list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } data_use_limitation Enum < [DS, GRU, HMB, other] >
#' \item \emph{ @param } data_use_limitation_modifiers Enum < [COL, GSO, IRB, MDS, NPU, PUB] >
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } samples list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } urls list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{InstitutionalCertificateResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : InstitutionalCertificateResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Labs } \emph{ List items in the Lab collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Lab items. Supports filtering on fields within Lab items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } awards_id list( character )
#' \item \emph{ @param } awards_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } awards_name list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } institute_label list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } pi list( character )
#' \item \emph{ @param } status Enum < [current, deleted, disabled] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{LabResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : LabResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ MatrixFiles } \emph{ List items in the MatrixFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate MatrixFile items. Supports filtering on fields within MatrixFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_summary list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } dimension1 Enum < [cell, fragment, gene, time, treatment, variant, genomic position] >
#' \item \emph{ @param } dimension2 Enum < [cell, fragment, gene, time, treatment, variant, genomic position] >
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [h5ad, hdf5, mtx, tar, hic] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } reference_files list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{MatrixFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : MatrixFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ MeasurementSets } \emph{ List items in the MeasurementSet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate MeasurementSet items. Supports filtering on fields within MeasurementSet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } assay_term_id list( character )
#' \item \emph{ @param } assay_term_term_name list( character )
#' \item \emph{ @param } auxiliary_sets_id list( character )
#' \item \emph{ @param } auxiliary_sets_accession list( character )
#' \item \emph{ @param } auxiliary_sets_aliases list( character )
#' \item \emph{ @param } auxiliary_sets_file_set_type Enum < [cell hashing, cell sorting, circularized RNA barcode detection, gRNA sequencing, oligo-conjugated lipids, quantification DNA barcode sequencing, variant sequencing] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_contact_pi_id list( character )
#' \item \emph{ @param } award_contact_pi_title list( character )
#' \item \emph{ @param } award_title list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_file_sets_id list( character )
#' \item \emph{ @param } control_file_sets_accession list( character )
#' \item \emph{ @param } control_file_sets_aliases list( character )
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors_id list( character )
#' \item \emph{ @param } donors_accession list( character )
#' \item \emph{ @param } donors_aliases list( character )
#' \item \emph{ @param } donors_sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } donors_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } donors_taxa list( character )
#' \item \emph{ @param } external_image_url list( character )
#' \item \emph{ @param } file_set_type Enum < [experimental data] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_creation_timestamp list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_file_size list( integer )
#' \item \emph{ @param } files_href list( character )
#' \item \emph{ @param } files_s3_uri list( character )
#' \item \emph{ @param } files_sequencing_platform_id list( character )
#' \item \emph{ @param } files_sequencing_platform_term_name list( character )
#' \item \emph{ @param } files_submitted_file_name list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } library_construction_platform_id list( character )
#' \item \emph{ @param } library_construction_platform_term_name list( character )
#' \item \emph{ @param } multiome_size list( integer )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } preferred_assay_title Enum < [10x multiome, 10x multiome with MULTI-seq, AAV-MPRA, ATAC-seq, CERES-seq, Cell painting, CRISPR FlowFISH, DOGMA-seq, Histone ChIP-seq, Hi-C, lentiMPRA, MERFISH, MIAA, mN2H, MPRA, MPRA (scQer), MULTI-seq, Parse SPLiT-seq, Perturb-seq, RNA-seq, SGE, scATAC-seq, scNT-seq, scNT-seq2, scRNA-seq, semi-qY2H, SHARE-seq, smFISH, snATAC-seq, snmC-Seq2, snMCT-seq, snM3C-seq, snRNA-seq, SUPERSTARR, TAP-seq, TF ChIP-seq, VAMP-seq, Variant FlowFISH, Variant painting, Y2H, yN2H] >
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } related_multiome_datasets_id list( character )
#' \item \emph{ @param } related_multiome_datasets_accession list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } samples_id list( character )
#' \item \emph{ @param } samples_accession list( character )
#' \item \emph{ @param } samples_aliases list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments_id list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } samples_cell_fate_change_treatments_summary list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } samples_classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } samples_construct_library_sets_id list( character )
#' \item \emph{ @param } samples_construct_library_sets_accession list( character )
#' \item \emph{ @param } samples_construct_library_sets_file_set_type Enum < [guide library, reporter library, expression vector library, editing template library] >
#' \item \emph{ @param } samples_construct_library_sets_small_scale_gene_list_id list( character )
#' \item \emph{ @param } samples_construct_library_sets_small_scale_gene_list_geneid list( character )
#' \item \emph{ @param } samples_construct_library_sets_small_scale_gene_list_name list( character )
#' \item \emph{ @param } samples_construct_library_sets_small_scale_gene_list_summary list( character )
#' \item \emph{ @param } samples_construct_library_sets_small_scale_gene_list_symbol list( character )
#' \item \emph{ @param } samples_construct_library_sets_summary list( character )
#' \item \emph{ @param } samples_disease_terms_id list( character )
#' \item \emph{ @param } samples_disease_terms_term_name list( character )
#' \item \emph{ @param } samples_modifications_id list( character )
#' \item \emph{ @param } samples_sample_terms_id list( character )
#' \item \emph{ @param } samples_sample_terms_aliases list( character )
#' \item \emph{ @param } samples_sample_terms_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } samples_sample_terms_summary list( character )
#' \item \emph{ @param } samples_sample_terms_term_name list( character )
#' \item \emph{ @param } samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } samples_summary list( character )
#' \item \emph{ @param } samples_targeted_sample_term_id list( character )
#' \item \emph{ @param } samples_targeted_sample_term_term_name list( character )
#' \item \emph{ @param } samples_taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } samples_treatments_id list( character )
#' \item \emph{ @param } samples_treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } samples_treatments_summary list( character )
#' \item \emph{ @param } samples_treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } sequencing_library_types Enum < [direct RNA, exome capture, mRNA enriched, rRNA depleted, polyA depleted, polyA enriched] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } targeted_genes_id list( character )
#' \item \emph{ @param } targeted_genes_geneid list( character )
#' \item \emph{ @param } targeted_genes_name list( character )
#' \item \emph{ @param } targeted_genes_symbol list( character )
#' \item \emph{ @param } targeted_genes_synonyms list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{MeasurementSetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : MeasurementSetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ ModelFiles } \emph{ List items in the ModelFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate ModelFile items. Supports filtering on fields within ModelFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } anvil_url list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } controlled_access list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [hdf5, json, tar, tsv] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{ModelFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ModelFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ ModelSets } \emph{ List items in the ModelSet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate ModelSet items. Supports filtering on fields within ModelSet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } assessed_genes list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_contact_pi_id list( character )
#' \item \emph{ @param } award_contact_pi_title list( character )
#' \item \emph{ @param } award_title list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors_id list( character )
#' \item \emph{ @param } donors_accession list( character )
#' \item \emph{ @param } donors_aliases list( character )
#' \item \emph{ @param } donors_sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } donors_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } donors_taxa list( character )
#' \item \emph{ @param } file_set_type Enum < [neural network, random forest, decision tree, support vector machine, variant binding effect] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_creation_timestamp list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_file_size list( integer )
#' \item \emph{ @param } files_href list( character )
#' \item \emph{ @param } files_s3_uri list( character )
#' \item \emph{ @param } files_sequencing_platform list( character )
#' \item \emph{ @param } files_submitted_file_name list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } input_file_sets_id list( character )
#' \item \emph{ @param } input_file_sets_accession list( character )
#' \item \emph{ @param } input_file_sets_aliases list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } model_name list( character )
#' \item \emph{ @param } model_version list( character )
#' \item \emph{ @param } model_zoo_location list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } prediction_objects Enum < [coding variants, non-coding variants, all variants, regulatory elements, genes] >
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } samples_id list( character )
#' \item \emph{ @param } samples_accession list( character )
#' \item \emph{ @param } samples_aliases list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments list( character )
#' \item \emph{ @param } samples_classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } samples_construct_library_sets list( character )
#' \item \emph{ @param } samples_disease_terms_id list( character )
#' \item \emph{ @param } samples_disease_terms_term_name list( character )
#' \item \emph{ @param } samples_modifications list( character )
#' \item \emph{ @param } samples_sample_terms_id list( character )
#' \item \emph{ @param } samples_sample_terms_aliases list( character )
#' \item \emph{ @param } samples_sample_terms_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } samples_sample_terms_summary list( character )
#' \item \emph{ @param } samples_sample_terms_term_name list( character )
#' \item \emph{ @param } samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } samples_summary list( character )
#' \item \emph{ @param } samples_targeted_sample_term_id list( character )
#' \item \emph{ @param } samples_targeted_sample_term_term_name list( character )
#' \item \emph{ @param } samples_taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } samples_treatments list( character )
#' \item \emph{ @param } software_version list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{ModelSetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ModelSetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ MultiplexedSamples } \emph{ List items in the MultiplexedSample collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate MultiplexedSample items. Supports filtering on fields within MultiplexedSample items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } barcode_sample_map list( character )
#' \item \emph{ @param } biomarkers list( character )
#' \item \emph{ @param } cellular_sub_pool list( character )
#' \item \emph{ @param } classifications list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } construct_library_sets_id list( character )
#' \item \emph{ @param } construct_library_sets_accession list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_obtained list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } disease_terms_id list( character )
#' \item \emph{ @param } disease_terms_term_name list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors list( character )
#' \item \emph{ @param } file_sets_id list( character )
#' \item \emph{ @param } file_sets_accession list( character )
#' \item \emph{ @param } file_sets_aliases list( character )
#' \item \emph{ @param } file_sets_assay_term_term_name list( character )
#' \item \emph{ @param } file_sets_lab_title list( character )
#' \item \emph{ @param } file_sets_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } file_sets_summary list( character )
#' \item \emph{ @param } institutional_certificates list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } modifications_id list( character )
#' \item \emph{ @param } modifications_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } modifications_summary list( character )
#' \item \emph{ @param } moi list( numeric )
#' \item \emph{ @param } multiplexed_in_id list( character )
#' \item \emph{ @param } multiplexed_in_accession list( character )
#' \item \emph{ @param } multiplexed_samples_id list( character )
#' \item \emph{ @param } multiplexed_samples_accession list( character )
#' \item \emph{ @param } multiplexed_samples_construct_library_sets list( character )
#' \item \emph{ @param } multiplexed_samples_disease_terms_id list( character )
#' \item \emph{ @param } multiplexed_samples_disease_terms_term_name list( character )
#' \item \emph{ @param } multiplexed_samples_donors_id list( character )
#' \item \emph{ @param } multiplexed_samples_donors_accession list( character )
#' \item \emph{ @param } multiplexed_samples_sample_terms_id list( character )
#' \item \emph{ @param } multiplexed_samples_sample_terms_term_name list( character )
#' \item \emph{ @param } multiplexed_samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } multiplexed_samples_summary list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } nucleic_acid_delivery Enum < [transfection, adenoviral transduction, lentiviral transduction] >
#' \item \emph{ @param } origin_of list( character )
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sample_terms_id list( character )
#' \item \emph{ @param } sample_terms_term_name list( character )
#' \item \emph{ @param } sorted_fractions list( character )
#' \item \emph{ @param } sorted_from_id list( character )
#' \item \emph{ @param } sorted_from_accession list( character )
#' \item \emph{ @param } sorted_from_detail list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } starting_amount list( numeric )
#' \item \emph{ @param } starting_amount_units Enum < [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } time_post_library_delivery list( numeric )
#' \item \emph{ @param } time_post_library_delivery_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } treatments_id list( character )
#' \item \emph{ @param } treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } treatments_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } treatments_summary list( character )
#' \item \emph{ @param } treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{MultiplexedSampleResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : MultiplexedSampleResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ OpenReadingFrames } \emph{ List items in the OpenReadingFrame collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate OpenReadingFrame items. Supports filtering on fields within OpenReadingFrame items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } gene_id list( character )
#' \item \emph{ @param } gene_geneid list( character )
#' \item \emph{ @param } gene_symbol list( character )
#' \item \emph{ @param } lab list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } orf_id list( character )
#' \item \emph{ @param } pct_coverage_orf list( numeric )
#' \item \emph{ @param } pct_coverage_protein list( numeric )
#' \item \emph{ @param } pct_identical_protein list( numeric )
#' \item \emph{ @param } protein_id list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{OpenReadingFrameResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : OpenReadingFrameResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Pages } \emph{ List items in the Page collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Page items. Supports filtering on fields within Page items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award list( character )
#' \item \emph{ @param } canonical_uri list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } lab list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } parent list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{PageResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PageResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ PhenotypeTerms } \emph{ List items in the PhenotypeTerm collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate PhenotypeTerm items. Supports filtering on fields within PhenotypeTerm items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } ancestors list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } deprecated_ntr_terms list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } is_a list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } ontology list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } synonyms list( character )
#' \item \emph{ @param } term_id list( character )
#' \item \emph{ @param } term_name list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{PhenotypeTermResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PhenotypeTermResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ PhenotypicFeatures } \emph{ List items in the PhenotypicFeature collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate PhenotypicFeature items. Supports filtering on fields within PhenotypicFeature items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } feature_id list( character )
#' \item \emph{ @param } feature_term_id list( character )
#' \item \emph{ @param } feature_term_name list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } observation_date list( character )
#' \item \emph{ @param } quantity list( numeric )
#' \item \emph{ @param } quantity_units Enum < [meter, micromole, nanogram, microgram, milligram, gram, kilogram, milli-International Unit per milliliter, picogram per milliliter, nanogram per milliliter, milligram per deciliter] >
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{PhenotypicFeatureResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PhenotypicFeatureResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ PlatformTerms } \emph{ List items in the PlatformTerm collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate PlatformTerm items. Supports filtering on fields within PlatformTerm items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } ancestors list( character )
#' \item \emph{ @param } company Enum < [10X Genomics, Illumina, Life Technologies, Oxford Nanopore Technologies, Pacific Biosciences, Parse Biosciences, Roche] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } deprecated_ntr_terms list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } is_a list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } ontology list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } sequencing_kits Enum < [HiSeq SBS Kit v4, HiSeq SR Cluster Kit v4-cBot-HS, HiSeq PE Cluster Kit v4-cBot-HS, HiSeq SR Rapid Cluster Kit v2, HiSeq PE Rapid Cluster Kit v2, HiSeq Rapid SBS Kit v2, HiSeq 3000/4000 SBS Kit, HiSeq 3000/4000 SR Cluster Kit, HiSeq 3000/4000 PE Cluster Kit, MiSeq Reagent Kit v2, NextSeq 500 Mid Output Kit, NextSeq 500 High Output Kit, NextSeq 500 Mid Output v2 Kit, NextSeq 500 High Output v2 Kit, NextSeq 500/550 Mid-Output v2.5 Kit, NextSeq 500/550 High-Output v2.5 Kit, TG NextSeq 500/550 Mid-Output Kit v2.5, TG NextSeq 500/550 High-Output Kit v2.5, NextSeq 1000/2000 P1 Reagent Kit, NextSeq 1000/2000 P2 Reagent Kit, NextSeq 1000/2000 P3 Reagent Kit, NextSeq 1000/2000 P1 XLEAP-SBS Reagent Kit, NextSeq 1000/2000 P2 XLEAP-SBS Reagent Kit, NextSeq 2000 P3 XLEAP-SBS Reagent Kit, NextSeq 2000 P4 XLEAP-SBS Reagent Kit, NovaSeq 6000 SP Reagent Kit v1.5, NovaSeq 6000 S1 Reagent Kit v1.5, NovaSeq 6000 S2 Reagent Kit v1.5, NovaSeq 6000 S4 Reagent Kit v1.5, NovaSeq X Series 10B Reagent Kit, ONT Ligation Sequencing Kit V14, Sequel sequencing kit 3.0, Sequel II sequencing kit 2.0] >
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } synonyms list( character )
#' \item \emph{ @param } term_id list( character )
#' \item \emph{ @param } term_name list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{PlatformTermResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PlatformTermResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ PredictionSets } \emph{ List items in the PredictionSet collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate PredictionSet items. Supports filtering on fields within PredictionSet items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } award_contact_pi_id list( character )
#' \item \emph{ @param } award_contact_pi_title list( character )
#' \item \emph{ @param } award_title list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } control_for_id list( character )
#' \item \emph{ @param } control_for_accession list( character )
#' \item \emph{ @param } control_for_aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors_id list( character )
#' \item \emph{ @param } donors_accession list( character )
#' \item \emph{ @param } donors_aliases list( character )
#' \item \emph{ @param } donors_sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } donors_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } donors_taxa list( character )
#' \item \emph{ @param } file_set_type Enum < [pathogenicity, functional effect, protein stability, activity level, binding effect] >
#' \item \emph{ @param } files_id list( character )
#' \item \emph{ @param } files_accession list( character )
#' \item \emph{ @param } files_aliases list( character )
#' \item \emph{ @param } files_content_type list( character )
#' \item \emph{ @param } files_creation_timestamp list( character )
#' \item \emph{ @param } files_file_format list( character )
#' \item \emph{ @param } files_file_size list( integer )
#' \item \emph{ @param } files_href list( character )
#' \item \emph{ @param } files_s3_uri list( character )
#' \item \emph{ @param } files_sequencing_platform list( character )
#' \item \emph{ @param } files_submitted_file_name list( character )
#' \item \emph{ @param } files_upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } input_file_set_for list( character )
#' \item \emph{ @param } input_file_sets list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } large_scale_gene_list_id list( character )
#' \item \emph{ @param } large_scale_gene_list_accession list( character )
#' \item \emph{ @param } large_scale_gene_list_aliases list( character )
#' \item \emph{ @param } large_scale_loci_list_id list( character )
#' \item \emph{ @param } large_scale_loci_list_accession list( character )
#' \item \emph{ @param } large_scale_loci_list_aliases list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } samples_id list( character )
#' \item \emph{ @param } samples_accession list( character )
#' \item \emph{ @param } samples_aliases list( character )
#' \item \emph{ @param } samples_cell_fate_change_treatments list( character )
#' \item \emph{ @param } samples_classifications Enum < [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen] >
#' \item \emph{ @param } samples_construct_library_sets_id list( character )
#' \item \emph{ @param } samples_construct_library_sets_accession list( character )
#' \item \emph{ @param } samples_construct_library_sets_summary list( character )
#' \item \emph{ @param } samples_disease_terms_id list( character )
#' \item \emph{ @param } samples_disease_terms_term_name list( character )
#' \item \emph{ @param } samples_modifications list( character )
#' \item \emph{ @param } samples_sample_terms_id list( character )
#' \item \emph{ @param } samples_sample_terms_aliases list( character )
#' \item \emph{ @param } samples_sample_terms_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } samples_sample_terms_summary list( character )
#' \item \emph{ @param } samples_sample_terms_term_name list( character )
#' \item \emph{ @param } samples_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } samples_summary list( character )
#' \item \emph{ @param } samples_targeted_sample_term_id list( character )
#' \item \emph{ @param } samples_targeted_sample_term_term_name list( character )
#' \item \emph{ @param } samples_taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } samples_treatments list( character )
#' \item \emph{ @param } scope Enum < [genes, loci, genome-wide] >
#' \item \emph{ @param } small_scale_gene_list_id list( character )
#' \item \emph{ @param } small_scale_gene_list_geneid list( character )
#' \item \emph{ @param } small_scale_gene_list_name list( character )
#' \item \emph{ @param } small_scale_gene_list_symbol list( character )
#' \item \emph{ @param } small_scale_gene_list_synonyms list( character )
#' \item \emph{ @param } small_scale_loci_list list( \link{Locus} )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_files_timestamp list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{PredictionSetResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PredictionSetResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ PrimaryCells } \emph{ List items in the PrimaryCell collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate PrimaryCell items. Supports filtering on fields within PrimaryCell items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } age list( character )
#' \item \emph{ @param } age_units Enum < [minute, hour, day, week, month, year] >
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biomarkers list( character )
#' \item \emph{ @param } cellular_sub_pool list( character )
#' \item \emph{ @param } classifications list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } construct_library_sets list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_obtained list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } disease_terms_id list( character )
#' \item \emph{ @param } disease_terms_term_name list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors list( character )
#' \item \emph{ @param } embryonic list( character )
#' \item \emph{ @param } file_sets_id list( character )
#' \item \emph{ @param } file_sets_accession list( character )
#' \item \emph{ @param } file_sets_aliases list( character )
#' \item \emph{ @param } file_sets_assay_term_term_name list( character )
#' \item \emph{ @param } file_sets_lab_title list( character )
#' \item \emph{ @param } file_sets_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } file_sets_summary list( character )
#' \item \emph{ @param } institutional_certificates_id list( character )
#' \item \emph{ @param } institutional_certificates_certificate_identifier list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } lower_bound_age list( numeric )
#' \item \emph{ @param } lower_bound_age_in_hours list( numeric )
#' \item \emph{ @param } modifications_id list( character )
#' \item \emph{ @param } modifications_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } modifications_summary list( character )
#' \item \emph{ @param } moi list( numeric )
#' \item \emph{ @param } multiplexed_in_id list( character )
#' \item \emph{ @param } multiplexed_in_accession list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } nucleic_acid_delivery Enum < [transfection, adenoviral transduction, lentiviral transduction] >
#' \item \emph{ @param } origin_of list( character )
#' \item \emph{ @param } originated_from list( character )
#' \item \emph{ @param } part_of list( character )
#' \item \emph{ @param } parts list( character )
#' \item \emph{ @param } passage_number list( integer )
#' \item \emph{ @param } pooled_from list( character )
#' \item \emph{ @param } pooled_in list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sample_terms_id list( character )
#' \item \emph{ @param } sample_terms_term_name list( character )
#' \item \emph{ @param } sex Enum < [female, male, mixed, unspecified] >
#' \item \emph{ @param } sorted_fractions list( character )
#' \item \emph{ @param } sorted_from_id list( character )
#' \item \emph{ @param } sorted_from_accession list( character )
#' \item \emph{ @param } sorted_from_detail list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } starting_amount list( numeric )
#' \item \emph{ @param } starting_amount_units Enum < [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } time_post_library_delivery list( numeric )
#' \item \emph{ @param } time_post_library_delivery_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } treatments_id list( character )
#' \item \emph{ @param } treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } treatments_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } treatments_summary list( character )
#' \item \emph{ @param } treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } upper_bound_age list( numeric )
#' \item \emph{ @param } upper_bound_age_in_hours list( numeric )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{PrimaryCellResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PrimaryCellResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Publications } \emph{ List items in the Publication collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Publication items. Supports filtering on fields within Publication items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } abstract list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } authors list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_published list( character )
#' \item \emph{ @param } date_revised list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } donors list( character )
#' \item \emph{ @param } file_sets list( character )
#' \item \emph{ @param } issue list( character )
#' \item \emph{ @param } journal list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } page list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publication_year list( integer )
#' \item \emph{ @param } published_by Enum < [community, IGVF, ENCODE] >
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } samples list( character )
#' \item \emph{ @param } software list( character )
#' \item \emph{ @param } software_versions list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } volume list( character )
#' \item \emph{ @param } workflows list( character )
#' \item \emph{ @returnType } \link{PublicationResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : PublicationResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ ReferenceFiles } \emph{ List items in the ReferenceFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate ReferenceFile items. Supports filtering on fields within ReferenceFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } anvil_url list( character )
#' \item \emph{ @param } assembly Enum < [GRCh38, hg19, GRCm39, mm10] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } controlled_access list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } external list( character )
#' \item \emph{ @param } external_id list( character )
#' \item \emph{ @param } file_format Enum < [bed, csv, dat, fasta, gaf, gds, gtf, obo, owl, PWM, tar, tsv, txt, vcf, xml] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_format_type Enum < [bed12, bed3, bed3+, bed5, bed6, bed6+, bed9, bed9+, mpra_starr] >
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } source_url list( character )
#' \item \emph{ @param } sources list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } transcriptome_annotation Enum < [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34] >
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{ReferenceFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : ReferenceFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Report } \emph{ Generate a report based on search query. All results are returned. }
#' Like /search endpoint but returns a TSV file instead of JSON. Must specify item type(s).
#'
#' \itemize{
#' \item \emph{ @param } query character
#' \item \emph{ @param } type list( character )
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } field_filters \link{object}
#' \item \emph{ @param } include_fields list( character )
#' \item \emph{ @param } frame Enum < [object, embedded] >
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : character
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ RodentDonors } \emph{ List items in the RodentDonor collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate RodentDonor items. Supports filtering on fields within RodentDonor items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } genotype list( character )
#' \item \emph{ @param } individual_rodent list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } phenotypic_features_id list( character )
#' \item \emph{ @param } phenotypic_features_feature_id list( character )
#' \item \emph{ @param } phenotypic_features_feature_term_id list( character )
#' \item \emph{ @param } phenotypic_features_feature_term_name list( character )
#' \item \emph{ @param } phenotypic_features_observation_date list( character )
#' \item \emph{ @param } phenotypic_features_quantity_units Enum < [meter, micromole, nanogram, microgram, milligram, gram, kilogram, milli-International Unit per milliliter, picogram per milliliter, nanogram per milliliter, milligram per deciliter] >
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } rodent_identifier list( character )
#' \item \emph{ @param } sex Enum < [male, female, unspecified] >
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } strain list( character )
#' \item \emph{ @param } strain_background Enum < [A/J (AJ), C57BL/6J (B6), 129S1/SvImJ (129), NOD/ShiLtJ (NOD), NZO/H1LtJ (NZO), CAST/EiJ (CAST), PWK/PhJ (PWK), WSB/EiJ (WSB), CAST (M. m. castaneus), WSB (M. m. domesticus), PWK (M. m. musculus)] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Mus musculus] >
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{RodentDonorResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : RodentDonorResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ SampleTerms } \emph{ List items in the SampleTerm collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate SampleTerm items. Supports filtering on fields within SampleTerm items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } ancestors list( character )
#' \item \emph{ @param } cell_slims list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } deprecated_ntr_terms list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } developmental_slims list( character )
#' \item \emph{ @param } is_a list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } ontology list( character )
#' \item \emph{ @param } organ_slims list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } synonyms list( character )
#' \item \emph{ @param } system_slims list( character )
#' \item \emph{ @param } term_id list( character )
#' \item \emph{ @param } term_name list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{SampleTermResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SampleTermResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ SchemaForItemType } \emph{ Retrieve JSON schema for item type }
#' Returns JSON schemas of all the item types defined in IGVF
#'
#' \itemize{
#' \item \emph{ @param } item_type \link{ItemType}
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Schemas } \emph{ Retrieve JSON schemas for all item types }
#' Returns JSON schemas of all the item types defined in IGVF
#'
#' \itemize{
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Search } \emph{ Search for items in the IGVF Project. }
#' Search endpoint that accepts various query parameters to filter, sort, and paginate results. Supports complex filtering on types and fields within JSON objects.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } type list( character )
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } field_filters \link{object}
#' \item \emph{ @returnType } \link{SearchResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SearchResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ SequenceFiles } \emph{ List items in the SequenceFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate SequenceFile items. Supports filtering on fields within SequenceFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } anvil_url list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } controlled_access list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [bam, fastq, pod5] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } flowcell_id list( character )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } illumina_read_type Enum < [R1, R2, R3, I1, I2] >
#' \item \emph{ @param } index list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lane list( integer )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } maximum_read_length list( integer )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } mean_read_length list( numeric )
#' \item \emph{ @param } minimum_read_length list( integer )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } read_count list( integer )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } seqspecs list( character )
#' \item \emph{ @param } sequencing_kit Enum < [HiSeq SBS Kit v4, HiSeq SR Cluster Kit v4-cBot-HS, HiSeq PE Cluster Kit v4-cBot-HS, HiSeq SR Rapid Cluster Kit v2, HiSeq PE Rapid Cluster Kit v2, HiSeq Rapid SBS Kit v2, HiSeq 3000/4000 SBS Kit, HiSeq 3000/4000 SR Cluster Kit, HiSeq 3000/4000 PE Cluster Kit, MiSeq Reagent Kit v2, NextSeq 500 Mid Output Kit, NextSeq 500 High Output Kit, NextSeq 500 Mid Output v2 Kit, NextSeq 500 High Output v2 Kit, NextSeq 500/550 Mid-Output v2.5 Kit, NextSeq 500/550 High-Output v2.5 Kit, TG NextSeq 500/550 Mid-Output Kit v2.5, TG NextSeq 500/550 High-Output Kit v2.5, NextSeq 1000/2000 P1 Reagent Kit, NextSeq 1000/2000 P2 Reagent Kit, NextSeq 1000/2000 P3 Reagent Kit, NextSeq 1000/2000 P1 XLEAP-SBS Reagent Kit, NextSeq 1000/2000 P2 XLEAP-SBS Reagent Kit, NextSeq 2000 P3 XLEAP-SBS Reagent Kit, NextSeq 2000 P4 XLEAP-SBS Reagent Kit, NovaSeq 6000 SP Reagent Kit v1.5, NovaSeq 6000 S1 Reagent Kit v1.5, NovaSeq 6000 S2 Reagent Kit v1.5, NovaSeq 6000 S4 Reagent Kit v1.5, NovaSeq X Series 10B Reagent Kit, ONT Ligation Sequencing Kit V14, Sequel sequencing kit 3.0, Sequel II sequencing kit 2.0] >
#' \item \emph{ @param } sequencing_platform list( character )
#' \item \emph{ @param } sequencing_run list( integer )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{SequenceFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SequenceFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ SignalFiles } \emph{ List items in the SignalFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate SignalFile items. Supports filtering on fields within SignalFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } assembly Enum < [GRCh38, GRCm39] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_summary list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [bigWig] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } filtered list( character )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } normalized list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } reference_files list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } start_view_position list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } strand_specificity Enum < [plus, minus, unstranded] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } transcriptome_annotation Enum < [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34] >
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{SignalFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SignalFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Software } \emph{ List items in the Software collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Software items. Supports filtering on fields within Software items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } source_url list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } used_by Enum < [consortium analysis, DACC, processing pipelines] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } versions list( character )
#' \item \emph{ @returnType } \link{SoftwareResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SoftwareResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ SoftwareVersions } \emph{ List items in the SoftwareVersion collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate SoftwareVersion items. Supports filtering on fields within SoftwareVersion items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } download_id list( character )
#' \item \emph{ @param } downloaded_url list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } software_id list( character )
#' \item \emph{ @param } software_title list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } version list( character )
#' \item \emph{ @returnType } \link{SoftwareVersionResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SoftwareVersionResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Sources } \emph{ List items in the Source collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Source items. Supports filtering on fields within Source items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{SourceResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : SourceResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ TabularFiles } \emph{ List items in the TabularFile collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate TabularFile items. Supports filtering on fields within TabularFile items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_step_version list( character )
#' \item \emph{ @param } anvil_url list( character )
#' \item \emph{ @param } assembly Enum < [GRCh38, GRCm39] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } content_md5sum list( character )
#' \item \emph{ @param } content_type list( character )
#' \item \emph{ @param } controlled_access list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } derived_from list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_format Enum < [bed, csv, gtf, tsv, txt, vcf] >
#' \item \emph{ @param } file_format_specifications list( character )
#' \item \emph{ @param } file_format_type Enum < [bed12, bed3, bed3+, bed5, bed6, bed6+, bed9, bed9+, mpra_starr] >
#' \item \emph{ @param } file_set list( character )
#' \item \emph{ @param } file_size list( integer )
#' \item \emph{ @param } gene_list_for list( character )
#' \item \emph{ @param } href list( character )
#' \item \emph{ @param } input_file_for list( character )
#' \item \emph{ @param } integrated_in list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } loci_list_for list( character )
#' \item \emph{ @param } md5sum list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } s3_uri list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitted_file_name list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } transcriptome_annotation Enum < [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34] >
#' \item \emph{ @param } upload_status Enum < [pending, file not found, invalidated, validated] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } validation_error_detail list( character )
#' \item \emph{ @returnType } \link{TabularFileResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : TabularFileResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ TechnicalSamples } \emph{ List items in the TechnicalSample collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate TechnicalSample items. Supports filtering on fields within TechnicalSample items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } classifications list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } construct_library_sets list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_obtained list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } file_sets_id list( character )
#' \item \emph{ @param } file_sets_accession list( character )
#' \item \emph{ @param } file_sets_aliases list( character )
#' \item \emph{ @param } file_sets_assay_term_term_name list( character )
#' \item \emph{ @param } file_sets_lab_title list( character )
#' \item \emph{ @param } file_sets_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } file_sets_summary list( character )
#' \item \emph{ @param } institutional_certificates list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } moi list( numeric )
#' \item \emph{ @param } multiplexed_in_id list( character )
#' \item \emph{ @param } multiplexed_in_accession list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } nucleic_acid_delivery Enum < [transfection, adenoviral transduction, lentiviral transduction] >
#' \item \emph{ @param } origin_of list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sample_material Enum < [undefined, inorganic, synthetic, organic] >
#' \item \emph{ @param } sample_terms_id list( character )
#' \item \emph{ @param } sample_terms_term_name list( character )
#' \item \emph{ @param } sorted_fractions list( character )
#' \item \emph{ @param } sorted_from_id list( character )
#' \item \emph{ @param } sorted_from_accession list( character )
#' \item \emph{ @param } sorted_from_detail list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } starting_amount list( numeric )
#' \item \emph{ @param } starting_amount_units Enum < [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Saccharomyces cerevisiae, Homo sapiens] >
#' \item \emph{ @param } time_post_library_delivery list( numeric )
#' \item \emph{ @param } time_post_library_delivery_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{TechnicalSampleResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : TechnicalSampleResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Tissues } \emph{ List items in the Tissue collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Tissue items. Supports filtering on fields within Tissue items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } age list( character )
#' \item \emph{ @param } age_units Enum < [minute, hour, day, week, month, year] >
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biomarkers list( character )
#' \item \emph{ @param } ccf_id list( character )
#' \item \emph{ @param } cellular_sub_pool list( character )
#' \item \emph{ @param } classifications list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } construct_library_sets list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_obtained list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } disease_terms_id list( character )
#' \item \emph{ @param } disease_terms_term_name list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors list( character )
#' \item \emph{ @param } embryonic list( character )
#' \item \emph{ @param } file_sets_id list( character )
#' \item \emph{ @param } file_sets_accession list( character )
#' \item \emph{ @param } file_sets_aliases list( character )
#' \item \emph{ @param } file_sets_assay_term_term_name list( character )
#' \item \emph{ @param } file_sets_lab_title list( character )
#' \item \emph{ @param } file_sets_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } file_sets_summary list( character )
#' \item \emph{ @param } institutional_certificates_id list( character )
#' \item \emph{ @param } institutional_certificates_certificate_identifier list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } lower_bound_age list( numeric )
#' \item \emph{ @param } lower_bound_age_in_hours list( numeric )
#' \item \emph{ @param } modifications_id list( character )
#' \item \emph{ @param } modifications_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } modifications_summary list( character )
#' \item \emph{ @param } moi list( numeric )
#' \item \emph{ @param } multiplexed_in_id list( character )
#' \item \emph{ @param } multiplexed_in_accession list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } nucleic_acid_delivery Enum < [transfection, adenoviral transduction, lentiviral transduction] >
#' \item \emph{ @param } origin_of list( character )
#' \item \emph{ @param } originated_from list( character )
#' \item \emph{ @param } part_of list( character )
#' \item \emph{ @param } parts list( character )
#' \item \emph{ @param } pmi list( integer )
#' \item \emph{ @param } pmi_units Enum < [second, minute, hour, day, week] >
#' \item \emph{ @param } pooled_from list( character )
#' \item \emph{ @param } pooled_in list( character )
#' \item \emph{ @param } preservation_method Enum < [cryopreservation, flash-freezing] >
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sample_terms_id list( character )
#' \item \emph{ @param } sample_terms_term_name list( character )
#' \item \emph{ @param } sex Enum < [female, male, mixed, unspecified] >
#' \item \emph{ @param } sorted_fractions list( character )
#' \item \emph{ @param } sorted_from_id list( character )
#' \item \emph{ @param } sorted_from_accession list( character )
#' \item \emph{ @param } sorted_from_detail list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } starting_amount list( numeric )
#' \item \emph{ @param } starting_amount_units Enum < [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } time_post_library_delivery list( numeric )
#' \item \emph{ @param } time_post_library_delivery_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } treatments_id list( character )
#' \item \emph{ @param } treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } treatments_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } treatments_summary list( character )
#' \item \emph{ @param } treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } upper_bound_age list( numeric )
#' \item \emph{ @param } upper_bound_age_in_hours list( numeric )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{TissueResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : TissueResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Treatments } \emph{ List items in the Treatment collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Treatment items. Supports filtering on fields within Treatment items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } amount list( numeric )
#' \item \emph{ @param } amount_units Enum < [mg/kg, mg/mL, mM, ng/mL, nM, percent, μg/kg, μg/kg, μg/mL, μM, kpa] >
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biosamples_treated list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } depletion list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } duration list( numeric )
#' \item \emph{ @param } duration_units Enum < [second, minute, hour, day] >
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } p_h list( numeric )
#' \item \emph{ @param } post_treatment_time list( numeric )
#' \item \emph{ @param } post_treatment_time_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } temperature list( numeric )
#' \item \emph{ @param } temperature_units Enum < [Celsius] >
#' \item \emph{ @param } treatment_term_id list( character )
#' \item \emph{ @param } treatment_term_name list( character )
#' \item \emph{ @param } treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @returnType } \link{TreatmentResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : TreatmentResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Users } \emph{ List items in the User collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate User items. Supports filtering on fields within User items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } email list( character )
#' \item \emph{ @param } first_name list( character )
#' \item \emph{ @param } groups Enum < [admin, read-only-admin, verified] >
#' \item \emph{ @param } job_title Enum < [Principal Investigator, Co-Investigator, Project Manager, Submitter, Post Doc, Data Wrangler, Scientist, Computational Scientist, Software Developer, NHGRI staff member, Other] >
#' \item \emph{ @param } lab list( character )
#' \item \emph{ @param } last_name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } status Enum < [current, deleted, disabled] >
#' \item \emph{ @param } submits_for list( character )
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } title list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } viewing_groups Enum < [community, IGVF] >
#' \item \emph{ @returnType } \link{UserResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : UserResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ WholeOrganisms } \emph{ List items in the WholeOrganism collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate WholeOrganism items. Supports filtering on fields within WholeOrganism items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } age list( character )
#' \item \emph{ @param } age_units Enum < [minute, hour, day, week, month, year] >
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } biomarkers list( character )
#' \item \emph{ @param } cellular_sub_pool list( character )
#' \item \emph{ @param } classifications list( character )
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } construct_library_sets list( character )
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } date_obtained list( character )
#' \item \emph{ @param } dbxrefs list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } disease_terms_id list( character )
#' \item \emph{ @param } disease_terms_term_name list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } donors list( character )
#' \item \emph{ @param } embryonic list( character )
#' \item \emph{ @param } file_sets_id list( character )
#' \item \emph{ @param } file_sets_accession list( character )
#' \item \emph{ @param } file_sets_aliases list( character )
#' \item \emph{ @param } file_sets_assay_term_term_name list( character )
#' \item \emph{ @param } file_sets_lab_title list( character )
#' \item \emph{ @param } file_sets_status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } file_sets_summary list( character )
#' \item \emph{ @param } institutional_certificates_id list( character )
#' \item \emph{ @param } institutional_certificates_certificate_identifier list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } lot_id list( character )
#' \item \emph{ @param } lower_bound_age list( numeric )
#' \item \emph{ @param } lower_bound_age_in_hours list( numeric )
#' \item \emph{ @param } modifications_id list( character )
#' \item \emph{ @param } modifications_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } modifications_summary list( character )
#' \item \emph{ @param } moi list( numeric )
#' \item \emph{ @param } multiplexed_in_id list( character )
#' \item \emph{ @param } multiplexed_in_accession list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } nucleic_acid_delivery Enum < [transfection, adenoviral transduction, lentiviral transduction] >
#' \item \emph{ @param } origin_of list( character )
#' \item \emph{ @param } originated_from list( character )
#' \item \emph{ @param } part_of list( character )
#' \item \emph{ @param } parts list( character )
#' \item \emph{ @param } pooled_from list( character )
#' \item \emph{ @param } pooled_in list( character )
#' \item \emph{ @param } product_id list( character )
#' \item \emph{ @param } protocols list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } sample_terms_id list( character )
#' \item \emph{ @param } sample_terms_term_name list( character )
#' \item \emph{ @param } sex Enum < [female, male, mixed, unspecified] >
#' \item \emph{ @param } sorted_fractions list( character )
#' \item \emph{ @param } sorted_from_id list( character )
#' \item \emph{ @param } sorted_from_accession list( character )
#' \item \emph{ @param } sorted_from_detail list( character )
#' \item \emph{ @param } sources_id list( character )
#' \item \emph{ @param } starting_amount list( numeric )
#' \item \emph{ @param } starting_amount_units Enum < [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng] >
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } taxa Enum < [Homo sapiens, Mus musculus] >
#' \item \emph{ @param } time_post_library_delivery list( numeric )
#' \item \emph{ @param } time_post_library_delivery_units Enum < [minute, hour, day, week, month] >
#' \item \emph{ @param } treatments_id list( character )
#' \item \emph{ @param } treatments_purpose Enum < [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation] >
#' \item \emph{ @param } treatments_status Enum < [archived, deleted, in progress, released] >
#' \item \emph{ @param } treatments_summary list( character )
#' \item \emph{ @param } treatments_treatment_type Enum < [chemical, protein, environmental] >
#' \item \emph{ @param } upper_bound_age list( numeric )
#' \item \emph{ @param } upper_bound_age_in_hours list( numeric )
#' \item \emph{ @param } url list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } virtual list( character )
#' \item \emph{ @returnType } \link{WholeOrganismResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : WholeOrganismResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' \strong{ Workflows } \emph{ List items in the Workflow collection. }
#' Collection endpoint that accepts various query parameters to filter, sort, and paginate Workflow items. Supports filtering on fields within Workflow items.
#'
#' \itemize{
#' \item \emph{ @param } frame Enum < [object] >
#' \item \emph{ @param } query character
#' \item \emph{ @param } limit \link{Limit}
#' \item \emph{ @param } sort list( character )
#' \item \emph{ @param } id list( character )
#' \item \emph{ @param } accession list( character )
#' \item \emph{ @param } aliases list( character )
#' \item \emph{ @param } alternate_accessions list( character )
#' \item \emph{ @param } analysis_steps list( character )
#' \item \emph{ @param } award_id list( character )
#' \item \emph{ @param } award_component Enum < [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling] >
#' \item \emph{ @param } collections Enum < [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista] >
#' \item \emph{ @param } creation_timestamp list( character )
#' \item \emph{ @param } description list( character )
#' \item \emph{ @param } documents list( character )
#' \item \emph{ @param } lab_id list( character )
#' \item \emph{ @param } lab_title list( character )
#' \item \emph{ @param } name list( character )
#' \item \emph{ @param } notes list( character )
#' \item \emph{ @param } publication_identifiers list( character )
#' \item \emph{ @param } publications list( character )
#' \item \emph{ @param } release_timestamp list( character )
#' \item \emph{ @param } revoke_detail list( character )
#' \item \emph{ @param } source_url list( character )
#' \item \emph{ @param } standards_page_id list( character )
#' \item \emph{ @param } standards_page_title list( character )
#' \item \emph{ @param } status Enum < [in progress, released, deleted, replaced, revoked, archived] >
#' \item \emph{ @param } submitted_by_id list( character )
#' \item \emph{ @param } submitted_by_title list( character )
#' \item \emph{ @param } submitter_comment list( character )
#' \item \emph{ @param } summary list( character )
#' \item \emph{ @param } uuid list( character )
#' \item \emph{ @param } workflow_repositories list( character )
#' \item \emph{ @param } workflow_version list( integer )
#' \item \emph{ @returnType } \link{WorkflowResults} \cr
#'
#'
#' \item status code : 200 | Successful response
#'
#' \item return type : WorkflowResults
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 400 | Bad request
#'
#' \item return type : object
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 404 | No results found
#'
#' \item return type : NoResultsResponse
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' \item status code : 500 | Internal server error
#'
#'
#' \item response headers :
#'
#' \tabular{ll}{
#' }
#' }
#'
#' }
#'
#'
#' @examples
#' \dontrun{
#' ####################  AccessKeys  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_access_key_id <- c("inner_example") # array[character] | Filter by access_key_id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_secret_access_key_hash <- c("inner_example") # array[character] | Filter by secret_access_key_hash (Optional)
#' var_status <- c("current") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_user <- c("inner_example") # array[character] | Filter by user (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the AccessKey collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AccessKeys(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, access_key_id = var_access_key_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, description = var_description, notes = var_notes, secret_access_key_hash = var_secret_access_key_hash, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, user = var_user, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$AccessKeys(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, access_key_id = var_access_key_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, description = var_description, notes = var_notes, secret_access_key_hash = var_secret_access_key_hash, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, user = var_user, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  AlignmentFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_anvil_url <- c("inner_example") # array[character] | Filter by anvil_url (Optional)
#' var_assembly <- c("GRCh38") # array[character] | Filter by assembly (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_summary <- c("inner_example") # array[character] | Filter by content_summary (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_controlled_access <- c("inner_example") # array[character] | Filter by controlled_access (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("bam") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_filtered <- c("inner_example") # array[character] | Filter by filtered (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_redacted <- c("inner_example") # array[character] | Filter by redacted (Optional)
#' var_reference_files <- c("inner_example") # array[character] | Filter by reference_files (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_transcriptome_annotation <- c("GENCODE 40") # array[character] | Filter by transcriptome_annotation (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the AlignmentFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AlignmentFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_summary = var_content_summary, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, filtered = var_filtered, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, redacted = var_redacted, reference_files = var_reference_files, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$AlignmentFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_summary = var_content_summary, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, filtered = var_filtered, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, redacted = var_redacted, reference_files = var_reference_files, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  AnalysisSets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_assay_titles <- c("inner_example") # array[character] | Filter by assay_titles (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_contact_pi_id <- c("inner_example") # array[character] | Filter by award.contact_pi.@id (Optional)
#' var_award_contact_pi_title <- c("inner_example") # array[character] | Filter by award.contact_pi.title (Optional)
#' var_award_title <- c("inner_example") # array[character] | Filter by award.title (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors_id <- c("inner_example") # array[character] | Filter by donors.@id (Optional)
#' var_donors_accession <- c("inner_example") # array[character] | Filter by donors.accession (Optional)
#' var_donors_aliases <- c("inner_example") # array[character] | Filter by donors.aliases (Optional)
#' var_donors_sex <- c("male") # array[character] | Filter by donors.sex (Optional)
#' var_donors_status <- c("in progress") # array[character] | Filter by donors.status (Optional)
#' var_donors_taxa <- c("inner_example") # array[character] | Filter by donors.taxa (Optional)
#' var_file_set_type <- c("intermediate analysis") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_creation_timestamp <- c("inner_example") # array[character] | Filter by files.creation_timestamp (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_file_size <- c(123) # array[integer] | Filter by files.file_size (Optional)
#' var_files_href <- c("inner_example") # array[character] | Filter by files.href (Optional)
#' var_files_s3_uri <- c("inner_example") # array[character] | Filter by files.s3_uri (Optional)
#' var_files_sequencing_platform <- c("inner_example") # array[character] | Filter by files.sequencing_platform (Optional)
#' var_files_submitted_file_name <- c("inner_example") # array[character] | Filter by files.submitted_file_name (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_input_file_sets_id <- c("inner_example") # array[character] | Filter by input_file_sets.@id (Optional)
#' var_input_file_sets_accession <- c("inner_example") # array[character] | Filter by input_file_sets.accession (Optional)
#' var_input_file_sets_aliases <- c("inner_example") # array[character] | Filter by input_file_sets.aliases (Optional)
#' var_input_file_sets_file_set_type <- c("pathogenicity") # array[character] | Filter by input_file_sets.file_set_type (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_samples_id <- c("inner_example") # array[character] | Filter by samples.@id (Optional)
#' var_samples_accession <- c("inner_example") # array[character] | Filter by samples.accession (Optional)
#' var_samples_aliases <- c("inner_example") # array[character] | Filter by samples.aliases (Optional)
#' var_samples_cell_fate_change_treatments <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments (Optional)
#' var_samples_classifications <- c("organoid") # array[character] | Filter by samples.classifications (Optional)
#' var_samples_construct_library_sets <- c("inner_example") # array[character] | Filter by samples.construct_library_sets (Optional)
#' var_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by samples.disease_terms.@id (Optional)
#' var_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by samples.disease_terms.term_name (Optional)
#' var_samples_modifications <- c("inner_example") # array[character] | Filter by samples.modifications (Optional)
#' var_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by samples.sample_terms.@id (Optional)
#' var_samples_sample_terms_aliases <- c("inner_example") # array[character] | Filter by samples.sample_terms.aliases (Optional)
#' var_samples_sample_terms_status <- c("archived") # array[character] | Filter by samples.sample_terms.status (Optional)
#' var_samples_sample_terms_summary <- c("inner_example") # array[character] | Filter by samples.sample_terms.summary (Optional)
#' var_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by samples.sample_terms.term_name (Optional)
#' var_samples_status <- c("in progress") # array[character] | Filter by samples.status (Optional)
#' var_samples_summary <- c("inner_example") # array[character] | Filter by samples.summary (Optional)
#' var_samples_targeted_sample_term_id <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.@id (Optional)
#' var_samples_targeted_sample_term_term_name <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.term_name (Optional)
#' var_samples_taxa <- c("Homo sapiens") # array[character] | Filter by samples.taxa (Optional)
#' var_samples_treatments <- c("inner_example") # array[character] | Filter by samples.treatments (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the AnalysisSet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AnalysisSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assay_titles = var_assay_titles, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, input_file_sets_id = var_input_file_sets_id, input_file_sets_accession = var_input_file_sets_accession, input_file_sets_aliases = var_input_file_sets_aliases, input_file_sets_file_set_type = var_input_file_sets_file_set_type, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$AnalysisSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assay_titles = var_assay_titles, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, input_file_sets_id = var_input_file_sets_id, input_file_sets_accession = var_input_file_sets_accession, input_file_sets_aliases = var_input_file_sets_aliases, input_file_sets_file_set_type = var_input_file_sets_file_set_type, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  AnalysisStepVersions  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_analysis_step_id <- c("inner_example") # array[character] | Filter by analysis_step.@id (Optional)
#' var_analysis_step_name <- c("inner_example") # array[character] | Filter by analysis_step.name (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_software_versions_id <- c("inner_example") # array[character] | Filter by software_versions.@id (Optional)
#' var_software_versions_name <- c("inner_example") # array[character] | Filter by software_versions.name (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the AnalysisStepVersion collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AnalysisStepVersions(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, analysis_step_id = var_analysis_step_id, analysis_step_name = var_analysis_step_name, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, release_timestamp = var_release_timestamp, software_versions_id = var_software_versions_id, software_versions_name = var_software_versions_name, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$AnalysisStepVersions(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, analysis_step_id = var_analysis_step_id, analysis_step_name = var_analysis_step_name, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, release_timestamp = var_release_timestamp, software_versions_id = var_software_versions_id, software_versions_name = var_software_versions_name, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  AnalysisSteps  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_analysis_step_types <- c("alignment") # array[character] | Filter by analysis_step_types (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_input_content_types <- c("inner_example") # array[character] | Filter by input_content_types (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_output_content_types <- c("inner_example") # array[character] | Filter by output_content_types (Optional)
#' var_parents_id <- c("inner_example") # array[character] | Filter by parents.@id (Optional)
#' var_parents_title <- c("inner_example") # array[character] | Filter by parents.title (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_step_label <- c("inner_example") # array[character] | Filter by step_label (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_workflow_id <- c("inner_example") # array[character] | Filter by workflow.@id (Optional)
#' var_workflow_accession <- c("inner_example") # array[character] | Filter by workflow.accession (Optional)
#'
#' #List items in the AnalysisStep collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AnalysisSteps(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, analysis_step_types = var_analysis_step_types, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, input_content_types = var_input_content_types, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, output_content_types = var_output_content_types, parents_id = var_parents_id, parents_title = var_parents_title, release_timestamp = var_release_timestamp, status = var_status, step_label = var_step_label, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid, workflow_id = var_workflow_id, workflow_accession = var_workflow_accessiondata_file = "result.txt")
#' result <- api_instance$AnalysisSteps(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, analysis_step_types = var_analysis_step_types, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, input_content_types = var_input_content_types, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, output_content_types = var_output_content_types, parents_id = var_parents_id, parents_title = var_parents_title, release_timestamp = var_release_timestamp, status = var_status, step_label = var_step_label, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid, workflow_id = var_workflow_id, workflow_accession = var_workflow_accession)
#' dput(result)
#'
#'
#' ####################  AssayTerms  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_ancestors <- c("inner_example") # array[character] | Filter by ancestors (Optional)
#' var_assay_slims <- c("inner_example") # array[character] | Filter by assay_slims (Optional)
#' var_category_slims <- c("inner_example") # array[character] | Filter by category_slims (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_deprecated_ntr_terms <- c("inner_example") # array[character] | Filter by deprecated_ntr_terms (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_is_a <- c("inner_example") # array[character] | Filter by is_a (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_objective_slims <- c("inner_example") # array[character] | Filter by objective_slims (Optional)
#' var_ontology <- c("inner_example") # array[character] | Filter by ontology (Optional)
#' var_preferred_assay_titles <- c("10x multiome") # array[character] | Filter by preferred_assay_titles (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_synonyms <- c("inner_example") # array[character] | Filter by synonyms (Optional)
#' var_term_id <- c("inner_example") # array[character] | Filter by term_id (Optional)
#' var_term_name <- c("inner_example") # array[character] | Filter by term_name (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the AssayTerm collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AssayTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, assay_slims = var_assay_slims, category_slims = var_category_slims, creation_timestamp = var_creation_timestamp, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, is_a = var_is_a, name = var_name, notes = var_notes, objective_slims = var_objective_slims, ontology = var_ontology, preferred_assay_titles = var_preferred_assay_titles, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, term_id = var_term_id, term_name = var_term_name, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$AssayTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, assay_slims = var_assay_slims, category_slims = var_category_slims, creation_timestamp = var_creation_timestamp, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, is_a = var_is_a, name = var_name, notes = var_notes, objective_slims = var_objective_slims, ontology = var_ontology, preferred_assay_titles = var_preferred_assay_titles, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, term_id = var_term_id, term_name = var_term_name, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  AuxiliarySets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_contact_pi_id <- c("inner_example") # array[character] | Filter by award.contact_pi.@id (Optional)
#' var_award_contact_pi_title <- c("inner_example") # array[character] | Filter by award.contact_pi.title (Optional)
#' var_award_title <- c("inner_example") # array[character] | Filter by award.title (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors_id <- c("inner_example") # array[character] | Filter by donors.@id (Optional)
#' var_donors_accession <- c("inner_example") # array[character] | Filter by donors.accession (Optional)
#' var_donors_aliases <- c("inner_example") # array[character] | Filter by donors.aliases (Optional)
#' var_donors_sex <- c("male") # array[character] | Filter by donors.sex (Optional)
#' var_donors_status <- c("in progress") # array[character] | Filter by donors.status (Optional)
#' var_donors_taxa <- c("inner_example") # array[character] | Filter by donors.taxa (Optional)
#' var_file_set_type <- c("cell hashing") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_creation_timestamp <- c("inner_example") # array[character] | Filter by files.creation_timestamp (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_file_size <- c(123) # array[integer] | Filter by files.file_size (Optional)
#' var_files_href <- c("inner_example") # array[character] | Filter by files.href (Optional)
#' var_files_s3_uri <- c("inner_example") # array[character] | Filter by files.s3_uri (Optional)
#' var_files_sequencing_platform <- c("inner_example") # array[character] | Filter by files.sequencing_platform (Optional)
#' var_files_submitted_file_name <- c("inner_example") # array[character] | Filter by files.submitted_file_name (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_library_construction_platform <- c("inner_example") # array[character] | Filter by library_construction_platform (Optional)
#' var_measurement_sets_id <- c("inner_example") # array[character] | Filter by measurement_sets.@id (Optional)
#' var_measurement_sets_accession <- c("inner_example") # array[character] | Filter by measurement_sets.accession (Optional)
#' var_measurement_sets_aliases <- c("inner_example") # array[character] | Filter by measurement_sets.aliases (Optional)
#' var_measurement_sets_preferred_assay_title <- c("10x multiome") # array[character] | Filter by measurement_sets.preferred_assay_title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_samples_id <- c("inner_example") # array[character] | Filter by samples.@id (Optional)
#' var_samples_accession <- c("inner_example") # array[character] | Filter by samples.accession (Optional)
#' var_samples_aliases <- c("inner_example") # array[character] | Filter by samples.aliases (Optional)
#' var_samples_cell_fate_change_treatments <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments (Optional)
#' var_samples_classifications <- c("organoid") # array[character] | Filter by samples.classifications (Optional)
#' var_samples_construct_library_sets <- c("inner_example") # array[character] | Filter by samples.construct_library_sets (Optional)
#' var_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by samples.disease_terms.@id (Optional)
#' var_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by samples.disease_terms.term_name (Optional)
#' var_samples_modifications <- c("inner_example") # array[character] | Filter by samples.modifications (Optional)
#' var_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by samples.sample_terms.@id (Optional)
#' var_samples_sample_terms_aliases <- c("inner_example") # array[character] | Filter by samples.sample_terms.aliases (Optional)
#' var_samples_sample_terms_status <- c("archived") # array[character] | Filter by samples.sample_terms.status (Optional)
#' var_samples_sample_terms_summary <- c("inner_example") # array[character] | Filter by samples.sample_terms.summary (Optional)
#' var_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by samples.sample_terms.term_name (Optional)
#' var_samples_status <- c("in progress") # array[character] | Filter by samples.status (Optional)
#' var_samples_summary <- c("inner_example") # array[character] | Filter by samples.summary (Optional)
#' var_samples_targeted_sample_term_id <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.@id (Optional)
#' var_samples_targeted_sample_term_term_name <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.term_name (Optional)
#' var_samples_taxa <- c("Homo sapiens") # array[character] | Filter by samples.taxa (Optional)
#' var_samples_treatments <- c("inner_example") # array[character] | Filter by samples.treatments (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the AuxiliarySet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$AuxiliarySets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, lab_id = var_lab_id, lab_title = var_lab_title, library_construction_platform = var_library_construction_platform, measurement_sets_id = var_measurement_sets_id, measurement_sets_accession = var_measurement_sets_accession, measurement_sets_aliases = var_measurement_sets_aliases, measurement_sets_preferred_assay_title = var_measurement_sets_preferred_assay_title, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, url = var_url, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$AuxiliarySets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, lab_id = var_lab_id, lab_title = var_lab_title, library_construction_platform = var_library_construction_platform, measurement_sets_id = var_measurement_sets_id, measurement_sets_accession = var_measurement_sets_accession, measurement_sets_aliases = var_measurement_sets_aliases, measurement_sets_preferred_assay_title = var_measurement_sets_preferred_assay_title, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, url = var_url, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  Awards  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_component <- c("affiliate") # array[character] | Filter by component (Optional)
#' var_contact_pi <- c("inner_example") # array[character] | Filter by contact_pi (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_end_date <- c("inner_example") # array[character] | Filter by end_date (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_pis <- c("inner_example") # array[character] | Filter by pis (Optional)
#' var_project <- c("community") # array[character] | Filter by project (Optional)
#' var_start_date <- c("inner_example") # array[character] | Filter by start_date (Optional)
#' var_status <- c("current") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_viewing_group <- c("community") # array[character] | Filter by viewing_group (Optional)
#'
#' #List items in the Award collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Awards(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, component = var_component, contact_pi = var_contact_pi, creation_timestamp = var_creation_timestamp, description = var_description, end_date = var_end_date, name = var_name, notes = var_notes, pis = var_pis, project = var_project, start_date = var_start_date, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, url = var_url, uuid = var_uuid, viewing_group = var_viewing_groupdata_file = "result.txt")
#' result <- api_instance$Awards(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, component = var_component, contact_pi = var_contact_pi, creation_timestamp = var_creation_timestamp, description = var_description, end_date = var_end_date, name = var_name, notes = var_notes, pis = var_pis, project = var_project, start_date = var_start_date, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, url = var_url, uuid = var_uuid, viewing_group = var_viewing_group)
#' dput(result)
#'
#'
#' ####################  BatchDownload  ####################
#'
#' library(igvfclient)
#' var_type <- c("inner_example") # array[character] | Type of objects to return. Can be repeated for multiple types.
#' var_query <- "query_example" # character | Query string for searching. (Optional)
#' var_field_filters <- c(key = TODO) # object | Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields. (Optional)
#'
#' #List files to download based on search query. All results are returned.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$BatchDownload(var_type, query = var_query, field_filters = var_field_filtersdata_file = "result.txt")
#' result <- api_instance$BatchDownload(var_type, query = var_query, field_filters = var_field_filters)
#' dput(result)
#'
#'
#' ####################  Biomarkers  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_name <- c("inner_example") # array[character] | Filter by award.name (Optional)
#' var_biomarker_for <- c("inner_example") # array[character] | Filter by biomarker_for (Optional)
#' var_classification <- c("cell surface protein") # array[character] | Filter by classification (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_gene <- c("inner_example") # array[character] | Filter by gene (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_name_quantification <- c("inner_example") # array[character] | Filter by name_quantification (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_quantification <- c("negative") # array[character] | Filter by quantification (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_synonyms <- c("inner_example") # array[character] | Filter by synonyms (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Biomarker collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Biomarkers(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, award_name = var_award_name, biomarker_for = var_biomarker_for, classification = var_classification, creation_timestamp = var_creation_timestamp, description = var_description, gene = var_gene, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, name_quantification = var_name_quantification, notes = var_notes, quantification = var_quantification, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Biomarkers(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, award_name = var_award_name, biomarker_for = var_biomarker_for, classification = var_classification, creation_timestamp = var_creation_timestamp, description = var_description, gene = var_gene, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, name_quantification = var_name_quantification, notes = var_notes, quantification = var_quantification, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  ConfigurationFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("yaml") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_seqspec_of <- c("inner_example") # array[character] | Filter by seqspec_of (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the ConfigurationFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$ConfigurationFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, seqspec_of = var_seqspec_of, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$ConfigurationFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, seqspec_of = var_seqspec_of, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  ConstructLibrarySets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_applied_to_samples_id <- c("inner_example") # array[character] | Filter by applied_to_samples.@id (Optional)
#' var_applied_to_samples_accession <- c("inner_example") # array[character] | Filter by applied_to_samples.accession (Optional)
#' var_applied_to_samples_aliases <- c("inner_example") # array[character] | Filter by applied_to_samples.aliases (Optional)
#' var_applied_to_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by applied_to_samples.disease_terms.@id (Optional)
#' var_applied_to_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by applied_to_samples.disease_terms.term_name (Optional)
#' var_applied_to_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by applied_to_samples.sample_terms.@id (Optional)
#' var_applied_to_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by applied_to_samples.sample_terms.term_name (Optional)
#' var_applied_to_samples_status <- c("in progress") # array[character] | Filter by applied_to_samples.status (Optional)
#' var_applied_to_samples_summary <- c("inner_example") # array[character] | Filter by applied_to_samples.summary (Optional)
#' var_associated_phenotypes_id <- c("inner_example") # array[character] | Filter by associated_phenotypes.@id (Optional)
#' var_associated_phenotypes_term_id <- c("inner_example") # array[character] | Filter by associated_phenotypes.term_id (Optional)
#' var_associated_phenotypes_term_name <- c("inner_example") # array[character] | Filter by associated_phenotypes.term_name (Optional)
#' var_average_guide_coverage <- c(123) # array[numeric] | Filter by average_guide_coverage (Optional)
#' var_average_insert_size <- c(123) # array[numeric] | Filter by average_insert_size (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_exon <- c("inner_example") # array[character] | Filter by exon (Optional)
#' var_file_set_type <- c("guide library") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_guide_type <- c("sgRNA") # array[character] | Filter by guide_type (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_integrated_content_files <- c("inner_example") # array[character] | Filter by integrated_content_files (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_large_scale_gene_list_id <- c("inner_example") # array[character] | Filter by large_scale_gene_list.@id (Optional)
#' var_large_scale_gene_list_accession <- c("inner_example") # array[character] | Filter by large_scale_gene_list.accession (Optional)
#' var_large_scale_gene_list_aliases <- c("inner_example") # array[character] | Filter by large_scale_gene_list.aliases (Optional)
#' var_large_scale_loci_list_id <- c("inner_example") # array[character] | Filter by large_scale_loci_list.@id (Optional)
#' var_large_scale_loci_list_accession <- c("inner_example") # array[character] | Filter by large_scale_loci_list.accession (Optional)
#' var_large_scale_loci_list_aliases <- c("inner_example") # array[character] | Filter by large_scale_loci_list.aliases (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_lower_bound_guide_coverage <- c(123) # array[integer] | Filter by lower_bound_guide_coverage (Optional)
#' var_lower_bound_insert_size <- c(123) # array[integer] | Filter by lower_bound_insert_size (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_orf_list_id <- c("inner_example") # array[character] | Filter by orf_list.@id (Optional)
#' var_orf_list_aliases <- c("inner_example") # array[character] | Filter by orf_list.aliases (Optional)
#' var_orf_list_gene <- c("inner_example") # array[character] | Filter by orf_list.gene (Optional)
#' var_orf_list_orf_id <- c("inner_example") # array[character] | Filter by orf_list.orf_id (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_scope <- c("tile") # array[character] | Filter by scope (Optional)
#' var_selection_criteria <- c("accessible genome regions") # array[character] | Filter by selection_criteria (Optional)
#' var_small_scale_gene_list_id <- c("inner_example") # array[character] | Filter by small_scale_gene_list.@id (Optional)
#' var_small_scale_gene_list_geneid <- c("inner_example") # array[character] | Filter by small_scale_gene_list.geneid (Optional)
#' var_small_scale_gene_list_name <- c("inner_example") # array[character] | Filter by small_scale_gene_list.name (Optional)
#' var_small_scale_gene_list_symbol <- c("inner_example") # array[character] | Filter by small_scale_gene_list.symbol (Optional)
#' var_small_scale_gene_list_synonyms <- c("inner_example") # array[character] | Filter by small_scale_gene_list.synonyms (Optional)
#' var_small_scale_loci_list <- c(Locus$new("GRCh38", "chromosome_example", 123, 123)) # array[Locus] | Filter by small_scale_loci_list (Optional)
#' var_sources <- c("inner_example") # array[character] | Filter by sources (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_targeton <- c("inner_example") # array[character] | Filter by targeton (Optional)
#' var_tiling_modality <- c("peak tiling") # array[character] | Filter by tiling_modality (Optional)
#' var_upper_bound_guide_coverage <- c(123) # array[integer] | Filter by upper_bound_guide_coverage (Optional)
#' var_upper_bound_insert_size <- c(123) # array[integer] | Filter by upper_bound_insert_size (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the ConstructLibrarySet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$ConstructLibrarySets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, applied_to_samples_id = var_applied_to_samples_id, applied_to_samples_accession = var_applied_to_samples_accession, applied_to_samples_aliases = var_applied_to_samples_aliases, applied_to_samples_disease_terms_id = var_applied_to_samples_disease_terms_id, applied_to_samples_disease_terms_term_name = var_applied_to_samples_disease_terms_term_name, applied_to_samples_sample_terms_id = var_applied_to_samples_sample_terms_id, applied_to_samples_sample_terms_term_name = var_applied_to_samples_sample_terms_term_name, applied_to_samples_status = var_applied_to_samples_status, applied_to_samples_summary = var_applied_to_samples_summary, associated_phenotypes_id = var_associated_phenotypes_id, associated_phenotypes_term_id = var_associated_phenotypes_term_id, associated_phenotypes_term_name = var_associated_phenotypes_term_name, average_guide_coverage = var_average_guide_coverage, average_insert_size = var_average_insert_size, award_id = var_award_id, award_component = var_award_component, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, description = var_description, documents = var_documents, exon = var_exon, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_file_format = var_files_file_format, files_upload_status = var_files_upload_status, guide_type = var_guide_type, input_file_set_for = var_input_file_set_for, integrated_content_files = var_integrated_content_files, lab_id = var_lab_id, lab_title = var_lab_title, large_scale_gene_list_id = var_large_scale_gene_list_id, large_scale_gene_list_accession = var_large_scale_gene_list_accession, large_scale_gene_list_aliases = var_large_scale_gene_list_aliases, large_scale_loci_list_id = var_large_scale_loci_list_id, large_scale_loci_list_accession = var_large_scale_loci_list_accession, large_scale_loci_list_aliases = var_large_scale_loci_list_aliases, lot_id = var_lot_id, lower_bound_guide_coverage = var_lower_bound_guide_coverage, lower_bound_insert_size = var_lower_bound_insert_size, notes = var_notes, orf_list_id = var_orf_list_id, orf_list_aliases = var_orf_list_aliases, orf_list_gene = var_orf_list_gene, orf_list_orf_id = var_orf_list_orf_id, product_id = var_product_id, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, scope = var_scope, selection_criteria = var_selection_criteria, small_scale_gene_list_id = var_small_scale_gene_list_id, small_scale_gene_list_geneid = var_small_scale_gene_list_geneid, small_scale_gene_list_name = var_small_scale_gene_list_name, small_scale_gene_list_symbol = var_small_scale_gene_list_symbol, small_scale_gene_list_synonyms = var_small_scale_gene_list_synonyms, small_scale_loci_list = var_small_scale_loci_list, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, targeton = var_targeton, tiling_modality = var_tiling_modality, upper_bound_guide_coverage = var_upper_bound_guide_coverage, upper_bound_insert_size = var_upper_bound_insert_size, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$ConstructLibrarySets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, applied_to_samples_id = var_applied_to_samples_id, applied_to_samples_accession = var_applied_to_samples_accession, applied_to_samples_aliases = var_applied_to_samples_aliases, applied_to_samples_disease_terms_id = var_applied_to_samples_disease_terms_id, applied_to_samples_disease_terms_term_name = var_applied_to_samples_disease_terms_term_name, applied_to_samples_sample_terms_id = var_applied_to_samples_sample_terms_id, applied_to_samples_sample_terms_term_name = var_applied_to_samples_sample_terms_term_name, applied_to_samples_status = var_applied_to_samples_status, applied_to_samples_summary = var_applied_to_samples_summary, associated_phenotypes_id = var_associated_phenotypes_id, associated_phenotypes_term_id = var_associated_phenotypes_term_id, associated_phenotypes_term_name = var_associated_phenotypes_term_name, average_guide_coverage = var_average_guide_coverage, average_insert_size = var_average_insert_size, award_id = var_award_id, award_component = var_award_component, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, description = var_description, documents = var_documents, exon = var_exon, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_file_format = var_files_file_format, files_upload_status = var_files_upload_status, guide_type = var_guide_type, input_file_set_for = var_input_file_set_for, integrated_content_files = var_integrated_content_files, lab_id = var_lab_id, lab_title = var_lab_title, large_scale_gene_list_id = var_large_scale_gene_list_id, large_scale_gene_list_accession = var_large_scale_gene_list_accession, large_scale_gene_list_aliases = var_large_scale_gene_list_aliases, large_scale_loci_list_id = var_large_scale_loci_list_id, large_scale_loci_list_accession = var_large_scale_loci_list_accession, large_scale_loci_list_aliases = var_large_scale_loci_list_aliases, lot_id = var_lot_id, lower_bound_guide_coverage = var_lower_bound_guide_coverage, lower_bound_insert_size = var_lower_bound_insert_size, notes = var_notes, orf_list_id = var_orf_list_id, orf_list_aliases = var_orf_list_aliases, orf_list_gene = var_orf_list_gene, orf_list_orf_id = var_orf_list_orf_id, product_id = var_product_id, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, scope = var_scope, selection_criteria = var_selection_criteria, small_scale_gene_list_id = var_small_scale_gene_list_id, small_scale_gene_list_geneid = var_small_scale_gene_list_geneid, small_scale_gene_list_name = var_small_scale_gene_list_name, small_scale_gene_list_symbol = var_small_scale_gene_list_symbol, small_scale_gene_list_synonyms = var_small_scale_gene_list_synonyms, small_scale_loci_list = var_small_scale_loci_list, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, targeton = var_targeton, tiling_modality = var_tiling_modality, upper_bound_guide_coverage = var_upper_bound_guide_coverage, upper_bound_insert_size = var_upper_bound_insert_size, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  CrisprModifications  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_activated <- c("inner_example") # array[character] | Filter by activated (Optional)
#' var_activating_agent_term_id <- c("inner_example") # array[character] | Filter by activating_agent_term_id (Optional)
#' var_activating_agent_term_name <- c("inner_example") # array[character] | Filter by activating_agent_term_name (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biosamples_modified <- c("inner_example") # array[character] | Filter by biosamples_modified (Optional)
#' var_cas <- c("Cas9") # array[character] | Filter by cas (Optional)
#' var_cas_species <- c("Streptococcus pyogenes (Sp)") # array[character] | Filter by cas_species (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_fused_domain <- c("2xVP64") # array[character] | Filter by fused_domain (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_modality <- c("activation") # array[character] | Filter by modality (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_sources <- c("inner_example") # array[character] | Filter by sources (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_tagged_protein <- c("inner_example") # array[character] | Filter by tagged_protein (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the CrisprModification collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$CrisprModifications(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, activated = var_activated, activating_agent_term_id = var_activating_agent_term_id, activating_agent_term_name = var_activating_agent_term_name, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, biosamples_modified = var_biosamples_modified, cas = var_cas, cas_species = var_cas_species, creation_timestamp = var_creation_timestamp, description = var_description, documents = var_documents, fused_domain = var_fused_domain, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, modality = var_modality, notes = var_notes, product_id = var_product_id, release_timestamp = var_release_timestamp, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, tagged_protein = var_tagged_protein, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$CrisprModifications(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, activated = var_activated, activating_agent_term_id = var_activating_agent_term_id, activating_agent_term_name = var_activating_agent_term_name, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, biosamples_modified = var_biosamples_modified, cas = var_cas, cas_species = var_cas_species, creation_timestamp = var_creation_timestamp, description = var_description, documents = var_documents, fused_domain = var_fused_domain, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, modality = var_modality, notes = var_notes, product_id = var_product_id, release_timestamp = var_release_timestamp, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, tagged_protein = var_tagged_protein, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  CuratedSets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_assemblies <- c("inner_example") # array[character] | Filter by assemblies (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_contact_pi_id <- c("inner_example") # array[character] | Filter by award.contact_pi.@id (Optional)
#' var_award_contact_pi_title <- c("inner_example") # array[character] | Filter by award.contact_pi.title (Optional)
#' var_award_title <- c("inner_example") # array[character] | Filter by award.title (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors_id <- c("inner_example") # array[character] | Filter by donors.@id (Optional)
#' var_donors_accession <- c("inner_example") # array[character] | Filter by donors.accession (Optional)
#' var_donors_aliases <- c("inner_example") # array[character] | Filter by donors.aliases (Optional)
#' var_donors_sex <- c("male") # array[character] | Filter by donors.sex (Optional)
#' var_donors_status <- c("in progress") # array[character] | Filter by donors.status (Optional)
#' var_donors_taxa <- c("inner_example") # array[character] | Filter by donors.taxa (Optional)
#' var_file_set_type <- c("barcodes") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_creation_timestamp <- c("inner_example") # array[character] | Filter by files.creation_timestamp (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_file_size <- c(123) # array[integer] | Filter by files.file_size (Optional)
#' var_files_href <- c("inner_example") # array[character] | Filter by files.href (Optional)
#' var_files_s3_uri <- c("inner_example") # array[character] | Filter by files.s3_uri (Optional)
#' var_files_sequencing_platform <- c("inner_example") # array[character] | Filter by files.sequencing_platform (Optional)
#' var_files_submitted_file_name <- c("inner_example") # array[character] | Filter by files.submitted_file_name (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_samples_id <- c("inner_example") # array[character] | Filter by samples.@id (Optional)
#' var_samples_accession <- c("inner_example") # array[character] | Filter by samples.accession (Optional)
#' var_samples_aliases <- c("inner_example") # array[character] | Filter by samples.aliases (Optional)
#' var_samples_cell_fate_change_treatments <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments (Optional)
#' var_samples_classifications <- c("organoid") # array[character] | Filter by samples.classifications (Optional)
#' var_samples_construct_library_sets <- c("inner_example") # array[character] | Filter by samples.construct_library_sets (Optional)
#' var_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by samples.disease_terms.@id (Optional)
#' var_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by samples.disease_terms.term_name (Optional)
#' var_samples_modifications <- c("inner_example") # array[character] | Filter by samples.modifications (Optional)
#' var_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by samples.sample_terms.@id (Optional)
#' var_samples_sample_terms_aliases <- c("inner_example") # array[character] | Filter by samples.sample_terms.aliases (Optional)
#' var_samples_sample_terms_status <- c("archived") # array[character] | Filter by samples.sample_terms.status (Optional)
#' var_samples_sample_terms_summary <- c("inner_example") # array[character] | Filter by samples.sample_terms.summary (Optional)
#' var_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by samples.sample_terms.term_name (Optional)
#' var_samples_status <- c("in progress") # array[character] | Filter by samples.status (Optional)
#' var_samples_summary <- c("inner_example") # array[character] | Filter by samples.summary (Optional)
#' var_samples_targeted_sample_term_id <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.@id (Optional)
#' var_samples_targeted_sample_term_term_name <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.term_name (Optional)
#' var_samples_taxa <- c("Homo sapiens") # array[character] | Filter by samples.taxa (Optional)
#' var_samples_treatments <- c("inner_example") # array[character] | Filter by samples.treatments (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_transcriptome_annotations <- c("inner_example") # array[character] | Filter by transcriptome_annotations (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the CuratedSet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$CuratedSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assemblies = var_assemblies, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, transcriptome_annotations = var_transcriptome_annotations, url = var_url, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$CuratedSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assemblies = var_assemblies, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, transcriptome_annotations = var_transcriptome_annotations, url = var_url, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  DegronModifications  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_activated <- c("inner_example") # array[character] | Filter by activated (Optional)
#' var_activating_agent_term_id <- c("inner_example") # array[character] | Filter by activating_agent_term_id (Optional)
#' var_activating_agent_term_name <- c("inner_example") # array[character] | Filter by activating_agent_term_name (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biosamples_modified <- c("inner_example") # array[character] | Filter by biosamples_modified (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_degron_system <- c("AID") # array[character] | Filter by degron_system (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_modality <- c("degradation") # array[character] | Filter by modality (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_sources <- c("inner_example") # array[character] | Filter by sources (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_tagged_proteins <- c("inner_example") # array[character] | Filter by tagged_proteins (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the DegronModification collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$DegronModifications(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, activated = var_activated, activating_agent_term_id = var_activating_agent_term_id, activating_agent_term_name = var_activating_agent_term_name, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, biosamples_modified = var_biosamples_modified, creation_timestamp = var_creation_timestamp, degron_system = var_degron_system, description = var_description, documents = var_documents, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, modality = var_modality, notes = var_notes, product_id = var_product_id, release_timestamp = var_release_timestamp, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, tagged_proteins = var_tagged_proteins, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$DegronModifications(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, activated = var_activated, activating_agent_term_id = var_activating_agent_term_id, activating_agent_term_name = var_activating_agent_term_name, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, biosamples_modified = var_biosamples_modified, creation_timestamp = var_creation_timestamp, degron_system = var_degron_system, description = var_description, documents = var_documents, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, modality = var_modality, notes = var_notes, product_id = var_product_id, release_timestamp = var_release_timestamp, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, tagged_proteins = var_tagged_proteins, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  Documents  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_characterization_method <- c("FACS") # array[character] | Filter by characterization_method (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_document_type <- c("cell fate change protocol") # array[character] | Filter by document_type (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_urls <- c("inner_example") # array[character] | Filter by urls (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Document collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Documents(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, characterization_method = var_characterization_method, creation_timestamp = var_creation_timestamp, description = var_description, document_type = var_document_type, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, urls = var_urls, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Documents(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, characterization_method = var_characterization_method, creation_timestamp = var_creation_timestamp, description = var_description, document_type = var_document_type, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, urls = var_urls, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  Download  ####################
#'
#' library(igvfclient)
#' var_file_id <- "file_id_example" # character | The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03).
#'
#' #Download file.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Download(var_file_iddata_file = "result.txt")
#' result <- api_instance$Download(var_file_id)
#' dput(result)
#'
#'
#' ####################  Genes  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_geneid <- c("inner_example") # array[character] | Filter by geneid (Optional)
#' var_geneid_with_version <- c("inner_example") # array[character] | Filter by geneid_with_version (Optional)
#' var_locations <- c(Gene_Location$new("GRCh38", "chromosome_example", 123, 123)) # array[GeneLocation] | Filter by locations (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_symbol <- c("inner_example") # array[character] | Filter by symbol (Optional)
#' var_synonyms <- c("inner_example") # array[character] | Filter by synonyms (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_transcriptome_annotation <- c("GENCODE 40") # array[character] | Filter by transcriptome_annotation (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_version_number <- c("inner_example") # array[character] | Filter by version_number (Optional)
#'
#' #List items in the Gene collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Genes(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, geneid = var_geneid, geneid_with_version = var_geneid_with_version, locations = var_locations, name = var_name, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, symbol = var_symbol, synonyms = var_synonyms, taxa = var_taxa, title = var_title, transcriptome_annotation = var_transcriptome_annotation, uuid = var_uuid, version_number = var_version_numberdata_file = "result.txt")
#' result <- api_instance$Genes(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, geneid = var_geneid, geneid_with_version = var_geneid_with_version, locations = var_locations, name = var_name, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, symbol = var_symbol, synonyms = var_synonyms, taxa = var_taxa, title = var_title, transcriptome_annotation = var_transcriptome_annotation, uuid = var_uuid, version_number = var_version_number)
#' dput(result)
#'
#'
#' ####################  GenomeBrowserAnnotationFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_assembly <- c("GRCh38") # array[character] | Filter by assembly (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("bigBed") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_format_type <- c("bed12") # array[character] | Filter by file_format_type (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_transcriptome_annotation <- c("GENCODE 40") # array[character] | Filter by transcriptome_annotation (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the GenomeBrowserAnnotationFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$GenomeBrowserAnnotationFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_format_type = var_file_format_type, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$GenomeBrowserAnnotationFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_format_type = var_file_format_type, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  GetById  ####################
#'
#' library(igvfclient)
#' var_resource_id <- "/sequence-files/IGVFFI1165AJSO/" # character | The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`).
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#'
#' #Get item information
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$GetById(var_resource_id, var_framedata_file = "result.txt")
#' result <- api_instance$GetById(var_resource_id, var_frame)
#' dput(result)
#'
#'
#' ####################  HumanDonors  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_ethnicities <- c("African American") # array[character] | Filter by ethnicities (Optional)
#' var_human_donor_identifiers <- c("inner_example") # array[character] | Filter by human_donor_identifiers (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_phenotypic_features_id <- c("inner_example") # array[character] | Filter by phenotypic_features.@id (Optional)
#' var_phenotypic_features_feature_id <- c("inner_example") # array[character] | Filter by phenotypic_features.feature.@id (Optional)
#' var_phenotypic_features_feature_term_id <- c("inner_example") # array[character] | Filter by phenotypic_features.feature.term_id (Optional)
#' var_phenotypic_features_feature_term_name <- c("inner_example") # array[character] | Filter by phenotypic_features.feature.term_name (Optional)
#' var_phenotypic_features_observation_date <- c("inner_example") # array[character] | Filter by phenotypic_features.observation_date (Optional)
#' var_phenotypic_features_quantity_units <- c("meter") # array[character] | Filter by phenotypic_features.quantity_units (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_related_donors_donor_id <- c("inner_example") # array[character] | Filter by related_donors.donor.@id (Optional)
#' var_related_donors_donor_accession <- c("inner_example") # array[character] | Filter by related_donors.donor.accession (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sex <- c("male") # array[character] | Filter by sex (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the HumanDonor collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$HumanDonors(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, collections = var_collections, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, ethnicities = var_ethnicities, human_donor_identifiers = var_human_donor_identifiers, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, phenotypic_features_id = var_phenotypic_features_id, phenotypic_features_feature_id = var_phenotypic_features_feature_id, phenotypic_features_feature_term_id = var_phenotypic_features_feature_term_id, phenotypic_features_feature_term_name = var_phenotypic_features_feature_term_name, phenotypic_features_observation_date = var_phenotypic_features_observation_date, phenotypic_features_quantity_units = var_phenotypic_features_quantity_units, publication_identifiers = var_publication_identifiers, publications = var_publications, related_donors_donor_id = var_related_donors_donor_id, related_donors_donor_accession = var_related_donors_donor_accession, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sex = var_sex, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$HumanDonors(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, collections = var_collections, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, ethnicities = var_ethnicities, human_donor_identifiers = var_human_donor_identifiers, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, phenotypic_features_id = var_phenotypic_features_id, phenotypic_features_feature_id = var_phenotypic_features_feature_id, phenotypic_features_feature_term_id = var_phenotypic_features_feature_term_id, phenotypic_features_feature_term_name = var_phenotypic_features_feature_term_name, phenotypic_features_observation_date = var_phenotypic_features_observation_date, phenotypic_features_quantity_units = var_phenotypic_features_quantity_units, publication_identifiers = var_publication_identifiers, publications = var_publications, related_donors_donor_id = var_related_donors_donor_id, related_donors_donor_accession = var_related_donors_donor_accession, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sex = var_sex, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  ImageFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("jpg") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the ImageFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$ImageFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$ImageFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  Images  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_caption <- c("inner_example") # array[character] | Filter by caption (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_download_url <- c("inner_example") # array[character] | Filter by download_url (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_thumb_nail <- c("inner_example") # array[character] | Filter by thumb_nail (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Image collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Images(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, caption = var_caption, creation_timestamp = var_creation_timestamp, description = var_description, download_url = var_download_url, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, thumb_nail = var_thumb_nail, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Images(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, caption = var_caption, creation_timestamp = var_creation_timestamp, description = var_description, download_url = var_download_url, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, thumb_nail = var_thumb_nail, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  InVitroSystems  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_age <- c("inner_example") # array[character] | Filter by age (Optional)
#' var_age_units <- c("minute") # array[character] | Filter by age_units (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biomarkers <- c("inner_example") # array[character] | Filter by biomarkers (Optional)
#' var_cell_fate_change_protocol <- c("inner_example") # array[character] | Filter by cell_fate_change_protocol (Optional)
#' var_cell_fate_change_treatments_id <- c("inner_example") # array[character] | Filter by cell_fate_change_treatments.@id (Optional)
#' var_cell_fate_change_treatments_purpose <- c("activation") # array[character] | Filter by cell_fate_change_treatments.purpose (Optional)
#' var_cell_fate_change_treatments_status <- c("archived") # array[character] | Filter by cell_fate_change_treatments.status (Optional)
#' var_cell_fate_change_treatments_summary <- c("inner_example") # array[character] | Filter by cell_fate_change_treatments.summary (Optional)
#' var_cell_fate_change_treatments_treatment_type <- c("chemical") # array[character] | Filter by cell_fate_change_treatments.treatment_type (Optional)
#' var_cellular_sub_pool <- c("inner_example") # array[character] | Filter by cellular_sub_pool (Optional)
#' var_classifications <- c("organoid") # array[character] | Filter by classifications (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_construct_library_sets <- c("inner_example") # array[character] | Filter by construct_library_sets (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_obtained <- c("inner_example") # array[character] | Filter by date_obtained (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_demultiplexed_from <- c("inner_example") # array[character] | Filter by demultiplexed_from (Optional)
#' var_demultiplexed_to <- c("inner_example") # array[character] | Filter by demultiplexed_to (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_disease_terms_id <- c("inner_example") # array[character] | Filter by disease_terms.@id (Optional)
#' var_disease_terms_term_name <- c("inner_example") # array[character] | Filter by disease_terms.term_name (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors <- c("inner_example") # array[character] | Filter by donors (Optional)
#' var_embryonic <- c("inner_example") # array[character] | Filter by embryonic (Optional)
#' var_file_sets_id <- c("inner_example") # array[character] | Filter by file_sets.@id (Optional)
#' var_file_sets_accession <- c("inner_example") # array[character] | Filter by file_sets.accession (Optional)
#' var_file_sets_aliases <- c("inner_example") # array[character] | Filter by file_sets.aliases (Optional)
#' var_file_sets_assay_term_term_name <- c("inner_example") # array[character] | Filter by file_sets.assay_term.term_name (Optional)
#' var_file_sets_lab_title <- c("inner_example") # array[character] | Filter by file_sets.lab.title (Optional)
#' var_file_sets_status <- c("in progress") # array[character] | Filter by file_sets.status (Optional)
#' var_file_sets_summary <- c("inner_example") # array[character] | Filter by file_sets.summary (Optional)
#' var_growth_medium <- c("DMEM with serum") # array[character] | Filter by growth_medium (Optional)
#' var_institutional_certificates_id <- c("inner_example") # array[character] | Filter by institutional_certificates.@id (Optional)
#' var_institutional_certificates_certificate_identifier <- c("inner_example") # array[character] | Filter by institutional_certificates.certificate_identifier (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_lower_bound_age <- c(123) # array[numeric] | Filter by lower_bound_age (Optional)
#' var_lower_bound_age_in_hours <- c(123) # array[numeric] | Filter by lower_bound_age_in_hours (Optional)
#' var_modifications_id <- c("inner_example") # array[character] | Filter by modifications.@id (Optional)
#' var_modifications_status <- c("archived") # array[character] | Filter by modifications.status (Optional)
#' var_modifications_summary <- c("inner_example") # array[character] | Filter by modifications.summary (Optional)
#' var_moi <- c(123) # array[numeric] | Filter by moi (Optional)
#' var_multiplexed_in_id <- c("inner_example") # array[character] | Filter by multiplexed_in.@id (Optional)
#' var_multiplexed_in_accession <- c("inner_example") # array[character] | Filter by multiplexed_in.accession (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_nucleic_acid_delivery <- c("transfection") # array[character] | Filter by nucleic_acid_delivery (Optional)
#' var_origin_of <- c("inner_example") # array[character] | Filter by origin_of (Optional)
#' var_originated_from_id <- c("inner_example") # array[character] | Filter by originated_from.@id (Optional)
#' var_originated_from_accession <- c("inner_example") # array[character] | Filter by originated_from.accession (Optional)
#' var_part_of <- c("inner_example") # array[character] | Filter by part_of (Optional)
#' var_parts <- c("inner_example") # array[character] | Filter by parts (Optional)
#' var_passage_number <- c(123) # array[integer] | Filter by passage_number (Optional)
#' var_pooled_from <- c("inner_example") # array[character] | Filter by pooled_from (Optional)
#' var_pooled_in <- c("inner_example") # array[character] | Filter by pooled_in (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sample_terms_id <- c("inner_example") # array[character] | Filter by sample_terms.@id (Optional)
#' var_sample_terms_term_name <- c("inner_example") # array[character] | Filter by sample_terms.term_name (Optional)
#' var_sex <- c("female") # array[character] | Filter by sex (Optional)
#' var_sorted_fractions <- c("inner_example") # array[character] | Filter by sorted_fractions (Optional)
#' var_sorted_from_id <- c("inner_example") # array[character] | Filter by sorted_from.@id (Optional)
#' var_sorted_from_accession <- c("inner_example") # array[character] | Filter by sorted_from.accession (Optional)
#' var_sorted_from_detail <- c("inner_example") # array[character] | Filter by sorted_from_detail (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_starting_amount <- c(123) # array[numeric] | Filter by starting_amount (Optional)
#' var_starting_amount_units <- c("cells") # array[character] | Filter by starting_amount_units (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_targeted_sample_term <- c("inner_example") # array[character] | Filter by targeted_sample_term (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_time_post_change <- c(123) # array[numeric] | Filter by time_post_change (Optional)
#' var_time_post_change_units <- c("minute") # array[character] | Filter by time_post_change_units (Optional)
#' var_time_post_library_delivery <- c(123) # array[numeric] | Filter by time_post_library_delivery (Optional)
#' var_time_post_library_delivery_units <- c("minute") # array[character] | Filter by time_post_library_delivery_units (Optional)
#' var_treatments_id <- c("inner_example") # array[character] | Filter by treatments.@id (Optional)
#' var_treatments_purpose <- c("activation") # array[character] | Filter by treatments.purpose (Optional)
#' var_treatments_status <- c("archived") # array[character] | Filter by treatments.status (Optional)
#' var_treatments_summary <- c("inner_example") # array[character] | Filter by treatments.summary (Optional)
#' var_treatments_treatment_type <- c("chemical") # array[character] | Filter by treatments.treatment_type (Optional)
#' var_upper_bound_age <- c(123) # array[numeric] | Filter by upper_bound_age (Optional)
#' var_upper_bound_age_in_hours <- c(123) # array[numeric] | Filter by upper_bound_age_in_hours (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the InVitroSystem collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$InVitroSystems(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, cell_fate_change_protocol = var_cell_fate_change_protocol, cell_fate_change_treatments_id = var_cell_fate_change_treatments_id, cell_fate_change_treatments_purpose = var_cell_fate_change_treatments_purpose, cell_fate_change_treatments_status = var_cell_fate_change_treatments_status, cell_fate_change_treatments_summary = var_cell_fate_change_treatments_summary, cell_fate_change_treatments_treatment_type = var_cell_fate_change_treatments_treatment_type, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, demultiplexed_from = var_demultiplexed_from, demultiplexed_to = var_demultiplexed_to, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, growth_medium = var_growth_medium, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from_id = var_originated_from_id, originated_from_accession = var_originated_from_accession, part_of = var_part_of, parts = var_parts, passage_number = var_passage_number, pooled_from = var_pooled_from, pooled_in = var_pooled_in, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, targeted_sample_term = var_targeted_sample_term, taxa = var_taxa, time_post_change = var_time_post_change, time_post_change_units = var_time_post_change_units, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$InVitroSystems(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, cell_fate_change_protocol = var_cell_fate_change_protocol, cell_fate_change_treatments_id = var_cell_fate_change_treatments_id, cell_fate_change_treatments_purpose = var_cell_fate_change_treatments_purpose, cell_fate_change_treatments_status = var_cell_fate_change_treatments_status, cell_fate_change_treatments_summary = var_cell_fate_change_treatments_summary, cell_fate_change_treatments_treatment_type = var_cell_fate_change_treatments_treatment_type, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, demultiplexed_from = var_demultiplexed_from, demultiplexed_to = var_demultiplexed_to, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, growth_medium = var_growth_medium, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from_id = var_originated_from_id, originated_from_accession = var_originated_from_accession, part_of = var_part_of, parts = var_parts, passage_number = var_passage_number, pooled_from = var_pooled_from, pooled_in = var_pooled_in, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, targeted_sample_term = var_targeted_sample_term, taxa = var_taxa, time_post_change = var_time_post_change, time_post_change_units = var_time_post_change_units, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  InstitutionalCertificates  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_certificate_identifier <- c("inner_example") # array[character] | Filter by certificate_identifier (Optional)
#' var_controlled_access <- c("inner_example") # array[character] | Filter by controlled_access (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_data_use_limitation <- c("DS") # array[character] | Filter by data_use_limitation (Optional)
#' var_data_use_limitation_modifiers <- c("COL") # array[character] | Filter by data_use_limitation_modifiers (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_samples <- c("inner_example") # array[character] | Filter by samples (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_urls <- c("inner_example") # array[character] | Filter by urls (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the InstitutionalCertificate collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$InstitutionalCertificates(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, certificate_identifier = var_certificate_identifier, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, data_use_limitation = var_data_use_limitation, data_use_limitation_modifiers = var_data_use_limitation_modifiers, description = var_description, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, release_timestamp = var_release_timestamp, samples = var_samples, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, urls = var_urls, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$InstitutionalCertificates(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, certificate_identifier = var_certificate_identifier, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, data_use_limitation = var_data_use_limitation, data_use_limitation_modifiers = var_data_use_limitation_modifiers, description = var_description, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, release_timestamp = var_release_timestamp, samples = var_samples, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, urls = var_urls, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  Labs  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_awards_id <- c("inner_example") # array[character] | Filter by awards.@id (Optional)
#' var_awards_component <- c("affiliate") # array[character] | Filter by awards.component (Optional)
#' var_awards_name <- c("inner_example") # array[character] | Filter by awards.name (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_institute_label <- c("inner_example") # array[character] | Filter by institute_label (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_pi <- c("inner_example") # array[character] | Filter by pi (Optional)
#' var_status <- c("current") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Lab collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Labs(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, awards_id = var_awards_id, awards_component = var_awards_component, awards_name = var_awards_name, creation_timestamp = var_creation_timestamp, description = var_description, institute_label = var_institute_label, name = var_name, notes = var_notes, pi = var_pi, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, url = var_url, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Labs(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, awards_id = var_awards_id, awards_component = var_awards_component, awards_name = var_awards_name, creation_timestamp = var_creation_timestamp, description = var_description, institute_label = var_institute_label, name = var_name, notes = var_notes, pi = var_pi, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, url = var_url, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  MatrixFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_summary <- c("inner_example") # array[character] | Filter by content_summary (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_dimension1 <- c("cell") # array[character] | Filter by dimension1 (Optional)
#' var_dimension2 <- c("cell") # array[character] | Filter by dimension2 (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("h5ad") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_reference_files <- c("inner_example") # array[character] | Filter by reference_files (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the MatrixFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$MatrixFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_summary = var_content_summary, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, dimension1 = var_dimension1, dimension2 = var_dimension2, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, reference_files = var_reference_files, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$MatrixFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_summary = var_content_summary, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, dimension1 = var_dimension1, dimension2 = var_dimension2, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, reference_files = var_reference_files, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  MeasurementSets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_assay_term_id <- c("inner_example") # array[character] | Filter by assay_term.@id (Optional)
#' var_assay_term_term_name <- c("inner_example") # array[character] | Filter by assay_term.term_name (Optional)
#' var_auxiliary_sets_id <- c("inner_example") # array[character] | Filter by auxiliary_sets.@id (Optional)
#' var_auxiliary_sets_accession <- c("inner_example") # array[character] | Filter by auxiliary_sets.accession (Optional)
#' var_auxiliary_sets_aliases <- c("inner_example") # array[character] | Filter by auxiliary_sets.aliases (Optional)
#' var_auxiliary_sets_file_set_type <- c("cell hashing") # array[character] | Filter by auxiliary_sets.file_set_type (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_contact_pi_id <- c("inner_example") # array[character] | Filter by award.contact_pi.@id (Optional)
#' var_award_contact_pi_title <- c("inner_example") # array[character] | Filter by award.contact_pi.title (Optional)
#' var_award_title <- c("inner_example") # array[character] | Filter by award.title (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_file_sets_id <- c("inner_example") # array[character] | Filter by control_file_sets.@id (Optional)
#' var_control_file_sets_accession <- c("inner_example") # array[character] | Filter by control_file_sets.accession (Optional)
#' var_control_file_sets_aliases <- c("inner_example") # array[character] | Filter by control_file_sets.aliases (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors_id <- c("inner_example") # array[character] | Filter by donors.@id (Optional)
#' var_donors_accession <- c("inner_example") # array[character] | Filter by donors.accession (Optional)
#' var_donors_aliases <- c("inner_example") # array[character] | Filter by donors.aliases (Optional)
#' var_donors_sex <- c("male") # array[character] | Filter by donors.sex (Optional)
#' var_donors_status <- c("in progress") # array[character] | Filter by donors.status (Optional)
#' var_donors_taxa <- c("inner_example") # array[character] | Filter by donors.taxa (Optional)
#' var_external_image_url <- c("inner_example") # array[character] | Filter by external_image_url (Optional)
#' var_file_set_type <- c("experimental data") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_creation_timestamp <- c("inner_example") # array[character] | Filter by files.creation_timestamp (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_file_size <- c(123) # array[integer] | Filter by files.file_size (Optional)
#' var_files_href <- c("inner_example") # array[character] | Filter by files.href (Optional)
#' var_files_s3_uri <- c("inner_example") # array[character] | Filter by files.s3_uri (Optional)
#' var_files_sequencing_platform_id <- c("inner_example") # array[character] | Filter by files.sequencing_platform.@id (Optional)
#' var_files_sequencing_platform_term_name <- c("inner_example") # array[character] | Filter by files.sequencing_platform.term_name (Optional)
#' var_files_submitted_file_name <- c("inner_example") # array[character] | Filter by files.submitted_file_name (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_library_construction_platform_id <- c("inner_example") # array[character] | Filter by library_construction_platform.@id (Optional)
#' var_library_construction_platform_term_name <- c("inner_example") # array[character] | Filter by library_construction_platform.term_name (Optional)
#' var_multiome_size <- c(123) # array[integer] | Filter by multiome_size (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_preferred_assay_title <- c("10x multiome") # array[character] | Filter by preferred_assay_title (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_related_multiome_datasets_id <- c("inner_example") # array[character] | Filter by related_multiome_datasets.@id (Optional)
#' var_related_multiome_datasets_accession <- c("inner_example") # array[character] | Filter by related_multiome_datasets.accession (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_samples_id <- c("inner_example") # array[character] | Filter by samples.@id (Optional)
#' var_samples_accession <- c("inner_example") # array[character] | Filter by samples.accession (Optional)
#' var_samples_aliases <- c("inner_example") # array[character] | Filter by samples.aliases (Optional)
#' var_samples_cell_fate_change_treatments_id <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments.@id (Optional)
#' var_samples_cell_fate_change_treatments_purpose <- c("activation") # array[character] | Filter by samples.cell_fate_change_treatments.purpose (Optional)
#' var_samples_cell_fate_change_treatments_summary <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments.summary (Optional)
#' var_samples_cell_fate_change_treatments_treatment_type <- c("chemical") # array[character] | Filter by samples.cell_fate_change_treatments.treatment_type (Optional)
#' var_samples_classifications <- c("organoid") # array[character] | Filter by samples.classifications (Optional)
#' var_samples_construct_library_sets_id <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.@id (Optional)
#' var_samples_construct_library_sets_accession <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.accession (Optional)
#' var_samples_construct_library_sets_file_set_type <- c("guide library") # array[character] | Filter by samples.construct_library_sets.file_set_type (Optional)
#' var_samples_construct_library_sets_small_scale_gene_list_id <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.small_scale_gene_list.@id (Optional)
#' var_samples_construct_library_sets_small_scale_gene_list_geneid <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.small_scale_gene_list.geneid (Optional)
#' var_samples_construct_library_sets_small_scale_gene_list_name <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.small_scale_gene_list.name (Optional)
#' var_samples_construct_library_sets_small_scale_gene_list_summary <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.small_scale_gene_list.summary (Optional)
#' var_samples_construct_library_sets_small_scale_gene_list_symbol <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.small_scale_gene_list.symbol (Optional)
#' var_samples_construct_library_sets_summary <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.summary (Optional)
#' var_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by samples.disease_terms.@id (Optional)
#' var_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by samples.disease_terms.term_name (Optional)
#' var_samples_modifications_id <- c("inner_example") # array[character] | Filter by samples.modifications.@id (Optional)
#' var_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by samples.sample_terms.@id (Optional)
#' var_samples_sample_terms_aliases <- c("inner_example") # array[character] | Filter by samples.sample_terms.aliases (Optional)
#' var_samples_sample_terms_status <- c("archived") # array[character] | Filter by samples.sample_terms.status (Optional)
#' var_samples_sample_terms_summary <- c("inner_example") # array[character] | Filter by samples.sample_terms.summary (Optional)
#' var_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by samples.sample_terms.term_name (Optional)
#' var_samples_status <- c("in progress") # array[character] | Filter by samples.status (Optional)
#' var_samples_summary <- c("inner_example") # array[character] | Filter by samples.summary (Optional)
#' var_samples_targeted_sample_term_id <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.@id (Optional)
#' var_samples_targeted_sample_term_term_name <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.term_name (Optional)
#' var_samples_taxa <- c("Homo sapiens") # array[character] | Filter by samples.taxa (Optional)
#' var_samples_treatments_id <- c("inner_example") # array[character] | Filter by samples.treatments.@id (Optional)
#' var_samples_treatments_purpose <- c("activation") # array[character] | Filter by samples.treatments.purpose (Optional)
#' var_samples_treatments_summary <- c("inner_example") # array[character] | Filter by samples.treatments.summary (Optional)
#' var_samples_treatments_treatment_type <- c("chemical") # array[character] | Filter by samples.treatments.treatment_type (Optional)
#' var_sequencing_library_types <- c("direct RNA") # array[character] | Filter by sequencing_library_types (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_targeted_genes_id <- c("inner_example") # array[character] | Filter by targeted_genes.@id (Optional)
#' var_targeted_genes_geneid <- c("inner_example") # array[character] | Filter by targeted_genes.geneid (Optional)
#' var_targeted_genes_name <- c("inner_example") # array[character] | Filter by targeted_genes.name (Optional)
#' var_targeted_genes_symbol <- c("inner_example") # array[character] | Filter by targeted_genes.symbol (Optional)
#' var_targeted_genes_synonyms <- c("inner_example") # array[character] | Filter by targeted_genes.synonyms (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the MeasurementSet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$MeasurementSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assay_term_id = var_assay_term_id, assay_term_term_name = var_assay_term_term_name, auxiliary_sets_id = var_auxiliary_sets_id, auxiliary_sets_accession = var_auxiliary_sets_accession, auxiliary_sets_aliases = var_auxiliary_sets_aliases, auxiliary_sets_file_set_type = var_auxiliary_sets_file_set_type, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_file_sets_id = var_control_file_sets_id, control_file_sets_accession = var_control_file_sets_accession, control_file_sets_aliases = var_control_file_sets_aliases, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, external_image_url = var_external_image_url, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform_id = var_files_sequencing_platform_id, files_sequencing_platform_term_name = var_files_sequencing_platform_term_name, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, lab_id = var_lab_id, lab_title = var_lab_title, library_construction_platform_id = var_library_construction_platform_id, library_construction_platform_term_name = var_library_construction_platform_term_name, multiome_size = var_multiome_size, notes = var_notes, preferred_assay_title = var_preferred_assay_title, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, related_multiome_datasets_id = var_related_multiome_datasets_id, related_multiome_datasets_accession = var_related_multiome_datasets_accession, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments_id = var_samples_cell_fate_change_treatments_id, samples_cell_fate_change_treatments_purpose = var_samples_cell_fate_change_treatments_purpose, samples_cell_fate_change_treatments_summary = var_samples_cell_fate_change_treatments_summary, samples_cell_fate_change_treatments_treatment_type = var_samples_cell_fate_change_treatments_treatment_type, samples_classifications = var_samples_classifications, samples_construct_library_sets_id = var_samples_construct_library_sets_id, samples_construct_library_sets_accession = var_samples_construct_library_sets_accession, samples_construct_library_sets_file_set_type = var_samples_construct_library_sets_file_set_type, samples_construct_library_sets_small_scale_gene_list_id = var_samples_construct_library_sets_small_scale_gene_list_id, samples_construct_library_sets_small_scale_gene_list_geneid = var_samples_construct_library_sets_small_scale_gene_list_geneid, samples_construct_library_sets_small_scale_gene_list_name = var_samples_construct_library_sets_small_scale_gene_list_name, samples_construct_library_sets_small_scale_gene_list_summary = var_samples_construct_library_sets_small_scale_gene_list_summary, samples_construct_library_sets_small_scale_gene_list_symbol = var_samples_construct_library_sets_small_scale_gene_list_symbol, samples_construct_library_sets_summary = var_samples_construct_library_sets_summary, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications_id = var_samples_modifications_id, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments_id = var_samples_treatments_id, samples_treatments_purpose = var_samples_treatments_purpose, samples_treatments_summary = var_samples_treatments_summary, samples_treatments_treatment_type = var_samples_treatments_treatment_type, sequencing_library_types = var_sequencing_library_types, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, targeted_genes_id = var_targeted_genes_id, targeted_genes_geneid = var_targeted_genes_geneid, targeted_genes_name = var_targeted_genes_name, targeted_genes_symbol = var_targeted_genes_symbol, targeted_genes_synonyms = var_targeted_genes_synonyms, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$MeasurementSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assay_term_id = var_assay_term_id, assay_term_term_name = var_assay_term_term_name, auxiliary_sets_id = var_auxiliary_sets_id, auxiliary_sets_accession = var_auxiliary_sets_accession, auxiliary_sets_aliases = var_auxiliary_sets_aliases, auxiliary_sets_file_set_type = var_auxiliary_sets_file_set_type, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_file_sets_id = var_control_file_sets_id, control_file_sets_accession = var_control_file_sets_accession, control_file_sets_aliases = var_control_file_sets_aliases, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, external_image_url = var_external_image_url, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform_id = var_files_sequencing_platform_id, files_sequencing_platform_term_name = var_files_sequencing_platform_term_name, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, lab_id = var_lab_id, lab_title = var_lab_title, library_construction_platform_id = var_library_construction_platform_id, library_construction_platform_term_name = var_library_construction_platform_term_name, multiome_size = var_multiome_size, notes = var_notes, preferred_assay_title = var_preferred_assay_title, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, related_multiome_datasets_id = var_related_multiome_datasets_id, related_multiome_datasets_accession = var_related_multiome_datasets_accession, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments_id = var_samples_cell_fate_change_treatments_id, samples_cell_fate_change_treatments_purpose = var_samples_cell_fate_change_treatments_purpose, samples_cell_fate_change_treatments_summary = var_samples_cell_fate_change_treatments_summary, samples_cell_fate_change_treatments_treatment_type = var_samples_cell_fate_change_treatments_treatment_type, samples_classifications = var_samples_classifications, samples_construct_library_sets_id = var_samples_construct_library_sets_id, samples_construct_library_sets_accession = var_samples_construct_library_sets_accession, samples_construct_library_sets_file_set_type = var_samples_construct_library_sets_file_set_type, samples_construct_library_sets_small_scale_gene_list_id = var_samples_construct_library_sets_small_scale_gene_list_id, samples_construct_library_sets_small_scale_gene_list_geneid = var_samples_construct_library_sets_small_scale_gene_list_geneid, samples_construct_library_sets_small_scale_gene_list_name = var_samples_construct_library_sets_small_scale_gene_list_name, samples_construct_library_sets_small_scale_gene_list_summary = var_samples_construct_library_sets_small_scale_gene_list_summary, samples_construct_library_sets_small_scale_gene_list_symbol = var_samples_construct_library_sets_small_scale_gene_list_symbol, samples_construct_library_sets_summary = var_samples_construct_library_sets_summary, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications_id = var_samples_modifications_id, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments_id = var_samples_treatments_id, samples_treatments_purpose = var_samples_treatments_purpose, samples_treatments_summary = var_samples_treatments_summary, samples_treatments_treatment_type = var_samples_treatments_treatment_type, sequencing_library_types = var_sequencing_library_types, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, targeted_genes_id = var_targeted_genes_id, targeted_genes_geneid = var_targeted_genes_geneid, targeted_genes_name = var_targeted_genes_name, targeted_genes_symbol = var_targeted_genes_symbol, targeted_genes_synonyms = var_targeted_genes_synonyms, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  ModelFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_anvil_url <- c("inner_example") # array[character] | Filter by anvil_url (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_controlled_access <- c("inner_example") # array[character] | Filter by controlled_access (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("hdf5") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the ModelFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$ModelFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$ModelFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  ModelSets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_assessed_genes <- c("inner_example") # array[character] | Filter by assessed_genes (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_contact_pi_id <- c("inner_example") # array[character] | Filter by award.contact_pi.@id (Optional)
#' var_award_contact_pi_title <- c("inner_example") # array[character] | Filter by award.contact_pi.title (Optional)
#' var_award_title <- c("inner_example") # array[character] | Filter by award.title (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors_id <- c("inner_example") # array[character] | Filter by donors.@id (Optional)
#' var_donors_accession <- c("inner_example") # array[character] | Filter by donors.accession (Optional)
#' var_donors_aliases <- c("inner_example") # array[character] | Filter by donors.aliases (Optional)
#' var_donors_sex <- c("male") # array[character] | Filter by donors.sex (Optional)
#' var_donors_status <- c("in progress") # array[character] | Filter by donors.status (Optional)
#' var_donors_taxa <- c("inner_example") # array[character] | Filter by donors.taxa (Optional)
#' var_file_set_type <- c("neural network") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_creation_timestamp <- c("inner_example") # array[character] | Filter by files.creation_timestamp (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_file_size <- c(123) # array[integer] | Filter by files.file_size (Optional)
#' var_files_href <- c("inner_example") # array[character] | Filter by files.href (Optional)
#' var_files_s3_uri <- c("inner_example") # array[character] | Filter by files.s3_uri (Optional)
#' var_files_sequencing_platform <- c("inner_example") # array[character] | Filter by files.sequencing_platform (Optional)
#' var_files_submitted_file_name <- c("inner_example") # array[character] | Filter by files.submitted_file_name (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_input_file_sets_id <- c("inner_example") # array[character] | Filter by input_file_sets.@id (Optional)
#' var_input_file_sets_accession <- c("inner_example") # array[character] | Filter by input_file_sets.accession (Optional)
#' var_input_file_sets_aliases <- c("inner_example") # array[character] | Filter by input_file_sets.aliases (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_model_name <- c("inner_example") # array[character] | Filter by model_name (Optional)
#' var_model_version <- c("inner_example") # array[character] | Filter by model_version (Optional)
#' var_model_zoo_location <- c("inner_example") # array[character] | Filter by model_zoo_location (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_prediction_objects <- c("coding variants") # array[character] | Filter by prediction_objects (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_samples_id <- c("inner_example") # array[character] | Filter by samples.@id (Optional)
#' var_samples_accession <- c("inner_example") # array[character] | Filter by samples.accession (Optional)
#' var_samples_aliases <- c("inner_example") # array[character] | Filter by samples.aliases (Optional)
#' var_samples_cell_fate_change_treatments <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments (Optional)
#' var_samples_classifications <- c("organoid") # array[character] | Filter by samples.classifications (Optional)
#' var_samples_construct_library_sets <- c("inner_example") # array[character] | Filter by samples.construct_library_sets (Optional)
#' var_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by samples.disease_terms.@id (Optional)
#' var_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by samples.disease_terms.term_name (Optional)
#' var_samples_modifications <- c("inner_example") # array[character] | Filter by samples.modifications (Optional)
#' var_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by samples.sample_terms.@id (Optional)
#' var_samples_sample_terms_aliases <- c("inner_example") # array[character] | Filter by samples.sample_terms.aliases (Optional)
#' var_samples_sample_terms_status <- c("archived") # array[character] | Filter by samples.sample_terms.status (Optional)
#' var_samples_sample_terms_summary <- c("inner_example") # array[character] | Filter by samples.sample_terms.summary (Optional)
#' var_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by samples.sample_terms.term_name (Optional)
#' var_samples_status <- c("in progress") # array[character] | Filter by samples.status (Optional)
#' var_samples_summary <- c("inner_example") # array[character] | Filter by samples.summary (Optional)
#' var_samples_targeted_sample_term_id <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.@id (Optional)
#' var_samples_targeted_sample_term_term_name <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.term_name (Optional)
#' var_samples_taxa <- c("Homo sapiens") # array[character] | Filter by samples.taxa (Optional)
#' var_samples_treatments <- c("inner_example") # array[character] | Filter by samples.treatments (Optional)
#' var_software_version <- c("inner_example") # array[character] | Filter by software_version (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the ModelSet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$ModelSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assessed_genes = var_assessed_genes, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, input_file_sets_id = var_input_file_sets_id, input_file_sets_accession = var_input_file_sets_accession, input_file_sets_aliases = var_input_file_sets_aliases, lab_id = var_lab_id, lab_title = var_lab_title, model_name = var_model_name, model_version = var_model_version, model_zoo_location = var_model_zoo_location, notes = var_notes, prediction_objects = var_prediction_objects, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, software_version = var_software_version, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, url = var_url, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$ModelSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, assessed_genes = var_assessed_genes, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, input_file_sets_id = var_input_file_sets_id, input_file_sets_accession = var_input_file_sets_accession, input_file_sets_aliases = var_input_file_sets_aliases, lab_id = var_lab_id, lab_title = var_lab_title, model_name = var_model_name, model_version = var_model_version, model_zoo_location = var_model_zoo_location, notes = var_notes, prediction_objects = var_prediction_objects, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets = var_samples_construct_library_sets, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, software_version = var_software_version, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, url = var_url, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  MultiplexedSamples  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_barcode_sample_map <- c("inner_example") # array[character] | Filter by barcode_sample_map (Optional)
#' var_biomarkers <- c("inner_example") # array[character] | Filter by biomarkers (Optional)
#' var_cellular_sub_pool <- c("inner_example") # array[character] | Filter by cellular_sub_pool (Optional)
#' var_classifications <- c("inner_example") # array[character] | Filter by classifications (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_construct_library_sets_id <- c("inner_example") # array[character] | Filter by construct_library_sets.@id (Optional)
#' var_construct_library_sets_accession <- c("inner_example") # array[character] | Filter by construct_library_sets.accession (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_obtained <- c("inner_example") # array[character] | Filter by date_obtained (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_disease_terms_id <- c("inner_example") # array[character] | Filter by disease_terms.@id (Optional)
#' var_disease_terms_term_name <- c("inner_example") # array[character] | Filter by disease_terms.term_name (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors <- c("inner_example") # array[character] | Filter by donors (Optional)
#' var_file_sets_id <- c("inner_example") # array[character] | Filter by file_sets.@id (Optional)
#' var_file_sets_accession <- c("inner_example") # array[character] | Filter by file_sets.accession (Optional)
#' var_file_sets_aliases <- c("inner_example") # array[character] | Filter by file_sets.aliases (Optional)
#' var_file_sets_assay_term_term_name <- c("inner_example") # array[character] | Filter by file_sets.assay_term.term_name (Optional)
#' var_file_sets_lab_title <- c("inner_example") # array[character] | Filter by file_sets.lab.title (Optional)
#' var_file_sets_status <- c("in progress") # array[character] | Filter by file_sets.status (Optional)
#' var_file_sets_summary <- c("inner_example") # array[character] | Filter by file_sets.summary (Optional)
#' var_institutional_certificates <- c("inner_example") # array[character] | Filter by institutional_certificates (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_modifications_id <- c("inner_example") # array[character] | Filter by modifications.@id (Optional)
#' var_modifications_status <- c("archived") # array[character] | Filter by modifications.status (Optional)
#' var_modifications_summary <- c("inner_example") # array[character] | Filter by modifications.summary (Optional)
#' var_moi <- c(123) # array[numeric] | Filter by moi (Optional)
#' var_multiplexed_in_id <- c("inner_example") # array[character] | Filter by multiplexed_in.@id (Optional)
#' var_multiplexed_in_accession <- c("inner_example") # array[character] | Filter by multiplexed_in.accession (Optional)
#' var_multiplexed_samples_id <- c("inner_example") # array[character] | Filter by multiplexed_samples.@id (Optional)
#' var_multiplexed_samples_accession <- c("inner_example") # array[character] | Filter by multiplexed_samples.accession (Optional)
#' var_multiplexed_samples_construct_library_sets <- c("inner_example") # array[character] | Filter by multiplexed_samples.construct_library_sets (Optional)
#' var_multiplexed_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by multiplexed_samples.disease_terms.@id (Optional)
#' var_multiplexed_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by multiplexed_samples.disease_terms.term_name (Optional)
#' var_multiplexed_samples_donors_id <- c("inner_example") # array[character] | Filter by multiplexed_samples.donors.@id (Optional)
#' var_multiplexed_samples_donors_accession <- c("inner_example") # array[character] | Filter by multiplexed_samples.donors.accession (Optional)
#' var_multiplexed_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by multiplexed_samples.sample_terms.@id (Optional)
#' var_multiplexed_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by multiplexed_samples.sample_terms.term_name (Optional)
#' var_multiplexed_samples_status <- c("in progress") # array[character] | Filter by multiplexed_samples.status (Optional)
#' var_multiplexed_samples_summary <- c("inner_example") # array[character] | Filter by multiplexed_samples.summary (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_nucleic_acid_delivery <- c("transfection") # array[character] | Filter by nucleic_acid_delivery (Optional)
#' var_origin_of <- c("inner_example") # array[character] | Filter by origin_of (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sample_terms_id <- c("inner_example") # array[character] | Filter by sample_terms.@id (Optional)
#' var_sample_terms_term_name <- c("inner_example") # array[character] | Filter by sample_terms.term_name (Optional)
#' var_sorted_fractions <- c("inner_example") # array[character] | Filter by sorted_fractions (Optional)
#' var_sorted_from_id <- c("inner_example") # array[character] | Filter by sorted_from.@id (Optional)
#' var_sorted_from_accession <- c("inner_example") # array[character] | Filter by sorted_from.accession (Optional)
#' var_sorted_from_detail <- c("inner_example") # array[character] | Filter by sorted_from_detail (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_starting_amount <- c(123) # array[numeric] | Filter by starting_amount (Optional)
#' var_starting_amount_units <- c("cells") # array[character] | Filter by starting_amount_units (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_time_post_library_delivery <- c(123) # array[numeric] | Filter by time_post_library_delivery (Optional)
#' var_time_post_library_delivery_units <- c("minute") # array[character] | Filter by time_post_library_delivery_units (Optional)
#' var_treatments_id <- c("inner_example") # array[character] | Filter by treatments.@id (Optional)
#' var_treatments_purpose <- c("activation") # array[character] | Filter by treatments.purpose (Optional)
#' var_treatments_status <- c("archived") # array[character] | Filter by treatments.status (Optional)
#' var_treatments_summary <- c("inner_example") # array[character] | Filter by treatments.summary (Optional)
#' var_treatments_treatment_type <- c("chemical") # array[character] | Filter by treatments.treatment_type (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the MultiplexedSample collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$MultiplexedSamples(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, barcode_sample_map = var_barcode_sample_map, biomarkers = var_biomarkers, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets_id = var_construct_library_sets_id, construct_library_sets_accession = var_construct_library_sets_accession, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates = var_institutional_certificates, lab_id = var_lab_id, lab_title = var_lab_title, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, multiplexed_samples_id = var_multiplexed_samples_id, multiplexed_samples_accession = var_multiplexed_samples_accession, multiplexed_samples_construct_library_sets = var_multiplexed_samples_construct_library_sets, multiplexed_samples_disease_terms_id = var_multiplexed_samples_disease_terms_id, multiplexed_samples_disease_terms_term_name = var_multiplexed_samples_disease_terms_term_name, multiplexed_samples_donors_id = var_multiplexed_samples_donors_id, multiplexed_samples_donors_accession = var_multiplexed_samples_donors_accession, multiplexed_samples_sample_terms_id = var_multiplexed_samples_sample_terms_id, multiplexed_samples_sample_terms_term_name = var_multiplexed_samples_sample_terms_term_name, multiplexed_samples_status = var_multiplexed_samples_status, multiplexed_samples_summary = var_multiplexed_samples_summary, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$MultiplexedSamples(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, barcode_sample_map = var_barcode_sample_map, biomarkers = var_biomarkers, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets_id = var_construct_library_sets_id, construct_library_sets_accession = var_construct_library_sets_accession, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates = var_institutional_certificates, lab_id = var_lab_id, lab_title = var_lab_title, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, multiplexed_samples_id = var_multiplexed_samples_id, multiplexed_samples_accession = var_multiplexed_samples_accession, multiplexed_samples_construct_library_sets = var_multiplexed_samples_construct_library_sets, multiplexed_samples_disease_terms_id = var_multiplexed_samples_disease_terms_id, multiplexed_samples_disease_terms_term_name = var_multiplexed_samples_disease_terms_term_name, multiplexed_samples_donors_id = var_multiplexed_samples_donors_id, multiplexed_samples_donors_accession = var_multiplexed_samples_donors_accession, multiplexed_samples_sample_terms_id = var_multiplexed_samples_sample_terms_id, multiplexed_samples_sample_terms_term_name = var_multiplexed_samples_sample_terms_term_name, multiplexed_samples_status = var_multiplexed_samples_status, multiplexed_samples_summary = var_multiplexed_samples_summary, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  OpenReadingFrames  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award <- c("inner_example") # array[character] | Filter by award (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_gene_id <- c("inner_example") # array[character] | Filter by gene.@id (Optional)
#' var_gene_geneid <- c("inner_example") # array[character] | Filter by gene.geneid (Optional)
#' var_gene_symbol <- c("inner_example") # array[character] | Filter by gene.symbol (Optional)
#' var_lab <- c("inner_example") # array[character] | Filter by lab (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_orf_id <- c("inner_example") # array[character] | Filter by orf_id (Optional)
#' var_pct_coverage_orf <- c(123) # array[numeric] | Filter by pct_coverage_orf (Optional)
#' var_pct_coverage_protein <- c(123) # array[numeric] | Filter by pct_coverage_protein (Optional)
#' var_pct_identical_protein <- c(123) # array[numeric] | Filter by pct_identical_protein (Optional)
#' var_protein_id <- c("inner_example") # array[character] | Filter by protein_id (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by <- c("inner_example") # array[character] | Filter by submitted_by (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the OpenReadingFrame collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$OpenReadingFrames(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award = var_award, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, gene_id = var_gene_id, gene_geneid = var_gene_geneid, gene_symbol = var_gene_symbol, lab = var_lab, notes = var_notes, orf_id = var_orf_id, pct_coverage_orf = var_pct_coverage_orf, pct_coverage_protein = var_pct_coverage_protein, pct_identical_protein = var_pct_identical_protein, protein_id = var_protein_id, release_timestamp = var_release_timestamp, status = var_status, submitted_by = var_submitted_by, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$OpenReadingFrames(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award = var_award, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, gene_id = var_gene_id, gene_geneid = var_gene_geneid, gene_symbol = var_gene_symbol, lab = var_lab, notes = var_notes, orf_id = var_orf_id, pct_coverage_orf = var_pct_coverage_orf, pct_coverage_protein = var_pct_coverage_protein, pct_identical_protein = var_pct_identical_protein, protein_id = var_protein_id, release_timestamp = var_release_timestamp, status = var_status, submitted_by = var_submitted_by, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  Pages  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award <- c("inner_example") # array[character] | Filter by award (Optional)
#' var_canonical_uri <- c("inner_example") # array[character] | Filter by canonical_uri (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_lab <- c("inner_example") # array[character] | Filter by lab (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_parent <- c("inner_example") # array[character] | Filter by parent (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Page collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Pages(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award = var_award, canonical_uri = var_canonical_uri, creation_timestamp = var_creation_timestamp, description = var_description, lab = var_lab, name = var_name, notes = var_notes, parent = var_parent, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Pages(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award = var_award, canonical_uri = var_canonical_uri, creation_timestamp = var_creation_timestamp, description = var_description, lab = var_lab, name = var_name, notes = var_notes, parent = var_parent, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  PhenotypeTerms  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_ancestors <- c("inner_example") # array[character] | Filter by ancestors (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_deprecated_ntr_terms <- c("inner_example") # array[character] | Filter by deprecated_ntr_terms (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_is_a <- c("inner_example") # array[character] | Filter by is_a (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_ontology <- c("inner_example") # array[character] | Filter by ontology (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_synonyms <- c("inner_example") # array[character] | Filter by synonyms (Optional)
#' var_term_id <- c("inner_example") # array[character] | Filter by term_id (Optional)
#' var_term_name <- c("inner_example") # array[character] | Filter by term_name (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the PhenotypeTerm collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$PhenotypeTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, creation_timestamp = var_creation_timestamp, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, is_a = var_is_a, name = var_name, notes = var_notes, ontology = var_ontology, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, term_id = var_term_id, term_name = var_term_name, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$PhenotypeTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, creation_timestamp = var_creation_timestamp, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, is_a = var_is_a, name = var_name, notes = var_notes, ontology = var_ontology, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, term_id = var_term_id, term_name = var_term_name, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  PhenotypicFeatures  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_feature_id <- c("inner_example") # array[character] | Filter by feature.@id (Optional)
#' var_feature_term_id <- c("inner_example") # array[character] | Filter by feature.term_id (Optional)
#' var_feature_term_name <- c("inner_example") # array[character] | Filter by feature.term_name (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_observation_date <- c("inner_example") # array[character] | Filter by observation_date (Optional)
#' var_quantity <- c(123) # array[numeric] | Filter by quantity (Optional)
#' var_quantity_units <- c("meter") # array[character] | Filter by quantity_units (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the PhenotypicFeature collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$PhenotypicFeatures(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, feature_id = var_feature_id, feature_term_id = var_feature_term_id, feature_term_name = var_feature_term_name, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, observation_date = var_observation_date, quantity = var_quantity, quantity_units = var_quantity_units, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$PhenotypicFeatures(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, feature_id = var_feature_id, feature_term_id = var_feature_term_id, feature_term_name = var_feature_term_name, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, observation_date = var_observation_date, quantity = var_quantity, quantity_units = var_quantity_units, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  PlatformTerms  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_ancestors <- c("inner_example") # array[character] | Filter by ancestors (Optional)
#' var_company <- c("10X Genomics") # array[character] | Filter by company (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_deprecated_ntr_terms <- c("inner_example") # array[character] | Filter by deprecated_ntr_terms (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_is_a <- c("inner_example") # array[character] | Filter by is_a (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_ontology <- c("inner_example") # array[character] | Filter by ontology (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_sequencing_kits <- c("HiSeq SBS Kit v4") # array[character] | Filter by sequencing_kits (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_synonyms <- c("inner_example") # array[character] | Filter by synonyms (Optional)
#' var_term_id <- c("inner_example") # array[character] | Filter by term_id (Optional)
#' var_term_name <- c("inner_example") # array[character] | Filter by term_name (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the PlatformTerm collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$PlatformTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, company = var_company, creation_timestamp = var_creation_timestamp, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, is_a = var_is_a, name = var_name, notes = var_notes, ontology = var_ontology, release_timestamp = var_release_timestamp, sequencing_kits = var_sequencing_kits, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, term_id = var_term_id, term_name = var_term_name, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$PlatformTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, company = var_company, creation_timestamp = var_creation_timestamp, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, is_a = var_is_a, name = var_name, notes = var_notes, ontology = var_ontology, release_timestamp = var_release_timestamp, sequencing_kits = var_sequencing_kits, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, term_id = var_term_id, term_name = var_term_name, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  PredictionSets  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_award_contact_pi_id <- c("inner_example") # array[character] | Filter by award.contact_pi.@id (Optional)
#' var_award_contact_pi_title <- c("inner_example") # array[character] | Filter by award.contact_pi.title (Optional)
#' var_award_title <- c("inner_example") # array[character] | Filter by award.title (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_control_for_id <- c("inner_example") # array[character] | Filter by control_for.@id (Optional)
#' var_control_for_accession <- c("inner_example") # array[character] | Filter by control_for.accession (Optional)
#' var_control_for_aliases <- c("inner_example") # array[character] | Filter by control_for.aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors_id <- c("inner_example") # array[character] | Filter by donors.@id (Optional)
#' var_donors_accession <- c("inner_example") # array[character] | Filter by donors.accession (Optional)
#' var_donors_aliases <- c("inner_example") # array[character] | Filter by donors.aliases (Optional)
#' var_donors_sex <- c("male") # array[character] | Filter by donors.sex (Optional)
#' var_donors_status <- c("in progress") # array[character] | Filter by donors.status (Optional)
#' var_donors_taxa <- c("inner_example") # array[character] | Filter by donors.taxa (Optional)
#' var_file_set_type <- c("pathogenicity") # array[character] | Filter by file_set_type (Optional)
#' var_files_id <- c("inner_example") # array[character] | Filter by files.@id (Optional)
#' var_files_accession <- c("inner_example") # array[character] | Filter by files.accession (Optional)
#' var_files_aliases <- c("inner_example") # array[character] | Filter by files.aliases (Optional)
#' var_files_content_type <- c("inner_example") # array[character] | Filter by files.content_type (Optional)
#' var_files_creation_timestamp <- c("inner_example") # array[character] | Filter by files.creation_timestamp (Optional)
#' var_files_file_format <- c("inner_example") # array[character] | Filter by files.file_format (Optional)
#' var_files_file_size <- c(123) # array[integer] | Filter by files.file_size (Optional)
#' var_files_href <- c("inner_example") # array[character] | Filter by files.href (Optional)
#' var_files_s3_uri <- c("inner_example") # array[character] | Filter by files.s3_uri (Optional)
#' var_files_sequencing_platform <- c("inner_example") # array[character] | Filter by files.sequencing_platform (Optional)
#' var_files_submitted_file_name <- c("inner_example") # array[character] | Filter by files.submitted_file_name (Optional)
#' var_files_upload_status <- c("pending") # array[character] | Filter by files.upload_status (Optional)
#' var_input_file_set_for <- c("inner_example") # array[character] | Filter by input_file_set_for (Optional)
#' var_input_file_sets <- c("inner_example") # array[character] | Filter by input_file_sets (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_large_scale_gene_list_id <- c("inner_example") # array[character] | Filter by large_scale_gene_list.@id (Optional)
#' var_large_scale_gene_list_accession <- c("inner_example") # array[character] | Filter by large_scale_gene_list.accession (Optional)
#' var_large_scale_gene_list_aliases <- c("inner_example") # array[character] | Filter by large_scale_gene_list.aliases (Optional)
#' var_large_scale_loci_list_id <- c("inner_example") # array[character] | Filter by large_scale_loci_list.@id (Optional)
#' var_large_scale_loci_list_accession <- c("inner_example") # array[character] | Filter by large_scale_loci_list.accession (Optional)
#' var_large_scale_loci_list_aliases <- c("inner_example") # array[character] | Filter by large_scale_loci_list.aliases (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_samples_id <- c("inner_example") # array[character] | Filter by samples.@id (Optional)
#' var_samples_accession <- c("inner_example") # array[character] | Filter by samples.accession (Optional)
#' var_samples_aliases <- c("inner_example") # array[character] | Filter by samples.aliases (Optional)
#' var_samples_cell_fate_change_treatments <- c("inner_example") # array[character] | Filter by samples.cell_fate_change_treatments (Optional)
#' var_samples_classifications <- c("organoid") # array[character] | Filter by samples.classifications (Optional)
#' var_samples_construct_library_sets_id <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.@id (Optional)
#' var_samples_construct_library_sets_accession <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.accession (Optional)
#' var_samples_construct_library_sets_summary <- c("inner_example") # array[character] | Filter by samples.construct_library_sets.summary (Optional)
#' var_samples_disease_terms_id <- c("inner_example") # array[character] | Filter by samples.disease_terms.@id (Optional)
#' var_samples_disease_terms_term_name <- c("inner_example") # array[character] | Filter by samples.disease_terms.term_name (Optional)
#' var_samples_modifications <- c("inner_example") # array[character] | Filter by samples.modifications (Optional)
#' var_samples_sample_terms_id <- c("inner_example") # array[character] | Filter by samples.sample_terms.@id (Optional)
#' var_samples_sample_terms_aliases <- c("inner_example") # array[character] | Filter by samples.sample_terms.aliases (Optional)
#' var_samples_sample_terms_status <- c("archived") # array[character] | Filter by samples.sample_terms.status (Optional)
#' var_samples_sample_terms_summary <- c("inner_example") # array[character] | Filter by samples.sample_terms.summary (Optional)
#' var_samples_sample_terms_term_name <- c("inner_example") # array[character] | Filter by samples.sample_terms.term_name (Optional)
#' var_samples_status <- c("in progress") # array[character] | Filter by samples.status (Optional)
#' var_samples_summary <- c("inner_example") # array[character] | Filter by samples.summary (Optional)
#' var_samples_targeted_sample_term_id <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.@id (Optional)
#' var_samples_targeted_sample_term_term_name <- c("inner_example") # array[character] | Filter by samples.targeted_sample_term.term_name (Optional)
#' var_samples_taxa <- c("Homo sapiens") # array[character] | Filter by samples.taxa (Optional)
#' var_samples_treatments <- c("inner_example") # array[character] | Filter by samples.treatments (Optional)
#' var_scope <- c("genes") # array[character] | Filter by scope (Optional)
#' var_small_scale_gene_list_id <- c("inner_example") # array[character] | Filter by small_scale_gene_list.@id (Optional)
#' var_small_scale_gene_list_geneid <- c("inner_example") # array[character] | Filter by small_scale_gene_list.geneid (Optional)
#' var_small_scale_gene_list_name <- c("inner_example") # array[character] | Filter by small_scale_gene_list.name (Optional)
#' var_small_scale_gene_list_symbol <- c("inner_example") # array[character] | Filter by small_scale_gene_list.symbol (Optional)
#' var_small_scale_gene_list_synonyms <- c("inner_example") # array[character] | Filter by small_scale_gene_list.synonyms (Optional)
#' var_small_scale_loci_list <- c(Locus$new("GRCh38", "chromosome_example", 123, 123)) # array[Locus] | Filter by small_scale_loci_list (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_files_timestamp <- c("inner_example") # array[character] | Filter by submitted_files_timestamp (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the PredictionSet collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$PredictionSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, input_file_sets = var_input_file_sets, lab_id = var_lab_id, lab_title = var_lab_title, large_scale_gene_list_id = var_large_scale_gene_list_id, large_scale_gene_list_accession = var_large_scale_gene_list_accession, large_scale_gene_list_aliases = var_large_scale_gene_list_aliases, large_scale_loci_list_id = var_large_scale_loci_list_id, large_scale_loci_list_accession = var_large_scale_loci_list_accession, large_scale_loci_list_aliases = var_large_scale_loci_list_aliases, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets_id = var_samples_construct_library_sets_id, samples_construct_library_sets_accession = var_samples_construct_library_sets_accession, samples_construct_library_sets_summary = var_samples_construct_library_sets_summary, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, scope = var_scope, small_scale_gene_list_id = var_small_scale_gene_list_id, small_scale_gene_list_geneid = var_small_scale_gene_list_geneid, small_scale_gene_list_name = var_small_scale_gene_list_name, small_scale_gene_list_symbol = var_small_scale_gene_list_symbol, small_scale_gene_list_synonyms = var_small_scale_gene_list_synonyms, small_scale_loci_list = var_small_scale_loci_list, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, url = var_url, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$PredictionSets(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, award_contact_pi_id = var_award_contact_pi_id, award_contact_pi_title = var_award_contact_pi_title, award_title = var_award_title, collections = var_collections, control_for_id = var_control_for_id, control_for_accession = var_control_for_accession, control_for_aliases = var_control_for_aliases, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, donors_id = var_donors_id, donors_accession = var_donors_accession, donors_aliases = var_donors_aliases, donors_sex = var_donors_sex, donors_status = var_donors_status, donors_taxa = var_donors_taxa, file_set_type = var_file_set_type, files_id = var_files_id, files_accession = var_files_accession, files_aliases = var_files_aliases, files_content_type = var_files_content_type, files_creation_timestamp = var_files_creation_timestamp, files_file_format = var_files_file_format, files_file_size = var_files_file_size, files_href = var_files_href, files_s3_uri = var_files_s3_uri, files_sequencing_platform = var_files_sequencing_platform, files_submitted_file_name = var_files_submitted_file_name, files_upload_status = var_files_upload_status, input_file_set_for = var_input_file_set_for, input_file_sets = var_input_file_sets, lab_id = var_lab_id, lab_title = var_lab_title, large_scale_gene_list_id = var_large_scale_gene_list_id, large_scale_gene_list_accession = var_large_scale_gene_list_accession, large_scale_gene_list_aliases = var_large_scale_gene_list_aliases, large_scale_loci_list_id = var_large_scale_loci_list_id, large_scale_loci_list_accession = var_large_scale_loci_list_accession, large_scale_loci_list_aliases = var_large_scale_loci_list_aliases, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, samples_id = var_samples_id, samples_accession = var_samples_accession, samples_aliases = var_samples_aliases, samples_cell_fate_change_treatments = var_samples_cell_fate_change_treatments, samples_classifications = var_samples_classifications, samples_construct_library_sets_id = var_samples_construct_library_sets_id, samples_construct_library_sets_accession = var_samples_construct_library_sets_accession, samples_construct_library_sets_summary = var_samples_construct_library_sets_summary, samples_disease_terms_id = var_samples_disease_terms_id, samples_disease_terms_term_name = var_samples_disease_terms_term_name, samples_modifications = var_samples_modifications, samples_sample_terms_id = var_samples_sample_terms_id, samples_sample_terms_aliases = var_samples_sample_terms_aliases, samples_sample_terms_status = var_samples_sample_terms_status, samples_sample_terms_summary = var_samples_sample_terms_summary, samples_sample_terms_term_name = var_samples_sample_terms_term_name, samples_status = var_samples_status, samples_summary = var_samples_summary, samples_targeted_sample_term_id = var_samples_targeted_sample_term_id, samples_targeted_sample_term_term_name = var_samples_targeted_sample_term_term_name, samples_taxa = var_samples_taxa, samples_treatments = var_samples_treatments, scope = var_scope, small_scale_gene_list_id = var_small_scale_gene_list_id, small_scale_gene_list_geneid = var_small_scale_gene_list_geneid, small_scale_gene_list_name = var_small_scale_gene_list_name, small_scale_gene_list_symbol = var_small_scale_gene_list_symbol, small_scale_gene_list_synonyms = var_small_scale_gene_list_synonyms, small_scale_loci_list = var_small_scale_loci_list, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_files_timestamp = var_submitted_files_timestamp, submitter_comment = var_submitter_comment, summary = var_summary, url = var_url, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  PrimaryCells  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_age <- c("inner_example") # array[character] | Filter by age (Optional)
#' var_age_units <- c("minute") # array[character] | Filter by age_units (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biomarkers <- c("inner_example") # array[character] | Filter by biomarkers (Optional)
#' var_cellular_sub_pool <- c("inner_example") # array[character] | Filter by cellular_sub_pool (Optional)
#' var_classifications <- c("inner_example") # array[character] | Filter by classifications (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_construct_library_sets <- c("inner_example") # array[character] | Filter by construct_library_sets (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_obtained <- c("inner_example") # array[character] | Filter by date_obtained (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_disease_terms_id <- c("inner_example") # array[character] | Filter by disease_terms.@id (Optional)
#' var_disease_terms_term_name <- c("inner_example") # array[character] | Filter by disease_terms.term_name (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors <- c("inner_example") # array[character] | Filter by donors (Optional)
#' var_embryonic <- c("inner_example") # array[character] | Filter by embryonic (Optional)
#' var_file_sets_id <- c("inner_example") # array[character] | Filter by file_sets.@id (Optional)
#' var_file_sets_accession <- c("inner_example") # array[character] | Filter by file_sets.accession (Optional)
#' var_file_sets_aliases <- c("inner_example") # array[character] | Filter by file_sets.aliases (Optional)
#' var_file_sets_assay_term_term_name <- c("inner_example") # array[character] | Filter by file_sets.assay_term.term_name (Optional)
#' var_file_sets_lab_title <- c("inner_example") # array[character] | Filter by file_sets.lab.title (Optional)
#' var_file_sets_status <- c("in progress") # array[character] | Filter by file_sets.status (Optional)
#' var_file_sets_summary <- c("inner_example") # array[character] | Filter by file_sets.summary (Optional)
#' var_institutional_certificates_id <- c("inner_example") # array[character] | Filter by institutional_certificates.@id (Optional)
#' var_institutional_certificates_certificate_identifier <- c("inner_example") # array[character] | Filter by institutional_certificates.certificate_identifier (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_lower_bound_age <- c(123) # array[numeric] | Filter by lower_bound_age (Optional)
#' var_lower_bound_age_in_hours <- c(123) # array[numeric] | Filter by lower_bound_age_in_hours (Optional)
#' var_modifications_id <- c("inner_example") # array[character] | Filter by modifications.@id (Optional)
#' var_modifications_status <- c("archived") # array[character] | Filter by modifications.status (Optional)
#' var_modifications_summary <- c("inner_example") # array[character] | Filter by modifications.summary (Optional)
#' var_moi <- c(123) # array[numeric] | Filter by moi (Optional)
#' var_multiplexed_in_id <- c("inner_example") # array[character] | Filter by multiplexed_in.@id (Optional)
#' var_multiplexed_in_accession <- c("inner_example") # array[character] | Filter by multiplexed_in.accession (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_nucleic_acid_delivery <- c("transfection") # array[character] | Filter by nucleic_acid_delivery (Optional)
#' var_origin_of <- c("inner_example") # array[character] | Filter by origin_of (Optional)
#' var_originated_from <- c("inner_example") # array[character] | Filter by originated_from (Optional)
#' var_part_of <- c("inner_example") # array[character] | Filter by part_of (Optional)
#' var_parts <- c("inner_example") # array[character] | Filter by parts (Optional)
#' var_passage_number <- c(123) # array[integer] | Filter by passage_number (Optional)
#' var_pooled_from <- c("inner_example") # array[character] | Filter by pooled_from (Optional)
#' var_pooled_in <- c("inner_example") # array[character] | Filter by pooled_in (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sample_terms_id <- c("inner_example") # array[character] | Filter by sample_terms.@id (Optional)
#' var_sample_terms_term_name <- c("inner_example") # array[character] | Filter by sample_terms.term_name (Optional)
#' var_sex <- c("female") # array[character] | Filter by sex (Optional)
#' var_sorted_fractions <- c("inner_example") # array[character] | Filter by sorted_fractions (Optional)
#' var_sorted_from_id <- c("inner_example") # array[character] | Filter by sorted_from.@id (Optional)
#' var_sorted_from_accession <- c("inner_example") # array[character] | Filter by sorted_from.accession (Optional)
#' var_sorted_from_detail <- c("inner_example") # array[character] | Filter by sorted_from_detail (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_starting_amount <- c(123) # array[numeric] | Filter by starting_amount (Optional)
#' var_starting_amount_units <- c("cells") # array[character] | Filter by starting_amount_units (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_time_post_library_delivery <- c(123) # array[numeric] | Filter by time_post_library_delivery (Optional)
#' var_time_post_library_delivery_units <- c("minute") # array[character] | Filter by time_post_library_delivery_units (Optional)
#' var_treatments_id <- c("inner_example") # array[character] | Filter by treatments.@id (Optional)
#' var_treatments_purpose <- c("activation") # array[character] | Filter by treatments.purpose (Optional)
#' var_treatments_status <- c("archived") # array[character] | Filter by treatments.status (Optional)
#' var_treatments_summary <- c("inner_example") # array[character] | Filter by treatments.summary (Optional)
#' var_treatments_treatment_type <- c("chemical") # array[character] | Filter by treatments.treatment_type (Optional)
#' var_upper_bound_age <- c(123) # array[numeric] | Filter by upper_bound_age (Optional)
#' var_upper_bound_age_in_hours <- c(123) # array[numeric] | Filter by upper_bound_age_in_hours (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the PrimaryCell collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$PrimaryCells(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from = var_originated_from, part_of = var_part_of, parts = var_parts, passage_number = var_passage_number, pooled_from = var_pooled_from, pooled_in = var_pooled_in, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$PrimaryCells(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from = var_originated_from, part_of = var_part_of, parts = var_parts, passage_number = var_passage_number, pooled_from = var_pooled_from, pooled_in = var_pooled_in, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  Publications  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_abstract <- c("inner_example") # array[character] | Filter by abstract (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_authors <- c("inner_example") # array[character] | Filter by authors (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_published <- c("inner_example") # array[character] | Filter by date_published (Optional)
#' var_date_revised <- c("inner_example") # array[character] | Filter by date_revised (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_donors <- c("inner_example") # array[character] | Filter by donors (Optional)
#' var_file_sets <- c("inner_example") # array[character] | Filter by file_sets (Optional)
#' var_issue <- c("inner_example") # array[character] | Filter by issue (Optional)
#' var_journal <- c("inner_example") # array[character] | Filter by journal (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_page <- c("inner_example") # array[character] | Filter by page (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publication_year <- c(123) # array[integer] | Filter by publication_year (Optional)
#' var_published_by <- c("community") # array[character] | Filter by published_by (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_samples <- c("inner_example") # array[character] | Filter by samples (Optional)
#' var_software <- c("inner_example") # array[character] | Filter by software (Optional)
#' var_software_versions <- c("inner_example") # array[character] | Filter by software_versions (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_volume <- c("inner_example") # array[character] | Filter by volume (Optional)
#' var_workflows <- c("inner_example") # array[character] | Filter by workflows (Optional)
#'
#' #List items in the Publication collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Publications(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, abstract = var_abstract, aliases = var_aliases, authors = var_authors, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, date_published = var_date_published, date_revised = var_date_revised, description = var_description, donors = var_donors, file_sets = var_file_sets, issue = var_issue, journal = var_journal, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, page = var_page, publication_identifiers = var_publication_identifiers, publication_year = var_publication_year, published_by = var_published_by, release_timestamp = var_release_timestamp, samples = var_samples, software = var_software, software_versions = var_software_versions, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid, volume = var_volume, workflows = var_workflowsdata_file = "result.txt")
#' result <- api_instance$Publications(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, abstract = var_abstract, aliases = var_aliases, authors = var_authors, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, date_published = var_date_published, date_revised = var_date_revised, description = var_description, donors = var_donors, file_sets = var_file_sets, issue = var_issue, journal = var_journal, lab_id = var_lab_id, lab_title = var_lab_title, notes = var_notes, page = var_page, publication_identifiers = var_publication_identifiers, publication_year = var_publication_year, published_by = var_published_by, release_timestamp = var_release_timestamp, samples = var_samples, software = var_software, software_versions = var_software_versions, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid, volume = var_volume, workflows = var_workflows)
#' dput(result)
#'
#'
#' ####################  ReferenceFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_anvil_url <- c("inner_example") # array[character] | Filter by anvil_url (Optional)
#' var_assembly <- c("GRCh38") # array[character] | Filter by assembly (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_controlled_access <- c("inner_example") # array[character] | Filter by controlled_access (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_external <- c("inner_example") # array[character] | Filter by external (Optional)
#' var_external_id <- c("inner_example") # array[character] | Filter by external_id (Optional)
#' var_file_format <- c("bed") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_format_type <- c("bed12") # array[character] | Filter by file_format_type (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_source_url <- c("inner_example") # array[character] | Filter by source_url (Optional)
#' var_sources <- c("inner_example") # array[character] | Filter by sources (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_transcriptome_annotation <- c("GENCODE 40") # array[character] | Filter by transcriptome_annotation (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the ReferenceFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$ReferenceFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, external = var_external, external_id = var_external_id, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_format_type = var_file_format_type, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, source_url = var_source_url, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$ReferenceFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, external = var_external, external_id = var_external_id, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_format_type = var_file_format_type, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, source_url = var_source_url, sources = var_sources, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  Report  ####################
#'
#' library(igvfclient)
#' var_query <- "query_example" # character | Query string for searching. (Optional)
#' var_type <- c("inner_example") # array[character] | Filter by item type. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all. (Optional)
#' var_field_filters <- c(key = TODO) # object | Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'} (Optional)
#' var_include_fields <- c("inner_example") # array[character] | Fields to include in the response. Can be repeated for multiple fields. (Optional)
#' var_frame <- "frame_example" # character | Object with links, or object with some links embedded. (Optional)
#'
#' #Generate a report based on search query. All results are returned.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Report(query = var_query, type = var_type, limit = var_limit, sort = var_sort, field_filters = var_field_filters, include_fields = var_include_fields, frame = var_framedata_file = "result.txt")
#' result <- api_instance$Report(query = var_query, type = var_type, limit = var_limit, sort = var_sort, field_filters = var_field_filters, include_fields = var_include_fields, frame = var_frame)
#' dput(result)
#'
#'
#' ####################  RodentDonors  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_genotype <- c("inner_example") # array[character] | Filter by genotype (Optional)
#' var_individual_rodent <- c("inner_example") # array[character] | Filter by individual_rodent (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_phenotypic_features_id <- c("inner_example") # array[character] | Filter by phenotypic_features.@id (Optional)
#' var_phenotypic_features_feature_id <- c("inner_example") # array[character] | Filter by phenotypic_features.feature.@id (Optional)
#' var_phenotypic_features_feature_term_id <- c("inner_example") # array[character] | Filter by phenotypic_features.feature.term_id (Optional)
#' var_phenotypic_features_feature_term_name <- c("inner_example") # array[character] | Filter by phenotypic_features.feature.term_name (Optional)
#' var_phenotypic_features_observation_date <- c("inner_example") # array[character] | Filter by phenotypic_features.observation_date (Optional)
#' var_phenotypic_features_quantity_units <- c("meter") # array[character] | Filter by phenotypic_features.quantity_units (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_rodent_identifier <- c("inner_example") # array[character] | Filter by rodent_identifier (Optional)
#' var_sex <- c("male") # array[character] | Filter by sex (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_strain <- c("inner_example") # array[character] | Filter by strain (Optional)
#' var_strain_background <- c("A/J (AJ)") # array[character] | Filter by strain_background (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Mus musculus") # array[character] | Filter by taxa (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the RodentDonor collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$RodentDonors(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, collections = var_collections, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, genotype = var_genotype, individual_rodent = var_individual_rodent, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, notes = var_notes, phenotypic_features_id = var_phenotypic_features_id, phenotypic_features_feature_id = var_phenotypic_features_feature_id, phenotypic_features_feature_term_id = var_phenotypic_features_feature_term_id, phenotypic_features_feature_term_name = var_phenotypic_features_feature_term_name, phenotypic_features_observation_date = var_phenotypic_features_observation_date, phenotypic_features_quantity_units = var_phenotypic_features_quantity_units, product_id = var_product_id, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, rodent_identifier = var_rodent_identifier, sex = var_sex, sources_id = var_sources_id, status = var_status, strain = var_strain, strain_background = var_strain_background, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$RodentDonors(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, collections = var_collections, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, genotype = var_genotype, individual_rodent = var_individual_rodent, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, notes = var_notes, phenotypic_features_id = var_phenotypic_features_id, phenotypic_features_feature_id = var_phenotypic_features_feature_id, phenotypic_features_feature_term_id = var_phenotypic_features_feature_term_id, phenotypic_features_feature_term_name = var_phenotypic_features_feature_term_name, phenotypic_features_observation_date = var_phenotypic_features_observation_date, phenotypic_features_quantity_units = var_phenotypic_features_quantity_units, product_id = var_product_id, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, rodent_identifier = var_rodent_identifier, sex = var_sex, sources_id = var_sources_id, status = var_status, strain = var_strain, strain_background = var_strain_background, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  SampleTerms  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_ancestors <- c("inner_example") # array[character] | Filter by ancestors (Optional)
#' var_cell_slims <- c("inner_example") # array[character] | Filter by cell_slims (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_deprecated_ntr_terms <- c("inner_example") # array[character] | Filter by deprecated_ntr_terms (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_developmental_slims <- c("inner_example") # array[character] | Filter by developmental_slims (Optional)
#' var_is_a <- c("inner_example") # array[character] | Filter by is_a (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_ontology <- c("inner_example") # array[character] | Filter by ontology (Optional)
#' var_organ_slims <- c("inner_example") # array[character] | Filter by organ_slims (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_synonyms <- c("inner_example") # array[character] | Filter by synonyms (Optional)
#' var_system_slims <- c("inner_example") # array[character] | Filter by system_slims (Optional)
#' var_term_id <- c("inner_example") # array[character] | Filter by term_id (Optional)
#' var_term_name <- c("inner_example") # array[character] | Filter by term_name (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the SampleTerm collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$SampleTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, cell_slims = var_cell_slims, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, developmental_slims = var_developmental_slims, is_a = var_is_a, name = var_name, notes = var_notes, ontology = var_ontology, organ_slims = var_organ_slims, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, system_slims = var_system_slims, term_id = var_term_id, term_name = var_term_name, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$SampleTerms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, ancestors = var_ancestors, cell_slims = var_cell_slims, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, deprecated_ntr_terms = var_deprecated_ntr_terms, description = var_description, developmental_slims = var_developmental_slims, is_a = var_is_a, name = var_name, notes = var_notes, ontology = var_ontology, organ_slims = var_organ_slims, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, synonyms = var_synonyms, system_slims = var_system_slims, term_id = var_term_id, term_name = var_term_name, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  SchemaForItemType  ####################
#'
#' library(igvfclient)
#' var_item_type <- ItemType$new() # ItemType | The name of the item type
#'
#' #Retrieve JSON schema for item type
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$SchemaForItemType(var_item_typedata_file = "result.txt")
#' result <- api_instance$SchemaForItemType(var_item_type)
#' dput(result)
#'
#'
#' ####################  Schemas  ####################
#'
#' library(igvfclient)
#'
#' #Retrieve JSON schemas for all item types
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Schemas(data_file = "result.txt")
#' result <- api_instance$Schemas()
#' dput(result)
#'
#'
#' ####################  Search  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "query_example" # character | Query string for searching. (Optional)
#' var_type <- c("inner_example") # array[character] | Filter by item type. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all. (Optional)
#' var_field_filters <- c(key = TODO) # object | Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'} (Optional)
#'
#' #Search for items in the IGVF Project.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Search(var_frame, query = var_query, type = var_type, limit = var_limit, sort = var_sort, field_filters = var_field_filtersdata_file = "result.txt")
#' result <- api_instance$Search(var_frame, query = var_query, type = var_type, limit = var_limit, sort = var_sort, field_filters = var_field_filters)
#' dput(result)
#'
#'
#' ####################  SequenceFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_anvil_url <- c("inner_example") # array[character] | Filter by anvil_url (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_controlled_access <- c("inner_example") # array[character] | Filter by controlled_access (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("bam") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_flowcell_id <- c("inner_example") # array[character] | Filter by flowcell_id (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_illumina_read_type <- c("R1") # array[character] | Filter by illumina_read_type (Optional)
#' var_index <- c("inner_example") # array[character] | Filter by index (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lane <- c(123) # array[integer] | Filter by lane (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_maximum_read_length <- c(123) # array[integer] | Filter by maximum_read_length (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_mean_read_length <- c(123) # array[numeric] | Filter by mean_read_length (Optional)
#' var_minimum_read_length <- c(123) # array[integer] | Filter by minimum_read_length (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_read_count <- c(123) # array[integer] | Filter by read_count (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_seqspecs <- c("inner_example") # array[character] | Filter by seqspecs (Optional)
#' var_sequencing_kit <- c("HiSeq SBS Kit v4") # array[character] | Filter by sequencing_kit (Optional)
#' var_sequencing_platform <- c("inner_example") # array[character] | Filter by sequencing_platform (Optional)
#' var_sequencing_run <- c(123) # array[integer] | Filter by sequencing_run (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the SequenceFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$SequenceFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, flowcell_id = var_flowcell_id, gene_list_for = var_gene_list_for, href = var_href, illumina_read_type = var_illumina_read_type, index = var_index, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, lane = var_lane, loci_list_for = var_loci_list_for, maximum_read_length = var_maximum_read_length, md5sum = var_md5sum, mean_read_length = var_mean_read_length, minimum_read_length = var_minimum_read_length, notes = var_notes, read_count = var_read_count, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, seqspecs = var_seqspecs, sequencing_kit = var_sequencing_kit, sequencing_platform = var_sequencing_platform, sequencing_run = var_sequencing_run, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$SequenceFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, flowcell_id = var_flowcell_id, gene_list_for = var_gene_list_for, href = var_href, illumina_read_type = var_illumina_read_type, index = var_index, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, lane = var_lane, loci_list_for = var_loci_list_for, maximum_read_length = var_maximum_read_length, md5sum = var_md5sum, mean_read_length = var_mean_read_length, minimum_read_length = var_minimum_read_length, notes = var_notes, read_count = var_read_count, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, seqspecs = var_seqspecs, sequencing_kit = var_sequencing_kit, sequencing_platform = var_sequencing_platform, sequencing_run = var_sequencing_run, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  SignalFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_assembly <- c("GRCh38") # array[character] | Filter by assembly (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_summary <- c("inner_example") # array[character] | Filter by content_summary (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("bigWig") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_filtered <- c("inner_example") # array[character] | Filter by filtered (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_normalized <- c("inner_example") # array[character] | Filter by normalized (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_reference_files <- c("inner_example") # array[character] | Filter by reference_files (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_start_view_position <- c("inner_example") # array[character] | Filter by start_view_position (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_strand_specificity <- c("plus") # array[character] | Filter by strand_specificity (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_transcriptome_annotation <- c("GENCODE 40") # array[character] | Filter by transcriptome_annotation (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the SignalFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$SignalFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_summary = var_content_summary, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, filtered = var_filtered, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, normalized = var_normalized, notes = var_notes, reference_files = var_reference_files, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, start_view_position = var_start_view_position, status = var_status, strand_specificity = var_strand_specificity, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$SignalFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_summary = var_content_summary, content_type = var_content_type, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_set = var_file_set, file_size = var_file_size, filtered = var_filtered, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, normalized = var_normalized, notes = var_notes, reference_files = var_reference_files, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, start_view_position = var_start_view_position, status = var_status, strand_specificity = var_strand_specificity, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  Software  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_source_url <- c("inner_example") # array[character] | Filter by source_url (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_used_by <- c("consortium analysis") # array[character] | Filter by used_by (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_versions <- c("inner_example") # array[character] | Filter by versions (Optional)
#'
#' #List items in the Software collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Software(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, source_url = var_source_url, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, used_by = var_used_by, uuid = var_uuid, versions = var_versionsdata_file = "result.txt")
#' result <- api_instance$Software(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, source_url = var_source_url, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, used_by = var_used_by, uuid = var_uuid, versions = var_versions)
#' dput(result)
#'
#'
#' ####################  SoftwareVersions  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_download_id <- c("inner_example") # array[character] | Filter by download_id (Optional)
#' var_downloaded_url <- c("inner_example") # array[character] | Filter by downloaded_url (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_software_id <- c("inner_example") # array[character] | Filter by software.@id (Optional)
#' var_software_title <- c("inner_example") # array[character] | Filter by software.title (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_version <- c("inner_example") # array[character] | Filter by version (Optional)
#'
#' #List items in the SoftwareVersion collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$SoftwareVersions(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, download_id = var_download_id, downloaded_url = var_downloaded_url, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, software_id = var_software_id, software_title = var_software_title, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid, version = var_versiondata_file = "result.txt")
#' result <- api_instance$SoftwareVersions(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, award_id = var_award_id, award_component = var_award_component, creation_timestamp = var_creation_timestamp, description = var_description, download_id = var_download_id, downloaded_url = var_downloaded_url, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, software_id = var_software_id, software_title = var_software_title, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid, version = var_version)
#' dput(result)
#'
#'
#' ####################  Sources  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Source collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Sources(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, description = var_description, name = var_name, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, url = var_url, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Sources(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, description = var_description, name = var_name, notes = var_notes, release_timestamp = var_release_timestamp, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, url = var_url, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  TabularFiles  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_step_version <- c("inner_example") # array[character] | Filter by analysis_step_version (Optional)
#' var_anvil_url <- c("inner_example") # array[character] | Filter by anvil_url (Optional)
#' var_assembly <- c("GRCh38") # array[character] | Filter by assembly (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_content_md5sum <- c("inner_example") # array[character] | Filter by content_md5sum (Optional)
#' var_content_type <- c("inner_example") # array[character] | Filter by content_type (Optional)
#' var_controlled_access <- c("inner_example") # array[character] | Filter by controlled_access (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_derived_from <- c("inner_example") # array[character] | Filter by derived_from (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_format <- c("bed") # array[character] | Filter by file_format (Optional)
#' var_file_format_specifications <- c("inner_example") # array[character] | Filter by file_format_specifications (Optional)
#' var_file_format_type <- c("bed12") # array[character] | Filter by file_format_type (Optional)
#' var_file_set <- c("inner_example") # array[character] | Filter by file_set (Optional)
#' var_file_size <- c(123) # array[integer] | Filter by file_size (Optional)
#' var_gene_list_for <- c("inner_example") # array[character] | Filter by gene_list_for (Optional)
#' var_href <- c("inner_example") # array[character] | Filter by href (Optional)
#' var_input_file_for <- c("inner_example") # array[character] | Filter by input_file_for (Optional)
#' var_integrated_in <- c("inner_example") # array[character] | Filter by integrated_in (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_loci_list_for <- c("inner_example") # array[character] | Filter by loci_list_for (Optional)
#' var_md5sum <- c("inner_example") # array[character] | Filter by md5sum (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_s3_uri <- c("inner_example") # array[character] | Filter by s3_uri (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitted_file_name <- c("inner_example") # array[character] | Filter by submitted_file_name (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_transcriptome_annotation <- c("GENCODE 40") # array[character] | Filter by transcriptome_annotation (Optional)
#' var_upload_status <- c("pending") # array[character] | Filter by upload_status (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_validation_error_detail <- c("inner_example") # array[character] | Filter by validation_error_detail (Optional)
#'
#' #List items in the TabularFile collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$TabularFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_format_type = var_file_format_type, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detaildata_file = "result.txt")
#' result <- api_instance$TabularFiles(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_step_version = var_analysis_step_version, anvil_url = var_anvil_url, assembly = var_assembly, award_id = var_award_id, award_component = var_award_component, collections = var_collections, content_md5sum = var_content_md5sum, content_type = var_content_type, controlled_access = var_controlled_access, creation_timestamp = var_creation_timestamp, dbxrefs = var_dbxrefs, derived_from = var_derived_from, description = var_description, documents = var_documents, file_format = var_file_format, file_format_specifications = var_file_format_specifications, file_format_type = var_file_format_type, file_set = var_file_set, file_size = var_file_size, gene_list_for = var_gene_list_for, href = var_href, input_file_for = var_input_file_for, integrated_in = var_integrated_in, lab_id = var_lab_id, lab_title = var_lab_title, loci_list_for = var_loci_list_for, md5sum = var_md5sum, notes = var_notes, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, s3_uri = var_s3_uri, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitted_file_name = var_submitted_file_name, submitter_comment = var_submitter_comment, summary = var_summary, transcriptome_annotation = var_transcriptome_annotation, upload_status = var_upload_status, uuid = var_uuid, validation_error_detail = var_validation_error_detail)
#' dput(result)
#'
#'
#' ####################  TechnicalSamples  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_classifications <- c("inner_example") # array[character] | Filter by classifications (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_construct_library_sets <- c("inner_example") # array[character] | Filter by construct_library_sets (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_obtained <- c("inner_example") # array[character] | Filter by date_obtained (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_file_sets_id <- c("inner_example") # array[character] | Filter by file_sets.@id (Optional)
#' var_file_sets_accession <- c("inner_example") # array[character] | Filter by file_sets.accession (Optional)
#' var_file_sets_aliases <- c("inner_example") # array[character] | Filter by file_sets.aliases (Optional)
#' var_file_sets_assay_term_term_name <- c("inner_example") # array[character] | Filter by file_sets.assay_term.term_name (Optional)
#' var_file_sets_lab_title <- c("inner_example") # array[character] | Filter by file_sets.lab.title (Optional)
#' var_file_sets_status <- c("in progress") # array[character] | Filter by file_sets.status (Optional)
#' var_file_sets_summary <- c("inner_example") # array[character] | Filter by file_sets.summary (Optional)
#' var_institutional_certificates <- c("inner_example") # array[character] | Filter by institutional_certificates (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_moi <- c(123) # array[numeric] | Filter by moi (Optional)
#' var_multiplexed_in_id <- c("inner_example") # array[character] | Filter by multiplexed_in.@id (Optional)
#' var_multiplexed_in_accession <- c("inner_example") # array[character] | Filter by multiplexed_in.accession (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_nucleic_acid_delivery <- c("transfection") # array[character] | Filter by nucleic_acid_delivery (Optional)
#' var_origin_of <- c("inner_example") # array[character] | Filter by origin_of (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sample_material <- c("undefined") # array[character] | Filter by sample_material (Optional)
#' var_sample_terms_id <- c("inner_example") # array[character] | Filter by sample_terms.@id (Optional)
#' var_sample_terms_term_name <- c("inner_example") # array[character] | Filter by sample_terms.term_name (Optional)
#' var_sorted_fractions <- c("inner_example") # array[character] | Filter by sorted_fractions (Optional)
#' var_sorted_from_id <- c("inner_example") # array[character] | Filter by sorted_from.@id (Optional)
#' var_sorted_from_accession <- c("inner_example") # array[character] | Filter by sorted_from.accession (Optional)
#' var_sorted_from_detail <- c("inner_example") # array[character] | Filter by sorted_from_detail (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_starting_amount <- c(123) # array[numeric] | Filter by starting_amount (Optional)
#' var_starting_amount_units <- c("cells") # array[character] | Filter by starting_amount_units (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Saccharomyces cerevisiae") # array[character] | Filter by taxa (Optional)
#' var_time_post_library_delivery <- c(123) # array[numeric] | Filter by time_post_library_delivery (Optional)
#' var_time_post_library_delivery_units <- c("minute") # array[character] | Filter by time_post_library_delivery_units (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the TechnicalSample collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$TechnicalSamples(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates = var_institutional_certificates, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_material = var_sample_material, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$TechnicalSamples(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, documents = var_documents, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates = var_institutional_certificates, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_material = var_sample_material, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  Tissues  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_age <- c("inner_example") # array[character] | Filter by age (Optional)
#' var_age_units <- c("minute") # array[character] | Filter by age_units (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biomarkers <- c("inner_example") # array[character] | Filter by biomarkers (Optional)
#' var_ccf_id <- c("inner_example") # array[character] | Filter by ccf_id (Optional)
#' var_cellular_sub_pool <- c("inner_example") # array[character] | Filter by cellular_sub_pool (Optional)
#' var_classifications <- c("inner_example") # array[character] | Filter by classifications (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_construct_library_sets <- c("inner_example") # array[character] | Filter by construct_library_sets (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_obtained <- c("inner_example") # array[character] | Filter by date_obtained (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_disease_terms_id <- c("inner_example") # array[character] | Filter by disease_terms.@id (Optional)
#' var_disease_terms_term_name <- c("inner_example") # array[character] | Filter by disease_terms.term_name (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors <- c("inner_example") # array[character] | Filter by donors (Optional)
#' var_embryonic <- c("inner_example") # array[character] | Filter by embryonic (Optional)
#' var_file_sets_id <- c("inner_example") # array[character] | Filter by file_sets.@id (Optional)
#' var_file_sets_accession <- c("inner_example") # array[character] | Filter by file_sets.accession (Optional)
#' var_file_sets_aliases <- c("inner_example") # array[character] | Filter by file_sets.aliases (Optional)
#' var_file_sets_assay_term_term_name <- c("inner_example") # array[character] | Filter by file_sets.assay_term.term_name (Optional)
#' var_file_sets_lab_title <- c("inner_example") # array[character] | Filter by file_sets.lab.title (Optional)
#' var_file_sets_status <- c("in progress") # array[character] | Filter by file_sets.status (Optional)
#' var_file_sets_summary <- c("inner_example") # array[character] | Filter by file_sets.summary (Optional)
#' var_institutional_certificates_id <- c("inner_example") # array[character] | Filter by institutional_certificates.@id (Optional)
#' var_institutional_certificates_certificate_identifier <- c("inner_example") # array[character] | Filter by institutional_certificates.certificate_identifier (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_lower_bound_age <- c(123) # array[numeric] | Filter by lower_bound_age (Optional)
#' var_lower_bound_age_in_hours <- c(123) # array[numeric] | Filter by lower_bound_age_in_hours (Optional)
#' var_modifications_id <- c("inner_example") # array[character] | Filter by modifications.@id (Optional)
#' var_modifications_status <- c("archived") # array[character] | Filter by modifications.status (Optional)
#' var_modifications_summary <- c("inner_example") # array[character] | Filter by modifications.summary (Optional)
#' var_moi <- c(123) # array[numeric] | Filter by moi (Optional)
#' var_multiplexed_in_id <- c("inner_example") # array[character] | Filter by multiplexed_in.@id (Optional)
#' var_multiplexed_in_accession <- c("inner_example") # array[character] | Filter by multiplexed_in.accession (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_nucleic_acid_delivery <- c("transfection") # array[character] | Filter by nucleic_acid_delivery (Optional)
#' var_origin_of <- c("inner_example") # array[character] | Filter by origin_of (Optional)
#' var_originated_from <- c("inner_example") # array[character] | Filter by originated_from (Optional)
#' var_part_of <- c("inner_example") # array[character] | Filter by part_of (Optional)
#' var_parts <- c("inner_example") # array[character] | Filter by parts (Optional)
#' var_pmi <- c(123) # array[integer] | Filter by pmi (Optional)
#' var_pmi_units <- c("second") # array[character] | Filter by pmi_units (Optional)
#' var_pooled_from <- c("inner_example") # array[character] | Filter by pooled_from (Optional)
#' var_pooled_in <- c("inner_example") # array[character] | Filter by pooled_in (Optional)
#' var_preservation_method <- c("cryopreservation") # array[character] | Filter by preservation_method (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sample_terms_id <- c("inner_example") # array[character] | Filter by sample_terms.@id (Optional)
#' var_sample_terms_term_name <- c("inner_example") # array[character] | Filter by sample_terms.term_name (Optional)
#' var_sex <- c("female") # array[character] | Filter by sex (Optional)
#' var_sorted_fractions <- c("inner_example") # array[character] | Filter by sorted_fractions (Optional)
#' var_sorted_from_id <- c("inner_example") # array[character] | Filter by sorted_from.@id (Optional)
#' var_sorted_from_accession <- c("inner_example") # array[character] | Filter by sorted_from.accession (Optional)
#' var_sorted_from_detail <- c("inner_example") # array[character] | Filter by sorted_from_detail (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_starting_amount <- c(123) # array[numeric] | Filter by starting_amount (Optional)
#' var_starting_amount_units <- c("cells") # array[character] | Filter by starting_amount_units (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_time_post_library_delivery <- c(123) # array[numeric] | Filter by time_post_library_delivery (Optional)
#' var_time_post_library_delivery_units <- c("minute") # array[character] | Filter by time_post_library_delivery_units (Optional)
#' var_treatments_id <- c("inner_example") # array[character] | Filter by treatments.@id (Optional)
#' var_treatments_purpose <- c("activation") # array[character] | Filter by treatments.purpose (Optional)
#' var_treatments_status <- c("archived") # array[character] | Filter by treatments.status (Optional)
#' var_treatments_summary <- c("inner_example") # array[character] | Filter by treatments.summary (Optional)
#' var_treatments_treatment_type <- c("chemical") # array[character] | Filter by treatments.treatment_type (Optional)
#' var_upper_bound_age <- c(123) # array[numeric] | Filter by upper_bound_age (Optional)
#' var_upper_bound_age_in_hours <- c(123) # array[numeric] | Filter by upper_bound_age_in_hours (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the Tissue collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Tissues(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, ccf_id = var_ccf_id, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from = var_originated_from, part_of = var_part_of, parts = var_parts, pmi = var_pmi, pmi_units = var_pmi_units, pooled_from = var_pooled_from, pooled_in = var_pooled_in, preservation_method = var_preservation_method, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$Tissues(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, ccf_id = var_ccf_id, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from = var_originated_from, part_of = var_part_of, parts = var_parts, pmi = var_pmi, pmi_units = var_pmi_units, pooled_from = var_pooled_from, pooled_in = var_pooled_in, preservation_method = var_preservation_method, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  Treatments  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_amount <- c(123) # array[numeric] | Filter by amount (Optional)
#' var_amount_units <- c("mg/kg") # array[character] | Filter by amount_units (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biosamples_treated <- c("inner_example") # array[character] | Filter by biosamples_treated (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_depletion <- c("inner_example") # array[character] | Filter by depletion (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_duration <- c(123) # array[numeric] | Filter by duration (Optional)
#' var_duration_units <- c("second") # array[character] | Filter by duration_units (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_p_h <- c(123) # array[numeric] | Filter by pH (Optional)
#' var_post_treatment_time <- c(123) # array[numeric] | Filter by post_treatment_time (Optional)
#' var_post_treatment_time_units <- c("minute") # array[character] | Filter by post_treatment_time_units (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_purpose <- c("activation") # array[character] | Filter by purpose (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_status <- c("archived") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_temperature <- c(123) # array[numeric] | Filter by temperature (Optional)
#' var_temperature_units <- c("Celsius") # array[character] | Filter by temperature_units (Optional)
#' var_treatment_term_id <- c("inner_example") # array[character] | Filter by treatment_term_id (Optional)
#' var_treatment_term_name <- c("inner_example") # array[character] | Filter by treatment_term_name (Optional)
#' var_treatment_type <- c("chemical") # array[character] | Filter by treatment_type (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#'
#' #List items in the Treatment collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Treatments(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, amount = var_amount, amount_units = var_amount_units, award_id = var_award_id, award_component = var_award_component, biosamples_treated = var_biosamples_treated, creation_timestamp = var_creation_timestamp, depletion = var_depletion, description = var_description, documents = var_documents, duration = var_duration, duration_units = var_duration_units, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, notes = var_notes, p_h = var_p_h, post_treatment_time = var_post_treatment_time, post_treatment_time_units = var_post_treatment_time_units, product_id = var_product_id, purpose = var_purpose, release_timestamp = var_release_timestamp, sources_id = var_sources_id, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, temperature = var_temperature, temperature_units = var_temperature_units, treatment_term_id = var_treatment_term_id, treatment_term_name = var_treatment_term_name, treatment_type = var_treatment_type, uuid = var_uuiddata_file = "result.txt")
#' result <- api_instance$Treatments(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, amount = var_amount, amount_units = var_amount_units, award_id = var_award_id, award_component = var_award_component, biosamples_treated = var_biosamples_treated, creation_timestamp = var_creation_timestamp, depletion = var_depletion, description = var_description, documents = var_documents, duration = var_duration, duration_units = var_duration_units, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, notes = var_notes, p_h = var_p_h, post_treatment_time = var_post_treatment_time, post_treatment_time_units = var_post_treatment_time_units, product_id = var_product_id, purpose = var_purpose, release_timestamp = var_release_timestamp, sources_id = var_sources_id, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, temperature = var_temperature, temperature_units = var_temperature_units, treatment_term_id = var_treatment_term_id, treatment_term_name = var_treatment_term_name, treatment_type = var_treatment_type, uuid = var_uuid)
#' dput(result)
#'
#'
#' ####################  Users  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_email <- c("inner_example") # array[character] | Filter by email (Optional)
#' var_first_name <- c("inner_example") # array[character] | Filter by first_name (Optional)
#' var_groups <- c("admin") # array[character] | Filter by groups (Optional)
#' var_job_title <- c("Principal Investigator") # array[character] | Filter by job_title (Optional)
#' var_lab <- c("inner_example") # array[character] | Filter by lab (Optional)
#' var_last_name <- c("inner_example") # array[character] | Filter by last_name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_status <- c("current") # array[character] | Filter by status (Optional)
#' var_submits_for <- c("inner_example") # array[character] | Filter by submits_for (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_title <- c("inner_example") # array[character] | Filter by title (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_viewing_groups <- c("community") # array[character] | Filter by viewing_groups (Optional)
#'
#' #List items in the User collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Users(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, description = var_description, email = var_email, first_name = var_first_name, groups = var_groups, job_title = var_job_title, lab = var_lab, last_name = var_last_name, notes = var_notes, status = var_status, submits_for = var_submits_for, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid, viewing_groups = var_viewing_groupsdata_file = "result.txt")
#' result <- api_instance$Users(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, aliases = var_aliases, creation_timestamp = var_creation_timestamp, description = var_description, email = var_email, first_name = var_first_name, groups = var_groups, job_title = var_job_title, lab = var_lab, last_name = var_last_name, notes = var_notes, status = var_status, submits_for = var_submits_for, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, title = var_title, uuid = var_uuid, viewing_groups = var_viewing_groups)
#' dput(result)
#'
#'
#' ####################  WholeOrganisms  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_age <- c("inner_example") # array[character] | Filter by age (Optional)
#' var_age_units <- c("minute") # array[character] | Filter by age_units (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_biomarkers <- c("inner_example") # array[character] | Filter by biomarkers (Optional)
#' var_cellular_sub_pool <- c("inner_example") # array[character] | Filter by cellular_sub_pool (Optional)
#' var_classifications <- c("inner_example") # array[character] | Filter by classifications (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_construct_library_sets <- c("inner_example") # array[character] | Filter by construct_library_sets (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_date_obtained <- c("inner_example") # array[character] | Filter by date_obtained (Optional)
#' var_dbxrefs <- c("inner_example") # array[character] | Filter by dbxrefs (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_disease_terms_id <- c("inner_example") # array[character] | Filter by disease_terms.@id (Optional)
#' var_disease_terms_term_name <- c("inner_example") # array[character] | Filter by disease_terms.term_name (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_donors <- c("inner_example") # array[character] | Filter by donors (Optional)
#' var_embryonic <- c("inner_example") # array[character] | Filter by embryonic (Optional)
#' var_file_sets_id <- c("inner_example") # array[character] | Filter by file_sets.@id (Optional)
#' var_file_sets_accession <- c("inner_example") # array[character] | Filter by file_sets.accession (Optional)
#' var_file_sets_aliases <- c("inner_example") # array[character] | Filter by file_sets.aliases (Optional)
#' var_file_sets_assay_term_term_name <- c("inner_example") # array[character] | Filter by file_sets.assay_term.term_name (Optional)
#' var_file_sets_lab_title <- c("inner_example") # array[character] | Filter by file_sets.lab.title (Optional)
#' var_file_sets_status <- c("in progress") # array[character] | Filter by file_sets.status (Optional)
#' var_file_sets_summary <- c("inner_example") # array[character] | Filter by file_sets.summary (Optional)
#' var_institutional_certificates_id <- c("inner_example") # array[character] | Filter by institutional_certificates.@id (Optional)
#' var_institutional_certificates_certificate_identifier <- c("inner_example") # array[character] | Filter by institutional_certificates.certificate_identifier (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_lot_id <- c("inner_example") # array[character] | Filter by lot_id (Optional)
#' var_lower_bound_age <- c(123) # array[numeric] | Filter by lower_bound_age (Optional)
#' var_lower_bound_age_in_hours <- c(123) # array[numeric] | Filter by lower_bound_age_in_hours (Optional)
#' var_modifications_id <- c("inner_example") # array[character] | Filter by modifications.@id (Optional)
#' var_modifications_status <- c("archived") # array[character] | Filter by modifications.status (Optional)
#' var_modifications_summary <- c("inner_example") # array[character] | Filter by modifications.summary (Optional)
#' var_moi <- c(123) # array[numeric] | Filter by moi (Optional)
#' var_multiplexed_in_id <- c("inner_example") # array[character] | Filter by multiplexed_in.@id (Optional)
#' var_multiplexed_in_accession <- c("inner_example") # array[character] | Filter by multiplexed_in.accession (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_nucleic_acid_delivery <- c("transfection") # array[character] | Filter by nucleic_acid_delivery (Optional)
#' var_origin_of <- c("inner_example") # array[character] | Filter by origin_of (Optional)
#' var_originated_from <- c("inner_example") # array[character] | Filter by originated_from (Optional)
#' var_part_of <- c("inner_example") # array[character] | Filter by part_of (Optional)
#' var_parts <- c("inner_example") # array[character] | Filter by parts (Optional)
#' var_pooled_from <- c("inner_example") # array[character] | Filter by pooled_from (Optional)
#' var_pooled_in <- c("inner_example") # array[character] | Filter by pooled_in (Optional)
#' var_product_id <- c("inner_example") # array[character] | Filter by product_id (Optional)
#' var_protocols <- c("inner_example") # array[character] | Filter by protocols (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_sample_terms_id <- c("inner_example") # array[character] | Filter by sample_terms.@id (Optional)
#' var_sample_terms_term_name <- c("inner_example") # array[character] | Filter by sample_terms.term_name (Optional)
#' var_sex <- c("female") # array[character] | Filter by sex (Optional)
#' var_sorted_fractions <- c("inner_example") # array[character] | Filter by sorted_fractions (Optional)
#' var_sorted_from_id <- c("inner_example") # array[character] | Filter by sorted_from.@id (Optional)
#' var_sorted_from_accession <- c("inner_example") # array[character] | Filter by sorted_from.accession (Optional)
#' var_sorted_from_detail <- c("inner_example") # array[character] | Filter by sorted_from_detail (Optional)
#' var_sources_id <- c("inner_example") # array[character] | Filter by sources.@id (Optional)
#' var_starting_amount <- c(123) # array[numeric] | Filter by starting_amount (Optional)
#' var_starting_amount_units <- c("cells") # array[character] | Filter by starting_amount_units (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_taxa <- c("Homo sapiens") # array[character] | Filter by taxa (Optional)
#' var_time_post_library_delivery <- c(123) # array[numeric] | Filter by time_post_library_delivery (Optional)
#' var_time_post_library_delivery_units <- c("minute") # array[character] | Filter by time_post_library_delivery_units (Optional)
#' var_treatments_id <- c("inner_example") # array[character] | Filter by treatments.@id (Optional)
#' var_treatments_purpose <- c("activation") # array[character] | Filter by treatments.purpose (Optional)
#' var_treatments_status <- c("archived") # array[character] | Filter by treatments.status (Optional)
#' var_treatments_summary <- c("inner_example") # array[character] | Filter by treatments.summary (Optional)
#' var_treatments_treatment_type <- c("chemical") # array[character] | Filter by treatments.treatment_type (Optional)
#' var_upper_bound_age <- c(123) # array[numeric] | Filter by upper_bound_age (Optional)
#' var_upper_bound_age_in_hours <- c(123) # array[numeric] | Filter by upper_bound_age_in_hours (Optional)
#' var_url <- c("inner_example") # array[character] | Filter by url (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_virtual <- c("inner_example") # array[character] | Filter by virtual (Optional)
#'
#' #List items in the WholeOrganism collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$WholeOrganisms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from = var_originated_from, part_of = var_part_of, parts = var_parts, pooled_from = var_pooled_from, pooled_in = var_pooled_in, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtualdata_file = "result.txt")
#' result <- api_instance$WholeOrganisms(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, age = var_age, age_units = var_age_units, aliases = var_aliases, alternate_accessions = var_alternate_accessions, award_id = var_award_id, award_component = var_award_component, biomarkers = var_biomarkers, cellular_sub_pool = var_cellular_sub_pool, classifications = var_classifications, collections = var_collections, construct_library_sets = var_construct_library_sets, creation_timestamp = var_creation_timestamp, date_obtained = var_date_obtained, dbxrefs = var_dbxrefs, description = var_description, disease_terms_id = var_disease_terms_id, disease_terms_term_name = var_disease_terms_term_name, documents = var_documents, donors = var_donors, embryonic = var_embryonic, file_sets_id = var_file_sets_id, file_sets_accession = var_file_sets_accession, file_sets_aliases = var_file_sets_aliases, file_sets_assay_term_term_name = var_file_sets_assay_term_term_name, file_sets_lab_title = var_file_sets_lab_title, file_sets_status = var_file_sets_status, file_sets_summary = var_file_sets_summary, institutional_certificates_id = var_institutional_certificates_id, institutional_certificates_certificate_identifier = var_institutional_certificates_certificate_identifier, lab_id = var_lab_id, lab_title = var_lab_title, lot_id = var_lot_id, lower_bound_age = var_lower_bound_age, lower_bound_age_in_hours = var_lower_bound_age_in_hours, modifications_id = var_modifications_id, modifications_status = var_modifications_status, modifications_summary = var_modifications_summary, moi = var_moi, multiplexed_in_id = var_multiplexed_in_id, multiplexed_in_accession = var_multiplexed_in_accession, notes = var_notes, nucleic_acid_delivery = var_nucleic_acid_delivery, origin_of = var_origin_of, originated_from = var_originated_from, part_of = var_part_of, parts = var_parts, pooled_from = var_pooled_from, pooled_in = var_pooled_in, product_id = var_product_id, protocols = var_protocols, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, sample_terms_id = var_sample_terms_id, sample_terms_term_name = var_sample_terms_term_name, sex = var_sex, sorted_fractions = var_sorted_fractions, sorted_from_id = var_sorted_from_id, sorted_from_accession = var_sorted_from_accession, sorted_from_detail = var_sorted_from_detail, sources_id = var_sources_id, starting_amount = var_starting_amount, starting_amount_units = var_starting_amount_units, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, taxa = var_taxa, time_post_library_delivery = var_time_post_library_delivery, time_post_library_delivery_units = var_time_post_library_delivery_units, treatments_id = var_treatments_id, treatments_purpose = var_treatments_purpose, treatments_status = var_treatments_status, treatments_summary = var_treatments_summary, treatments_treatment_type = var_treatments_treatment_type, upper_bound_age = var_upper_bound_age, upper_bound_age_in_hours = var_upper_bound_age_in_hours, url = var_url, uuid = var_uuid, virtual = var_virtual)
#' dput(result)
#'
#'
#' ####################  Workflows  ####################
#'
#' library(igvfclient)
#' var_frame <- "frame_example" # character | Constant value. Do not set.
#' var_query <- "variant flowfish jurkat" # character | Query string for searching. (Optional)
#' var_limit <- Limit$new() # Limit | Maximum number of results to return. Default is 25. Use 'all' for all results. (Optional)
#' var_sort <- c("inner_example") # array[character] | Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all. (Optional)
#' var_id <- c("inner_example") # array[character] | Filter by @id (Optional)
#' var_accession <- c("inner_example") # array[character] | Filter by accession (Optional)
#' var_aliases <- c("inner_example") # array[character] | Filter by aliases (Optional)
#' var_alternate_accessions <- c("inner_example") # array[character] | Filter by alternate_accessions (Optional)
#' var_analysis_steps <- c("inner_example") # array[character] | Filter by analysis_steps (Optional)
#' var_award_id <- c("inner_example") # array[character] | Filter by award.@id (Optional)
#' var_award_component <- c("affiliate") # array[character] | Filter by award.component (Optional)
#' var_collections <- c("ClinGen") # array[character] | Filter by collections (Optional)
#' var_creation_timestamp <- c("inner_example") # array[character] | Filter by creation_timestamp (Optional)
#' var_description <- c("inner_example") # array[character] | Filter by description (Optional)
#' var_documents <- c("inner_example") # array[character] | Filter by documents (Optional)
#' var_lab_id <- c("inner_example") # array[character] | Filter by lab.@id (Optional)
#' var_lab_title <- c("inner_example") # array[character] | Filter by lab.title (Optional)
#' var_name <- c("inner_example") # array[character] | Filter by name (Optional)
#' var_notes <- c("inner_example") # array[character] | Filter by notes (Optional)
#' var_publication_identifiers <- c("inner_example") # array[character] | Filter by publication_identifiers (Optional)
#' var_publications <- c("inner_example") # array[character] | Filter by publications (Optional)
#' var_release_timestamp <- c("inner_example") # array[character] | Filter by release_timestamp (Optional)
#' var_revoke_detail <- c("inner_example") # array[character] | Filter by revoke_detail (Optional)
#' var_source_url <- c("inner_example") # array[character] | Filter by source_url (Optional)
#' var_standards_page_id <- c("inner_example") # array[character] | Filter by standards_page.@id (Optional)
#' var_standards_page_title <- c("inner_example") # array[character] | Filter by standards_page.title (Optional)
#' var_status <- c("in progress") # array[character] | Filter by status (Optional)
#' var_submitted_by_id <- c("inner_example") # array[character] | Filter by submitted_by.@id (Optional)
#' var_submitted_by_title <- c("inner_example") # array[character] | Filter by submitted_by.title (Optional)
#' var_submitter_comment <- c("inner_example") # array[character] | Filter by submitter_comment (Optional)
#' var_summary <- c("inner_example") # array[character] | Filter by summary (Optional)
#' var_uuid <- c("inner_example") # array[character] | Filter by uuid (Optional)
#' var_workflow_repositories <- c("inner_example") # array[character] | Filter by workflow_repositories (Optional)
#' var_workflow_version <- c(123) # array[integer] | Filter by workflow_version (Optional)
#'
#' #List items in the Workflow collection.
#' api_instance <- IgvfApi$new()
#'
#' # Configure HTTP basic authorization: basicAuth
#' api_instance$api_client$username <- Sys.getenv("USERNAME")
#' api_instance$api_client$password <- Sys.getenv("PASSWORD")
#'
#' # to save the result into a file, simply add the optional `data_file` parameter, e.g.
#' # result <- api_instance$Workflows(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_steps = var_analysis_steps, award_id = var_award_id, award_component = var_award_component, collections = var_collections, creation_timestamp = var_creation_timestamp, description = var_description, documents = var_documents, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, source_url = var_source_url, standards_page_id = var_standards_page_id, standards_page_title = var_standards_page_title, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid, workflow_repositories = var_workflow_repositories, workflow_version = var_workflow_versiondata_file = "result.txt")
#' result <- api_instance$Workflows(var_frame, query = var_query, limit = var_limit, sort = var_sort, id = var_id, accession = var_accession, aliases = var_aliases, alternate_accessions = var_alternate_accessions, analysis_steps = var_analysis_steps, award_id = var_award_id, award_component = var_award_component, collections = var_collections, creation_timestamp = var_creation_timestamp, description = var_description, documents = var_documents, lab_id = var_lab_id, lab_title = var_lab_title, name = var_name, notes = var_notes, publication_identifiers = var_publication_identifiers, publications = var_publications, release_timestamp = var_release_timestamp, revoke_detail = var_revoke_detail, source_url = var_source_url, standards_page_id = var_standards_page_id, standards_page_title = var_standards_page_title, status = var_status, submitted_by_id = var_submitted_by_id, submitted_by_title = var_submitted_by_title, submitter_comment = var_submitter_comment, summary = var_summary, uuid = var_uuid, workflow_repositories = var_workflow_repositories, workflow_version = var_workflow_version)
#' dput(result)
#'
#'
#' }
#' @importFrom R6 R6Class
#' @importFrom base64enc base64encode
#' @export
IgvfApi <- R6::R6Class(
  "IgvfApi",
  public = list(
    api_client = NULL,
    #' Initialize a new IgvfApi.
    #'
    #' @description
    #' Initialize a new IgvfApi.
    #'
    #' @param api_client An instance of API client.
    #' @export
    initialize = function(api_client) {
      if (!missing(api_client)) {
        self$api_client <- api_client
      } else {
        self$api_client <- ApiClient$new()
      }
    },
    #' List items in the AccessKey collection.
    #'
    #' @description
    #' List items in the AccessKey collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param access_key_id (optional) Filter by access_key_id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param notes (optional) Filter by notes
    #' @param secret_access_key_hash (optional) Filter by secret_access_key_hash
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param user (optional) Filter by user
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AccessKeyResults
    #' @export
    AccessKeys = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, access_key_id = NULL, aliases = NULL, creation_timestamp = NULL, description = NULL, notes = NULL, secret_access_key_hash = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, user = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$AccessKeysWithHttpInfo(frame, query, limit, sort, id, access_key_id, aliases, creation_timestamp, description, notes, secret_access_key_hash, status, submitted_by_id, submitted_by_title, submitter_comment, summary, user, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AccessKey collection.
    #'
    #' @description
    #' List items in the AccessKey collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param access_key_id (optional) Filter by access_key_id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param notes (optional) Filter by notes
    #' @param secret_access_key_hash (optional) Filter by secret_access_key_hash
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param user (optional) Filter by user
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AccessKeyResults) with additional information such as HTTP status code, headers
    #' @export
    AccessKeysWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, access_key_id = NULL, aliases = NULL, creation_timestamp = NULL, description = NULL, notes = NULL, secret_access_key_hash = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, user = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AccessKeys. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `access_key_id`) {
        query_params[["access_key_id"]] <- c(query_params[["access_key_id"]], list(`access_key_id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `secret_access_key_hash`) {
        query_params[["secret_access_key_hash"]] <- c(query_params[["secret_access_key_hash"]], list(`secret_access_key_hash` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("current", "deleted"))) {
          stop("Invalid value for status when calling IgvfApi$AccessKeys. Must be [current, deleted].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `user`) {
        query_params[["user"]] <- c(query_params[["user"]], list(`user` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/access-keys/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AccessKeyResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the AlignmentFile collection.
    #'
    #' @description
    #' List items in the AlignmentFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_summary (optional) Filter by content_summary
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param filtered (optional) Filter by filtered
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param redacted (optional) Filter by redacted
    #' @param reference_files (optional) Filter by reference_files
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AlignmentFileResults
    #' @export
    AlignmentFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_summary = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, filtered = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, redacted = NULL, reference_files = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$AlignmentFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, anvil_url, assembly, award_id, award_component, collections, content_md5sum, content_summary, content_type, controlled_access, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_set, file_size, filtered, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, redacted, reference_files, release_timestamp, revoke_detail, s3_uri, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, transcriptome_annotation, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AlignmentFile collection.
    #'
    #' @description
    #' List items in the AlignmentFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_summary (optional) Filter by content_summary
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param filtered (optional) Filter by filtered
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param redacted (optional) Filter by redacted
    #' @param reference_files (optional) Filter by reference_files
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AlignmentFileResults) with additional information such as HTTP status code, headers
    #' @export
    AlignmentFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_summary = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, filtered = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, redacted = NULL, reference_files = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AlignmentFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `anvil_url`) {
        query_params[["anvil_url"]] <- c(query_params[["anvil_url"]], list(`anvil_url` = query_item))
      }

      # explore
      for (query_item in `assembly`) {
        # validate enum values
        if (!(query_item %in% c("GRCh38", "GRCm39"))) {
          stop("Invalid value for assembly when calling IgvfApi$AlignmentFiles. Must be [GRCh38, GRCm39].")
        }
        query_params[["assembly"]] <- c(query_params[["assembly"]], list(`assembly` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$AlignmentFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$AlignmentFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_summary`) {
        query_params[["content_summary"]] <- c(query_params[["content_summary"]], list(`content_summary` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `controlled_access`) {
        query_params[["controlled_access"]] <- c(query_params[["controlled_access"]], list(`controlled_access` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("bam"))) {
          stop("Invalid value for file_format when calling IgvfApi$AlignmentFiles. Must be [bam].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `filtered`) {
        query_params[["filtered"]] <- c(query_params[["filtered"]], list(`filtered` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `redacted`) {
        query_params[["redacted"]] <- c(query_params[["redacted"]], list(`redacted` = query_item))
      }

      # explore
      for (query_item in `reference_files`) {
        query_params[["reference_files"]] <- c(query_params[["reference_files"]], list(`reference_files` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$AlignmentFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotation`) {
        # validate enum values
        if (!(query_item %in% c("GENCODE 40", "GENCODE 41", "GENCODE 42", "GENCODE 43", "GENCODE 44", "GENCODE 45", "GENCODE M30", "GENCODE M31", "GENCODE M32", "GENCODE M33", "GENCODE M34"))) {
          stop("Invalid value for transcriptome_annotation when calling IgvfApi$AlignmentFiles. Must be [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34].")
        }
        query_params[["transcriptome_annotation"]] <- c(query_params[["transcriptome_annotation"]], list(`transcriptome_annotation` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$AlignmentFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/alignment-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AlignmentFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the AnalysisSet collection.
    #'
    #' @description
    #' List items in the AnalysisSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assay_titles (optional) Filter by assay_titles
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param input_file_sets_id (optional) Filter by input_file_sets.@id
    #' @param input_file_sets_accession (optional) Filter by input_file_sets.accession
    #' @param input_file_sets_aliases (optional) Filter by input_file_sets.aliases
    #' @param input_file_sets_file_set_type (optional) Filter by input_file_sets.file_set_type
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AnalysisSetResults
    #' @export
    AnalysisSets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assay_titles = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, input_file_sets_id = NULL, input_file_sets_accession = NULL, input_file_sets_aliases = NULL, input_file_sets_file_set_type = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$AnalysisSetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, assay_titles, award_id, award_component, award_contact_pi_id, award_contact_pi_title, award_title, collections, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, dbxrefs, description, documents, donors_id, donors_accession, donors_aliases, donors_sex, donors_status, donors_taxa, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_creation_timestamp, files_file_format, files_file_size, files_href, files_s3_uri, files_sequencing_platform, files_submitted_file_name, files_upload_status, input_file_set_for, input_file_sets_id, input_file_sets_accession, input_file_sets_aliases, input_file_sets_file_set_type, lab_id, lab_title, notes, publication_identifiers, publications, release_timestamp, revoke_detail, samples_id, samples_accession, samples_aliases, samples_cell_fate_change_treatments, samples_classifications, samples_construct_library_sets, samples_disease_terms_id, samples_disease_terms_term_name, samples_modifications, samples_sample_terms_id, samples_sample_terms_aliases, samples_sample_terms_status, samples_sample_terms_summary, samples_sample_terms_term_name, samples_status, samples_summary, samples_targeted_sample_term_id, samples_targeted_sample_term_term_name, samples_taxa, samples_treatments, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AnalysisSet collection.
    #'
    #' @description
    #' List items in the AnalysisSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assay_titles (optional) Filter by assay_titles
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param input_file_sets_id (optional) Filter by input_file_sets.@id
    #' @param input_file_sets_accession (optional) Filter by input_file_sets.accession
    #' @param input_file_sets_aliases (optional) Filter by input_file_sets.aliases
    #' @param input_file_sets_file_set_type (optional) Filter by input_file_sets.file_set_type
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AnalysisSetResults) with additional information such as HTTP status code, headers
    #' @export
    AnalysisSetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assay_titles = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, input_file_sets_id = NULL, input_file_sets_accession = NULL, input_file_sets_aliases = NULL, input_file_sets_file_set_type = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }


























































      if (length(`samples_classifications`) > 2) {
        stop("Invalid length for `samples_classifications` when calling IgvfApi$AnalysisSets, number of items must be less than or equal to 2.")
      }




      if (length(`samples_modifications`) > 2) {
        stop("Invalid length for `samples_modifications` when calling IgvfApi$AnalysisSets, number of items must be less than or equal to 2.")
      }



















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AnalysisSets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `assay_titles`) {
        query_params[["assay_titles"]] <- c(query_params[["assay_titles"]], list(`assay_titles` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$AnalysisSets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_id`) {
        query_params[["award.contact_pi.@id"]] <- c(query_params[["award.contact_pi.@id"]], list(`award.contact_pi.@id` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_title`) {
        query_params[["award.contact_pi.title"]] <- c(query_params[["award.contact_pi.title"]], list(`award.contact_pi.title` = query_item))
      }

      # explore
      for (query_item in `award_title`) {
        query_params[["award.title"]] <- c(query_params[["award.title"]], list(`award.title` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$AnalysisSets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors_id`) {
        query_params[["donors.@id"]] <- c(query_params[["donors.@id"]], list(`donors.@id` = query_item))
      }

      # explore
      for (query_item in `donors_accession`) {
        query_params[["donors.accession"]] <- c(query_params[["donors.accession"]], list(`donors.accession` = query_item))
      }

      # explore
      for (query_item in `donors_aliases`) {
        query_params[["donors.aliases"]] <- c(query_params[["donors.aliases"]], list(`donors.aliases` = query_item))
      }

      # explore
      for (query_item in `donors_sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for donors_sex when calling IgvfApi$AnalysisSets. Must be [male, female, unspecified].")
        }
        query_params[["donors.sex"]] <- c(query_params[["donors.sex"]], list(`donors.sex` = query_item))
      }

      # explore
      for (query_item in `donors_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for donors_status when calling IgvfApi$AnalysisSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["donors.status"]] <- c(query_params[["donors.status"]], list(`donors.status` = query_item))
      }

      # explore
      for (query_item in `donors_taxa`) {
        query_params[["donors.taxa"]] <- c(query_params[["donors.taxa"]], list(`donors.taxa` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("intermediate analysis", "principal analysis"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$AnalysisSets. Must be [intermediate analysis, principal analysis].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_creation_timestamp`) {
        query_params[["files.creation_timestamp"]] <- c(query_params[["files.creation_timestamp"]], list(`files.creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_file_size`) {
        query_params[["files.file_size"]] <- c(query_params[["files.file_size"]], list(`files.file_size` = query_item))
      }

      # explore
      for (query_item in `files_href`) {
        query_params[["files.href"]] <- c(query_params[["files.href"]], list(`files.href` = query_item))
      }

      # explore
      for (query_item in `files_s3_uri`) {
        query_params[["files.s3_uri"]] <- c(query_params[["files.s3_uri"]], list(`files.s3_uri` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform`) {
        query_params[["files.sequencing_platform"]] <- c(query_params[["files.sequencing_platform"]], list(`files.sequencing_platform` = query_item))
      }

      # explore
      for (query_item in `files_submitted_file_name`) {
        query_params[["files.submitted_file_name"]] <- c(query_params[["files.submitted_file_name"]], list(`files.submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$AnalysisSets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_id`) {
        query_params[["input_file_sets.@id"]] <- c(query_params[["input_file_sets.@id"]], list(`input_file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_accession`) {
        query_params[["input_file_sets.accession"]] <- c(query_params[["input_file_sets.accession"]], list(`input_file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_aliases`) {
        query_params[["input_file_sets.aliases"]] <- c(query_params[["input_file_sets.aliases"]], list(`input_file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("pathogenicity", "functional effect", "protein stability", "activity level", "binding effect"))) {
          stop("Invalid value for input_file_sets_file_set_type when calling IgvfApi$AnalysisSets. Must be [pathogenicity, functional effect, protein stability, activity level, binding effect].")
        }
        query_params[["input_file_sets.file_set_type"]] <- c(query_params[["input_file_sets.file_set_type"]], list(`input_file_sets.file_set_type` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `samples_id`) {
        query_params[["samples.@id"]] <- c(query_params[["samples.@id"]], list(`samples.@id` = query_item))
      }

      # explore
      for (query_item in `samples_accession`) {
        query_params[["samples.accession"]] <- c(query_params[["samples.accession"]], list(`samples.accession` = query_item))
      }

      # explore
      for (query_item in `samples_aliases`) {
        query_params[["samples.aliases"]] <- c(query_params[["samples.aliases"]], list(`samples.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments`) {
        query_params[["samples.cell_fate_change_treatments"]] <- c(query_params[["samples.cell_fate_change_treatments"]], list(`samples.cell_fate_change_treatments` = query_item))
      }

      # explore
      for (query_item in `samples_classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for samples_classifications when calling IgvfApi$AnalysisSets. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["samples.classifications"]] <- c(query_params[["samples.classifications"]], list(`samples.classifications` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets`) {
        query_params[["samples.construct_library_sets"]] <- c(query_params[["samples.construct_library_sets"]], list(`samples.construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_id`) {
        query_params[["samples.disease_terms.@id"]] <- c(query_params[["samples.disease_terms.@id"]], list(`samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_term_name`) {
        query_params[["samples.disease_terms.term_name"]] <- c(query_params[["samples.disease_terms.term_name"]], list(`samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_modifications`) {
        query_params[["samples.modifications"]] <- c(query_params[["samples.modifications"]], list(`samples.modifications` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_id`) {
        query_params[["samples.sample_terms.@id"]] <- c(query_params[["samples.sample_terms.@id"]], list(`samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_aliases`) {
        query_params[["samples.sample_terms.aliases"]] <- c(query_params[["samples.sample_terms.aliases"]], list(`samples.sample_terms.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for samples_sample_terms_status when calling IgvfApi$AnalysisSets. Must be [archived, deleted, in progress, released].")
        }
        query_params[["samples.sample_terms.status"]] <- c(query_params[["samples.sample_terms.status"]], list(`samples.sample_terms.status` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_summary`) {
        query_params[["samples.sample_terms.summary"]] <- c(query_params[["samples.sample_terms.summary"]], list(`samples.sample_terms.summary` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_term_name`) {
        query_params[["samples.sample_terms.term_name"]] <- c(query_params[["samples.sample_terms.term_name"]], list(`samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for samples_status when calling IgvfApi$AnalysisSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["samples.status"]] <- c(query_params[["samples.status"]], list(`samples.status` = query_item))
      }

      # explore
      for (query_item in `samples_summary`) {
        query_params[["samples.summary"]] <- c(query_params[["samples.summary"]], list(`samples.summary` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_id`) {
        query_params[["samples.targeted_sample_term.@id"]] <- c(query_params[["samples.targeted_sample_term.@id"]], list(`samples.targeted_sample_term.@id` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_term_name`) {
        query_params[["samples.targeted_sample_term.term_name"]] <- c(query_params[["samples.targeted_sample_term.term_name"]], list(`samples.targeted_sample_term.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for samples_taxa when calling IgvfApi$AnalysisSets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["samples.taxa"]] <- c(query_params[["samples.taxa"]], list(`samples.taxa` = query_item))
      }

      # explore
      for (query_item in `samples_treatments`) {
        query_params[["samples.treatments"]] <- c(query_params[["samples.treatments"]], list(`samples.treatments` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$AnalysisSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/analysis-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AnalysisSetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the AnalysisStepVersion collection.
    #'
    #' @description
    #' List items in the AnalysisStepVersion collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param analysis_step_id (optional) Filter by analysis_step.@id
    #' @param analysis_step_name (optional) Filter by analysis_step.name
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param software_versions_id (optional) Filter by software_versions.@id
    #' @param software_versions_name (optional) Filter by software_versions.name
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AnalysisStepVersionResults
    #' @export
    AnalysisStepVersions = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, analysis_step_id = NULL, analysis_step_name = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, release_timestamp = NULL, software_versions_id = NULL, software_versions_name = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$AnalysisStepVersionsWithHttpInfo(frame, query, limit, sort, id, aliases, analysis_step_id, analysis_step_name, award_id, award_component, creation_timestamp, description, lab_id, lab_title, notes, release_timestamp, software_versions_id, software_versions_name, status, submitted_by_id, submitted_by_title, submitter_comment, summary, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AnalysisStepVersion collection.
    #'
    #' @description
    #' List items in the AnalysisStepVersion collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param analysis_step_id (optional) Filter by analysis_step.@id
    #' @param analysis_step_name (optional) Filter by analysis_step.name
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param software_versions_id (optional) Filter by software_versions.@id
    #' @param software_versions_name (optional) Filter by software_versions.name
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AnalysisStepVersionResults) with additional information such as HTTP status code, headers
    #' @export
    AnalysisStepVersionsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, analysis_step_id = NULL, analysis_step_name = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, release_timestamp = NULL, software_versions_id = NULL, software_versions_name = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }

























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AnalysisStepVersions. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `analysis_step_id`) {
        query_params[["analysis_step.@id"]] <- c(query_params[["analysis_step.@id"]], list(`analysis_step.@id` = query_item))
      }

      # explore
      for (query_item in `analysis_step_name`) {
        query_params[["analysis_step.name"]] <- c(query_params[["analysis_step.name"]], list(`analysis_step.name` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$AnalysisStepVersions. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `software_versions_id`) {
        query_params[["software_versions.@id"]] <- c(query_params[["software_versions.@id"]], list(`software_versions.@id` = query_item))
      }

      # explore
      for (query_item in `software_versions_name`) {
        query_params[["software_versions.name"]] <- c(query_params[["software_versions.name"]], list(`software_versions.name` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$AnalysisStepVersions. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/analysis-step-versions/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AnalysisStepVersionResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the AnalysisStep collection.
    #'
    #' @description
    #' List items in the AnalysisStep collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param analysis_step_types (optional) Filter by analysis_step_types
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param input_content_types (optional) Filter by input_content_types
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param output_content_types (optional) Filter by output_content_types
    #' @param parents_id (optional) Filter by parents.@id
    #' @param parents_title (optional) Filter by parents.title
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param step_label (optional) Filter by step_label
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param workflow_id (optional) Filter by workflow.@id
    #' @param workflow_accession (optional) Filter by workflow.accession
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AnalysisStepResults
    #' @export
    AnalysisSteps = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, analysis_step_types = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, input_content_types = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, output_content_types = NULL, parents_id = NULL, parents_title = NULL, release_timestamp = NULL, status = NULL, step_label = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, workflow_id = NULL, workflow_accession = NULL, data_file = NULL, ...) {
      local_var_response <- self$AnalysisStepsWithHttpInfo(frame, query, limit, sort, id, aliases, analysis_step_types, award_id, award_component, creation_timestamp, description, input_content_types, lab_id, lab_title, name, notes, output_content_types, parents_id, parents_title, release_timestamp, status, step_label, submitted_by_id, submitted_by_title, submitter_comment, summary, title, uuid, workflow_id, workflow_accession, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AnalysisStep collection.
    #'
    #' @description
    #' List items in the AnalysisStep collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param analysis_step_types (optional) Filter by analysis_step_types
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param input_content_types (optional) Filter by input_content_types
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param output_content_types (optional) Filter by output_content_types
    #' @param parents_id (optional) Filter by parents.@id
    #' @param parents_title (optional) Filter by parents.title
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param step_label (optional) Filter by step_label
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param workflow_id (optional) Filter by workflow.@id
    #' @param workflow_accession (optional) Filter by workflow.accession
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AnalysisStepResults) with additional information such as HTTP status code, headers
    #' @export
    AnalysisStepsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, analysis_step_types = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, input_content_types = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, output_content_types = NULL, parents_id = NULL, parents_title = NULL, release_timestamp = NULL, status = NULL, step_label = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, workflow_id = NULL, workflow_accession = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AnalysisSteps. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `analysis_step_types`) {
        # validate enum values
        if (!(query_item %in% c("alignment", "file format conversion", "signal generation"))) {
          stop("Invalid value for analysis_step_types when calling IgvfApi$AnalysisSteps. Must be [alignment, file format conversion, signal generation].")
        }
        query_params[["analysis_step_types"]] <- c(query_params[["analysis_step_types"]], list(`analysis_step_types` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$AnalysisSteps. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `input_content_types`) {
        query_params[["input_content_types"]] <- c(query_params[["input_content_types"]], list(`input_content_types` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `output_content_types`) {
        query_params[["output_content_types"]] <- c(query_params[["output_content_types"]], list(`output_content_types` = query_item))
      }

      # explore
      for (query_item in `parents_id`) {
        query_params[["parents.@id"]] <- c(query_params[["parents.@id"]], list(`parents.@id` = query_item))
      }

      # explore
      for (query_item in `parents_title`) {
        query_params[["parents.title"]] <- c(query_params[["parents.title"]], list(`parents.title` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$AnalysisSteps. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `step_label`) {
        query_params[["step_label"]] <- c(query_params[["step_label"]], list(`step_label` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `workflow_id`) {
        query_params[["workflow.@id"]] <- c(query_params[["workflow.@id"]], list(`workflow.@id` = query_item))
      }

      # explore
      for (query_item in `workflow_accession`) {
        query_params[["workflow.accession"]] <- c(query_params[["workflow.accession"]], list(`workflow.accession` = query_item))
      }

      local_var_url_path <- "/analysis-steps/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AnalysisStepResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the AssayTerm collection.
    #'
    #' @description
    #' List items in the AssayTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param assay_slims (optional) Filter by assay_slims
    #' @param category_slims (optional) Filter by category_slims
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param objective_slims (optional) Filter by objective_slims
    #' @param ontology (optional) Filter by ontology
    #' @param preferred_assay_titles (optional) Filter by preferred_assay_titles
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AssayTermResults
    #' @export
    AssayTerms = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, assay_slims = NULL, category_slims = NULL, creation_timestamp = NULL, deprecated_ntr_terms = NULL, description = NULL, is_a = NULL, name = NULL, notes = NULL, objective_slims = NULL, ontology = NULL, preferred_assay_titles = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$AssayTermsWithHttpInfo(frame, query, limit, sort, id, aliases, ancestors, assay_slims, category_slims, creation_timestamp, deprecated_ntr_terms, description, is_a, name, notes, objective_slims, ontology, preferred_assay_titles, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, synonyms, term_id, term_name, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AssayTerm collection.
    #'
    #' @description
    #' List items in the AssayTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param assay_slims (optional) Filter by assay_slims
    #' @param category_slims (optional) Filter by category_slims
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param objective_slims (optional) Filter by objective_slims
    #' @param ontology (optional) Filter by ontology
    #' @param preferred_assay_titles (optional) Filter by preferred_assay_titles
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AssayTermResults) with additional information such as HTTP status code, headers
    #' @export
    AssayTermsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, assay_slims = NULL, category_slims = NULL, creation_timestamp = NULL, deprecated_ntr_terms = NULL, description = NULL, is_a = NULL, name = NULL, notes = NULL, objective_slims = NULL, ontology = NULL, preferred_assay_titles = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AssayTerms. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `ancestors`) {
        query_params[["ancestors"]] <- c(query_params[["ancestors"]], list(`ancestors` = query_item))
      }

      # explore
      for (query_item in `assay_slims`) {
        query_params[["assay_slims"]] <- c(query_params[["assay_slims"]], list(`assay_slims` = query_item))
      }

      # explore
      for (query_item in `category_slims`) {
        query_params[["category_slims"]] <- c(query_params[["category_slims"]], list(`category_slims` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `deprecated_ntr_terms`) {
        query_params[["deprecated_ntr_terms"]] <- c(query_params[["deprecated_ntr_terms"]], list(`deprecated_ntr_terms` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `is_a`) {
        query_params[["is_a"]] <- c(query_params[["is_a"]], list(`is_a` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `objective_slims`) {
        query_params[["objective_slims"]] <- c(query_params[["objective_slims"]], list(`objective_slims` = query_item))
      }

      # explore
      for (query_item in `ontology`) {
        query_params[["ontology"]] <- c(query_params[["ontology"]], list(`ontology` = query_item))
      }

      # explore
      for (query_item in `preferred_assay_titles`) {
        # validate enum values
        if (!(query_item %in% c("10x multiome", "10x multiome with MULTI-seq", "AAV-MPRA", "ATAC-seq", "CERES-seq", "Cell painting", "CRISPR FlowFISH", "DOGMA-seq", "Histone ChIP-seq", "Hi-C", "lentiMPRA", "MERFISH", "MIAA", "mN2H", "MPRA", "MPRA (scQer)", "MULTI-seq", "Parse SPLiT-seq", "Perturb-seq", "RNA-seq", "SGE", "scATAC-seq", "scNT-seq", "scNT-seq2", "scRNA-seq", "semi-qY2H", "SHARE-seq", "smFISH", "snATAC-seq", "snmC-Seq2", "snMCT-seq", "snM3C-seq", "snRNA-seq", "SUPERSTARR", "TAP-seq", "TF ChIP-seq", "VAMP-seq", "Variant FlowFISH", "Variant painting", "Y2H", "yN2H"))) {
          stop("Invalid value for preferred_assay_titles when calling IgvfApi$AssayTerms. Must be [10x multiome, 10x multiome with MULTI-seq, AAV-MPRA, ATAC-seq, CERES-seq, Cell painting, CRISPR FlowFISH, DOGMA-seq, Histone ChIP-seq, Hi-C, lentiMPRA, MERFISH, MIAA, mN2H, MPRA, MPRA (scQer), MULTI-seq, Parse SPLiT-seq, Perturb-seq, RNA-seq, SGE, scATAC-seq, scNT-seq, scNT-seq2, scRNA-seq, semi-qY2H, SHARE-seq, smFISH, snATAC-seq, snmC-Seq2, snMCT-seq, snM3C-seq, snRNA-seq, SUPERSTARR, TAP-seq, TF ChIP-seq, VAMP-seq, Variant FlowFISH, Variant painting, Y2H, yN2H].")
        }
        query_params[["preferred_assay_titles"]] <- c(query_params[["preferred_assay_titles"]], list(`preferred_assay_titles` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$AssayTerms. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `synonyms`) {
        query_params[["synonyms"]] <- c(query_params[["synonyms"]], list(`synonyms` = query_item))
      }

      # explore
      for (query_item in `term_id`) {
        query_params[["term_id"]] <- c(query_params[["term_id"]], list(`term_id` = query_item))
      }

      # explore
      for (query_item in `term_name`) {
        query_params[["term_name"]] <- c(query_params[["term_name"]], list(`term_name` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/assay-terms/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AssayTermResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the AuxiliarySet collection.
    #'
    #' @description
    #' List items in the AuxiliarySet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param library_construction_platform (optional) Filter by library_construction_platform
    #' @param measurement_sets_id (optional) Filter by measurement_sets.@id
    #' @param measurement_sets_accession (optional) Filter by measurement_sets.accession
    #' @param measurement_sets_aliases (optional) Filter by measurement_sets.aliases
    #' @param measurement_sets_preferred_assay_title (optional) Filter by measurement_sets.preferred_assay_title
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AuxiliarySetResults
    #' @export
    AuxiliarySets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, lab_id = NULL, lab_title = NULL, library_construction_platform = NULL, measurement_sets_id = NULL, measurement_sets_accession = NULL, measurement_sets_aliases = NULL, measurement_sets_preferred_assay_title = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$AuxiliarySetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, award_id, award_component, award_contact_pi_id, award_contact_pi_title, award_title, collections, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, dbxrefs, description, documents, donors_id, donors_accession, donors_aliases, donors_sex, donors_status, donors_taxa, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_creation_timestamp, files_file_format, files_file_size, files_href, files_s3_uri, files_sequencing_platform, files_submitted_file_name, files_upload_status, input_file_set_for, lab_id, lab_title, library_construction_platform, measurement_sets_id, measurement_sets_accession, measurement_sets_aliases, measurement_sets_preferred_assay_title, notes, publication_identifiers, publications, release_timestamp, revoke_detail, samples_id, samples_accession, samples_aliases, samples_cell_fate_change_treatments, samples_classifications, samples_construct_library_sets, samples_disease_terms_id, samples_disease_terms_term_name, samples_modifications, samples_sample_terms_id, samples_sample_terms_aliases, samples_sample_terms_status, samples_sample_terms_summary, samples_sample_terms_term_name, samples_status, samples_summary, samples_targeted_sample_term_id, samples_targeted_sample_term_term_name, samples_taxa, samples_treatments, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, url, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the AuxiliarySet collection.
    #'
    #' @description
    #' List items in the AuxiliarySet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param library_construction_platform (optional) Filter by library_construction_platform
    #' @param measurement_sets_id (optional) Filter by measurement_sets.@id
    #' @param measurement_sets_accession (optional) Filter by measurement_sets.accession
    #' @param measurement_sets_aliases (optional) Filter by measurement_sets.aliases
    #' @param measurement_sets_preferred_assay_title (optional) Filter by measurement_sets.preferred_assay_title
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AuxiliarySetResults) with additional information such as HTTP status code, headers
    #' @export
    AuxiliarySetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, lab_id = NULL, lab_title = NULL, library_construction_platform = NULL, measurement_sets_id = NULL, measurement_sets_accession = NULL, measurement_sets_aliases = NULL, measurement_sets_preferred_assay_title = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }


























































      if (length(`samples_classifications`) > 2) {
        stop("Invalid length for `samples_classifications` when calling IgvfApi$AuxiliarySets, number of items must be less than or equal to 2.")
      }




      if (length(`samples_modifications`) > 2) {
        stop("Invalid length for `samples_modifications` when calling IgvfApi$AuxiliarySets, number of items must be less than or equal to 2.")
      }




















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$AuxiliarySets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$AuxiliarySets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_id`) {
        query_params[["award.contact_pi.@id"]] <- c(query_params[["award.contact_pi.@id"]], list(`award.contact_pi.@id` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_title`) {
        query_params[["award.contact_pi.title"]] <- c(query_params[["award.contact_pi.title"]], list(`award.contact_pi.title` = query_item))
      }

      # explore
      for (query_item in `award_title`) {
        query_params[["award.title"]] <- c(query_params[["award.title"]], list(`award.title` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$AuxiliarySets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors_id`) {
        query_params[["donors.@id"]] <- c(query_params[["donors.@id"]], list(`donors.@id` = query_item))
      }

      # explore
      for (query_item in `donors_accession`) {
        query_params[["donors.accession"]] <- c(query_params[["donors.accession"]], list(`donors.accession` = query_item))
      }

      # explore
      for (query_item in `donors_aliases`) {
        query_params[["donors.aliases"]] <- c(query_params[["donors.aliases"]], list(`donors.aliases` = query_item))
      }

      # explore
      for (query_item in `donors_sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for donors_sex when calling IgvfApi$AuxiliarySets. Must be [male, female, unspecified].")
        }
        query_params[["donors.sex"]] <- c(query_params[["donors.sex"]], list(`donors.sex` = query_item))
      }

      # explore
      for (query_item in `donors_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for donors_status when calling IgvfApi$AuxiliarySets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["donors.status"]] <- c(query_params[["donors.status"]], list(`donors.status` = query_item))
      }

      # explore
      for (query_item in `donors_taxa`) {
        query_params[["donors.taxa"]] <- c(query_params[["donors.taxa"]], list(`donors.taxa` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("cell hashing", "cell sorting", "circularized RNA barcode detection", "gRNA sequencing", "oligo-conjugated lipids", "quantification DNA barcode sequencing", "variant sequencing"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$AuxiliarySets. Must be [cell hashing, cell sorting, circularized RNA barcode detection, gRNA sequencing, oligo-conjugated lipids, quantification DNA barcode sequencing, variant sequencing].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_creation_timestamp`) {
        query_params[["files.creation_timestamp"]] <- c(query_params[["files.creation_timestamp"]], list(`files.creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_file_size`) {
        query_params[["files.file_size"]] <- c(query_params[["files.file_size"]], list(`files.file_size` = query_item))
      }

      # explore
      for (query_item in `files_href`) {
        query_params[["files.href"]] <- c(query_params[["files.href"]], list(`files.href` = query_item))
      }

      # explore
      for (query_item in `files_s3_uri`) {
        query_params[["files.s3_uri"]] <- c(query_params[["files.s3_uri"]], list(`files.s3_uri` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform`) {
        query_params[["files.sequencing_platform"]] <- c(query_params[["files.sequencing_platform"]], list(`files.sequencing_platform` = query_item))
      }

      # explore
      for (query_item in `files_submitted_file_name`) {
        query_params[["files.submitted_file_name"]] <- c(query_params[["files.submitted_file_name"]], list(`files.submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$AuxiliarySets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `library_construction_platform`) {
        query_params[["library_construction_platform"]] <- c(query_params[["library_construction_platform"]], list(`library_construction_platform` = query_item))
      }

      # explore
      for (query_item in `measurement_sets_id`) {
        query_params[["measurement_sets.@id"]] <- c(query_params[["measurement_sets.@id"]], list(`measurement_sets.@id` = query_item))
      }

      # explore
      for (query_item in `measurement_sets_accession`) {
        query_params[["measurement_sets.accession"]] <- c(query_params[["measurement_sets.accession"]], list(`measurement_sets.accession` = query_item))
      }

      # explore
      for (query_item in `measurement_sets_aliases`) {
        query_params[["measurement_sets.aliases"]] <- c(query_params[["measurement_sets.aliases"]], list(`measurement_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `measurement_sets_preferred_assay_title`) {
        # validate enum values
        if (!(query_item %in% c("10x multiome", "10x multiome with MULTI-seq", "AAV-MPRA", "ATAC-seq", "CERES-seq", "Cell painting", "CRISPR FlowFISH", "DOGMA-seq", "Histone ChIP-seq", "Hi-C", "lentiMPRA", "MERFISH", "MIAA", "mN2H", "MPRA", "MPRA (scQer)", "MULTI-seq", "Parse SPLiT-seq", "Perturb-seq", "RNA-seq", "SGE", "scATAC-seq", "scNT-seq", "scNT-seq2", "scRNA-seq", "semi-qY2H", "SHARE-seq", "smFISH", "snATAC-seq", "snmC-Seq2", "snMCT-seq", "snM3C-seq", "snRNA-seq", "SUPERSTARR", "TAP-seq", "TF ChIP-seq", "VAMP-seq", "Variant FlowFISH", "Variant painting", "Y2H", "yN2H"))) {
          stop("Invalid value for measurement_sets_preferred_assay_title when calling IgvfApi$AuxiliarySets. Must be [10x multiome, 10x multiome with MULTI-seq, AAV-MPRA, ATAC-seq, CERES-seq, Cell painting, CRISPR FlowFISH, DOGMA-seq, Histone ChIP-seq, Hi-C, lentiMPRA, MERFISH, MIAA, mN2H, MPRA, MPRA (scQer), MULTI-seq, Parse SPLiT-seq, Perturb-seq, RNA-seq, SGE, scATAC-seq, scNT-seq, scNT-seq2, scRNA-seq, semi-qY2H, SHARE-seq, smFISH, snATAC-seq, snmC-Seq2, snMCT-seq, snM3C-seq, snRNA-seq, SUPERSTARR, TAP-seq, TF ChIP-seq, VAMP-seq, Variant FlowFISH, Variant painting, Y2H, yN2H].")
        }
        query_params[["measurement_sets.preferred_assay_title"]] <- c(query_params[["measurement_sets.preferred_assay_title"]], list(`measurement_sets.preferred_assay_title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `samples_id`) {
        query_params[["samples.@id"]] <- c(query_params[["samples.@id"]], list(`samples.@id` = query_item))
      }

      # explore
      for (query_item in `samples_accession`) {
        query_params[["samples.accession"]] <- c(query_params[["samples.accession"]], list(`samples.accession` = query_item))
      }

      # explore
      for (query_item in `samples_aliases`) {
        query_params[["samples.aliases"]] <- c(query_params[["samples.aliases"]], list(`samples.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments`) {
        query_params[["samples.cell_fate_change_treatments"]] <- c(query_params[["samples.cell_fate_change_treatments"]], list(`samples.cell_fate_change_treatments` = query_item))
      }

      # explore
      for (query_item in `samples_classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for samples_classifications when calling IgvfApi$AuxiliarySets. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["samples.classifications"]] <- c(query_params[["samples.classifications"]], list(`samples.classifications` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets`) {
        query_params[["samples.construct_library_sets"]] <- c(query_params[["samples.construct_library_sets"]], list(`samples.construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_id`) {
        query_params[["samples.disease_terms.@id"]] <- c(query_params[["samples.disease_terms.@id"]], list(`samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_term_name`) {
        query_params[["samples.disease_terms.term_name"]] <- c(query_params[["samples.disease_terms.term_name"]], list(`samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_modifications`) {
        query_params[["samples.modifications"]] <- c(query_params[["samples.modifications"]], list(`samples.modifications` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_id`) {
        query_params[["samples.sample_terms.@id"]] <- c(query_params[["samples.sample_terms.@id"]], list(`samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_aliases`) {
        query_params[["samples.sample_terms.aliases"]] <- c(query_params[["samples.sample_terms.aliases"]], list(`samples.sample_terms.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for samples_sample_terms_status when calling IgvfApi$AuxiliarySets. Must be [archived, deleted, in progress, released].")
        }
        query_params[["samples.sample_terms.status"]] <- c(query_params[["samples.sample_terms.status"]], list(`samples.sample_terms.status` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_summary`) {
        query_params[["samples.sample_terms.summary"]] <- c(query_params[["samples.sample_terms.summary"]], list(`samples.sample_terms.summary` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_term_name`) {
        query_params[["samples.sample_terms.term_name"]] <- c(query_params[["samples.sample_terms.term_name"]], list(`samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for samples_status when calling IgvfApi$AuxiliarySets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["samples.status"]] <- c(query_params[["samples.status"]], list(`samples.status` = query_item))
      }

      # explore
      for (query_item in `samples_summary`) {
        query_params[["samples.summary"]] <- c(query_params[["samples.summary"]], list(`samples.summary` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_id`) {
        query_params[["samples.targeted_sample_term.@id"]] <- c(query_params[["samples.targeted_sample_term.@id"]], list(`samples.targeted_sample_term.@id` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_term_name`) {
        query_params[["samples.targeted_sample_term.term_name"]] <- c(query_params[["samples.targeted_sample_term.term_name"]], list(`samples.targeted_sample_term.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for samples_taxa when calling IgvfApi$AuxiliarySets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["samples.taxa"]] <- c(query_params[["samples.taxa"]], list(`samples.taxa` = query_item))
      }

      # explore
      for (query_item in `samples_treatments`) {
        query_params[["samples.treatments"]] <- c(query_params[["samples.treatments"]], list(`samples.treatments` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$AuxiliarySets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/auxiliary-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AuxiliarySetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Award collection.
    #'
    #' @description
    #' List items in the Award collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param component (optional) Filter by component
    #' @param contact_pi (optional) Filter by contact_pi
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param end_date (optional) Filter by end_date
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param pis (optional) Filter by pis
    #' @param project (optional) Filter by project
    #' @param start_date (optional) Filter by start_date
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param viewing_group (optional) Filter by viewing_group
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return AwardResults
    #' @export
    Awards = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, component = NULL, contact_pi = NULL, creation_timestamp = NULL, description = NULL, end_date = NULL, name = NULL, notes = NULL, pis = NULL, project = NULL, start_date = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, url = NULL, uuid = NULL, viewing_group = NULL, data_file = NULL, ...) {
      local_var_response <- self$AwardsWithHttpInfo(frame, query, limit, sort, id, aliases, component, contact_pi, creation_timestamp, description, end_date, name, notes, pis, project, start_date, status, submitted_by_id, submitted_by_title, submitter_comment, summary, title, url, uuid, viewing_group, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Award collection.
    #'
    #' @description
    #' List items in the Award collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param component (optional) Filter by component
    #' @param contact_pi (optional) Filter by contact_pi
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param end_date (optional) Filter by end_date
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param pis (optional) Filter by pis
    #' @param project (optional) Filter by project
    #' @param start_date (optional) Filter by start_date
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param viewing_group (optional) Filter by viewing_group
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (AwardResults) with additional information such as HTTP status code, headers
    #' @export
    AwardsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, component = NULL, contact_pi = NULL, creation_timestamp = NULL, description = NULL, end_date = NULL, name = NULL, notes = NULL, pis = NULL, project = NULL, start_date = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, url = NULL, uuid = NULL, viewing_group = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }


























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Awards. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for component when calling IgvfApi$Awards. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["component"]] <- c(query_params[["component"]], list(`component` = query_item))
      }

      # explore
      for (query_item in `contact_pi`) {
        query_params[["contact_pi"]] <- c(query_params[["contact_pi"]], list(`contact_pi` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `end_date`) {
        query_params[["end_date"]] <- c(query_params[["end_date"]], list(`end_date` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `pis`) {
        query_params[["pis"]] <- c(query_params[["pis"]], list(`pis` = query_item))
      }

      # explore
      for (query_item in `project`) {
        # validate enum values
        if (!(query_item %in% c("community", "ENCODE", "IGVF"))) {
          stop("Invalid value for project when calling IgvfApi$Awards. Must be [community, ENCODE, IGVF].")
        }
        query_params[["project"]] <- c(query_params[["project"]], list(`project` = query_item))
      }

      # explore
      for (query_item in `start_date`) {
        query_params[["start_date"]] <- c(query_params[["start_date"]], list(`start_date` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("current", "deleted", "disabled"))) {
          stop("Invalid value for status when calling IgvfApi$Awards. Must be [current, deleted, disabled].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `viewing_group`) {
        # validate enum values
        if (!(query_item %in% c("community", "IGVF"))) {
          stop("Invalid value for viewing_group when calling IgvfApi$Awards. Must be [community, IGVF].")
        }
        query_params[["viewing_group"]] <- c(query_params[["viewing_group"]], list(`viewing_group` = query_item))
      }

      local_var_url_path <- "/awards/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "AwardResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List files to download based on search query. All results are returned.
    #'
    #' @description
    #' List files to download based on search query. All results are returned.
    #'
    #' @param type Type of objects to return. Can be repeated for multiple types.
    #' @param query (optional) Query string for searching.
    #' @param field_filters (optional) Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return character
    #' @export
    BatchDownload = function(type, query = NULL, field_filters = NULL, data_file = NULL, ...) {
      local_var_response <- self$BatchDownloadWithHttpInfo(type, query, field_filters, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List files to download based on search query. All results are returned.
    #'
    #' @description
    #' List files to download based on search query. All results are returned.
    #'
    #' @param type Type of objects to return. Can be repeated for multiple types.
    #' @param query (optional) Query string for searching.
    #' @param field_filters (optional) Any field from any object type can be used as a filter. Use '!' for negation, '*' as a wildcard, and 'lt:', 'lte:', 'gt:', 'gte:' for range queries on numeric fields.
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (character) with additional information such as HTTP status code, headers
    #' @export
    BatchDownloadWithHttpInfo = function(type, query = NULL, field_filters = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`type`)) {
        stop("Missing required parameter `type`.")
      }




      # explore
      for (query_item in `type`) {
        query_params[["type"]] <- c(query_params[["type"]], list(`type` = query_item))
      }

      query_params[["query"]] <- `query`

      for (i in seq_along(`field_filters`)) {
          k <- names(`field_filters`)[[i]]
          v <- `field_filters`[[i]]
          query_params[[k]] <- v
      }

      local_var_url_path <- "/batch-download"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values", "application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "character", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Biomarker collection.
    #'
    #' @description
    #' List items in the Biomarker collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_name (optional) Filter by award.name
    #' @param biomarker_for (optional) Filter by biomarker_for
    #' @param classification (optional) Filter by classification
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param gene (optional) Filter by gene
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param name_quantification (optional) Filter by name_quantification
    #' @param notes (optional) Filter by notes
    #' @param quantification (optional) Filter by quantification
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return BiomarkerResults
    #' @export
    Biomarkers = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, award_name = NULL, biomarker_for = NULL, classification = NULL, creation_timestamp = NULL, description = NULL, gene = NULL, lab_id = NULL, lab_title = NULL, name = NULL, name_quantification = NULL, notes = NULL, quantification = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$BiomarkersWithHttpInfo(frame, query, limit, sort, id, aliases, award_id, award_component, award_name, biomarker_for, classification, creation_timestamp, description, gene, lab_id, lab_title, name, name_quantification, notes, quantification, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, synonyms, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Biomarker collection.
    #'
    #' @description
    #' List items in the Biomarker collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_name (optional) Filter by award.name
    #' @param biomarker_for (optional) Filter by biomarker_for
    #' @param classification (optional) Filter by classification
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param gene (optional) Filter by gene
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param name_quantification (optional) Filter by name_quantification
    #' @param notes (optional) Filter by notes
    #' @param quantification (optional) Filter by quantification
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (BiomarkerResults) with additional information such as HTTP status code, headers
    #' @export
    BiomarkersWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, award_name = NULL, biomarker_for = NULL, classification = NULL, creation_timestamp = NULL, description = NULL, gene = NULL, lab_id = NULL, lab_title = NULL, name = NULL, name_quantification = NULL, notes = NULL, quantification = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Biomarkers. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Biomarkers. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_name`) {
        query_params[["award.name"]] <- c(query_params[["award.name"]], list(`award.name` = query_item))
      }

      # explore
      for (query_item in `biomarker_for`) {
        query_params[["biomarker_for"]] <- c(query_params[["biomarker_for"]], list(`biomarker_for` = query_item))
      }

      # explore
      for (query_item in `classification`) {
        # validate enum values
        if (!(query_item %in% c("cell surface protein", "marker gene"))) {
          stop("Invalid value for classification when calling IgvfApi$Biomarkers. Must be [cell surface protein, marker gene].")
        }
        query_params[["classification"]] <- c(query_params[["classification"]], list(`classification` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `gene`) {
        query_params[["gene"]] <- c(query_params[["gene"]], list(`gene` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `name_quantification`) {
        query_params[["name_quantification"]] <- c(query_params[["name_quantification"]], list(`name_quantification` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `quantification`) {
        # validate enum values
        if (!(query_item %in% c("negative", "positive", "low", "intermediate", "high"))) {
          stop("Invalid value for quantification when calling IgvfApi$Biomarkers. Must be [negative, positive, low, intermediate, high].")
        }
        query_params[["quantification"]] <- c(query_params[["quantification"]], list(`quantification` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Biomarkers. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `synonyms`) {
        query_params[["synonyms"]] <- c(query_params[["synonyms"]], list(`synonyms` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/biomarkers/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "BiomarkerResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the ConfigurationFile collection.
    #'
    #' @description
    #' List items in the ConfigurationFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param seqspec_of (optional) Filter by seqspec_of
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ConfigurationFileResults
    #' @export
    ConfigurationFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, seqspec_of = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$ConfigurationFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, award_id, award_component, collections, content_md5sum, content_type, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, release_timestamp, revoke_detail, s3_uri, seqspec_of, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the ConfigurationFile collection.
    #'
    #' @description
    #' List items in the ConfigurationFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param seqspec_of (optional) Filter by seqspec_of
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ConfigurationFileResults) with additional information such as HTTP status code, headers
    #' @export
    ConfigurationFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, seqspec_of = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }














































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$ConfigurationFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$ConfigurationFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$ConfigurationFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("yaml"))) {
          stop("Invalid value for file_format when calling IgvfApi$ConfigurationFiles. Must be [yaml].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `seqspec_of`) {
        query_params[["seqspec_of"]] <- c(query_params[["seqspec_of"]], list(`seqspec_of` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$ConfigurationFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$ConfigurationFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/configuration-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ConfigurationFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the ConstructLibrarySet collection.
    #'
    #' @description
    #' List items in the ConstructLibrarySet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param applied_to_samples_id (optional) Filter by applied_to_samples.@id
    #' @param applied_to_samples_accession (optional) Filter by applied_to_samples.accession
    #' @param applied_to_samples_aliases (optional) Filter by applied_to_samples.aliases
    #' @param applied_to_samples_disease_terms_id (optional) Filter by applied_to_samples.disease_terms.@id
    #' @param applied_to_samples_disease_terms_term_name (optional) Filter by applied_to_samples.disease_terms.term_name
    #' @param applied_to_samples_sample_terms_id (optional) Filter by applied_to_samples.sample_terms.@id
    #' @param applied_to_samples_sample_terms_term_name (optional) Filter by applied_to_samples.sample_terms.term_name
    #' @param applied_to_samples_status (optional) Filter by applied_to_samples.status
    #' @param applied_to_samples_summary (optional) Filter by applied_to_samples.summary
    #' @param associated_phenotypes_id (optional) Filter by associated_phenotypes.@id
    #' @param associated_phenotypes_term_id (optional) Filter by associated_phenotypes.term_id
    #' @param associated_phenotypes_term_name (optional) Filter by associated_phenotypes.term_name
    #' @param average_guide_coverage (optional) Filter by average_guide_coverage
    #' @param average_insert_size (optional) Filter by average_insert_size
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param exon (optional) Filter by exon
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param guide_type (optional) Filter by guide_type
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param integrated_content_files (optional) Filter by integrated_content_files
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param large_scale_gene_list_id (optional) Filter by large_scale_gene_list.@id
    #' @param large_scale_gene_list_accession (optional) Filter by large_scale_gene_list.accession
    #' @param large_scale_gene_list_aliases (optional) Filter by large_scale_gene_list.aliases
    #' @param large_scale_loci_list_id (optional) Filter by large_scale_loci_list.@id
    #' @param large_scale_loci_list_accession (optional) Filter by large_scale_loci_list.accession
    #' @param large_scale_loci_list_aliases (optional) Filter by large_scale_loci_list.aliases
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_guide_coverage (optional) Filter by lower_bound_guide_coverage
    #' @param lower_bound_insert_size (optional) Filter by lower_bound_insert_size
    #' @param notes (optional) Filter by notes
    #' @param orf_list_id (optional) Filter by orf_list.@id
    #' @param orf_list_aliases (optional) Filter by orf_list.aliases
    #' @param orf_list_gene (optional) Filter by orf_list.gene
    #' @param orf_list_orf_id (optional) Filter by orf_list.orf_id
    #' @param product_id (optional) Filter by product_id
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param scope (optional) Filter by scope
    #' @param selection_criteria (optional) Filter by selection_criteria
    #' @param small_scale_gene_list_id (optional) Filter by small_scale_gene_list.@id
    #' @param small_scale_gene_list_geneid (optional) Filter by small_scale_gene_list.geneid
    #' @param small_scale_gene_list_name (optional) Filter by small_scale_gene_list.name
    #' @param small_scale_gene_list_symbol (optional) Filter by small_scale_gene_list.symbol
    #' @param small_scale_gene_list_synonyms (optional) Filter by small_scale_gene_list.synonyms
    #' @param small_scale_loci_list (optional) Filter by small_scale_loci_list
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param targeton (optional) Filter by targeton
    #' @param tiling_modality (optional) Filter by tiling_modality
    #' @param upper_bound_guide_coverage (optional) Filter by upper_bound_guide_coverage
    #' @param upper_bound_insert_size (optional) Filter by upper_bound_insert_size
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ConstructLibrarySetResults
    #' @export
    ConstructLibrarySets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, applied_to_samples_id = NULL, applied_to_samples_accession = NULL, applied_to_samples_aliases = NULL, applied_to_samples_disease_terms_id = NULL, applied_to_samples_disease_terms_term_name = NULL, applied_to_samples_sample_terms_id = NULL, applied_to_samples_sample_terms_term_name = NULL, applied_to_samples_status = NULL, applied_to_samples_summary = NULL, associated_phenotypes_id = NULL, associated_phenotypes_term_id = NULL, associated_phenotypes_term_name = NULL, average_guide_coverage = NULL, average_insert_size = NULL, award_id = NULL, award_component = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, description = NULL, documents = NULL, exon = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_file_format = NULL, files_upload_status = NULL, guide_type = NULL, input_file_set_for = NULL, integrated_content_files = NULL, lab_id = NULL, lab_title = NULL, large_scale_gene_list_id = NULL, large_scale_gene_list_accession = NULL, large_scale_gene_list_aliases = NULL, large_scale_loci_list_id = NULL, large_scale_loci_list_accession = NULL, large_scale_loci_list_aliases = NULL, lot_id = NULL, lower_bound_guide_coverage = NULL, lower_bound_insert_size = NULL, notes = NULL, orf_list_id = NULL, orf_list_aliases = NULL, orf_list_gene = NULL, orf_list_orf_id = NULL, product_id = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, scope = NULL, selection_criteria = NULL, small_scale_gene_list_id = NULL, small_scale_gene_list_geneid = NULL, small_scale_gene_list_name = NULL, small_scale_gene_list_symbol = NULL, small_scale_gene_list_synonyms = NULL, small_scale_loci_list = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, targeton = NULL, tiling_modality = NULL, upper_bound_guide_coverage = NULL, upper_bound_insert_size = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$ConstructLibrarySetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, applied_to_samples_id, applied_to_samples_accession, applied_to_samples_aliases, applied_to_samples_disease_terms_id, applied_to_samples_disease_terms_term_name, applied_to_samples_sample_terms_id, applied_to_samples_sample_terms_term_name, applied_to_samples_status, applied_to_samples_summary, associated_phenotypes_id, associated_phenotypes_term_id, associated_phenotypes_term_name, average_guide_coverage, average_insert_size, award_id, award_component, collections, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, description, documents, exon, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_file_format, files_upload_status, guide_type, input_file_set_for, integrated_content_files, lab_id, lab_title, large_scale_gene_list_id, large_scale_gene_list_accession, large_scale_gene_list_aliases, large_scale_loci_list_id, large_scale_loci_list_accession, large_scale_loci_list_aliases, lot_id, lower_bound_guide_coverage, lower_bound_insert_size, notes, orf_list_id, orf_list_aliases, orf_list_gene, orf_list_orf_id, product_id, publication_identifiers, publications, release_timestamp, revoke_detail, scope, selection_criteria, small_scale_gene_list_id, small_scale_gene_list_geneid, small_scale_gene_list_name, small_scale_gene_list_symbol, small_scale_gene_list_synonyms, small_scale_loci_list, sources, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, targeton, tiling_modality, upper_bound_guide_coverage, upper_bound_insert_size, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the ConstructLibrarySet collection.
    #'
    #' @description
    #' List items in the ConstructLibrarySet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param applied_to_samples_id (optional) Filter by applied_to_samples.@id
    #' @param applied_to_samples_accession (optional) Filter by applied_to_samples.accession
    #' @param applied_to_samples_aliases (optional) Filter by applied_to_samples.aliases
    #' @param applied_to_samples_disease_terms_id (optional) Filter by applied_to_samples.disease_terms.@id
    #' @param applied_to_samples_disease_terms_term_name (optional) Filter by applied_to_samples.disease_terms.term_name
    #' @param applied_to_samples_sample_terms_id (optional) Filter by applied_to_samples.sample_terms.@id
    #' @param applied_to_samples_sample_terms_term_name (optional) Filter by applied_to_samples.sample_terms.term_name
    #' @param applied_to_samples_status (optional) Filter by applied_to_samples.status
    #' @param applied_to_samples_summary (optional) Filter by applied_to_samples.summary
    #' @param associated_phenotypes_id (optional) Filter by associated_phenotypes.@id
    #' @param associated_phenotypes_term_id (optional) Filter by associated_phenotypes.term_id
    #' @param associated_phenotypes_term_name (optional) Filter by associated_phenotypes.term_name
    #' @param average_guide_coverage (optional) Filter by average_guide_coverage
    #' @param average_insert_size (optional) Filter by average_insert_size
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param exon (optional) Filter by exon
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param guide_type (optional) Filter by guide_type
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param integrated_content_files (optional) Filter by integrated_content_files
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param large_scale_gene_list_id (optional) Filter by large_scale_gene_list.@id
    #' @param large_scale_gene_list_accession (optional) Filter by large_scale_gene_list.accession
    #' @param large_scale_gene_list_aliases (optional) Filter by large_scale_gene_list.aliases
    #' @param large_scale_loci_list_id (optional) Filter by large_scale_loci_list.@id
    #' @param large_scale_loci_list_accession (optional) Filter by large_scale_loci_list.accession
    #' @param large_scale_loci_list_aliases (optional) Filter by large_scale_loci_list.aliases
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_guide_coverage (optional) Filter by lower_bound_guide_coverage
    #' @param lower_bound_insert_size (optional) Filter by lower_bound_insert_size
    #' @param notes (optional) Filter by notes
    #' @param orf_list_id (optional) Filter by orf_list.@id
    #' @param orf_list_aliases (optional) Filter by orf_list.aliases
    #' @param orf_list_gene (optional) Filter by orf_list.gene
    #' @param orf_list_orf_id (optional) Filter by orf_list.orf_id
    #' @param product_id (optional) Filter by product_id
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param scope (optional) Filter by scope
    #' @param selection_criteria (optional) Filter by selection_criteria
    #' @param small_scale_gene_list_id (optional) Filter by small_scale_gene_list.@id
    #' @param small_scale_gene_list_geneid (optional) Filter by small_scale_gene_list.geneid
    #' @param small_scale_gene_list_name (optional) Filter by small_scale_gene_list.name
    #' @param small_scale_gene_list_symbol (optional) Filter by small_scale_gene_list.symbol
    #' @param small_scale_gene_list_synonyms (optional) Filter by small_scale_gene_list.synonyms
    #' @param small_scale_loci_list (optional) Filter by small_scale_loci_list
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param targeton (optional) Filter by targeton
    #' @param tiling_modality (optional) Filter by tiling_modality
    #' @param upper_bound_guide_coverage (optional) Filter by upper_bound_guide_coverage
    #' @param upper_bound_insert_size (optional) Filter by upper_bound_insert_size
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ConstructLibrarySetResults) with additional information such as HTTP status code, headers
    #' @export
    ConstructLibrarySetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, applied_to_samples_id = NULL, applied_to_samples_accession = NULL, applied_to_samples_aliases = NULL, applied_to_samples_disease_terms_id = NULL, applied_to_samples_disease_terms_term_name = NULL, applied_to_samples_sample_terms_id = NULL, applied_to_samples_sample_terms_term_name = NULL, applied_to_samples_status = NULL, applied_to_samples_summary = NULL, associated_phenotypes_id = NULL, associated_phenotypes_term_id = NULL, associated_phenotypes_term_name = NULL, average_guide_coverage = NULL, average_insert_size = NULL, award_id = NULL, award_component = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, description = NULL, documents = NULL, exon = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_file_format = NULL, files_upload_status = NULL, guide_type = NULL, input_file_set_for = NULL, integrated_content_files = NULL, lab_id = NULL, lab_title = NULL, large_scale_gene_list_id = NULL, large_scale_gene_list_accession = NULL, large_scale_gene_list_aliases = NULL, large_scale_loci_list_id = NULL, large_scale_loci_list_accession = NULL, large_scale_loci_list_aliases = NULL, lot_id = NULL, lower_bound_guide_coverage = NULL, lower_bound_insert_size = NULL, notes = NULL, orf_list_id = NULL, orf_list_aliases = NULL, orf_list_gene = NULL, orf_list_orf_id = NULL, product_id = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, scope = NULL, selection_criteria = NULL, small_scale_gene_list_id = NULL, small_scale_gene_list_geneid = NULL, small_scale_gene_list_name = NULL, small_scale_gene_list_symbol = NULL, small_scale_gene_list_synonyms = NULL, small_scale_loci_list = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, targeton = NULL, tiling_modality = NULL, upper_bound_guide_coverage = NULL, upper_bound_insert_size = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }




















































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$ConstructLibrarySets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_id`) {
        query_params[["applied_to_samples.@id"]] <- c(query_params[["applied_to_samples.@id"]], list(`applied_to_samples.@id` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_accession`) {
        query_params[["applied_to_samples.accession"]] <- c(query_params[["applied_to_samples.accession"]], list(`applied_to_samples.accession` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_aliases`) {
        query_params[["applied_to_samples.aliases"]] <- c(query_params[["applied_to_samples.aliases"]], list(`applied_to_samples.aliases` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_disease_terms_id`) {
        query_params[["applied_to_samples.disease_terms.@id"]] <- c(query_params[["applied_to_samples.disease_terms.@id"]], list(`applied_to_samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_disease_terms_term_name`) {
        query_params[["applied_to_samples.disease_terms.term_name"]] <- c(query_params[["applied_to_samples.disease_terms.term_name"]], list(`applied_to_samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_sample_terms_id`) {
        query_params[["applied_to_samples.sample_terms.@id"]] <- c(query_params[["applied_to_samples.sample_terms.@id"]], list(`applied_to_samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_sample_terms_term_name`) {
        query_params[["applied_to_samples.sample_terms.term_name"]] <- c(query_params[["applied_to_samples.sample_terms.term_name"]], list(`applied_to_samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for applied_to_samples_status when calling IgvfApi$ConstructLibrarySets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["applied_to_samples.status"]] <- c(query_params[["applied_to_samples.status"]], list(`applied_to_samples.status` = query_item))
      }

      # explore
      for (query_item in `applied_to_samples_summary`) {
        query_params[["applied_to_samples.summary"]] <- c(query_params[["applied_to_samples.summary"]], list(`applied_to_samples.summary` = query_item))
      }

      # explore
      for (query_item in `associated_phenotypes_id`) {
        query_params[["associated_phenotypes.@id"]] <- c(query_params[["associated_phenotypes.@id"]], list(`associated_phenotypes.@id` = query_item))
      }

      # explore
      for (query_item in `associated_phenotypes_term_id`) {
        query_params[["associated_phenotypes.term_id"]] <- c(query_params[["associated_phenotypes.term_id"]], list(`associated_phenotypes.term_id` = query_item))
      }

      # explore
      for (query_item in `associated_phenotypes_term_name`) {
        query_params[["associated_phenotypes.term_name"]] <- c(query_params[["associated_phenotypes.term_name"]], list(`associated_phenotypes.term_name` = query_item))
      }

      # explore
      for (query_item in `average_guide_coverage`) {
        query_params[["average_guide_coverage"]] <- c(query_params[["average_guide_coverage"]], list(`average_guide_coverage` = query_item))
      }

      # explore
      for (query_item in `average_insert_size`) {
        query_params[["average_insert_size"]] <- c(query_params[["average_insert_size"]], list(`average_insert_size` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$ConstructLibrarySets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$ConstructLibrarySets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `exon`) {
        query_params[["exon"]] <- c(query_params[["exon"]], list(`exon` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("guide library", "reporter library", "expression vector library", "editing template library"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$ConstructLibrarySets. Must be [guide library, reporter library, expression vector library, editing template library].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$ConstructLibrarySets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `guide_type`) {
        # validate enum values
        if (!(query_item %in% c("sgRNA", "pgRNA"))) {
          stop("Invalid value for guide_type when calling IgvfApi$ConstructLibrarySets. Must be [sgRNA, pgRNA].")
        }
        query_params[["guide_type"]] <- c(query_params[["guide_type"]], list(`guide_type` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `integrated_content_files`) {
        query_params[["integrated_content_files"]] <- c(query_params[["integrated_content_files"]], list(`integrated_content_files` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `large_scale_gene_list_id`) {
        query_params[["large_scale_gene_list.@id"]] <- c(query_params[["large_scale_gene_list.@id"]], list(`large_scale_gene_list.@id` = query_item))
      }

      # explore
      for (query_item in `large_scale_gene_list_accession`) {
        query_params[["large_scale_gene_list.accession"]] <- c(query_params[["large_scale_gene_list.accession"]], list(`large_scale_gene_list.accession` = query_item))
      }

      # explore
      for (query_item in `large_scale_gene_list_aliases`) {
        query_params[["large_scale_gene_list.aliases"]] <- c(query_params[["large_scale_gene_list.aliases"]], list(`large_scale_gene_list.aliases` = query_item))
      }

      # explore
      for (query_item in `large_scale_loci_list_id`) {
        query_params[["large_scale_loci_list.@id"]] <- c(query_params[["large_scale_loci_list.@id"]], list(`large_scale_loci_list.@id` = query_item))
      }

      # explore
      for (query_item in `large_scale_loci_list_accession`) {
        query_params[["large_scale_loci_list.accession"]] <- c(query_params[["large_scale_loci_list.accession"]], list(`large_scale_loci_list.accession` = query_item))
      }

      # explore
      for (query_item in `large_scale_loci_list_aliases`) {
        query_params[["large_scale_loci_list.aliases"]] <- c(query_params[["large_scale_loci_list.aliases"]], list(`large_scale_loci_list.aliases` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `lower_bound_guide_coverage`) {
        query_params[["lower_bound_guide_coverage"]] <- c(query_params[["lower_bound_guide_coverage"]], list(`lower_bound_guide_coverage` = query_item))
      }

      # explore
      for (query_item in `lower_bound_insert_size`) {
        query_params[["lower_bound_insert_size"]] <- c(query_params[["lower_bound_insert_size"]], list(`lower_bound_insert_size` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `orf_list_id`) {
        query_params[["orf_list.@id"]] <- c(query_params[["orf_list.@id"]], list(`orf_list.@id` = query_item))
      }

      # explore
      for (query_item in `orf_list_aliases`) {
        query_params[["orf_list.aliases"]] <- c(query_params[["orf_list.aliases"]], list(`orf_list.aliases` = query_item))
      }

      # explore
      for (query_item in `orf_list_gene`) {
        query_params[["orf_list.gene"]] <- c(query_params[["orf_list.gene"]], list(`orf_list.gene` = query_item))
      }

      # explore
      for (query_item in `orf_list_orf_id`) {
        query_params[["orf_list.orf_id"]] <- c(query_params[["orf_list.orf_id"]], list(`orf_list.orf_id` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `scope`) {
        # validate enum values
        if (!(query_item %in% c("tile", "exon", "genes", "loci", "genome-wide", "interactors", "alleles", "targeton"))) {
          stop("Invalid value for scope when calling IgvfApi$ConstructLibrarySets. Must be [tile, exon, genes, loci, genome-wide, interactors, alleles, targeton].")
        }
        query_params[["scope"]] <- c(query_params[["scope"]], list(`scope` = query_item))
      }

      # explore
      for (query_item in `selection_criteria`) {
        # validate enum values
        if (!(query_item %in% c("accessible genome regions", "candidate cis-regulatory elements", "chromatin states", "phenotype-associated variants", "DNase hypersensitive sites", "genes", "histone modifications", "protein interactors", "sequence variants", "synthetic elements", "transcription start sites", "TF binding sites"))) {
          stop("Invalid value for selection_criteria when calling IgvfApi$ConstructLibrarySets. Must be [accessible genome regions, candidate cis-regulatory elements, chromatin states, phenotype-associated variants, DNase hypersensitive sites, genes, histone modifications, protein interactors, sequence variants, synthetic elements, transcription start sites, TF binding sites].")
        }
        query_params[["selection_criteria"]] <- c(query_params[["selection_criteria"]], list(`selection_criteria` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_id`) {
        query_params[["small_scale_gene_list.@id"]] <- c(query_params[["small_scale_gene_list.@id"]], list(`small_scale_gene_list.@id` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_geneid`) {
        query_params[["small_scale_gene_list.geneid"]] <- c(query_params[["small_scale_gene_list.geneid"]], list(`small_scale_gene_list.geneid` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_name`) {
        query_params[["small_scale_gene_list.name"]] <- c(query_params[["small_scale_gene_list.name"]], list(`small_scale_gene_list.name` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_symbol`) {
        query_params[["small_scale_gene_list.symbol"]] <- c(query_params[["small_scale_gene_list.symbol"]], list(`small_scale_gene_list.symbol` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_synonyms`) {
        query_params[["small_scale_gene_list.synonyms"]] <- c(query_params[["small_scale_gene_list.synonyms"]], list(`small_scale_gene_list.synonyms` = query_item))
      }

      # explore
      for (query_item in `small_scale_loci_list`) {
        query_params[["small_scale_loci_list"]] <- c(query_params[["small_scale_loci_list"]], list(`small_scale_loci_list` = query_item))
      }

      # explore
      for (query_item in `sources`) {
        query_params[["sources"]] <- c(query_params[["sources"]], list(`sources` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$ConstructLibrarySets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `targeton`) {
        query_params[["targeton"]] <- c(query_params[["targeton"]], list(`targeton` = query_item))
      }

      # explore
      for (query_item in `tiling_modality`) {
        # validate enum values
        if (!(query_item %in% c("peak tiling", "full tiling", "sparse peaks"))) {
          stop("Invalid value for tiling_modality when calling IgvfApi$ConstructLibrarySets. Must be [peak tiling, full tiling, sparse peaks].")
        }
        query_params[["tiling_modality"]] <- c(query_params[["tiling_modality"]], list(`tiling_modality` = query_item))
      }

      # explore
      for (query_item in `upper_bound_guide_coverage`) {
        query_params[["upper_bound_guide_coverage"]] <- c(query_params[["upper_bound_guide_coverage"]], list(`upper_bound_guide_coverage` = query_item))
      }

      # explore
      for (query_item in `upper_bound_insert_size`) {
        query_params[["upper_bound_insert_size"]] <- c(query_params[["upper_bound_insert_size"]], list(`upper_bound_insert_size` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/construct-library-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ConstructLibrarySetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the CrisprModification collection.
    #'
    #' @description
    #' List items in the CrisprModification collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param activated (optional) Filter by activated
    #' @param activating_agent_term_id (optional) Filter by activating_agent_term_id
    #' @param activating_agent_term_name (optional) Filter by activating_agent_term_name
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biosamples_modified (optional) Filter by biosamples_modified
    #' @param cas (optional) Filter by cas
    #' @param cas_species (optional) Filter by cas_species
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param fused_domain (optional) Filter by fused_domain
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param modality (optional) Filter by modality
    #' @param notes (optional) Filter by notes
    #' @param product_id (optional) Filter by product_id
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param tagged_protein (optional) Filter by tagged_protein
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return CrisprModificationResults
    #' @export
    CrisprModifications = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, activated = NULL, activating_agent_term_id = NULL, activating_agent_term_name = NULL, aliases = NULL, award_id = NULL, award_component = NULL, biosamples_modified = NULL, cas = NULL, cas_species = NULL, creation_timestamp = NULL, description = NULL, documents = NULL, fused_domain = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, modality = NULL, notes = NULL, product_id = NULL, release_timestamp = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, tagged_protein = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$CrisprModificationsWithHttpInfo(frame, query, limit, sort, id, activated, activating_agent_term_id, activating_agent_term_name, aliases, award_id, award_component, biosamples_modified, cas, cas_species, creation_timestamp, description, documents, fused_domain, lab_id, lab_title, lot_id, modality, notes, product_id, release_timestamp, sources, status, submitted_by_id, submitted_by_title, submitter_comment, summary, tagged_protein, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the CrisprModification collection.
    #'
    #' @description
    #' List items in the CrisprModification collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param activated (optional) Filter by activated
    #' @param activating_agent_term_id (optional) Filter by activating_agent_term_id
    #' @param activating_agent_term_name (optional) Filter by activating_agent_term_name
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biosamples_modified (optional) Filter by biosamples_modified
    #' @param cas (optional) Filter by cas
    #' @param cas_species (optional) Filter by cas_species
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param fused_domain (optional) Filter by fused_domain
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param modality (optional) Filter by modality
    #' @param notes (optional) Filter by notes
    #' @param product_id (optional) Filter by product_id
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param tagged_protein (optional) Filter by tagged_protein
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (CrisprModificationResults) with additional information such as HTTP status code, headers
    #' @export
    CrisprModificationsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, activated = NULL, activating_agent_term_id = NULL, activating_agent_term_name = NULL, aliases = NULL, award_id = NULL, award_component = NULL, biosamples_modified = NULL, cas = NULL, cas_species = NULL, creation_timestamp = NULL, description = NULL, documents = NULL, fused_domain = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, modality = NULL, notes = NULL, product_id = NULL, release_timestamp = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, tagged_protein = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }


































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$CrisprModifications. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `activated`) {
        query_params[["activated"]] <- c(query_params[["activated"]], list(`activated` = query_item))
      }

      # explore
      for (query_item in `activating_agent_term_id`) {
        query_params[["activating_agent_term_id"]] <- c(query_params[["activating_agent_term_id"]], list(`activating_agent_term_id` = query_item))
      }

      # explore
      for (query_item in `activating_agent_term_name`) {
        query_params[["activating_agent_term_name"]] <- c(query_params[["activating_agent_term_name"]], list(`activating_agent_term_name` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$CrisprModifications. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biosamples_modified`) {
        query_params[["biosamples_modified"]] <- c(query_params[["biosamples_modified"]], list(`biosamples_modified` = query_item))
      }

      # explore
      for (query_item in `cas`) {
        # validate enum values
        if (!(query_item %in% c("Cas9", "Cas12a", "Cas13", "dCas9", "nCas9", "SpG", "SpRY"))) {
          stop("Invalid value for cas when calling IgvfApi$CrisprModifications. Must be [Cas9, Cas12a, Cas13, dCas9, nCas9, SpG, SpRY].")
        }
        query_params[["cas"]] <- c(query_params[["cas"]], list(`cas` = query_item))
      }

      # explore
      for (query_item in `cas_species`) {
        # validate enum values
        if (!(query_item %in% c("Streptococcus pyogenes (Sp)", "Staphylococcus aureus (Sa)", "Campylobacter jejuni (Cj)", "Neisseria meningitidis (Nm)"))) {
          stop("Invalid value for cas_species when calling IgvfApi$CrisprModifications. Must be [Streptococcus pyogenes (Sp), Staphylococcus aureus (Sa), Campylobacter jejuni (Cj), Neisseria meningitidis (Nm)].")
        }
        query_params[["cas_species"]] <- c(query_params[["cas_species"]], list(`cas_species` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `fused_domain`) {
        # validate enum values
        if (!(query_item %in% c("2xVP64", "3xVP64", "ABE8e", "ABE8.20", "ANTI-FLAG", "BE4", "BE4max", "eA3A", "eA3A-T31A", "eA3A-T44D-S45A", "KOX1-KRAB", "M-MLV RT (PE2)", "p300", "TdCBE", "TdCGBE", "TdDE", "VPH", "VP64", "VP64-p65-Rta (VPR)", "ZIM3-KRAB"))) {
          stop("Invalid value for fused_domain when calling IgvfApi$CrisprModifications. Must be [2xVP64, 3xVP64, ABE8e, ABE8.20, ANTI-FLAG, BE4, BE4max, eA3A, eA3A-T31A, eA3A-T44D-S45A, KOX1-KRAB, M-MLV RT (PE2), p300, TdCBE, TdCGBE, TdDE, VPH, VP64, VP64-p65-Rta (VPR), ZIM3-KRAB].")
        }
        query_params[["fused_domain"]] <- c(query_params[["fused_domain"]], list(`fused_domain` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `modality`) {
        # validate enum values
        if (!(query_item %in% c("activation", "base editing", "cutting", "interference", "knockout", "localizing", "prime editing"))) {
          stop("Invalid value for modality when calling IgvfApi$CrisprModifications. Must be [activation, base editing, cutting, interference, knockout, localizing, prime editing].")
        }
        query_params[["modality"]] <- c(query_params[["modality"]], list(`modality` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `sources`) {
        query_params[["sources"]] <- c(query_params[["sources"]], list(`sources` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$CrisprModifications. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `tagged_protein`) {
        query_params[["tagged_protein"]] <- c(query_params[["tagged_protein"]], list(`tagged_protein` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/crispr-modifications/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "CrisprModificationResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the CuratedSet collection.
    #'
    #' @description
    #' List items in the CuratedSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assemblies (optional) Filter by assemblies
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param transcriptome_annotations (optional) Filter by transcriptome_annotations
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return CuratedSetResults
    #' @export
    CuratedSets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assemblies = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, transcriptome_annotations = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$CuratedSetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, assemblies, award_id, award_component, award_contact_pi_id, award_contact_pi_title, award_title, collections, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, dbxrefs, description, documents, donors_id, donors_accession, donors_aliases, donors_sex, donors_status, donors_taxa, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_creation_timestamp, files_file_format, files_file_size, files_href, files_s3_uri, files_sequencing_platform, files_submitted_file_name, files_upload_status, input_file_set_for, lab_id, lab_title, notes, publication_identifiers, publications, release_timestamp, revoke_detail, samples_id, samples_accession, samples_aliases, samples_cell_fate_change_treatments, samples_classifications, samples_construct_library_sets, samples_disease_terms_id, samples_disease_terms_term_name, samples_modifications, samples_sample_terms_id, samples_sample_terms_aliases, samples_sample_terms_status, samples_sample_terms_summary, samples_sample_terms_term_name, samples_status, samples_summary, samples_targeted_sample_term_id, samples_targeted_sample_term_term_name, samples_taxa, samples_treatments, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, taxa, transcriptome_annotations, url, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the CuratedSet collection.
    #'
    #' @description
    #' List items in the CuratedSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assemblies (optional) Filter by assemblies
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param transcriptome_annotations (optional) Filter by transcriptome_annotations
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (CuratedSetResults) with additional information such as HTTP status code, headers
    #' @export
    CuratedSetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assemblies = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, transcriptome_annotations = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }






















































      if (length(`samples_classifications`) > 2) {
        stop("Invalid length for `samples_classifications` when calling IgvfApi$CuratedSets, number of items must be less than or equal to 2.")
      }




      if (length(`samples_modifications`) > 2) {
        stop("Invalid length for `samples_modifications` when calling IgvfApi$CuratedSets, number of items must be less than or equal to 2.")
      }






















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$CuratedSets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `assemblies`) {
        query_params[["assemblies"]] <- c(query_params[["assemblies"]], list(`assemblies` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$CuratedSets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_id`) {
        query_params[["award.contact_pi.@id"]] <- c(query_params[["award.contact_pi.@id"]], list(`award.contact_pi.@id` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_title`) {
        query_params[["award.contact_pi.title"]] <- c(query_params[["award.contact_pi.title"]], list(`award.contact_pi.title` = query_item))
      }

      # explore
      for (query_item in `award_title`) {
        query_params[["award.title"]] <- c(query_params[["award.title"]], list(`award.title` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$CuratedSets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors_id`) {
        query_params[["donors.@id"]] <- c(query_params[["donors.@id"]], list(`donors.@id` = query_item))
      }

      # explore
      for (query_item in `donors_accession`) {
        query_params[["donors.accession"]] <- c(query_params[["donors.accession"]], list(`donors.accession` = query_item))
      }

      # explore
      for (query_item in `donors_aliases`) {
        query_params[["donors.aliases"]] <- c(query_params[["donors.aliases"]], list(`donors.aliases` = query_item))
      }

      # explore
      for (query_item in `donors_sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for donors_sex when calling IgvfApi$CuratedSets. Must be [male, female, unspecified].")
        }
        query_params[["donors.sex"]] <- c(query_params[["donors.sex"]], list(`donors.sex` = query_item))
      }

      # explore
      for (query_item in `donors_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for donors_status when calling IgvfApi$CuratedSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["donors.status"]] <- c(query_params[["donors.status"]], list(`donors.status` = query_item))
      }

      # explore
      for (query_item in `donors_taxa`) {
        query_params[["donors.taxa"]] <- c(query_params[["donors.taxa"]], list(`donors.taxa` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("barcodes", "editing templates", "elements", "external data for catalog", "genome", "genes", "guide RNAs", "transcriptome", "variants"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$CuratedSets. Must be [barcodes, editing templates, elements, external data for catalog, genome, genes, guide RNAs, transcriptome, variants].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_creation_timestamp`) {
        query_params[["files.creation_timestamp"]] <- c(query_params[["files.creation_timestamp"]], list(`files.creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_file_size`) {
        query_params[["files.file_size"]] <- c(query_params[["files.file_size"]], list(`files.file_size` = query_item))
      }

      # explore
      for (query_item in `files_href`) {
        query_params[["files.href"]] <- c(query_params[["files.href"]], list(`files.href` = query_item))
      }

      # explore
      for (query_item in `files_s3_uri`) {
        query_params[["files.s3_uri"]] <- c(query_params[["files.s3_uri"]], list(`files.s3_uri` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform`) {
        query_params[["files.sequencing_platform"]] <- c(query_params[["files.sequencing_platform"]], list(`files.sequencing_platform` = query_item))
      }

      # explore
      for (query_item in `files_submitted_file_name`) {
        query_params[["files.submitted_file_name"]] <- c(query_params[["files.submitted_file_name"]], list(`files.submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$CuratedSets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `samples_id`) {
        query_params[["samples.@id"]] <- c(query_params[["samples.@id"]], list(`samples.@id` = query_item))
      }

      # explore
      for (query_item in `samples_accession`) {
        query_params[["samples.accession"]] <- c(query_params[["samples.accession"]], list(`samples.accession` = query_item))
      }

      # explore
      for (query_item in `samples_aliases`) {
        query_params[["samples.aliases"]] <- c(query_params[["samples.aliases"]], list(`samples.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments`) {
        query_params[["samples.cell_fate_change_treatments"]] <- c(query_params[["samples.cell_fate_change_treatments"]], list(`samples.cell_fate_change_treatments` = query_item))
      }

      # explore
      for (query_item in `samples_classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for samples_classifications when calling IgvfApi$CuratedSets. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["samples.classifications"]] <- c(query_params[["samples.classifications"]], list(`samples.classifications` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets`) {
        query_params[["samples.construct_library_sets"]] <- c(query_params[["samples.construct_library_sets"]], list(`samples.construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_id`) {
        query_params[["samples.disease_terms.@id"]] <- c(query_params[["samples.disease_terms.@id"]], list(`samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_term_name`) {
        query_params[["samples.disease_terms.term_name"]] <- c(query_params[["samples.disease_terms.term_name"]], list(`samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_modifications`) {
        query_params[["samples.modifications"]] <- c(query_params[["samples.modifications"]], list(`samples.modifications` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_id`) {
        query_params[["samples.sample_terms.@id"]] <- c(query_params[["samples.sample_terms.@id"]], list(`samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_aliases`) {
        query_params[["samples.sample_terms.aliases"]] <- c(query_params[["samples.sample_terms.aliases"]], list(`samples.sample_terms.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for samples_sample_terms_status when calling IgvfApi$CuratedSets. Must be [archived, deleted, in progress, released].")
        }
        query_params[["samples.sample_terms.status"]] <- c(query_params[["samples.sample_terms.status"]], list(`samples.sample_terms.status` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_summary`) {
        query_params[["samples.sample_terms.summary"]] <- c(query_params[["samples.sample_terms.summary"]], list(`samples.sample_terms.summary` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_term_name`) {
        query_params[["samples.sample_terms.term_name"]] <- c(query_params[["samples.sample_terms.term_name"]], list(`samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for samples_status when calling IgvfApi$CuratedSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["samples.status"]] <- c(query_params[["samples.status"]], list(`samples.status` = query_item))
      }

      # explore
      for (query_item in `samples_summary`) {
        query_params[["samples.summary"]] <- c(query_params[["samples.summary"]], list(`samples.summary` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_id`) {
        query_params[["samples.targeted_sample_term.@id"]] <- c(query_params[["samples.targeted_sample_term.@id"]], list(`samples.targeted_sample_term.@id` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_term_name`) {
        query_params[["samples.targeted_sample_term.term_name"]] <- c(query_params[["samples.targeted_sample_term.term_name"]], list(`samples.targeted_sample_term.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for samples_taxa when calling IgvfApi$CuratedSets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["samples.taxa"]] <- c(query_params[["samples.taxa"]], list(`samples.taxa` = query_item))
      }

      # explore
      for (query_item in `samples_treatments`) {
        query_params[["samples.treatments"]] <- c(query_params[["samples.treatments"]], list(`samples.treatments` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$CuratedSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$CuratedSets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotations`) {
        query_params[["transcriptome_annotations"]] <- c(query_params[["transcriptome_annotations"]], list(`transcriptome_annotations` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/curated-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "CuratedSetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the DegronModification collection.
    #'
    #' @description
    #' List items in the DegronModification collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param activated (optional) Filter by activated
    #' @param activating_agent_term_id (optional) Filter by activating_agent_term_id
    #' @param activating_agent_term_name (optional) Filter by activating_agent_term_name
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biosamples_modified (optional) Filter by biosamples_modified
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param degron_system (optional) Filter by degron_system
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param modality (optional) Filter by modality
    #' @param notes (optional) Filter by notes
    #' @param product_id (optional) Filter by product_id
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param tagged_proteins (optional) Filter by tagged_proteins
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return DegronModificationResults
    #' @export
    DegronModifications = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, activated = NULL, activating_agent_term_id = NULL, activating_agent_term_name = NULL, aliases = NULL, award_id = NULL, award_component = NULL, biosamples_modified = NULL, creation_timestamp = NULL, degron_system = NULL, description = NULL, documents = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, modality = NULL, notes = NULL, product_id = NULL, release_timestamp = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, tagged_proteins = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$DegronModificationsWithHttpInfo(frame, query, limit, sort, id, activated, activating_agent_term_id, activating_agent_term_name, aliases, award_id, award_component, biosamples_modified, creation_timestamp, degron_system, description, documents, lab_id, lab_title, lot_id, modality, notes, product_id, release_timestamp, sources, status, submitted_by_id, submitted_by_title, submitter_comment, summary, tagged_proteins, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the DegronModification collection.
    #'
    #' @description
    #' List items in the DegronModification collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param activated (optional) Filter by activated
    #' @param activating_agent_term_id (optional) Filter by activating_agent_term_id
    #' @param activating_agent_term_name (optional) Filter by activating_agent_term_name
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biosamples_modified (optional) Filter by biosamples_modified
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param degron_system (optional) Filter by degron_system
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param modality (optional) Filter by modality
    #' @param notes (optional) Filter by notes
    #' @param product_id (optional) Filter by product_id
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param tagged_proteins (optional) Filter by tagged_proteins
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (DegronModificationResults) with additional information such as HTTP status code, headers
    #' @export
    DegronModificationsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, activated = NULL, activating_agent_term_id = NULL, activating_agent_term_name = NULL, aliases = NULL, award_id = NULL, award_component = NULL, biosamples_modified = NULL, creation_timestamp = NULL, degron_system = NULL, description = NULL, documents = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, modality = NULL, notes = NULL, product_id = NULL, release_timestamp = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, tagged_proteins = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }
































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$DegronModifications. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `activated`) {
        query_params[["activated"]] <- c(query_params[["activated"]], list(`activated` = query_item))
      }

      # explore
      for (query_item in `activating_agent_term_id`) {
        query_params[["activating_agent_term_id"]] <- c(query_params[["activating_agent_term_id"]], list(`activating_agent_term_id` = query_item))
      }

      # explore
      for (query_item in `activating_agent_term_name`) {
        query_params[["activating_agent_term_name"]] <- c(query_params[["activating_agent_term_name"]], list(`activating_agent_term_name` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$DegronModifications. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biosamples_modified`) {
        query_params[["biosamples_modified"]] <- c(query_params[["biosamples_modified"]], list(`biosamples_modified` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `degron_system`) {
        # validate enum values
        if (!(query_item %in% c("AID", "AlissAid", "ssAID"))) {
          stop("Invalid value for degron_system when calling IgvfApi$DegronModifications. Must be [AID, AlissAid, ssAID].")
        }
        query_params[["degron_system"]] <- c(query_params[["degron_system"]], list(`degron_system` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `modality`) {
        # validate enum values
        if (!(query_item %in% c("degradation"))) {
          stop("Invalid value for modality when calling IgvfApi$DegronModifications. Must be [degradation].")
        }
        query_params[["modality"]] <- c(query_params[["modality"]], list(`modality` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `sources`) {
        query_params[["sources"]] <- c(query_params[["sources"]], list(`sources` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$DegronModifications. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `tagged_proteins`) {
        query_params[["tagged_proteins"]] <- c(query_params[["tagged_proteins"]], list(`tagged_proteins` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/degron-modifications/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "DegronModificationResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Document collection.
    #'
    #' @description
    #' List items in the Document collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param characterization_method (optional) Filter by characterization_method
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param document_type (optional) Filter by document_type
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param urls (optional) Filter by urls
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return DocumentResults
    #' @export
    Documents = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, characterization_method = NULL, creation_timestamp = NULL, description = NULL, document_type = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, urls = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$DocumentsWithHttpInfo(frame, query, limit, sort, id, aliases, award_id, award_component, characterization_method, creation_timestamp, description, document_type, lab_id, lab_title, notes, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, urls, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Document collection.
    #'
    #' @description
    #' List items in the Document collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param characterization_method (optional) Filter by characterization_method
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param document_type (optional) Filter by document_type
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param urls (optional) Filter by urls
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (DocumentResults) with additional information such as HTTP status code, headers
    #' @export
    DocumentsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, characterization_method = NULL, creation_timestamp = NULL, description = NULL, document_type = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, urls = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }
























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Documents. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Documents. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `characterization_method`) {
        # validate enum values
        if (!(query_item %in% c("FACS", "immunoblot", "immunofluorescence", "immunoprecipitation", "mass spectrometry", "PCR", "restriction digest", "RT-qPCR", "sequencing"))) {
          stop("Invalid value for characterization_method when calling IgvfApi$Documents. Must be [FACS, immunoblot, immunofluorescence, immunoprecipitation, mass spectrometry, PCR, restriction digest, RT-qPCR, sequencing].")
        }
        query_params[["characterization_method"]] <- c(query_params[["characterization_method"]], list(`characterization_method` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `document_type`) {
        # validate enum values
        if (!(query_item %in% c("cell fate change protocol", "characterization", "computational protocol", "experimental protocol", "file format specification", "image", "model source data", "plate map", "plasmid map", "plasmid sequence", "standards"))) {
          stop("Invalid value for document_type when calling IgvfApi$Documents. Must be [cell fate change protocol, characterization, computational protocol, experimental protocol, file format specification, image, model source data, plate map, plasmid map, plasmid sequence, standards].")
        }
        query_params[["document_type"]] <- c(query_params[["document_type"]], list(`document_type` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Documents. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `urls`) {
        query_params[["urls"]] <- c(query_params[["urls"]], list(`urls` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/documents/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "DocumentResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Download file.
    #'
    #' @description
    #' Download file.
    #'
    #' @param file_id The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03).
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return data.frame
    #' @export
    Download = function(file_id, data_file = NULL, ...) {
      local_var_response <- self$DownloadWithHttpInfo(file_id, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Download file.
    #'
    #' @description
    #' Download file.
    #'
    #' @param file_id The unique identifier for the file to download, e.g. @id (/tabular-files/IGVFFI8092FZKL/), accession (IGVFFI8092FZKL), or UUID (fdbdc159-e5b9-40a8-b788-3f72c9886b03).
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (data.frame) with additional information such as HTTP status code, headers
    #' @export
    DownloadWithHttpInfo = function(file_id, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`file_id`)) {
        stop("Missing required parameter `file_id`.")
      }


      local_var_url_path <- "/{file_id}/@@download"
      if (!missing(`file_id`)) {
        local_var_url_path <- gsub("\\{file_id\\}", URLencode(as.character(`file_id`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/octet-stream")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "data.frame", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Gene collection.
    #'
    #' @description
    #' List items in the Gene collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param geneid (optional) Filter by geneid
    #' @param geneid_with_version (optional) Filter by geneid_with_version
    #' @param locations (optional) Filter by locations
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param symbol (optional) Filter by symbol
    #' @param synonyms (optional) Filter by synonyms
    #' @param taxa (optional) Filter by taxa
    #' @param title (optional) Filter by title
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param uuid (optional) Filter by uuid
    #' @param version_number (optional) Filter by version_number
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return GeneResults
    #' @export
    Genes = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, geneid = NULL, geneid_with_version = NULL, locations = NULL, name = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, symbol = NULL, synonyms = NULL, taxa = NULL, title = NULL, transcriptome_annotation = NULL, uuid = NULL, version_number = NULL, data_file = NULL, ...) {
      local_var_response <- self$GenesWithHttpInfo(frame, query, limit, sort, id, aliases, creation_timestamp, dbxrefs, description, geneid, geneid_with_version, locations, name, notes, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, symbol, synonyms, taxa, title, transcriptome_annotation, uuid, version_number, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Gene collection.
    #'
    #' @description
    #' List items in the Gene collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param geneid (optional) Filter by geneid
    #' @param geneid_with_version (optional) Filter by geneid_with_version
    #' @param locations (optional) Filter by locations
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param symbol (optional) Filter by symbol
    #' @param synonyms (optional) Filter by synonyms
    #' @param taxa (optional) Filter by taxa
    #' @param title (optional) Filter by title
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param uuid (optional) Filter by uuid
    #' @param version_number (optional) Filter by version_number
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (GeneResults) with additional information such as HTTP status code, headers
    #' @export
    GenesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, geneid = NULL, geneid_with_version = NULL, locations = NULL, name = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, symbol = NULL, synonyms = NULL, taxa = NULL, title = NULL, transcriptome_annotation = NULL, uuid = NULL, version_number = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }




























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Genes. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `geneid`) {
        query_params[["geneid"]] <- c(query_params[["geneid"]], list(`geneid` = query_item))
      }

      # explore
      for (query_item in `geneid_with_version`) {
        query_params[["geneid_with_version"]] <- c(query_params[["geneid_with_version"]], list(`geneid_with_version` = query_item))
      }

      # explore
      for (query_item in `locations`) {
        query_params[["locations"]] <- c(query_params[["locations"]], list(`locations` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Genes. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `symbol`) {
        query_params[["symbol"]] <- c(query_params[["symbol"]], list(`symbol` = query_item))
      }

      # explore
      for (query_item in `synonyms`) {
        query_params[["synonyms"]] <- c(query_params[["synonyms"]], list(`synonyms` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$Genes. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotation`) {
        # validate enum values
        if (!(query_item %in% c("GENCODE 40", "GENCODE 41", "GENCODE 42", "GENCODE 43", "GENCODE 44", "GENCODE 45", "GENCODE M30", "GENCODE M31", "GENCODE M32", "GENCODE M33", "GENCODE M34"))) {
          stop("Invalid value for transcriptome_annotation when calling IgvfApi$Genes. Must be [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34].")
        }
        query_params[["transcriptome_annotation"]] <- c(query_params[["transcriptome_annotation"]], list(`transcriptome_annotation` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `version_number`) {
        query_params[["version_number"]] <- c(query_params[["version_number"]], list(`version_number` = query_item))
      }

      local_var_url_path <- "/genes/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "GeneResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the GenomeBrowserAnnotationFile collection.
    #'
    #' @description
    #' List items in the GenomeBrowserAnnotationFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_format_type (optional) Filter by file_format_type
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return GenomeBrowserAnnotationFileResults
    #' @export
    GenomeBrowserAnnotationFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_format_type = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$GenomeBrowserAnnotationFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, assembly, award_id, award_component, collections, content_md5sum, content_type, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_format_type, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, release_timestamp, revoke_detail, s3_uri, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, transcriptome_annotation, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the GenomeBrowserAnnotationFile collection.
    #'
    #' @description
    #' List items in the GenomeBrowserAnnotationFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_format_type (optional) Filter by file_format_type
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (GenomeBrowserAnnotationFileResults) with additional information such as HTTP status code, headers
    #' @export
    GenomeBrowserAnnotationFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_format_type = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }
















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `assembly`) {
        # validate enum values
        if (!(query_item %in% c("GRCh38", "GRCm39"))) {
          stop("Invalid value for assembly when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [GRCh38, GRCm39].")
        }
        query_params[["assembly"]] <- c(query_params[["assembly"]], list(`assembly` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("bigBed", "tabix"))) {
          stop("Invalid value for file_format when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [bigBed, tabix].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_format_type`) {
        # validate enum values
        if (!(query_item %in% c("bed12", "bed3", "bed3+", "bed5", "bed6", "bed6+", "bed9", "bed9+", "mpra_starr"))) {
          stop("Invalid value for file_format_type when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [bed12, bed3, bed3+, bed5, bed6, bed6+, bed9, bed9+, mpra_starr].")
        }
        query_params[["file_format_type"]] <- c(query_params[["file_format_type"]], list(`file_format_type` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotation`) {
        # validate enum values
        if (!(query_item %in% c("GENCODE 40", "GENCODE 41", "GENCODE 42", "GENCODE 43", "GENCODE 44", "GENCODE 45", "GENCODE M30", "GENCODE M31", "GENCODE M32", "GENCODE M33", "GENCODE M34"))) {
          stop("Invalid value for transcriptome_annotation when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34].")
        }
        query_params[["transcriptome_annotation"]] <- c(query_params[["transcriptome_annotation"]], list(`transcriptome_annotation` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$GenomeBrowserAnnotationFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/genome-browser-annotation-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "GenomeBrowserAnnotationFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Get item information
    #'
    #' @description
    #' Get item information
    #'
    #' @param resource_id The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`).
    #' @param frame Constant value. Do not set.
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return Item
    #' @export
    GetById = function(resource_id, frame, data_file = NULL, ...) {
      local_var_response <- self$GetByIdWithHttpInfo(resource_id, frame, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Get item information
    #'
    #' @description
    #' Get item information
    #'
    #' @param resource_id The unique identifier for the resource i.e. @id (`/sequence-files/IGVFFI1165AJSO/`), accession (`IGVFFI1165AJSO`) or UUID (`fffcd64e-af02-4675-8953-7352459ee06a`).
    #' @param frame Constant value. Do not set.
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (Item) with additional information such as HTTP status code, headers
    #' @export
    GetByIdWithHttpInfo = function(resource_id, frame, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`resource_id`)) {
        stop("Missing required parameter `resource_id`.")
      }

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$GetById. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      local_var_url_path <- "/{resource_id}"
      if (!missing(`resource_id`)) {
        local_var_url_path <- gsub("\\{resource_id\\}", URLencode(as.character(`resource_id`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "Item", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the HumanDonor collection.
    #'
    #' @description
    #' List items in the HumanDonor collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param ethnicities (optional) Filter by ethnicities
    #' @param human_donor_identifiers (optional) Filter by human_donor_identifiers
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param phenotypic_features_id (optional) Filter by phenotypic_features.@id
    #' @param phenotypic_features_feature_id (optional) Filter by phenotypic_features.feature.@id
    #' @param phenotypic_features_feature_term_id (optional) Filter by phenotypic_features.feature.term_id
    #' @param phenotypic_features_feature_term_name (optional) Filter by phenotypic_features.feature.term_name
    #' @param phenotypic_features_observation_date (optional) Filter by phenotypic_features.observation_date
    #' @param phenotypic_features_quantity_units (optional) Filter by phenotypic_features.quantity_units
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param related_donors_donor_id (optional) Filter by related_donors.donor.@id
    #' @param related_donors_donor_accession (optional) Filter by related_donors.donor.accession
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sex (optional) Filter by sex
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return HumanDonorResults
    #' @export
    HumanDonors = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, collections = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, ethnicities = NULL, human_donor_identifiers = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, phenotypic_features_id = NULL, phenotypic_features_feature_id = NULL, phenotypic_features_feature_term_id = NULL, phenotypic_features_feature_term_name = NULL, phenotypic_features_observation_date = NULL, phenotypic_features_quantity_units = NULL, publication_identifiers = NULL, publications = NULL, related_donors_donor_id = NULL, related_donors_donor_accession = NULL, release_timestamp = NULL, revoke_detail = NULL, sex = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$HumanDonorsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, award_id, award_component, collections, creation_timestamp, dbxrefs, description, documents, ethnicities, human_donor_identifiers, lab_id, lab_title, notes, phenotypic_features_id, phenotypic_features_feature_id, phenotypic_features_feature_term_id, phenotypic_features_feature_term_name, phenotypic_features_observation_date, phenotypic_features_quantity_units, publication_identifiers, publications, related_donors_donor_id, related_donors_donor_accession, release_timestamp, revoke_detail, sex, status, submitted_by_id, submitted_by_title, submitter_comment, summary, taxa, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the HumanDonor collection.
    #'
    #' @description
    #' List items in the HumanDonor collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param ethnicities (optional) Filter by ethnicities
    #' @param human_donor_identifiers (optional) Filter by human_donor_identifiers
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param phenotypic_features_id (optional) Filter by phenotypic_features.@id
    #' @param phenotypic_features_feature_id (optional) Filter by phenotypic_features.feature.@id
    #' @param phenotypic_features_feature_term_id (optional) Filter by phenotypic_features.feature.term_id
    #' @param phenotypic_features_feature_term_name (optional) Filter by phenotypic_features.feature.term_name
    #' @param phenotypic_features_observation_date (optional) Filter by phenotypic_features.observation_date
    #' @param phenotypic_features_quantity_units (optional) Filter by phenotypic_features.quantity_units
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param related_donors_donor_id (optional) Filter by related_donors.donor.@id
    #' @param related_donors_donor_accession (optional) Filter by related_donors.donor.accession
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sex (optional) Filter by sex
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (HumanDonorResults) with additional information such as HTTP status code, headers
    #' @export
    HumanDonorsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, collections = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, ethnicities = NULL, human_donor_identifiers = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, phenotypic_features_id = NULL, phenotypic_features_feature_id = NULL, phenotypic_features_feature_term_id = NULL, phenotypic_features_feature_term_name = NULL, phenotypic_features_observation_date = NULL, phenotypic_features_quantity_units = NULL, publication_identifiers = NULL, publications = NULL, related_donors_donor_id = NULL, related_donors_donor_accession = NULL, release_timestamp = NULL, revoke_detail = NULL, sex = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }











































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$HumanDonors. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$HumanDonors. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$HumanDonors. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `ethnicities`) {
        # validate enum values
        if (!(query_item %in% c("African American", "African Caribbean", "Arab", "Asian", "Black", "Black African", "Chinese", "Colombian", "Dai Chinese", "Esan", "Eskimo", "European", "Gambian", "Han Chinese", "Hispanic", "Indian", "Japanese", "Kinh Vietnamese", "Luhya", "Maasai", "Mende", "Native Hawaiian", "Pacific Islander", "Puerto Rican", "Yoruba"))) {
          stop("Invalid value for ethnicities when calling IgvfApi$HumanDonors. Must be [African American, African Caribbean, Arab, Asian, Black, Black African, Chinese, Colombian, Dai Chinese, Esan, Eskimo, European, Gambian, Han Chinese, Hispanic, Indian, Japanese, Kinh Vietnamese, Luhya, Maasai, Mende, Native Hawaiian, Pacific Islander, Puerto Rican, Yoruba].")
        }
        query_params[["ethnicities"]] <- c(query_params[["ethnicities"]], list(`ethnicities` = query_item))
      }

      # explore
      for (query_item in `human_donor_identifiers`) {
        query_params[["human_donor_identifiers"]] <- c(query_params[["human_donor_identifiers"]], list(`human_donor_identifiers` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_id`) {
        query_params[["phenotypic_features.@id"]] <- c(query_params[["phenotypic_features.@id"]], list(`phenotypic_features.@id` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_feature_id`) {
        query_params[["phenotypic_features.feature.@id"]] <- c(query_params[["phenotypic_features.feature.@id"]], list(`phenotypic_features.feature.@id` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_feature_term_id`) {
        query_params[["phenotypic_features.feature.term_id"]] <- c(query_params[["phenotypic_features.feature.term_id"]], list(`phenotypic_features.feature.term_id` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_feature_term_name`) {
        query_params[["phenotypic_features.feature.term_name"]] <- c(query_params[["phenotypic_features.feature.term_name"]], list(`phenotypic_features.feature.term_name` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_observation_date`) {
        query_params[["phenotypic_features.observation_date"]] <- c(query_params[["phenotypic_features.observation_date"]], list(`phenotypic_features.observation_date` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_quantity_units`) {
        # validate enum values
        if (!(query_item %in% c("meter", "micromole", "nanogram", "microgram", "milligram", "gram", "kilogram", "milli-International Unit per milliliter", "picogram per milliliter", "nanogram per milliliter", "milligram per deciliter"))) {
          stop("Invalid value for phenotypic_features_quantity_units when calling IgvfApi$HumanDonors. Must be [meter, micromole, nanogram, microgram, milligram, gram, kilogram, milli-International Unit per milliliter, picogram per milliliter, nanogram per milliliter, milligram per deciliter].")
        }
        query_params[["phenotypic_features.quantity_units"]] <- c(query_params[["phenotypic_features.quantity_units"]], list(`phenotypic_features.quantity_units` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `related_donors_donor_id`) {
        query_params[["related_donors.donor.@id"]] <- c(query_params[["related_donors.donor.@id"]], list(`related_donors.donor.@id` = query_item))
      }

      # explore
      for (query_item in `related_donors_donor_accession`) {
        query_params[["related_donors.donor.accession"]] <- c(query_params[["related_donors.donor.accession"]], list(`related_donors.donor.accession` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for sex when calling IgvfApi$HumanDonors. Must be [male, female, unspecified].")
        }
        query_params[["sex"]] <- c(query_params[["sex"]], list(`sex` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$HumanDonors. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens"))) {
          stop("Invalid value for taxa when calling IgvfApi$HumanDonors. Must be [Homo sapiens].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/human-donors/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "HumanDonorResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the ImageFile collection.
    #'
    #' @description
    #' List items in the ImageFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ImageFileResults
    #' @export
    ImageFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$ImageFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, award_id, award_component, collections, content_md5sum, content_type, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, release_timestamp, revoke_detail, s3_uri, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the ImageFile collection.
    #'
    #' @description
    #' List items in the ImageFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ImageFileResults) with additional information such as HTTP status code, headers
    #' @export
    ImageFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }













































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$ImageFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$ImageFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$ImageFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("jpg", "png"))) {
          stop("Invalid value for file_format when calling IgvfApi$ImageFiles. Must be [jpg, png].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$ImageFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$ImageFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/image-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ImageFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Image collection.
    #'
    #' @description
    #' List items in the Image collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param caption (optional) Filter by caption
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param download_url (optional) Filter by download_url
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param thumb_nail (optional) Filter by thumb_nail
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ImageResults
    #' @export
    Images = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, caption = NULL, creation_timestamp = NULL, description = NULL, download_url = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, thumb_nail = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$ImagesWithHttpInfo(frame, query, limit, sort, id, aliases, caption, creation_timestamp, description, download_url, notes, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, thumb_nail, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Image collection.
    #'
    #' @description
    #' List items in the Image collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param caption (optional) Filter by caption
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param download_url (optional) Filter by download_url
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param thumb_nail (optional) Filter by thumb_nail
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ImageResults) with additional information such as HTTP status code, headers
    #' @export
    ImagesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, caption = NULL, creation_timestamp = NULL, description = NULL, download_url = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, thumb_nail = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }




















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Images. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `caption`) {
        query_params[["caption"]] <- c(query_params[["caption"]], list(`caption` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `download_url`) {
        query_params[["download_url"]] <- c(query_params[["download_url"]], list(`download_url` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Images. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `thumb_nail`) {
        query_params[["thumb_nail"]] <- c(query_params[["thumb_nail"]], list(`thumb_nail` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/images/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ImageResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the InVitroSystem collection.
    #'
    #' @description
    #' List items in the InVitroSystem collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cell_fate_change_protocol (optional) Filter by cell_fate_change_protocol
    #' @param cell_fate_change_treatments_id (optional) Filter by cell_fate_change_treatments.@id
    #' @param cell_fate_change_treatments_purpose (optional) Filter by cell_fate_change_treatments.purpose
    #' @param cell_fate_change_treatments_status (optional) Filter by cell_fate_change_treatments.status
    #' @param cell_fate_change_treatments_summary (optional) Filter by cell_fate_change_treatments.summary
    #' @param cell_fate_change_treatments_treatment_type (optional) Filter by cell_fate_change_treatments.treatment_type
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param demultiplexed_from (optional) Filter by demultiplexed_from
    #' @param demultiplexed_to (optional) Filter by demultiplexed_to
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param growth_medium (optional) Filter by growth_medium
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from_id (optional) Filter by originated_from.@id
    #' @param originated_from_accession (optional) Filter by originated_from.accession
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param passage_number (optional) Filter by passage_number
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param targeted_sample_term (optional) Filter by targeted_sample_term
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_change (optional) Filter by time_post_change
    #' @param time_post_change_units (optional) Filter by time_post_change_units
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return InVitroSystemResults
    #' @export
    InVitroSystems = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, cell_fate_change_protocol = NULL, cell_fate_change_treatments_id = NULL, cell_fate_change_treatments_purpose = NULL, cell_fate_change_treatments_status = NULL, cell_fate_change_treatments_summary = NULL, cell_fate_change_treatments_treatment_type = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, demultiplexed_from = NULL, demultiplexed_to = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, growth_medium = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from_id = NULL, originated_from_accession = NULL, part_of = NULL, parts = NULL, passage_number = NULL, pooled_from = NULL, pooled_in = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, targeted_sample_term = NULL, taxa = NULL, time_post_change = NULL, time_post_change_units = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$InVitroSystemsWithHttpInfo(frame, query, limit, sort, id, accession, age, age_units, aliases, alternate_accessions, award_id, award_component, biomarkers, cell_fate_change_protocol, cell_fate_change_treatments_id, cell_fate_change_treatments_purpose, cell_fate_change_treatments_status, cell_fate_change_treatments_summary, cell_fate_change_treatments_treatment_type, cellular_sub_pool, classifications, collections, construct_library_sets, creation_timestamp, date_obtained, dbxrefs, demultiplexed_from, demultiplexed_to, description, disease_terms_id, disease_terms_term_name, documents, donors, embryonic, file_sets_id, file_sets_accession, file_sets_aliases, file_sets_assay_term_term_name, file_sets_lab_title, file_sets_status, file_sets_summary, growth_medium, institutional_certificates_id, institutional_certificates_certificate_identifier, lab_id, lab_title, lot_id, lower_bound_age, lower_bound_age_in_hours, modifications_id, modifications_status, modifications_summary, moi, multiplexed_in_id, multiplexed_in_accession, notes, nucleic_acid_delivery, origin_of, originated_from_id, originated_from_accession, part_of, parts, passage_number, pooled_from, pooled_in, product_id, protocols, publication_identifiers, publications, release_timestamp, revoke_detail, sample_terms_id, sample_terms_term_name, sex, sorted_fractions, sorted_from_id, sorted_from_accession, sorted_from_detail, sources_id, starting_amount, starting_amount_units, status, submitted_by_id, submitted_by_title, submitter_comment, summary, targeted_sample_term, taxa, time_post_change, time_post_change_units, time_post_library_delivery, time_post_library_delivery_units, treatments_id, treatments_purpose, treatments_status, treatments_summary, treatments_treatment_type, upper_bound_age, upper_bound_age_in_hours, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the InVitroSystem collection.
    #'
    #' @description
    #' List items in the InVitroSystem collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cell_fate_change_protocol (optional) Filter by cell_fate_change_protocol
    #' @param cell_fate_change_treatments_id (optional) Filter by cell_fate_change_treatments.@id
    #' @param cell_fate_change_treatments_purpose (optional) Filter by cell_fate_change_treatments.purpose
    #' @param cell_fate_change_treatments_status (optional) Filter by cell_fate_change_treatments.status
    #' @param cell_fate_change_treatments_summary (optional) Filter by cell_fate_change_treatments.summary
    #' @param cell_fate_change_treatments_treatment_type (optional) Filter by cell_fate_change_treatments.treatment_type
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param demultiplexed_from (optional) Filter by demultiplexed_from
    #' @param demultiplexed_to (optional) Filter by demultiplexed_to
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param growth_medium (optional) Filter by growth_medium
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from_id (optional) Filter by originated_from.@id
    #' @param originated_from_accession (optional) Filter by originated_from.accession
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param passage_number (optional) Filter by passage_number
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param targeted_sample_term (optional) Filter by targeted_sample_term
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_change (optional) Filter by time_post_change
    #' @param time_post_change_units (optional) Filter by time_post_change_units
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (InVitroSystemResults) with additional information such as HTTP status code, headers
    #' @export
    InVitroSystemsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, cell_fate_change_protocol = NULL, cell_fate_change_treatments_id = NULL, cell_fate_change_treatments_purpose = NULL, cell_fate_change_treatments_status = NULL, cell_fate_change_treatments_summary = NULL, cell_fate_change_treatments_treatment_type = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, demultiplexed_from = NULL, demultiplexed_to = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, growth_medium = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from_id = NULL, originated_from_accession = NULL, part_of = NULL, parts = NULL, passage_number = NULL, pooled_from = NULL, pooled_in = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, targeted_sample_term = NULL, taxa = NULL, time_post_change = NULL, time_post_change_units = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }







































































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$InVitroSystems. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `age`) {
        query_params[["age"]] <- c(query_params[["age"]], list(`age` = query_item))
      }

      # explore
      for (query_item in `age_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month", "year"))) {
          stop("Invalid value for age_units when calling IgvfApi$InVitroSystems. Must be [minute, hour, day, week, month, year].")
        }
        query_params[["age_units"]] <- c(query_params[["age_units"]], list(`age_units` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$InVitroSystems. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biomarkers`) {
        query_params[["biomarkers"]] <- c(query_params[["biomarkers"]], list(`biomarkers` = query_item))
      }

      # explore
      for (query_item in `cell_fate_change_protocol`) {
        query_params[["cell_fate_change_protocol"]] <- c(query_params[["cell_fate_change_protocol"]], list(`cell_fate_change_protocol` = query_item))
      }

      # explore
      for (query_item in `cell_fate_change_treatments_id`) {
        query_params[["cell_fate_change_treatments.@id"]] <- c(query_params[["cell_fate_change_treatments.@id"]], list(`cell_fate_change_treatments.@id` = query_item))
      }

      # explore
      for (query_item in `cell_fate_change_treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for cell_fate_change_treatments_purpose when calling IgvfApi$InVitroSystems. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["cell_fate_change_treatments.purpose"]] <- c(query_params[["cell_fate_change_treatments.purpose"]], list(`cell_fate_change_treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `cell_fate_change_treatments_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for cell_fate_change_treatments_status when calling IgvfApi$InVitroSystems. Must be [archived, deleted, in progress, released].")
        }
        query_params[["cell_fate_change_treatments.status"]] <- c(query_params[["cell_fate_change_treatments.status"]], list(`cell_fate_change_treatments.status` = query_item))
      }

      # explore
      for (query_item in `cell_fate_change_treatments_summary`) {
        query_params[["cell_fate_change_treatments.summary"]] <- c(query_params[["cell_fate_change_treatments.summary"]], list(`cell_fate_change_treatments.summary` = query_item))
      }

      # explore
      for (query_item in `cell_fate_change_treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for cell_fate_change_treatments_treatment_type when calling IgvfApi$InVitroSystems. Must be [chemical, protein, environmental].")
        }
        query_params[["cell_fate_change_treatments.treatment_type"]] <- c(query_params[["cell_fate_change_treatments.treatment_type"]], list(`cell_fate_change_treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `cellular_sub_pool`) {
        query_params[["cellular_sub_pool"]] <- c(query_params[["cellular_sub_pool"]], list(`cellular_sub_pool` = query_item))
      }

      # explore
      for (query_item in `classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for classifications when calling IgvfApi$InVitroSystems. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["classifications"]] <- c(query_params[["classifications"]], list(`classifications` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$InVitroSystems. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets`) {
        query_params[["construct_library_sets"]] <- c(query_params[["construct_library_sets"]], list(`construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_obtained`) {
        query_params[["date_obtained"]] <- c(query_params[["date_obtained"]], list(`date_obtained` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `demultiplexed_from`) {
        query_params[["demultiplexed_from"]] <- c(query_params[["demultiplexed_from"]], list(`demultiplexed_from` = query_item))
      }

      # explore
      for (query_item in `demultiplexed_to`) {
        query_params[["demultiplexed_to"]] <- c(query_params[["demultiplexed_to"]], list(`demultiplexed_to` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `disease_terms_id`) {
        query_params[["disease_terms.@id"]] <- c(query_params[["disease_terms.@id"]], list(`disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `disease_terms_term_name`) {
        query_params[["disease_terms.term_name"]] <- c(query_params[["disease_terms.term_name"]], list(`disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors`) {
        query_params[["donors"]] <- c(query_params[["donors"]], list(`donors` = query_item))
      }

      # explore
      for (query_item in `embryonic`) {
        query_params[["embryonic"]] <- c(query_params[["embryonic"]], list(`embryonic` = query_item))
      }

      # explore
      for (query_item in `file_sets_id`) {
        query_params[["file_sets.@id"]] <- c(query_params[["file_sets.@id"]], list(`file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `file_sets_accession`) {
        query_params[["file_sets.accession"]] <- c(query_params[["file_sets.accession"]], list(`file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `file_sets_aliases`) {
        query_params[["file_sets.aliases"]] <- c(query_params[["file_sets.aliases"]], list(`file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `file_sets_assay_term_term_name`) {
        query_params[["file_sets.assay_term.term_name"]] <- c(query_params[["file_sets.assay_term.term_name"]], list(`file_sets.assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `file_sets_lab_title`) {
        query_params[["file_sets.lab.title"]] <- c(query_params[["file_sets.lab.title"]], list(`file_sets.lab.title` = query_item))
      }

      # explore
      for (query_item in `file_sets_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for file_sets_status when calling IgvfApi$InVitroSystems. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["file_sets.status"]] <- c(query_params[["file_sets.status"]], list(`file_sets.status` = query_item))
      }

      # explore
      for (query_item in `file_sets_summary`) {
        query_params[["file_sets.summary"]] <- c(query_params[["file_sets.summary"]], list(`file_sets.summary` = query_item))
      }

      # explore
      for (query_item in `growth_medium`) {
        # validate enum values
        if (!(query_item %in% c("DMEM with serum", "DMEM without serum", "SMBM with serum", "SMBM without serum"))) {
          stop("Invalid value for growth_medium when calling IgvfApi$InVitroSystems. Must be [DMEM with serum, DMEM without serum, SMBM with serum, SMBM without serum].")
        }
        query_params[["growth_medium"]] <- c(query_params[["growth_medium"]], list(`growth_medium` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_id`) {
        query_params[["institutional_certificates.@id"]] <- c(query_params[["institutional_certificates.@id"]], list(`institutional_certificates.@id` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_certificate_identifier`) {
        query_params[["institutional_certificates.certificate_identifier"]] <- c(query_params[["institutional_certificates.certificate_identifier"]], list(`institutional_certificates.certificate_identifier` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age`) {
        query_params[["lower_bound_age"]] <- c(query_params[["lower_bound_age"]], list(`lower_bound_age` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age_in_hours`) {
        query_params[["lower_bound_age_in_hours"]] <- c(query_params[["lower_bound_age_in_hours"]], list(`lower_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `modifications_id`) {
        query_params[["modifications.@id"]] <- c(query_params[["modifications.@id"]], list(`modifications.@id` = query_item))
      }

      # explore
      for (query_item in `modifications_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for modifications_status when calling IgvfApi$InVitroSystems. Must be [archived, deleted, in progress, released].")
        }
        query_params[["modifications.status"]] <- c(query_params[["modifications.status"]], list(`modifications.status` = query_item))
      }

      # explore
      for (query_item in `modifications_summary`) {
        query_params[["modifications.summary"]] <- c(query_params[["modifications.summary"]], list(`modifications.summary` = query_item))
      }

      # explore
      for (query_item in `moi`) {
        query_params[["moi"]] <- c(query_params[["moi"]], list(`moi` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_id`) {
        query_params[["multiplexed_in.@id"]] <- c(query_params[["multiplexed_in.@id"]], list(`multiplexed_in.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_accession`) {
        query_params[["multiplexed_in.accession"]] <- c(query_params[["multiplexed_in.accession"]], list(`multiplexed_in.accession` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `nucleic_acid_delivery`) {
        # validate enum values
        if (!(query_item %in% c("transfection", "adenoviral transduction", "lentiviral transduction"))) {
          stop("Invalid value for nucleic_acid_delivery when calling IgvfApi$InVitroSystems. Must be [transfection, adenoviral transduction, lentiviral transduction].")
        }
        query_params[["nucleic_acid_delivery"]] <- c(query_params[["nucleic_acid_delivery"]], list(`nucleic_acid_delivery` = query_item))
      }

      # explore
      for (query_item in `origin_of`) {
        query_params[["origin_of"]] <- c(query_params[["origin_of"]], list(`origin_of` = query_item))
      }

      # explore
      for (query_item in `originated_from_id`) {
        query_params[["originated_from.@id"]] <- c(query_params[["originated_from.@id"]], list(`originated_from.@id` = query_item))
      }

      # explore
      for (query_item in `originated_from_accession`) {
        query_params[["originated_from.accession"]] <- c(query_params[["originated_from.accession"]], list(`originated_from.accession` = query_item))
      }

      # explore
      for (query_item in `part_of`) {
        query_params[["part_of"]] <- c(query_params[["part_of"]], list(`part_of` = query_item))
      }

      # explore
      for (query_item in `parts`) {
        query_params[["parts"]] <- c(query_params[["parts"]], list(`parts` = query_item))
      }

      # explore
      for (query_item in `passage_number`) {
        query_params[["passage_number"]] <- c(query_params[["passage_number"]], list(`passage_number` = query_item))
      }

      # explore
      for (query_item in `pooled_from`) {
        query_params[["pooled_from"]] <- c(query_params[["pooled_from"]], list(`pooled_from` = query_item))
      }

      # explore
      for (query_item in `pooled_in`) {
        query_params[["pooled_in"]] <- c(query_params[["pooled_in"]], list(`pooled_in` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sample_terms_id`) {
        query_params[["sample_terms.@id"]] <- c(query_params[["sample_terms.@id"]], list(`sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `sample_terms_term_name`) {
        query_params[["sample_terms.term_name"]] <- c(query_params[["sample_terms.term_name"]], list(`sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `sex`) {
        # validate enum values
        if (!(query_item %in% c("female", "male", "mixed", "unspecified"))) {
          stop("Invalid value for sex when calling IgvfApi$InVitroSystems. Must be [female, male, mixed, unspecified].")
        }
        query_params[["sex"]] <- c(query_params[["sex"]], list(`sex` = query_item))
      }

      # explore
      for (query_item in `sorted_fractions`) {
        query_params[["sorted_fractions"]] <- c(query_params[["sorted_fractions"]], list(`sorted_fractions` = query_item))
      }

      # explore
      for (query_item in `sorted_from_id`) {
        query_params[["sorted_from.@id"]] <- c(query_params[["sorted_from.@id"]], list(`sorted_from.@id` = query_item))
      }

      # explore
      for (query_item in `sorted_from_accession`) {
        query_params[["sorted_from.accession"]] <- c(query_params[["sorted_from.accession"]], list(`sorted_from.accession` = query_item))
      }

      # explore
      for (query_item in `sorted_from_detail`) {
        query_params[["sorted_from_detail"]] <- c(query_params[["sorted_from_detail"]], list(`sorted_from_detail` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `starting_amount`) {
        query_params[["starting_amount"]] <- c(query_params[["starting_amount"]], list(`starting_amount` = query_item))
      }

      # explore
      for (query_item in `starting_amount_units`) {
        # validate enum values
        if (!(query_item %in% c("cells", "cells/ml", "g", "items", "mg", "whole animals", "whole embryos", "μg", "ng"))) {
          stop("Invalid value for starting_amount_units when calling IgvfApi$InVitroSystems. Must be [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng].")
        }
        query_params[["starting_amount_units"]] <- c(query_params[["starting_amount_units"]], list(`starting_amount_units` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$InVitroSystems. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `targeted_sample_term`) {
        query_params[["targeted_sample_term"]] <- c(query_params[["targeted_sample_term"]], list(`targeted_sample_term` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$InVitroSystems. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `time_post_change`) {
        query_params[["time_post_change"]] <- c(query_params[["time_post_change"]], list(`time_post_change` = query_item))
      }

      # explore
      for (query_item in `time_post_change_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_change_units when calling IgvfApi$InVitroSystems. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_change_units"]] <- c(query_params[["time_post_change_units"]], list(`time_post_change_units` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery`) {
        query_params[["time_post_library_delivery"]] <- c(query_params[["time_post_library_delivery"]], list(`time_post_library_delivery` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_library_delivery_units when calling IgvfApi$InVitroSystems. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_library_delivery_units"]] <- c(query_params[["time_post_library_delivery_units"]], list(`time_post_library_delivery_units` = query_item))
      }

      # explore
      for (query_item in `treatments_id`) {
        query_params[["treatments.@id"]] <- c(query_params[["treatments.@id"]], list(`treatments.@id` = query_item))
      }

      # explore
      for (query_item in `treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for treatments_purpose when calling IgvfApi$InVitroSystems. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["treatments.purpose"]] <- c(query_params[["treatments.purpose"]], list(`treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `treatments_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for treatments_status when calling IgvfApi$InVitroSystems. Must be [archived, deleted, in progress, released].")
        }
        query_params[["treatments.status"]] <- c(query_params[["treatments.status"]], list(`treatments.status` = query_item))
      }

      # explore
      for (query_item in `treatments_summary`) {
        query_params[["treatments.summary"]] <- c(query_params[["treatments.summary"]], list(`treatments.summary` = query_item))
      }

      # explore
      for (query_item in `treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for treatments_treatment_type when calling IgvfApi$InVitroSystems. Must be [chemical, protein, environmental].")
        }
        query_params[["treatments.treatment_type"]] <- c(query_params[["treatments.treatment_type"]], list(`treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age`) {
        query_params[["upper_bound_age"]] <- c(query_params[["upper_bound_age"]], list(`upper_bound_age` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age_in_hours`) {
        query_params[["upper_bound_age_in_hours"]] <- c(query_params[["upper_bound_age_in_hours"]], list(`upper_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/in-vitro-systems/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "InVitroSystemResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the InstitutionalCertificate collection.
    #'
    #' @description
    #' List items in the InstitutionalCertificate collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param certificate_identifier (optional) Filter by certificate_identifier
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param data_use_limitation (optional) Filter by data_use_limitation
    #' @param data_use_limitation_modifiers (optional) Filter by data_use_limitation_modifiers
    #' @param description (optional) Filter by description
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param samples (optional) Filter by samples
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param urls (optional) Filter by urls
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return InstitutionalCertificateResults
    #' @export
    InstitutionalCertificates = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, certificate_identifier = NULL, controlled_access = NULL, creation_timestamp = NULL, data_use_limitation = NULL, data_use_limitation_modifiers = NULL, description = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, release_timestamp = NULL, samples = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, urls = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$InstitutionalCertificatesWithHttpInfo(frame, query, limit, sort, id, aliases, award_id, award_component, certificate_identifier, controlled_access, creation_timestamp, data_use_limitation, data_use_limitation_modifiers, description, lab_id, lab_title, notes, release_timestamp, samples, status, submitted_by_id, submitted_by_title, submitter_comment, summary, urls, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the InstitutionalCertificate collection.
    #'
    #' @description
    #' List items in the InstitutionalCertificate collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param certificate_identifier (optional) Filter by certificate_identifier
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param data_use_limitation (optional) Filter by data_use_limitation
    #' @param data_use_limitation_modifiers (optional) Filter by data_use_limitation_modifiers
    #' @param description (optional) Filter by description
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param samples (optional) Filter by samples
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param urls (optional) Filter by urls
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (InstitutionalCertificateResults) with additional information such as HTTP status code, headers
    #' @export
    InstitutionalCertificatesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, certificate_identifier = NULL, controlled_access = NULL, creation_timestamp = NULL, data_use_limitation = NULL, data_use_limitation_modifiers = NULL, description = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, release_timestamp = NULL, samples = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, urls = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$InstitutionalCertificates. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$InstitutionalCertificates. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `certificate_identifier`) {
        query_params[["certificate_identifier"]] <- c(query_params[["certificate_identifier"]], list(`certificate_identifier` = query_item))
      }

      # explore
      for (query_item in `controlled_access`) {
        query_params[["controlled_access"]] <- c(query_params[["controlled_access"]], list(`controlled_access` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `data_use_limitation`) {
        # validate enum values
        if (!(query_item %in% c("DS", "GRU", "HMB", "other"))) {
          stop("Invalid value for data_use_limitation when calling IgvfApi$InstitutionalCertificates. Must be [DS, GRU, HMB, other].")
        }
        query_params[["data_use_limitation"]] <- c(query_params[["data_use_limitation"]], list(`data_use_limitation` = query_item))
      }

      # explore
      for (query_item in `data_use_limitation_modifiers`) {
        # validate enum values
        if (!(query_item %in% c("COL", "GSO", "IRB", "MDS", "NPU", "PUB"))) {
          stop("Invalid value for data_use_limitation_modifiers when calling IgvfApi$InstitutionalCertificates. Must be [COL, GSO, IRB, MDS, NPU, PUB].")
        }
        query_params[["data_use_limitation_modifiers"]] <- c(query_params[["data_use_limitation_modifiers"]], list(`data_use_limitation_modifiers` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `samples`) {
        query_params[["samples"]] <- c(query_params[["samples"]], list(`samples` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$InstitutionalCertificates. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `urls`) {
        query_params[["urls"]] <- c(query_params[["urls"]], list(`urls` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/institutional-certificates/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "InstitutionalCertificateResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Lab collection.
    #'
    #' @description
    #' List items in the Lab collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param awards_id (optional) Filter by awards.@id
    #' @param awards_component (optional) Filter by awards.component
    #' @param awards_name (optional) Filter by awards.name
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param institute_label (optional) Filter by institute_label
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param pi (optional) Filter by pi
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return LabResults
    #' @export
    Labs = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, awards_id = NULL, awards_component = NULL, awards_name = NULL, creation_timestamp = NULL, description = NULL, institute_label = NULL, name = NULL, notes = NULL, pi = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$LabsWithHttpInfo(frame, query, limit, sort, id, aliases, awards_id, awards_component, awards_name, creation_timestamp, description, institute_label, name, notes, pi, status, submitted_by_id, submitted_by_title, submitter_comment, summary, title, url, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Lab collection.
    #'
    #' @description
    #' List items in the Lab collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param awards_id (optional) Filter by awards.@id
    #' @param awards_component (optional) Filter by awards.component
    #' @param awards_name (optional) Filter by awards.name
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param institute_label (optional) Filter by institute_label
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param pi (optional) Filter by pi
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (LabResults) with additional information such as HTTP status code, headers
    #' @export
    LabsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, awards_id = NULL, awards_component = NULL, awards_name = NULL, creation_timestamp = NULL, description = NULL, institute_label = NULL, name = NULL, notes = NULL, pi = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }
























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Labs. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `awards_id`) {
        query_params[["awards.@id"]] <- c(query_params[["awards.@id"]], list(`awards.@id` = query_item))
      }

      # explore
      for (query_item in `awards_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for awards_component when calling IgvfApi$Labs. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["awards.component"]] <- c(query_params[["awards.component"]], list(`awards.component` = query_item))
      }

      # explore
      for (query_item in `awards_name`) {
        query_params[["awards.name"]] <- c(query_params[["awards.name"]], list(`awards.name` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `institute_label`) {
        query_params[["institute_label"]] <- c(query_params[["institute_label"]], list(`institute_label` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `pi`) {
        query_params[["pi"]] <- c(query_params[["pi"]], list(`pi` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("current", "deleted", "disabled"))) {
          stop("Invalid value for status when calling IgvfApi$Labs. Must be [current, deleted, disabled].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/labs/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "LabResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the MatrixFile collection.
    #'
    #' @description
    #' List items in the MatrixFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_summary (optional) Filter by content_summary
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param dimension1 (optional) Filter by dimension1
    #' @param dimension2 (optional) Filter by dimension2
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param reference_files (optional) Filter by reference_files
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return MatrixFileResults
    #' @export
    MatrixFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_summary = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, dimension1 = NULL, dimension2 = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, reference_files = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$MatrixFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, award_id, award_component, collections, content_md5sum, content_summary, content_type, creation_timestamp, dbxrefs, derived_from, description, dimension1, dimension2, documents, file_format, file_format_specifications, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, reference_files, release_timestamp, revoke_detail, s3_uri, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the MatrixFile collection.
    #'
    #' @description
    #' List items in the MatrixFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_summary (optional) Filter by content_summary
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param dimension1 (optional) Filter by dimension1
    #' @param dimension2 (optional) Filter by dimension2
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param reference_files (optional) Filter by reference_files
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (MatrixFileResults) with additional information such as HTTP status code, headers
    #' @export
    MatrixFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_summary = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, dimension1 = NULL, dimension2 = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, reference_files = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }

















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$MatrixFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$MatrixFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$MatrixFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_summary`) {
        query_params[["content_summary"]] <- c(query_params[["content_summary"]], list(`content_summary` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `dimension1`) {
        # validate enum values
        if (!(query_item %in% c("cell", "fragment", "gene", "time", "treatment", "variant", "genomic position"))) {
          stop("Invalid value for dimension1 when calling IgvfApi$MatrixFiles. Must be [cell, fragment, gene, time, treatment, variant, genomic position].")
        }
        query_params[["dimension1"]] <- c(query_params[["dimension1"]], list(`dimension1` = query_item))
      }

      # explore
      for (query_item in `dimension2`) {
        # validate enum values
        if (!(query_item %in% c("cell", "fragment", "gene", "time", "treatment", "variant", "genomic position"))) {
          stop("Invalid value for dimension2 when calling IgvfApi$MatrixFiles. Must be [cell, fragment, gene, time, treatment, variant, genomic position].")
        }
        query_params[["dimension2"]] <- c(query_params[["dimension2"]], list(`dimension2` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("h5ad", "hdf5", "mtx", "tar", "hic"))) {
          stop("Invalid value for file_format when calling IgvfApi$MatrixFiles. Must be [h5ad, hdf5, mtx, tar, hic].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `reference_files`) {
        query_params[["reference_files"]] <- c(query_params[["reference_files"]], list(`reference_files` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$MatrixFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$MatrixFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/matrix-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "MatrixFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the MeasurementSet collection.
    #'
    #' @description
    #' List items in the MeasurementSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assay_term_id (optional) Filter by assay_term.@id
    #' @param assay_term_term_name (optional) Filter by assay_term.term_name
    #' @param auxiliary_sets_id (optional) Filter by auxiliary_sets.@id
    #' @param auxiliary_sets_accession (optional) Filter by auxiliary_sets.accession
    #' @param auxiliary_sets_aliases (optional) Filter by auxiliary_sets.aliases
    #' @param auxiliary_sets_file_set_type (optional) Filter by auxiliary_sets.file_set_type
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_file_sets_id (optional) Filter by control_file_sets.@id
    #' @param control_file_sets_accession (optional) Filter by control_file_sets.accession
    #' @param control_file_sets_aliases (optional) Filter by control_file_sets.aliases
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param external_image_url (optional) Filter by external_image_url
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform_id (optional) Filter by files.sequencing_platform.@id
    #' @param files_sequencing_platform_term_name (optional) Filter by files.sequencing_platform.term_name
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param library_construction_platform_id (optional) Filter by library_construction_platform.@id
    #' @param library_construction_platform_term_name (optional) Filter by library_construction_platform.term_name
    #' @param multiome_size (optional) Filter by multiome_size
    #' @param notes (optional) Filter by notes
    #' @param preferred_assay_title (optional) Filter by preferred_assay_title
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param related_multiome_datasets_id (optional) Filter by related_multiome_datasets.@id
    #' @param related_multiome_datasets_accession (optional) Filter by related_multiome_datasets.accession
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments_id (optional) Filter by samples.cell_fate_change_treatments.@id
    #' @param samples_cell_fate_change_treatments_purpose (optional) Filter by samples.cell_fate_change_treatments.purpose
    #' @param samples_cell_fate_change_treatments_summary (optional) Filter by samples.cell_fate_change_treatments.summary
    #' @param samples_cell_fate_change_treatments_treatment_type (optional) Filter by samples.cell_fate_change_treatments.treatment_type
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets_id (optional) Filter by samples.construct_library_sets.@id
    #' @param samples_construct_library_sets_accession (optional) Filter by samples.construct_library_sets.accession
    #' @param samples_construct_library_sets_file_set_type (optional) Filter by samples.construct_library_sets.file_set_type
    #' @param samples_construct_library_sets_small_scale_gene_list_id (optional) Filter by samples.construct_library_sets.small_scale_gene_list.@id
    #' @param samples_construct_library_sets_small_scale_gene_list_geneid (optional) Filter by samples.construct_library_sets.small_scale_gene_list.geneid
    #' @param samples_construct_library_sets_small_scale_gene_list_name (optional) Filter by samples.construct_library_sets.small_scale_gene_list.name
    #' @param samples_construct_library_sets_small_scale_gene_list_summary (optional) Filter by samples.construct_library_sets.small_scale_gene_list.summary
    #' @param samples_construct_library_sets_small_scale_gene_list_symbol (optional) Filter by samples.construct_library_sets.small_scale_gene_list.symbol
    #' @param samples_construct_library_sets_summary (optional) Filter by samples.construct_library_sets.summary
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications_id (optional) Filter by samples.modifications.@id
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments_id (optional) Filter by samples.treatments.@id
    #' @param samples_treatments_purpose (optional) Filter by samples.treatments.purpose
    #' @param samples_treatments_summary (optional) Filter by samples.treatments.summary
    #' @param samples_treatments_treatment_type (optional) Filter by samples.treatments.treatment_type
    #' @param sequencing_library_types (optional) Filter by sequencing_library_types
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param targeted_genes_id (optional) Filter by targeted_genes.@id
    #' @param targeted_genes_geneid (optional) Filter by targeted_genes.geneid
    #' @param targeted_genes_name (optional) Filter by targeted_genes.name
    #' @param targeted_genes_symbol (optional) Filter by targeted_genes.symbol
    #' @param targeted_genes_synonyms (optional) Filter by targeted_genes.synonyms
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return MeasurementSetResults
    #' @export
    MeasurementSets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assay_term_id = NULL, assay_term_term_name = NULL, auxiliary_sets_id = NULL, auxiliary_sets_accession = NULL, auxiliary_sets_aliases = NULL, auxiliary_sets_file_set_type = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_file_sets_id = NULL, control_file_sets_accession = NULL, control_file_sets_aliases = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, external_image_url = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform_id = NULL, files_sequencing_platform_term_name = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, lab_id = NULL, lab_title = NULL, library_construction_platform_id = NULL, library_construction_platform_term_name = NULL, multiome_size = NULL, notes = NULL, preferred_assay_title = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, related_multiome_datasets_id = NULL, related_multiome_datasets_accession = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments_id = NULL, samples_cell_fate_change_treatments_purpose = NULL, samples_cell_fate_change_treatments_summary = NULL, samples_cell_fate_change_treatments_treatment_type = NULL, samples_classifications = NULL, samples_construct_library_sets_id = NULL, samples_construct_library_sets_accession = NULL, samples_construct_library_sets_file_set_type = NULL, samples_construct_library_sets_small_scale_gene_list_id = NULL, samples_construct_library_sets_small_scale_gene_list_geneid = NULL, samples_construct_library_sets_small_scale_gene_list_name = NULL, samples_construct_library_sets_small_scale_gene_list_summary = NULL, samples_construct_library_sets_small_scale_gene_list_symbol = NULL, samples_construct_library_sets_summary = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications_id = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments_id = NULL, samples_treatments_purpose = NULL, samples_treatments_summary = NULL, samples_treatments_treatment_type = NULL, sequencing_library_types = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, targeted_genes_id = NULL, targeted_genes_geneid = NULL, targeted_genes_name = NULL, targeted_genes_symbol = NULL, targeted_genes_synonyms = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$MeasurementSetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, assay_term_id, assay_term_term_name, auxiliary_sets_id, auxiliary_sets_accession, auxiliary_sets_aliases, auxiliary_sets_file_set_type, award_id, award_component, award_contact_pi_id, award_contact_pi_title, award_title, collections, control_file_sets_id, control_file_sets_accession, control_file_sets_aliases, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, dbxrefs, description, documents, donors_id, donors_accession, donors_aliases, donors_sex, donors_status, donors_taxa, external_image_url, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_creation_timestamp, files_file_format, files_file_size, files_href, files_s3_uri, files_sequencing_platform_id, files_sequencing_platform_term_name, files_submitted_file_name, files_upload_status, input_file_set_for, lab_id, lab_title, library_construction_platform_id, library_construction_platform_term_name, multiome_size, notes, preferred_assay_title, protocols, publication_identifiers, publications, related_multiome_datasets_id, related_multiome_datasets_accession, release_timestamp, revoke_detail, samples_id, samples_accession, samples_aliases, samples_cell_fate_change_treatments_id, samples_cell_fate_change_treatments_purpose, samples_cell_fate_change_treatments_summary, samples_cell_fate_change_treatments_treatment_type, samples_classifications, samples_construct_library_sets_id, samples_construct_library_sets_accession, samples_construct_library_sets_file_set_type, samples_construct_library_sets_small_scale_gene_list_id, samples_construct_library_sets_small_scale_gene_list_geneid, samples_construct_library_sets_small_scale_gene_list_name, samples_construct_library_sets_small_scale_gene_list_summary, samples_construct_library_sets_small_scale_gene_list_symbol, samples_construct_library_sets_summary, samples_disease_terms_id, samples_disease_terms_term_name, samples_modifications_id, samples_sample_terms_id, samples_sample_terms_aliases, samples_sample_terms_status, samples_sample_terms_summary, samples_sample_terms_term_name, samples_status, samples_summary, samples_targeted_sample_term_id, samples_targeted_sample_term_term_name, samples_taxa, samples_treatments_id, samples_treatments_purpose, samples_treatments_summary, samples_treatments_treatment_type, sequencing_library_types, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, targeted_genes_id, targeted_genes_geneid, targeted_genes_name, targeted_genes_symbol, targeted_genes_synonyms, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the MeasurementSet collection.
    #'
    #' @description
    #' List items in the MeasurementSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assay_term_id (optional) Filter by assay_term.@id
    #' @param assay_term_term_name (optional) Filter by assay_term.term_name
    #' @param auxiliary_sets_id (optional) Filter by auxiliary_sets.@id
    #' @param auxiliary_sets_accession (optional) Filter by auxiliary_sets.accession
    #' @param auxiliary_sets_aliases (optional) Filter by auxiliary_sets.aliases
    #' @param auxiliary_sets_file_set_type (optional) Filter by auxiliary_sets.file_set_type
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_file_sets_id (optional) Filter by control_file_sets.@id
    #' @param control_file_sets_accession (optional) Filter by control_file_sets.accession
    #' @param control_file_sets_aliases (optional) Filter by control_file_sets.aliases
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param external_image_url (optional) Filter by external_image_url
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform_id (optional) Filter by files.sequencing_platform.@id
    #' @param files_sequencing_platform_term_name (optional) Filter by files.sequencing_platform.term_name
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param library_construction_platform_id (optional) Filter by library_construction_platform.@id
    #' @param library_construction_platform_term_name (optional) Filter by library_construction_platform.term_name
    #' @param multiome_size (optional) Filter by multiome_size
    #' @param notes (optional) Filter by notes
    #' @param preferred_assay_title (optional) Filter by preferred_assay_title
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param related_multiome_datasets_id (optional) Filter by related_multiome_datasets.@id
    #' @param related_multiome_datasets_accession (optional) Filter by related_multiome_datasets.accession
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments_id (optional) Filter by samples.cell_fate_change_treatments.@id
    #' @param samples_cell_fate_change_treatments_purpose (optional) Filter by samples.cell_fate_change_treatments.purpose
    #' @param samples_cell_fate_change_treatments_summary (optional) Filter by samples.cell_fate_change_treatments.summary
    #' @param samples_cell_fate_change_treatments_treatment_type (optional) Filter by samples.cell_fate_change_treatments.treatment_type
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets_id (optional) Filter by samples.construct_library_sets.@id
    #' @param samples_construct_library_sets_accession (optional) Filter by samples.construct_library_sets.accession
    #' @param samples_construct_library_sets_file_set_type (optional) Filter by samples.construct_library_sets.file_set_type
    #' @param samples_construct_library_sets_small_scale_gene_list_id (optional) Filter by samples.construct_library_sets.small_scale_gene_list.@id
    #' @param samples_construct_library_sets_small_scale_gene_list_geneid (optional) Filter by samples.construct_library_sets.small_scale_gene_list.geneid
    #' @param samples_construct_library_sets_small_scale_gene_list_name (optional) Filter by samples.construct_library_sets.small_scale_gene_list.name
    #' @param samples_construct_library_sets_small_scale_gene_list_summary (optional) Filter by samples.construct_library_sets.small_scale_gene_list.summary
    #' @param samples_construct_library_sets_small_scale_gene_list_symbol (optional) Filter by samples.construct_library_sets.small_scale_gene_list.symbol
    #' @param samples_construct_library_sets_summary (optional) Filter by samples.construct_library_sets.summary
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications_id (optional) Filter by samples.modifications.@id
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments_id (optional) Filter by samples.treatments.@id
    #' @param samples_treatments_purpose (optional) Filter by samples.treatments.purpose
    #' @param samples_treatments_summary (optional) Filter by samples.treatments.summary
    #' @param samples_treatments_treatment_type (optional) Filter by samples.treatments.treatment_type
    #' @param sequencing_library_types (optional) Filter by sequencing_library_types
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param targeted_genes_id (optional) Filter by targeted_genes.@id
    #' @param targeted_genes_geneid (optional) Filter by targeted_genes.geneid
    #' @param targeted_genes_name (optional) Filter by targeted_genes.name
    #' @param targeted_genes_symbol (optional) Filter by targeted_genes.symbol
    #' @param targeted_genes_synonyms (optional) Filter by targeted_genes.synonyms
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (MeasurementSetResults) with additional information such as HTTP status code, headers
    #' @export
    MeasurementSetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assay_term_id = NULL, assay_term_term_name = NULL, auxiliary_sets_id = NULL, auxiliary_sets_accession = NULL, auxiliary_sets_aliases = NULL, auxiliary_sets_file_set_type = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_file_sets_id = NULL, control_file_sets_accession = NULL, control_file_sets_aliases = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, external_image_url = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform_id = NULL, files_sequencing_platform_term_name = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, lab_id = NULL, lab_title = NULL, library_construction_platform_id = NULL, library_construction_platform_term_name = NULL, multiome_size = NULL, notes = NULL, preferred_assay_title = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, related_multiome_datasets_id = NULL, related_multiome_datasets_accession = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments_id = NULL, samples_cell_fate_change_treatments_purpose = NULL, samples_cell_fate_change_treatments_summary = NULL, samples_cell_fate_change_treatments_treatment_type = NULL, samples_classifications = NULL, samples_construct_library_sets_id = NULL, samples_construct_library_sets_accession = NULL, samples_construct_library_sets_file_set_type = NULL, samples_construct_library_sets_small_scale_gene_list_id = NULL, samples_construct_library_sets_small_scale_gene_list_geneid = NULL, samples_construct_library_sets_small_scale_gene_list_name = NULL, samples_construct_library_sets_small_scale_gene_list_summary = NULL, samples_construct_library_sets_small_scale_gene_list_symbol = NULL, samples_construct_library_sets_summary = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications_id = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments_id = NULL, samples_treatments_purpose = NULL, samples_treatments_summary = NULL, samples_treatments_treatment_type = NULL, sequencing_library_types = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, targeted_genes_id = NULL, targeted_genes_geneid = NULL, targeted_genes_name = NULL, targeted_genes_symbol = NULL, targeted_genes_synonyms = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }










































































      if (length(`samples_classifications`) > 2) {
        stop("Invalid length for `samples_classifications` when calling IgvfApi$MeasurementSets, number of items must be less than or equal to 2.")
      }








































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$MeasurementSets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `assay_term_id`) {
        query_params[["assay_term.@id"]] <- c(query_params[["assay_term.@id"]], list(`assay_term.@id` = query_item))
      }

      # explore
      for (query_item in `assay_term_term_name`) {
        query_params[["assay_term.term_name"]] <- c(query_params[["assay_term.term_name"]], list(`assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `auxiliary_sets_id`) {
        query_params[["auxiliary_sets.@id"]] <- c(query_params[["auxiliary_sets.@id"]], list(`auxiliary_sets.@id` = query_item))
      }

      # explore
      for (query_item in `auxiliary_sets_accession`) {
        query_params[["auxiliary_sets.accession"]] <- c(query_params[["auxiliary_sets.accession"]], list(`auxiliary_sets.accession` = query_item))
      }

      # explore
      for (query_item in `auxiliary_sets_aliases`) {
        query_params[["auxiliary_sets.aliases"]] <- c(query_params[["auxiliary_sets.aliases"]], list(`auxiliary_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `auxiliary_sets_file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("cell hashing", "cell sorting", "circularized RNA barcode detection", "gRNA sequencing", "oligo-conjugated lipids", "quantification DNA barcode sequencing", "variant sequencing"))) {
          stop("Invalid value for auxiliary_sets_file_set_type when calling IgvfApi$MeasurementSets. Must be [cell hashing, cell sorting, circularized RNA barcode detection, gRNA sequencing, oligo-conjugated lipids, quantification DNA barcode sequencing, variant sequencing].")
        }
        query_params[["auxiliary_sets.file_set_type"]] <- c(query_params[["auxiliary_sets.file_set_type"]], list(`auxiliary_sets.file_set_type` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$MeasurementSets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_id`) {
        query_params[["award.contact_pi.@id"]] <- c(query_params[["award.contact_pi.@id"]], list(`award.contact_pi.@id` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_title`) {
        query_params[["award.contact_pi.title"]] <- c(query_params[["award.contact_pi.title"]], list(`award.contact_pi.title` = query_item))
      }

      # explore
      for (query_item in `award_title`) {
        query_params[["award.title"]] <- c(query_params[["award.title"]], list(`award.title` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$MeasurementSets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_file_sets_id`) {
        query_params[["control_file_sets.@id"]] <- c(query_params[["control_file_sets.@id"]], list(`control_file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `control_file_sets_accession`) {
        query_params[["control_file_sets.accession"]] <- c(query_params[["control_file_sets.accession"]], list(`control_file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `control_file_sets_aliases`) {
        query_params[["control_file_sets.aliases"]] <- c(query_params[["control_file_sets.aliases"]], list(`control_file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors_id`) {
        query_params[["donors.@id"]] <- c(query_params[["donors.@id"]], list(`donors.@id` = query_item))
      }

      # explore
      for (query_item in `donors_accession`) {
        query_params[["donors.accession"]] <- c(query_params[["donors.accession"]], list(`donors.accession` = query_item))
      }

      # explore
      for (query_item in `donors_aliases`) {
        query_params[["donors.aliases"]] <- c(query_params[["donors.aliases"]], list(`donors.aliases` = query_item))
      }

      # explore
      for (query_item in `donors_sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for donors_sex when calling IgvfApi$MeasurementSets. Must be [male, female, unspecified].")
        }
        query_params[["donors.sex"]] <- c(query_params[["donors.sex"]], list(`donors.sex` = query_item))
      }

      # explore
      for (query_item in `donors_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for donors_status when calling IgvfApi$MeasurementSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["donors.status"]] <- c(query_params[["donors.status"]], list(`donors.status` = query_item))
      }

      # explore
      for (query_item in `donors_taxa`) {
        query_params[["donors.taxa"]] <- c(query_params[["donors.taxa"]], list(`donors.taxa` = query_item))
      }

      # explore
      for (query_item in `external_image_url`) {
        query_params[["external_image_url"]] <- c(query_params[["external_image_url"]], list(`external_image_url` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("experimental data"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$MeasurementSets. Must be [experimental data].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_creation_timestamp`) {
        query_params[["files.creation_timestamp"]] <- c(query_params[["files.creation_timestamp"]], list(`files.creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_file_size`) {
        query_params[["files.file_size"]] <- c(query_params[["files.file_size"]], list(`files.file_size` = query_item))
      }

      # explore
      for (query_item in `files_href`) {
        query_params[["files.href"]] <- c(query_params[["files.href"]], list(`files.href` = query_item))
      }

      # explore
      for (query_item in `files_s3_uri`) {
        query_params[["files.s3_uri"]] <- c(query_params[["files.s3_uri"]], list(`files.s3_uri` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform_id`) {
        query_params[["files.sequencing_platform.@id"]] <- c(query_params[["files.sequencing_platform.@id"]], list(`files.sequencing_platform.@id` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform_term_name`) {
        query_params[["files.sequencing_platform.term_name"]] <- c(query_params[["files.sequencing_platform.term_name"]], list(`files.sequencing_platform.term_name` = query_item))
      }

      # explore
      for (query_item in `files_submitted_file_name`) {
        query_params[["files.submitted_file_name"]] <- c(query_params[["files.submitted_file_name"]], list(`files.submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$MeasurementSets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `library_construction_platform_id`) {
        query_params[["library_construction_platform.@id"]] <- c(query_params[["library_construction_platform.@id"]], list(`library_construction_platform.@id` = query_item))
      }

      # explore
      for (query_item in `library_construction_platform_term_name`) {
        query_params[["library_construction_platform.term_name"]] <- c(query_params[["library_construction_platform.term_name"]], list(`library_construction_platform.term_name` = query_item))
      }

      # explore
      for (query_item in `multiome_size`) {
        query_params[["multiome_size"]] <- c(query_params[["multiome_size"]], list(`multiome_size` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `preferred_assay_title`) {
        # validate enum values
        if (!(query_item %in% c("10x multiome", "10x multiome with MULTI-seq", "AAV-MPRA", "ATAC-seq", "CERES-seq", "Cell painting", "CRISPR FlowFISH", "DOGMA-seq", "Histone ChIP-seq", "Hi-C", "lentiMPRA", "MERFISH", "MIAA", "mN2H", "MPRA", "MPRA (scQer)", "MULTI-seq", "Parse SPLiT-seq", "Perturb-seq", "RNA-seq", "SGE", "scATAC-seq", "scNT-seq", "scNT-seq2", "scRNA-seq", "semi-qY2H", "SHARE-seq", "smFISH", "snATAC-seq", "snmC-Seq2", "snMCT-seq", "snM3C-seq", "snRNA-seq", "SUPERSTARR", "TAP-seq", "TF ChIP-seq", "VAMP-seq", "Variant FlowFISH", "Variant painting", "Y2H", "yN2H"))) {
          stop("Invalid value for preferred_assay_title when calling IgvfApi$MeasurementSets. Must be [10x multiome, 10x multiome with MULTI-seq, AAV-MPRA, ATAC-seq, CERES-seq, Cell painting, CRISPR FlowFISH, DOGMA-seq, Histone ChIP-seq, Hi-C, lentiMPRA, MERFISH, MIAA, mN2H, MPRA, MPRA (scQer), MULTI-seq, Parse SPLiT-seq, Perturb-seq, RNA-seq, SGE, scATAC-seq, scNT-seq, scNT-seq2, scRNA-seq, semi-qY2H, SHARE-seq, smFISH, snATAC-seq, snmC-Seq2, snMCT-seq, snM3C-seq, snRNA-seq, SUPERSTARR, TAP-seq, TF ChIP-seq, VAMP-seq, Variant FlowFISH, Variant painting, Y2H, yN2H].")
        }
        query_params[["preferred_assay_title"]] <- c(query_params[["preferred_assay_title"]], list(`preferred_assay_title` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `related_multiome_datasets_id`) {
        query_params[["related_multiome_datasets.@id"]] <- c(query_params[["related_multiome_datasets.@id"]], list(`related_multiome_datasets.@id` = query_item))
      }

      # explore
      for (query_item in `related_multiome_datasets_accession`) {
        query_params[["related_multiome_datasets.accession"]] <- c(query_params[["related_multiome_datasets.accession"]], list(`related_multiome_datasets.accession` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `samples_id`) {
        query_params[["samples.@id"]] <- c(query_params[["samples.@id"]], list(`samples.@id` = query_item))
      }

      # explore
      for (query_item in `samples_accession`) {
        query_params[["samples.accession"]] <- c(query_params[["samples.accession"]], list(`samples.accession` = query_item))
      }

      # explore
      for (query_item in `samples_aliases`) {
        query_params[["samples.aliases"]] <- c(query_params[["samples.aliases"]], list(`samples.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments_id`) {
        query_params[["samples.cell_fate_change_treatments.@id"]] <- c(query_params[["samples.cell_fate_change_treatments.@id"]], list(`samples.cell_fate_change_treatments.@id` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for samples_cell_fate_change_treatments_purpose when calling IgvfApi$MeasurementSets. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["samples.cell_fate_change_treatments.purpose"]] <- c(query_params[["samples.cell_fate_change_treatments.purpose"]], list(`samples.cell_fate_change_treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments_summary`) {
        query_params[["samples.cell_fate_change_treatments.summary"]] <- c(query_params[["samples.cell_fate_change_treatments.summary"]], list(`samples.cell_fate_change_treatments.summary` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for samples_cell_fate_change_treatments_treatment_type when calling IgvfApi$MeasurementSets. Must be [chemical, protein, environmental].")
        }
        query_params[["samples.cell_fate_change_treatments.treatment_type"]] <- c(query_params[["samples.cell_fate_change_treatments.treatment_type"]], list(`samples.cell_fate_change_treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `samples_classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for samples_classifications when calling IgvfApi$MeasurementSets. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["samples.classifications"]] <- c(query_params[["samples.classifications"]], list(`samples.classifications` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_id`) {
        query_params[["samples.construct_library_sets.@id"]] <- c(query_params[["samples.construct_library_sets.@id"]], list(`samples.construct_library_sets.@id` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_accession`) {
        query_params[["samples.construct_library_sets.accession"]] <- c(query_params[["samples.construct_library_sets.accession"]], list(`samples.construct_library_sets.accession` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("guide library", "reporter library", "expression vector library", "editing template library"))) {
          stop("Invalid value for samples_construct_library_sets_file_set_type when calling IgvfApi$MeasurementSets. Must be [guide library, reporter library, expression vector library, editing template library].")
        }
        query_params[["samples.construct_library_sets.file_set_type"]] <- c(query_params[["samples.construct_library_sets.file_set_type"]], list(`samples.construct_library_sets.file_set_type` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_small_scale_gene_list_id`) {
        query_params[["samples.construct_library_sets.small_scale_gene_list.@id"]] <- c(query_params[["samples.construct_library_sets.small_scale_gene_list.@id"]], list(`samples.construct_library_sets.small_scale_gene_list.@id` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_small_scale_gene_list_geneid`) {
        query_params[["samples.construct_library_sets.small_scale_gene_list.geneid"]] <- c(query_params[["samples.construct_library_sets.small_scale_gene_list.geneid"]], list(`samples.construct_library_sets.small_scale_gene_list.geneid` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_small_scale_gene_list_name`) {
        query_params[["samples.construct_library_sets.small_scale_gene_list.name"]] <- c(query_params[["samples.construct_library_sets.small_scale_gene_list.name"]], list(`samples.construct_library_sets.small_scale_gene_list.name` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_small_scale_gene_list_summary`) {
        query_params[["samples.construct_library_sets.small_scale_gene_list.summary"]] <- c(query_params[["samples.construct_library_sets.small_scale_gene_list.summary"]], list(`samples.construct_library_sets.small_scale_gene_list.summary` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_small_scale_gene_list_symbol`) {
        query_params[["samples.construct_library_sets.small_scale_gene_list.symbol"]] <- c(query_params[["samples.construct_library_sets.small_scale_gene_list.symbol"]], list(`samples.construct_library_sets.small_scale_gene_list.symbol` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_summary`) {
        query_params[["samples.construct_library_sets.summary"]] <- c(query_params[["samples.construct_library_sets.summary"]], list(`samples.construct_library_sets.summary` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_id`) {
        query_params[["samples.disease_terms.@id"]] <- c(query_params[["samples.disease_terms.@id"]], list(`samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_term_name`) {
        query_params[["samples.disease_terms.term_name"]] <- c(query_params[["samples.disease_terms.term_name"]], list(`samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_modifications_id`) {
        query_params[["samples.modifications.@id"]] <- c(query_params[["samples.modifications.@id"]], list(`samples.modifications.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_id`) {
        query_params[["samples.sample_terms.@id"]] <- c(query_params[["samples.sample_terms.@id"]], list(`samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_aliases`) {
        query_params[["samples.sample_terms.aliases"]] <- c(query_params[["samples.sample_terms.aliases"]], list(`samples.sample_terms.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for samples_sample_terms_status when calling IgvfApi$MeasurementSets. Must be [archived, deleted, in progress, released].")
        }
        query_params[["samples.sample_terms.status"]] <- c(query_params[["samples.sample_terms.status"]], list(`samples.sample_terms.status` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_summary`) {
        query_params[["samples.sample_terms.summary"]] <- c(query_params[["samples.sample_terms.summary"]], list(`samples.sample_terms.summary` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_term_name`) {
        query_params[["samples.sample_terms.term_name"]] <- c(query_params[["samples.sample_terms.term_name"]], list(`samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for samples_status when calling IgvfApi$MeasurementSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["samples.status"]] <- c(query_params[["samples.status"]], list(`samples.status` = query_item))
      }

      # explore
      for (query_item in `samples_summary`) {
        query_params[["samples.summary"]] <- c(query_params[["samples.summary"]], list(`samples.summary` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_id`) {
        query_params[["samples.targeted_sample_term.@id"]] <- c(query_params[["samples.targeted_sample_term.@id"]], list(`samples.targeted_sample_term.@id` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_term_name`) {
        query_params[["samples.targeted_sample_term.term_name"]] <- c(query_params[["samples.targeted_sample_term.term_name"]], list(`samples.targeted_sample_term.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for samples_taxa when calling IgvfApi$MeasurementSets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["samples.taxa"]] <- c(query_params[["samples.taxa"]], list(`samples.taxa` = query_item))
      }

      # explore
      for (query_item in `samples_treatments_id`) {
        query_params[["samples.treatments.@id"]] <- c(query_params[["samples.treatments.@id"]], list(`samples.treatments.@id` = query_item))
      }

      # explore
      for (query_item in `samples_treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for samples_treatments_purpose when calling IgvfApi$MeasurementSets. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["samples.treatments.purpose"]] <- c(query_params[["samples.treatments.purpose"]], list(`samples.treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `samples_treatments_summary`) {
        query_params[["samples.treatments.summary"]] <- c(query_params[["samples.treatments.summary"]], list(`samples.treatments.summary` = query_item))
      }

      # explore
      for (query_item in `samples_treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for samples_treatments_treatment_type when calling IgvfApi$MeasurementSets. Must be [chemical, protein, environmental].")
        }
        query_params[["samples.treatments.treatment_type"]] <- c(query_params[["samples.treatments.treatment_type"]], list(`samples.treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `sequencing_library_types`) {
        # validate enum values
        if (!(query_item %in% c("direct RNA", "exome capture", "mRNA enriched", "rRNA depleted", "polyA depleted", "polyA enriched"))) {
          stop("Invalid value for sequencing_library_types when calling IgvfApi$MeasurementSets. Must be [direct RNA, exome capture, mRNA enriched, rRNA depleted, polyA depleted, polyA enriched].")
        }
        query_params[["sequencing_library_types"]] <- c(query_params[["sequencing_library_types"]], list(`sequencing_library_types` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$MeasurementSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `targeted_genes_id`) {
        query_params[["targeted_genes.@id"]] <- c(query_params[["targeted_genes.@id"]], list(`targeted_genes.@id` = query_item))
      }

      # explore
      for (query_item in `targeted_genes_geneid`) {
        query_params[["targeted_genes.geneid"]] <- c(query_params[["targeted_genes.geneid"]], list(`targeted_genes.geneid` = query_item))
      }

      # explore
      for (query_item in `targeted_genes_name`) {
        query_params[["targeted_genes.name"]] <- c(query_params[["targeted_genes.name"]], list(`targeted_genes.name` = query_item))
      }

      # explore
      for (query_item in `targeted_genes_symbol`) {
        query_params[["targeted_genes.symbol"]] <- c(query_params[["targeted_genes.symbol"]], list(`targeted_genes.symbol` = query_item))
      }

      # explore
      for (query_item in `targeted_genes_synonyms`) {
        query_params[["targeted_genes.synonyms"]] <- c(query_params[["targeted_genes.synonyms"]], list(`targeted_genes.synonyms` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/measurement-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "MeasurementSetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the ModelFile collection.
    #'
    #' @description
    #' List items in the ModelFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ModelFileResults
    #' @export
    ModelFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$ModelFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, anvil_url, award_id, award_component, collections, content_md5sum, content_type, controlled_access, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, release_timestamp, revoke_detail, s3_uri, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the ModelFile collection.
    #'
    #' @description
    #' List items in the ModelFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ModelFileResults) with additional information such as HTTP status code, headers
    #' @export
    ModelFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$ModelFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `anvil_url`) {
        query_params[["anvil_url"]] <- c(query_params[["anvil_url"]], list(`anvil_url` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$ModelFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$ModelFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `controlled_access`) {
        query_params[["controlled_access"]] <- c(query_params[["controlled_access"]], list(`controlled_access` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("hdf5", "json", "tar", "tsv"))) {
          stop("Invalid value for file_format when calling IgvfApi$ModelFiles. Must be [hdf5, json, tar, tsv].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$ModelFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$ModelFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/model-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ModelFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the ModelSet collection.
    #'
    #' @description
    #' List items in the ModelSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assessed_genes (optional) Filter by assessed_genes
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param input_file_sets_id (optional) Filter by input_file_sets.@id
    #' @param input_file_sets_accession (optional) Filter by input_file_sets.accession
    #' @param input_file_sets_aliases (optional) Filter by input_file_sets.aliases
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param model_name (optional) Filter by model_name
    #' @param model_version (optional) Filter by model_version
    #' @param model_zoo_location (optional) Filter by model_zoo_location
    #' @param notes (optional) Filter by notes
    #' @param prediction_objects (optional) Filter by prediction_objects
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param software_version (optional) Filter by software_version
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ModelSetResults
    #' @export
    ModelSets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assessed_genes = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, input_file_sets_id = NULL, input_file_sets_accession = NULL, input_file_sets_aliases = NULL, lab_id = NULL, lab_title = NULL, model_name = NULL, model_version = NULL, model_zoo_location = NULL, notes = NULL, prediction_objects = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, software_version = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$ModelSetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, assessed_genes, award_id, award_component, award_contact_pi_id, award_contact_pi_title, award_title, collections, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, dbxrefs, description, documents, donors_id, donors_accession, donors_aliases, donors_sex, donors_status, donors_taxa, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_creation_timestamp, files_file_format, files_file_size, files_href, files_s3_uri, files_sequencing_platform, files_submitted_file_name, files_upload_status, input_file_set_for, input_file_sets_id, input_file_sets_accession, input_file_sets_aliases, lab_id, lab_title, model_name, model_version, model_zoo_location, notes, prediction_objects, publication_identifiers, publications, release_timestamp, revoke_detail, samples_id, samples_accession, samples_aliases, samples_cell_fate_change_treatments, samples_classifications, samples_construct_library_sets, samples_disease_terms_id, samples_disease_terms_term_name, samples_modifications, samples_sample_terms_id, samples_sample_terms_aliases, samples_sample_terms_status, samples_sample_terms_summary, samples_sample_terms_term_name, samples_status, samples_summary, samples_targeted_sample_term_id, samples_targeted_sample_term_term_name, samples_taxa, samples_treatments, software_version, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, url, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the ModelSet collection.
    #'
    #' @description
    #' List items in the ModelSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param assessed_genes (optional) Filter by assessed_genes
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param input_file_sets_id (optional) Filter by input_file_sets.@id
    #' @param input_file_sets_accession (optional) Filter by input_file_sets.accession
    #' @param input_file_sets_aliases (optional) Filter by input_file_sets.aliases
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param model_name (optional) Filter by model_name
    #' @param model_version (optional) Filter by model_version
    #' @param model_zoo_location (optional) Filter by model_zoo_location
    #' @param notes (optional) Filter by notes
    #' @param prediction_objects (optional) Filter by prediction_objects
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets (optional) Filter by samples.construct_library_sets
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param software_version (optional) Filter by software_version
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ModelSetResults) with additional information such as HTTP status code, headers
    #' @export
    ModelSetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, assessed_genes = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, input_file_sets_id = NULL, input_file_sets_accession = NULL, input_file_sets_aliases = NULL, lab_id = NULL, lab_title = NULL, model_name = NULL, model_version = NULL, model_zoo_location = NULL, notes = NULL, prediction_objects = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, software_version = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





























































      if (length(`samples_classifications`) > 2) {
        stop("Invalid length for `samples_classifications` when calling IgvfApi$ModelSets, number of items must be less than or equal to 2.")
      }




      if (length(`samples_modifications`) > 2) {
        stop("Invalid length for `samples_modifications` when calling IgvfApi$ModelSets, number of items must be less than or equal to 2.")
      }





















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$ModelSets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `assessed_genes`) {
        query_params[["assessed_genes"]] <- c(query_params[["assessed_genes"]], list(`assessed_genes` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$ModelSets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_id`) {
        query_params[["award.contact_pi.@id"]] <- c(query_params[["award.contact_pi.@id"]], list(`award.contact_pi.@id` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_title`) {
        query_params[["award.contact_pi.title"]] <- c(query_params[["award.contact_pi.title"]], list(`award.contact_pi.title` = query_item))
      }

      # explore
      for (query_item in `award_title`) {
        query_params[["award.title"]] <- c(query_params[["award.title"]], list(`award.title` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$ModelSets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors_id`) {
        query_params[["donors.@id"]] <- c(query_params[["donors.@id"]], list(`donors.@id` = query_item))
      }

      # explore
      for (query_item in `donors_accession`) {
        query_params[["donors.accession"]] <- c(query_params[["donors.accession"]], list(`donors.accession` = query_item))
      }

      # explore
      for (query_item in `donors_aliases`) {
        query_params[["donors.aliases"]] <- c(query_params[["donors.aliases"]], list(`donors.aliases` = query_item))
      }

      # explore
      for (query_item in `donors_sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for donors_sex when calling IgvfApi$ModelSets. Must be [male, female, unspecified].")
        }
        query_params[["donors.sex"]] <- c(query_params[["donors.sex"]], list(`donors.sex` = query_item))
      }

      # explore
      for (query_item in `donors_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for donors_status when calling IgvfApi$ModelSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["donors.status"]] <- c(query_params[["donors.status"]], list(`donors.status` = query_item))
      }

      # explore
      for (query_item in `donors_taxa`) {
        query_params[["donors.taxa"]] <- c(query_params[["donors.taxa"]], list(`donors.taxa` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("neural network", "random forest", "decision tree", "support vector machine", "variant binding effect"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$ModelSets. Must be [neural network, random forest, decision tree, support vector machine, variant binding effect].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_creation_timestamp`) {
        query_params[["files.creation_timestamp"]] <- c(query_params[["files.creation_timestamp"]], list(`files.creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_file_size`) {
        query_params[["files.file_size"]] <- c(query_params[["files.file_size"]], list(`files.file_size` = query_item))
      }

      # explore
      for (query_item in `files_href`) {
        query_params[["files.href"]] <- c(query_params[["files.href"]], list(`files.href` = query_item))
      }

      # explore
      for (query_item in `files_s3_uri`) {
        query_params[["files.s3_uri"]] <- c(query_params[["files.s3_uri"]], list(`files.s3_uri` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform`) {
        query_params[["files.sequencing_platform"]] <- c(query_params[["files.sequencing_platform"]], list(`files.sequencing_platform` = query_item))
      }

      # explore
      for (query_item in `files_submitted_file_name`) {
        query_params[["files.submitted_file_name"]] <- c(query_params[["files.submitted_file_name"]], list(`files.submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$ModelSets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_id`) {
        query_params[["input_file_sets.@id"]] <- c(query_params[["input_file_sets.@id"]], list(`input_file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_accession`) {
        query_params[["input_file_sets.accession"]] <- c(query_params[["input_file_sets.accession"]], list(`input_file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `input_file_sets_aliases`) {
        query_params[["input_file_sets.aliases"]] <- c(query_params[["input_file_sets.aliases"]], list(`input_file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `model_name`) {
        query_params[["model_name"]] <- c(query_params[["model_name"]], list(`model_name` = query_item))
      }

      # explore
      for (query_item in `model_version`) {
        query_params[["model_version"]] <- c(query_params[["model_version"]], list(`model_version` = query_item))
      }

      # explore
      for (query_item in `model_zoo_location`) {
        query_params[["model_zoo_location"]] <- c(query_params[["model_zoo_location"]], list(`model_zoo_location` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `prediction_objects`) {
        # validate enum values
        if (!(query_item %in% c("coding variants", "non-coding variants", "all variants", "regulatory elements", "genes"))) {
          stop("Invalid value for prediction_objects when calling IgvfApi$ModelSets. Must be [coding variants, non-coding variants, all variants, regulatory elements, genes].")
        }
        query_params[["prediction_objects"]] <- c(query_params[["prediction_objects"]], list(`prediction_objects` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `samples_id`) {
        query_params[["samples.@id"]] <- c(query_params[["samples.@id"]], list(`samples.@id` = query_item))
      }

      # explore
      for (query_item in `samples_accession`) {
        query_params[["samples.accession"]] <- c(query_params[["samples.accession"]], list(`samples.accession` = query_item))
      }

      # explore
      for (query_item in `samples_aliases`) {
        query_params[["samples.aliases"]] <- c(query_params[["samples.aliases"]], list(`samples.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments`) {
        query_params[["samples.cell_fate_change_treatments"]] <- c(query_params[["samples.cell_fate_change_treatments"]], list(`samples.cell_fate_change_treatments` = query_item))
      }

      # explore
      for (query_item in `samples_classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for samples_classifications when calling IgvfApi$ModelSets. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["samples.classifications"]] <- c(query_params[["samples.classifications"]], list(`samples.classifications` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets`) {
        query_params[["samples.construct_library_sets"]] <- c(query_params[["samples.construct_library_sets"]], list(`samples.construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_id`) {
        query_params[["samples.disease_terms.@id"]] <- c(query_params[["samples.disease_terms.@id"]], list(`samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_term_name`) {
        query_params[["samples.disease_terms.term_name"]] <- c(query_params[["samples.disease_terms.term_name"]], list(`samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_modifications`) {
        query_params[["samples.modifications"]] <- c(query_params[["samples.modifications"]], list(`samples.modifications` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_id`) {
        query_params[["samples.sample_terms.@id"]] <- c(query_params[["samples.sample_terms.@id"]], list(`samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_aliases`) {
        query_params[["samples.sample_terms.aliases"]] <- c(query_params[["samples.sample_terms.aliases"]], list(`samples.sample_terms.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for samples_sample_terms_status when calling IgvfApi$ModelSets. Must be [archived, deleted, in progress, released].")
        }
        query_params[["samples.sample_terms.status"]] <- c(query_params[["samples.sample_terms.status"]], list(`samples.sample_terms.status` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_summary`) {
        query_params[["samples.sample_terms.summary"]] <- c(query_params[["samples.sample_terms.summary"]], list(`samples.sample_terms.summary` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_term_name`) {
        query_params[["samples.sample_terms.term_name"]] <- c(query_params[["samples.sample_terms.term_name"]], list(`samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for samples_status when calling IgvfApi$ModelSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["samples.status"]] <- c(query_params[["samples.status"]], list(`samples.status` = query_item))
      }

      # explore
      for (query_item in `samples_summary`) {
        query_params[["samples.summary"]] <- c(query_params[["samples.summary"]], list(`samples.summary` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_id`) {
        query_params[["samples.targeted_sample_term.@id"]] <- c(query_params[["samples.targeted_sample_term.@id"]], list(`samples.targeted_sample_term.@id` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_term_name`) {
        query_params[["samples.targeted_sample_term.term_name"]] <- c(query_params[["samples.targeted_sample_term.term_name"]], list(`samples.targeted_sample_term.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for samples_taxa when calling IgvfApi$ModelSets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["samples.taxa"]] <- c(query_params[["samples.taxa"]], list(`samples.taxa` = query_item))
      }

      # explore
      for (query_item in `samples_treatments`) {
        query_params[["samples.treatments"]] <- c(query_params[["samples.treatments"]], list(`samples.treatments` = query_item))
      }

      # explore
      for (query_item in `software_version`) {
        query_params[["software_version"]] <- c(query_params[["software_version"]], list(`software_version` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$ModelSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/model-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ModelSetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the MultiplexedSample collection.
    #'
    #' @description
    #' List items in the MultiplexedSample collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param barcode_sample_map (optional) Filter by barcode_sample_map
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets_id (optional) Filter by construct_library_sets.@id
    #' @param construct_library_sets_accession (optional) Filter by construct_library_sets.accession
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates (optional) Filter by institutional_certificates
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param multiplexed_samples_id (optional) Filter by multiplexed_samples.@id
    #' @param multiplexed_samples_accession (optional) Filter by multiplexed_samples.accession
    #' @param multiplexed_samples_construct_library_sets (optional) Filter by multiplexed_samples.construct_library_sets
    #' @param multiplexed_samples_disease_terms_id (optional) Filter by multiplexed_samples.disease_terms.@id
    #' @param multiplexed_samples_disease_terms_term_name (optional) Filter by multiplexed_samples.disease_terms.term_name
    #' @param multiplexed_samples_donors_id (optional) Filter by multiplexed_samples.donors.@id
    #' @param multiplexed_samples_donors_accession (optional) Filter by multiplexed_samples.donors.accession
    #' @param multiplexed_samples_sample_terms_id (optional) Filter by multiplexed_samples.sample_terms.@id
    #' @param multiplexed_samples_sample_terms_term_name (optional) Filter by multiplexed_samples.sample_terms.term_name
    #' @param multiplexed_samples_status (optional) Filter by multiplexed_samples.status
    #' @param multiplexed_samples_summary (optional) Filter by multiplexed_samples.summary
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return MultiplexedSampleResults
    #' @export
    MultiplexedSamples = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, barcode_sample_map = NULL, biomarkers = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets_id = NULL, construct_library_sets_accession = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates = NULL, lab_id = NULL, lab_title = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, multiplexed_samples_id = NULL, multiplexed_samples_accession = NULL, multiplexed_samples_construct_library_sets = NULL, multiplexed_samples_disease_terms_id = NULL, multiplexed_samples_disease_terms_term_name = NULL, multiplexed_samples_donors_id = NULL, multiplexed_samples_donors_accession = NULL, multiplexed_samples_sample_terms_id = NULL, multiplexed_samples_sample_terms_term_name = NULL, multiplexed_samples_status = NULL, multiplexed_samples_summary = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$MultiplexedSamplesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, award_id, award_component, barcode_sample_map, biomarkers, cellular_sub_pool, classifications, collections, construct_library_sets_id, construct_library_sets_accession, creation_timestamp, date_obtained, dbxrefs, description, disease_terms_id, disease_terms_term_name, documents, donors, file_sets_id, file_sets_accession, file_sets_aliases, file_sets_assay_term_term_name, file_sets_lab_title, file_sets_status, file_sets_summary, institutional_certificates, lab_id, lab_title, modifications_id, modifications_status, modifications_summary, moi, multiplexed_in_id, multiplexed_in_accession, multiplexed_samples_id, multiplexed_samples_accession, multiplexed_samples_construct_library_sets, multiplexed_samples_disease_terms_id, multiplexed_samples_disease_terms_term_name, multiplexed_samples_donors_id, multiplexed_samples_donors_accession, multiplexed_samples_sample_terms_id, multiplexed_samples_sample_terms_term_name, multiplexed_samples_status, multiplexed_samples_summary, notes, nucleic_acid_delivery, origin_of, protocols, publication_identifiers, publications, release_timestamp, revoke_detail, sample_terms_id, sample_terms_term_name, sorted_fractions, sorted_from_id, sorted_from_accession, sorted_from_detail, sources_id, starting_amount, starting_amount_units, status, submitted_by_id, submitted_by_title, submitter_comment, summary, time_post_library_delivery, time_post_library_delivery_units, treatments_id, treatments_purpose, treatments_status, treatments_summary, treatments_treatment_type, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the MultiplexedSample collection.
    #'
    #' @description
    #' List items in the MultiplexedSample collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param barcode_sample_map (optional) Filter by barcode_sample_map
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets_id (optional) Filter by construct_library_sets.@id
    #' @param construct_library_sets_accession (optional) Filter by construct_library_sets.accession
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates (optional) Filter by institutional_certificates
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param multiplexed_samples_id (optional) Filter by multiplexed_samples.@id
    #' @param multiplexed_samples_accession (optional) Filter by multiplexed_samples.accession
    #' @param multiplexed_samples_construct_library_sets (optional) Filter by multiplexed_samples.construct_library_sets
    #' @param multiplexed_samples_disease_terms_id (optional) Filter by multiplexed_samples.disease_terms.@id
    #' @param multiplexed_samples_disease_terms_term_name (optional) Filter by multiplexed_samples.disease_terms.term_name
    #' @param multiplexed_samples_donors_id (optional) Filter by multiplexed_samples.donors.@id
    #' @param multiplexed_samples_donors_accession (optional) Filter by multiplexed_samples.donors.accession
    #' @param multiplexed_samples_sample_terms_id (optional) Filter by multiplexed_samples.sample_terms.@id
    #' @param multiplexed_samples_sample_terms_term_name (optional) Filter by multiplexed_samples.sample_terms.term_name
    #' @param multiplexed_samples_status (optional) Filter by multiplexed_samples.status
    #' @param multiplexed_samples_summary (optional) Filter by multiplexed_samples.summary
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (MultiplexedSampleResults) with additional information such as HTTP status code, headers
    #' @export
    MultiplexedSamplesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, barcode_sample_map = NULL, biomarkers = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets_id = NULL, construct_library_sets_accession = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates = NULL, lab_id = NULL, lab_title = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, multiplexed_samples_id = NULL, multiplexed_samples_accession = NULL, multiplexed_samples_construct_library_sets = NULL, multiplexed_samples_disease_terms_id = NULL, multiplexed_samples_disease_terms_term_name = NULL, multiplexed_samples_donors_id = NULL, multiplexed_samples_donors_accession = NULL, multiplexed_samples_sample_terms_id = NULL, multiplexed_samples_sample_terms_term_name = NULL, multiplexed_samples_status = NULL, multiplexed_samples_summary = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





















































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$MultiplexedSamples. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$MultiplexedSamples. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `barcode_sample_map`) {
        query_params[["barcode_sample_map"]] <- c(query_params[["barcode_sample_map"]], list(`barcode_sample_map` = query_item))
      }

      # explore
      for (query_item in `biomarkers`) {
        query_params[["biomarkers"]] <- c(query_params[["biomarkers"]], list(`biomarkers` = query_item))
      }

      # explore
      for (query_item in `cellular_sub_pool`) {
        query_params[["cellular_sub_pool"]] <- c(query_params[["cellular_sub_pool"]], list(`cellular_sub_pool` = query_item))
      }

      # explore
      for (query_item in `classifications`) {
        query_params[["classifications"]] <- c(query_params[["classifications"]], list(`classifications` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$MultiplexedSamples. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets_id`) {
        query_params[["construct_library_sets.@id"]] <- c(query_params[["construct_library_sets.@id"]], list(`construct_library_sets.@id` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets_accession`) {
        query_params[["construct_library_sets.accession"]] <- c(query_params[["construct_library_sets.accession"]], list(`construct_library_sets.accession` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_obtained`) {
        query_params[["date_obtained"]] <- c(query_params[["date_obtained"]], list(`date_obtained` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `disease_terms_id`) {
        query_params[["disease_terms.@id"]] <- c(query_params[["disease_terms.@id"]], list(`disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `disease_terms_term_name`) {
        query_params[["disease_terms.term_name"]] <- c(query_params[["disease_terms.term_name"]], list(`disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors`) {
        query_params[["donors"]] <- c(query_params[["donors"]], list(`donors` = query_item))
      }

      # explore
      for (query_item in `file_sets_id`) {
        query_params[["file_sets.@id"]] <- c(query_params[["file_sets.@id"]], list(`file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `file_sets_accession`) {
        query_params[["file_sets.accession"]] <- c(query_params[["file_sets.accession"]], list(`file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `file_sets_aliases`) {
        query_params[["file_sets.aliases"]] <- c(query_params[["file_sets.aliases"]], list(`file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `file_sets_assay_term_term_name`) {
        query_params[["file_sets.assay_term.term_name"]] <- c(query_params[["file_sets.assay_term.term_name"]], list(`file_sets.assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `file_sets_lab_title`) {
        query_params[["file_sets.lab.title"]] <- c(query_params[["file_sets.lab.title"]], list(`file_sets.lab.title` = query_item))
      }

      # explore
      for (query_item in `file_sets_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for file_sets_status when calling IgvfApi$MultiplexedSamples. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["file_sets.status"]] <- c(query_params[["file_sets.status"]], list(`file_sets.status` = query_item))
      }

      # explore
      for (query_item in `file_sets_summary`) {
        query_params[["file_sets.summary"]] <- c(query_params[["file_sets.summary"]], list(`file_sets.summary` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates`) {
        query_params[["institutional_certificates"]] <- c(query_params[["institutional_certificates"]], list(`institutional_certificates` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `modifications_id`) {
        query_params[["modifications.@id"]] <- c(query_params[["modifications.@id"]], list(`modifications.@id` = query_item))
      }

      # explore
      for (query_item in `modifications_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for modifications_status when calling IgvfApi$MultiplexedSamples. Must be [archived, deleted, in progress, released].")
        }
        query_params[["modifications.status"]] <- c(query_params[["modifications.status"]], list(`modifications.status` = query_item))
      }

      # explore
      for (query_item in `modifications_summary`) {
        query_params[["modifications.summary"]] <- c(query_params[["modifications.summary"]], list(`modifications.summary` = query_item))
      }

      # explore
      for (query_item in `moi`) {
        query_params[["moi"]] <- c(query_params[["moi"]], list(`moi` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_id`) {
        query_params[["multiplexed_in.@id"]] <- c(query_params[["multiplexed_in.@id"]], list(`multiplexed_in.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_accession`) {
        query_params[["multiplexed_in.accession"]] <- c(query_params[["multiplexed_in.accession"]], list(`multiplexed_in.accession` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_id`) {
        query_params[["multiplexed_samples.@id"]] <- c(query_params[["multiplexed_samples.@id"]], list(`multiplexed_samples.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_accession`) {
        query_params[["multiplexed_samples.accession"]] <- c(query_params[["multiplexed_samples.accession"]], list(`multiplexed_samples.accession` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_construct_library_sets`) {
        query_params[["multiplexed_samples.construct_library_sets"]] <- c(query_params[["multiplexed_samples.construct_library_sets"]], list(`multiplexed_samples.construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_disease_terms_id`) {
        query_params[["multiplexed_samples.disease_terms.@id"]] <- c(query_params[["multiplexed_samples.disease_terms.@id"]], list(`multiplexed_samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_disease_terms_term_name`) {
        query_params[["multiplexed_samples.disease_terms.term_name"]] <- c(query_params[["multiplexed_samples.disease_terms.term_name"]], list(`multiplexed_samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_donors_id`) {
        query_params[["multiplexed_samples.donors.@id"]] <- c(query_params[["multiplexed_samples.donors.@id"]], list(`multiplexed_samples.donors.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_donors_accession`) {
        query_params[["multiplexed_samples.donors.accession"]] <- c(query_params[["multiplexed_samples.donors.accession"]], list(`multiplexed_samples.donors.accession` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_sample_terms_id`) {
        query_params[["multiplexed_samples.sample_terms.@id"]] <- c(query_params[["multiplexed_samples.sample_terms.@id"]], list(`multiplexed_samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_sample_terms_term_name`) {
        query_params[["multiplexed_samples.sample_terms.term_name"]] <- c(query_params[["multiplexed_samples.sample_terms.term_name"]], list(`multiplexed_samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for multiplexed_samples_status when calling IgvfApi$MultiplexedSamples. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["multiplexed_samples.status"]] <- c(query_params[["multiplexed_samples.status"]], list(`multiplexed_samples.status` = query_item))
      }

      # explore
      for (query_item in `multiplexed_samples_summary`) {
        query_params[["multiplexed_samples.summary"]] <- c(query_params[["multiplexed_samples.summary"]], list(`multiplexed_samples.summary` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `nucleic_acid_delivery`) {
        # validate enum values
        if (!(query_item %in% c("transfection", "adenoviral transduction", "lentiviral transduction"))) {
          stop("Invalid value for nucleic_acid_delivery when calling IgvfApi$MultiplexedSamples. Must be [transfection, adenoviral transduction, lentiviral transduction].")
        }
        query_params[["nucleic_acid_delivery"]] <- c(query_params[["nucleic_acid_delivery"]], list(`nucleic_acid_delivery` = query_item))
      }

      # explore
      for (query_item in `origin_of`) {
        query_params[["origin_of"]] <- c(query_params[["origin_of"]], list(`origin_of` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sample_terms_id`) {
        query_params[["sample_terms.@id"]] <- c(query_params[["sample_terms.@id"]], list(`sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `sample_terms_term_name`) {
        query_params[["sample_terms.term_name"]] <- c(query_params[["sample_terms.term_name"]], list(`sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `sorted_fractions`) {
        query_params[["sorted_fractions"]] <- c(query_params[["sorted_fractions"]], list(`sorted_fractions` = query_item))
      }

      # explore
      for (query_item in `sorted_from_id`) {
        query_params[["sorted_from.@id"]] <- c(query_params[["sorted_from.@id"]], list(`sorted_from.@id` = query_item))
      }

      # explore
      for (query_item in `sorted_from_accession`) {
        query_params[["sorted_from.accession"]] <- c(query_params[["sorted_from.accession"]], list(`sorted_from.accession` = query_item))
      }

      # explore
      for (query_item in `sorted_from_detail`) {
        query_params[["sorted_from_detail"]] <- c(query_params[["sorted_from_detail"]], list(`sorted_from_detail` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `starting_amount`) {
        query_params[["starting_amount"]] <- c(query_params[["starting_amount"]], list(`starting_amount` = query_item))
      }

      # explore
      for (query_item in `starting_amount_units`) {
        # validate enum values
        if (!(query_item %in% c("cells", "cells/ml", "g", "items", "mg", "whole animals", "whole embryos", "μg", "ng"))) {
          stop("Invalid value for starting_amount_units when calling IgvfApi$MultiplexedSamples. Must be [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng].")
        }
        query_params[["starting_amount_units"]] <- c(query_params[["starting_amount_units"]], list(`starting_amount_units` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$MultiplexedSamples. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery`) {
        query_params[["time_post_library_delivery"]] <- c(query_params[["time_post_library_delivery"]], list(`time_post_library_delivery` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_library_delivery_units when calling IgvfApi$MultiplexedSamples. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_library_delivery_units"]] <- c(query_params[["time_post_library_delivery_units"]], list(`time_post_library_delivery_units` = query_item))
      }

      # explore
      for (query_item in `treatments_id`) {
        query_params[["treatments.@id"]] <- c(query_params[["treatments.@id"]], list(`treatments.@id` = query_item))
      }

      # explore
      for (query_item in `treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for treatments_purpose when calling IgvfApi$MultiplexedSamples. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["treatments.purpose"]] <- c(query_params[["treatments.purpose"]], list(`treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `treatments_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for treatments_status when calling IgvfApi$MultiplexedSamples. Must be [archived, deleted, in progress, released].")
        }
        query_params[["treatments.status"]] <- c(query_params[["treatments.status"]], list(`treatments.status` = query_item))
      }

      # explore
      for (query_item in `treatments_summary`) {
        query_params[["treatments.summary"]] <- c(query_params[["treatments.summary"]], list(`treatments.summary` = query_item))
      }

      # explore
      for (query_item in `treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for treatments_treatment_type when calling IgvfApi$MultiplexedSamples. Must be [chemical, protein, environmental].")
        }
        query_params[["treatments.treatment_type"]] <- c(query_params[["treatments.treatment_type"]], list(`treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/multiplexed-samples/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "MultiplexedSampleResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the OpenReadingFrame collection.
    #'
    #' @description
    #' List items in the OpenReadingFrame collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award (optional) Filter by award
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param gene_id (optional) Filter by gene.@id
    #' @param gene_geneid (optional) Filter by gene.geneid
    #' @param gene_symbol (optional) Filter by gene.symbol
    #' @param lab (optional) Filter by lab
    #' @param notes (optional) Filter by notes
    #' @param orf_id (optional) Filter by orf_id
    #' @param pct_coverage_orf (optional) Filter by pct_coverage_orf
    #' @param pct_coverage_protein (optional) Filter by pct_coverage_protein
    #' @param pct_identical_protein (optional) Filter by pct_identical_protein
    #' @param protein_id (optional) Filter by protein_id
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by (optional) Filter by submitted_by
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return OpenReadingFrameResults
    #' @export
    OpenReadingFrames = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, gene_id = NULL, gene_geneid = NULL, gene_symbol = NULL, lab = NULL, notes = NULL, orf_id = NULL, pct_coverage_orf = NULL, pct_coverage_protein = NULL, pct_identical_protein = NULL, protein_id = NULL, release_timestamp = NULL, status = NULL, submitted_by = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$OpenReadingFramesWithHttpInfo(frame, query, limit, sort, id, aliases, award, creation_timestamp, dbxrefs, description, gene_id, gene_geneid, gene_symbol, lab, notes, orf_id, pct_coverage_orf, pct_coverage_protein, pct_identical_protein, protein_id, release_timestamp, status, submitted_by, submitter_comment, summary, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the OpenReadingFrame collection.
    #'
    #' @description
    #' List items in the OpenReadingFrame collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award (optional) Filter by award
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param gene_id (optional) Filter by gene.@id
    #' @param gene_geneid (optional) Filter by gene.geneid
    #' @param gene_symbol (optional) Filter by gene.symbol
    #' @param lab (optional) Filter by lab
    #' @param notes (optional) Filter by notes
    #' @param orf_id (optional) Filter by orf_id
    #' @param pct_coverage_orf (optional) Filter by pct_coverage_orf
    #' @param pct_coverage_protein (optional) Filter by pct_coverage_protein
    #' @param pct_identical_protein (optional) Filter by pct_identical_protein
    #' @param protein_id (optional) Filter by protein_id
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by (optional) Filter by submitted_by
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (OpenReadingFrameResults) with additional information such as HTTP status code, headers
    #' @export
    OpenReadingFramesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, gene_id = NULL, gene_geneid = NULL, gene_symbol = NULL, lab = NULL, notes = NULL, orf_id = NULL, pct_coverage_orf = NULL, pct_coverage_protein = NULL, pct_identical_protein = NULL, protein_id = NULL, release_timestamp = NULL, status = NULL, submitted_by = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$OpenReadingFrames. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award`) {
        query_params[["award"]] <- c(query_params[["award"]], list(`award` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `gene_id`) {
        query_params[["gene.@id"]] <- c(query_params[["gene.@id"]], list(`gene.@id` = query_item))
      }

      # explore
      for (query_item in `gene_geneid`) {
        query_params[["gene.geneid"]] <- c(query_params[["gene.geneid"]], list(`gene.geneid` = query_item))
      }

      # explore
      for (query_item in `gene_symbol`) {
        query_params[["gene.symbol"]] <- c(query_params[["gene.symbol"]], list(`gene.symbol` = query_item))
      }

      # explore
      for (query_item in `lab`) {
        query_params[["lab"]] <- c(query_params[["lab"]], list(`lab` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `orf_id`) {
        query_params[["orf_id"]] <- c(query_params[["orf_id"]], list(`orf_id` = query_item))
      }

      # explore
      for (query_item in `pct_coverage_orf`) {
        query_params[["pct_coverage_orf"]] <- c(query_params[["pct_coverage_orf"]], list(`pct_coverage_orf` = query_item))
      }

      # explore
      for (query_item in `pct_coverage_protein`) {
        query_params[["pct_coverage_protein"]] <- c(query_params[["pct_coverage_protein"]], list(`pct_coverage_protein` = query_item))
      }

      # explore
      for (query_item in `pct_identical_protein`) {
        query_params[["pct_identical_protein"]] <- c(query_params[["pct_identical_protein"]], list(`pct_identical_protein` = query_item))
      }

      # explore
      for (query_item in `protein_id`) {
        query_params[["protein_id"]] <- c(query_params[["protein_id"]], list(`protein_id` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$OpenReadingFrames. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by`) {
        query_params[["submitted_by"]] <- c(query_params[["submitted_by"]], list(`submitted_by` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/open-reading-frames/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "OpenReadingFrameResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Page collection.
    #'
    #' @description
    #' List items in the Page collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award (optional) Filter by award
    #' @param canonical_uri (optional) Filter by canonical_uri
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param lab (optional) Filter by lab
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param parent (optional) Filter by parent
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PageResults
    #' @export
    Pages = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award = NULL, canonical_uri = NULL, creation_timestamp = NULL, description = NULL, lab = NULL, name = NULL, notes = NULL, parent = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$PagesWithHttpInfo(frame, query, limit, sort, id, aliases, award, canonical_uri, creation_timestamp, description, lab, name, notes, parent, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, title, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Page collection.
    #'
    #' @description
    #' List items in the Page collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award (optional) Filter by award
    #' @param canonical_uri (optional) Filter by canonical_uri
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param lab (optional) Filter by lab
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param parent (optional) Filter by parent
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PageResults) with additional information such as HTTP status code, headers
    #' @export
    PagesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award = NULL, canonical_uri = NULL, creation_timestamp = NULL, description = NULL, lab = NULL, name = NULL, notes = NULL, parent = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Pages. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award`) {
        query_params[["award"]] <- c(query_params[["award"]], list(`award` = query_item))
      }

      # explore
      for (query_item in `canonical_uri`) {
        query_params[["canonical_uri"]] <- c(query_params[["canonical_uri"]], list(`canonical_uri` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `lab`) {
        query_params[["lab"]] <- c(query_params[["lab"]], list(`lab` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `parent`) {
        query_params[["parent"]] <- c(query_params[["parent"]], list(`parent` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Pages. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/pages/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PageResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the PhenotypeTerm collection.
    #'
    #' @description
    #' List items in the PhenotypeTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param ontology (optional) Filter by ontology
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PhenotypeTermResults
    #' @export
    PhenotypeTerms = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, creation_timestamp = NULL, deprecated_ntr_terms = NULL, description = NULL, is_a = NULL, name = NULL, notes = NULL, ontology = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$PhenotypeTermsWithHttpInfo(frame, query, limit, sort, id, aliases, ancestors, creation_timestamp, deprecated_ntr_terms, description, is_a, name, notes, ontology, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, synonyms, term_id, term_name, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the PhenotypeTerm collection.
    #'
    #' @description
    #' List items in the PhenotypeTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param ontology (optional) Filter by ontology
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PhenotypeTermResults) with additional information such as HTTP status code, headers
    #' @export
    PhenotypeTermsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, creation_timestamp = NULL, deprecated_ntr_terms = NULL, description = NULL, is_a = NULL, name = NULL, notes = NULL, ontology = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }

























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$PhenotypeTerms. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `ancestors`) {
        query_params[["ancestors"]] <- c(query_params[["ancestors"]], list(`ancestors` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `deprecated_ntr_terms`) {
        query_params[["deprecated_ntr_terms"]] <- c(query_params[["deprecated_ntr_terms"]], list(`deprecated_ntr_terms` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `is_a`) {
        query_params[["is_a"]] <- c(query_params[["is_a"]], list(`is_a` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `ontology`) {
        query_params[["ontology"]] <- c(query_params[["ontology"]], list(`ontology` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$PhenotypeTerms. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `synonyms`) {
        query_params[["synonyms"]] <- c(query_params[["synonyms"]], list(`synonyms` = query_item))
      }

      # explore
      for (query_item in `term_id`) {
        query_params[["term_id"]] <- c(query_params[["term_id"]], list(`term_id` = query_item))
      }

      # explore
      for (query_item in `term_name`) {
        query_params[["term_name"]] <- c(query_params[["term_name"]], list(`term_name` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/phenotype-terms/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PhenotypeTermResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the PhenotypicFeature collection.
    #'
    #' @description
    #' List items in the PhenotypicFeature collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param feature_id (optional) Filter by feature.@id
    #' @param feature_term_id (optional) Filter by feature.term_id
    #' @param feature_term_name (optional) Filter by feature.term_name
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param observation_date (optional) Filter by observation_date
    #' @param quantity (optional) Filter by quantity
    #' @param quantity_units (optional) Filter by quantity_units
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PhenotypicFeatureResults
    #' @export
    PhenotypicFeatures = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, feature_id = NULL, feature_term_id = NULL, feature_term_name = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, observation_date = NULL, quantity = NULL, quantity_units = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$PhenotypicFeaturesWithHttpInfo(frame, query, limit, sort, id, aliases, award_id, award_component, creation_timestamp, description, feature_id, feature_term_id, feature_term_name, lab_id, lab_title, notes, observation_date, quantity, quantity_units, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the PhenotypicFeature collection.
    #'
    #' @description
    #' List items in the PhenotypicFeature collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param feature_id (optional) Filter by feature.@id
    #' @param feature_term_id (optional) Filter by feature.term_id
    #' @param feature_term_name (optional) Filter by feature.term_name
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param observation_date (optional) Filter by observation_date
    #' @param quantity (optional) Filter by quantity
    #' @param quantity_units (optional) Filter by quantity_units
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PhenotypicFeatureResults) with additional information such as HTTP status code, headers
    #' @export
    PhenotypicFeaturesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, feature_id = NULL, feature_term_id = NULL, feature_term_name = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, observation_date = NULL, quantity = NULL, quantity_units = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$PhenotypicFeatures. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$PhenotypicFeatures. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `feature_id`) {
        query_params[["feature.@id"]] <- c(query_params[["feature.@id"]], list(`feature.@id` = query_item))
      }

      # explore
      for (query_item in `feature_term_id`) {
        query_params[["feature.term_id"]] <- c(query_params[["feature.term_id"]], list(`feature.term_id` = query_item))
      }

      # explore
      for (query_item in `feature_term_name`) {
        query_params[["feature.term_name"]] <- c(query_params[["feature.term_name"]], list(`feature.term_name` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `observation_date`) {
        query_params[["observation_date"]] <- c(query_params[["observation_date"]], list(`observation_date` = query_item))
      }

      # explore
      for (query_item in `quantity`) {
        query_params[["quantity"]] <- c(query_params[["quantity"]], list(`quantity` = query_item))
      }

      # explore
      for (query_item in `quantity_units`) {
        # validate enum values
        if (!(query_item %in% c("meter", "micromole", "nanogram", "microgram", "milligram", "gram", "kilogram", "milli-International Unit per milliliter", "picogram per milliliter", "nanogram per milliliter", "milligram per deciliter"))) {
          stop("Invalid value for quantity_units when calling IgvfApi$PhenotypicFeatures. Must be [meter, micromole, nanogram, microgram, milligram, gram, kilogram, milli-International Unit per milliliter, picogram per milliliter, nanogram per milliliter, milligram per deciliter].")
        }
        query_params[["quantity_units"]] <- c(query_params[["quantity_units"]], list(`quantity_units` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$PhenotypicFeatures. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/phenotypic-features/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PhenotypicFeatureResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the PlatformTerm collection.
    #'
    #' @description
    #' List items in the PlatformTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param company (optional) Filter by company
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param ontology (optional) Filter by ontology
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sequencing_kits (optional) Filter by sequencing_kits
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PlatformTermResults
    #' @export
    PlatformTerms = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, company = NULL, creation_timestamp = NULL, deprecated_ntr_terms = NULL, description = NULL, is_a = NULL, name = NULL, notes = NULL, ontology = NULL, release_timestamp = NULL, sequencing_kits = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$PlatformTermsWithHttpInfo(frame, query, limit, sort, id, aliases, ancestors, company, creation_timestamp, deprecated_ntr_terms, description, is_a, name, notes, ontology, release_timestamp, sequencing_kits, status, submitted_by_id, submitted_by_title, submitter_comment, summary, synonyms, term_id, term_name, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the PlatformTerm collection.
    #'
    #' @description
    #' List items in the PlatformTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param company (optional) Filter by company
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param ontology (optional) Filter by ontology
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sequencing_kits (optional) Filter by sequencing_kits
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PlatformTermResults) with additional information such as HTTP status code, headers
    #' @export
    PlatformTermsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, company = NULL, creation_timestamp = NULL, deprecated_ntr_terms = NULL, description = NULL, is_a = NULL, name = NULL, notes = NULL, ontology = NULL, release_timestamp = NULL, sequencing_kits = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$PlatformTerms. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `ancestors`) {
        query_params[["ancestors"]] <- c(query_params[["ancestors"]], list(`ancestors` = query_item))
      }

      # explore
      for (query_item in `company`) {
        # validate enum values
        if (!(query_item %in% c("10X Genomics", "Illumina", "Life Technologies", "Oxford Nanopore Technologies", "Pacific Biosciences", "Parse Biosciences", "Roche"))) {
          stop("Invalid value for company when calling IgvfApi$PlatformTerms. Must be [10X Genomics, Illumina, Life Technologies, Oxford Nanopore Technologies, Pacific Biosciences, Parse Biosciences, Roche].")
        }
        query_params[["company"]] <- c(query_params[["company"]], list(`company` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `deprecated_ntr_terms`) {
        query_params[["deprecated_ntr_terms"]] <- c(query_params[["deprecated_ntr_terms"]], list(`deprecated_ntr_terms` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `is_a`) {
        query_params[["is_a"]] <- c(query_params[["is_a"]], list(`is_a` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `ontology`) {
        query_params[["ontology"]] <- c(query_params[["ontology"]], list(`ontology` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `sequencing_kits`) {
        # validate enum values
        if (!(query_item %in% c("HiSeq SBS Kit v4", "HiSeq SR Cluster Kit v4-cBot-HS", "HiSeq PE Cluster Kit v4-cBot-HS", "HiSeq SR Rapid Cluster Kit v2", "HiSeq PE Rapid Cluster Kit v2", "HiSeq Rapid SBS Kit v2", "HiSeq 3000/4000 SBS Kit", "HiSeq 3000/4000 SR Cluster Kit", "HiSeq 3000/4000 PE Cluster Kit", "MiSeq Reagent Kit v2", "NextSeq 500 Mid Output Kit", "NextSeq 500 High Output Kit", "NextSeq 500 Mid Output v2 Kit", "NextSeq 500 High Output v2 Kit", "NextSeq 500/550 Mid-Output v2.5 Kit", "NextSeq 500/550 High-Output v2.5 Kit", "TG NextSeq 500/550 Mid-Output Kit v2.5", "TG NextSeq 500/550 High-Output Kit v2.5", "NextSeq 1000/2000 P1 Reagent Kit", "NextSeq 1000/2000 P2 Reagent Kit", "NextSeq 1000/2000 P3 Reagent Kit", "NextSeq 1000/2000 P1 XLEAP-SBS Reagent Kit", "NextSeq 1000/2000 P2 XLEAP-SBS Reagent Kit", "NextSeq 2000 P3 XLEAP-SBS Reagent Kit", "NextSeq 2000 P4 XLEAP-SBS Reagent Kit", "NovaSeq 6000 SP Reagent Kit v1.5", "NovaSeq 6000 S1 Reagent Kit v1.5", "NovaSeq 6000 S2 Reagent Kit v1.5", "NovaSeq 6000 S4 Reagent Kit v1.5", "NovaSeq X Series 10B Reagent Kit", "ONT Ligation Sequencing Kit V14", "Sequel sequencing kit 3.0", "Sequel II sequencing kit 2.0"))) {
          stop("Invalid value for sequencing_kits when calling IgvfApi$PlatformTerms. Must be [HiSeq SBS Kit v4, HiSeq SR Cluster Kit v4-cBot-HS, HiSeq PE Cluster Kit v4-cBot-HS, HiSeq SR Rapid Cluster Kit v2, HiSeq PE Rapid Cluster Kit v2, HiSeq Rapid SBS Kit v2, HiSeq 3000/4000 SBS Kit, HiSeq 3000/4000 SR Cluster Kit, HiSeq 3000/4000 PE Cluster Kit, MiSeq Reagent Kit v2, NextSeq 500 Mid Output Kit, NextSeq 500 High Output Kit, NextSeq 500 Mid Output v2 Kit, NextSeq 500 High Output v2 Kit, NextSeq 500/550 Mid-Output v2.5 Kit, NextSeq 500/550 High-Output v2.5 Kit, TG NextSeq 500/550 Mid-Output Kit v2.5, TG NextSeq 500/550 High-Output Kit v2.5, NextSeq 1000/2000 P1 Reagent Kit, NextSeq 1000/2000 P2 Reagent Kit, NextSeq 1000/2000 P3 Reagent Kit, NextSeq 1000/2000 P1 XLEAP-SBS Reagent Kit, NextSeq 1000/2000 P2 XLEAP-SBS Reagent Kit, NextSeq 2000 P3 XLEAP-SBS Reagent Kit, NextSeq 2000 P4 XLEAP-SBS Reagent Kit, NovaSeq 6000 SP Reagent Kit v1.5, NovaSeq 6000 S1 Reagent Kit v1.5, NovaSeq 6000 S2 Reagent Kit v1.5, NovaSeq 6000 S4 Reagent Kit v1.5, NovaSeq X Series 10B Reagent Kit, ONT Ligation Sequencing Kit V14, Sequel sequencing kit 3.0, Sequel II sequencing kit 2.0].")
        }
        query_params[["sequencing_kits"]] <- c(query_params[["sequencing_kits"]], list(`sequencing_kits` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$PlatformTerms. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `synonyms`) {
        query_params[["synonyms"]] <- c(query_params[["synonyms"]], list(`synonyms` = query_item))
      }

      # explore
      for (query_item in `term_id`) {
        query_params[["term_id"]] <- c(query_params[["term_id"]], list(`term_id` = query_item))
      }

      # explore
      for (query_item in `term_name`) {
        query_params[["term_name"]] <- c(query_params[["term_name"]], list(`term_name` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/platform-terms/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PlatformTermResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the PredictionSet collection.
    #'
    #' @description
    #' List items in the PredictionSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param input_file_sets (optional) Filter by input_file_sets
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param large_scale_gene_list_id (optional) Filter by large_scale_gene_list.@id
    #' @param large_scale_gene_list_accession (optional) Filter by large_scale_gene_list.accession
    #' @param large_scale_gene_list_aliases (optional) Filter by large_scale_gene_list.aliases
    #' @param large_scale_loci_list_id (optional) Filter by large_scale_loci_list.@id
    #' @param large_scale_loci_list_accession (optional) Filter by large_scale_loci_list.accession
    #' @param large_scale_loci_list_aliases (optional) Filter by large_scale_loci_list.aliases
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets_id (optional) Filter by samples.construct_library_sets.@id
    #' @param samples_construct_library_sets_accession (optional) Filter by samples.construct_library_sets.accession
    #' @param samples_construct_library_sets_summary (optional) Filter by samples.construct_library_sets.summary
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param scope (optional) Filter by scope
    #' @param small_scale_gene_list_id (optional) Filter by small_scale_gene_list.@id
    #' @param small_scale_gene_list_geneid (optional) Filter by small_scale_gene_list.geneid
    #' @param small_scale_gene_list_name (optional) Filter by small_scale_gene_list.name
    #' @param small_scale_gene_list_symbol (optional) Filter by small_scale_gene_list.symbol
    #' @param small_scale_gene_list_synonyms (optional) Filter by small_scale_gene_list.synonyms
    #' @param small_scale_loci_list (optional) Filter by small_scale_loci_list
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PredictionSetResults
    #' @export
    PredictionSets = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, input_file_sets = NULL, lab_id = NULL, lab_title = NULL, large_scale_gene_list_id = NULL, large_scale_gene_list_accession = NULL, large_scale_gene_list_aliases = NULL, large_scale_loci_list_id = NULL, large_scale_loci_list_accession = NULL, large_scale_loci_list_aliases = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets_id = NULL, samples_construct_library_sets_accession = NULL, samples_construct_library_sets_summary = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, scope = NULL, small_scale_gene_list_id = NULL, small_scale_gene_list_geneid = NULL, small_scale_gene_list_name = NULL, small_scale_gene_list_symbol = NULL, small_scale_gene_list_synonyms = NULL, small_scale_loci_list = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$PredictionSetsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, award_id, award_component, award_contact_pi_id, award_contact_pi_title, award_title, collections, control_for_id, control_for_accession, control_for_aliases, creation_timestamp, dbxrefs, description, documents, donors_id, donors_accession, donors_aliases, donors_sex, donors_status, donors_taxa, file_set_type, files_id, files_accession, files_aliases, files_content_type, files_creation_timestamp, files_file_format, files_file_size, files_href, files_s3_uri, files_sequencing_platform, files_submitted_file_name, files_upload_status, input_file_set_for, input_file_sets, lab_id, lab_title, large_scale_gene_list_id, large_scale_gene_list_accession, large_scale_gene_list_aliases, large_scale_loci_list_id, large_scale_loci_list_accession, large_scale_loci_list_aliases, notes, publication_identifiers, publications, release_timestamp, revoke_detail, samples_id, samples_accession, samples_aliases, samples_cell_fate_change_treatments, samples_classifications, samples_construct_library_sets_id, samples_construct_library_sets_accession, samples_construct_library_sets_summary, samples_disease_terms_id, samples_disease_terms_term_name, samples_modifications, samples_sample_terms_id, samples_sample_terms_aliases, samples_sample_terms_status, samples_sample_terms_summary, samples_sample_terms_term_name, samples_status, samples_summary, samples_targeted_sample_term_id, samples_targeted_sample_term_term_name, samples_taxa, samples_treatments, scope, small_scale_gene_list_id, small_scale_gene_list_geneid, small_scale_gene_list_name, small_scale_gene_list_symbol, small_scale_gene_list_synonyms, small_scale_loci_list, status, submitted_by_id, submitted_by_title, submitted_files_timestamp, submitter_comment, summary, url, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the PredictionSet collection.
    #'
    #' @description
    #' List items in the PredictionSet collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param award_contact_pi_id (optional) Filter by award.contact_pi.@id
    #' @param award_contact_pi_title (optional) Filter by award.contact_pi.title
    #' @param award_title (optional) Filter by award.title
    #' @param collections (optional) Filter by collections
    #' @param control_for_id (optional) Filter by control_for.@id
    #' @param control_for_accession (optional) Filter by control_for.accession
    #' @param control_for_aliases (optional) Filter by control_for.aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param donors_id (optional) Filter by donors.@id
    #' @param donors_accession (optional) Filter by donors.accession
    #' @param donors_aliases (optional) Filter by donors.aliases
    #' @param donors_sex (optional) Filter by donors.sex
    #' @param donors_status (optional) Filter by donors.status
    #' @param donors_taxa (optional) Filter by donors.taxa
    #' @param file_set_type (optional) Filter by file_set_type
    #' @param files_id (optional) Filter by files.@id
    #' @param files_accession (optional) Filter by files.accession
    #' @param files_aliases (optional) Filter by files.aliases
    #' @param files_content_type (optional) Filter by files.content_type
    #' @param files_creation_timestamp (optional) Filter by files.creation_timestamp
    #' @param files_file_format (optional) Filter by files.file_format
    #' @param files_file_size (optional) Filter by files.file_size
    #' @param files_href (optional) Filter by files.href
    #' @param files_s3_uri (optional) Filter by files.s3_uri
    #' @param files_sequencing_platform (optional) Filter by files.sequencing_platform
    #' @param files_submitted_file_name (optional) Filter by files.submitted_file_name
    #' @param files_upload_status (optional) Filter by files.upload_status
    #' @param input_file_set_for (optional) Filter by input_file_set_for
    #' @param input_file_sets (optional) Filter by input_file_sets
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param large_scale_gene_list_id (optional) Filter by large_scale_gene_list.@id
    #' @param large_scale_gene_list_accession (optional) Filter by large_scale_gene_list.accession
    #' @param large_scale_gene_list_aliases (optional) Filter by large_scale_gene_list.aliases
    #' @param large_scale_loci_list_id (optional) Filter by large_scale_loci_list.@id
    #' @param large_scale_loci_list_accession (optional) Filter by large_scale_loci_list.accession
    #' @param large_scale_loci_list_aliases (optional) Filter by large_scale_loci_list.aliases
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param samples_id (optional) Filter by samples.@id
    #' @param samples_accession (optional) Filter by samples.accession
    #' @param samples_aliases (optional) Filter by samples.aliases
    #' @param samples_cell_fate_change_treatments (optional) Filter by samples.cell_fate_change_treatments
    #' @param samples_classifications (optional) Filter by samples.classifications
    #' @param samples_construct_library_sets_id (optional) Filter by samples.construct_library_sets.@id
    #' @param samples_construct_library_sets_accession (optional) Filter by samples.construct_library_sets.accession
    #' @param samples_construct_library_sets_summary (optional) Filter by samples.construct_library_sets.summary
    #' @param samples_disease_terms_id (optional) Filter by samples.disease_terms.@id
    #' @param samples_disease_terms_term_name (optional) Filter by samples.disease_terms.term_name
    #' @param samples_modifications (optional) Filter by samples.modifications
    #' @param samples_sample_terms_id (optional) Filter by samples.sample_terms.@id
    #' @param samples_sample_terms_aliases (optional) Filter by samples.sample_terms.aliases
    #' @param samples_sample_terms_status (optional) Filter by samples.sample_terms.status
    #' @param samples_sample_terms_summary (optional) Filter by samples.sample_terms.summary
    #' @param samples_sample_terms_term_name (optional) Filter by samples.sample_terms.term_name
    #' @param samples_status (optional) Filter by samples.status
    #' @param samples_summary (optional) Filter by samples.summary
    #' @param samples_targeted_sample_term_id (optional) Filter by samples.targeted_sample_term.@id
    #' @param samples_targeted_sample_term_term_name (optional) Filter by samples.targeted_sample_term.term_name
    #' @param samples_taxa (optional) Filter by samples.taxa
    #' @param samples_treatments (optional) Filter by samples.treatments
    #' @param scope (optional) Filter by scope
    #' @param small_scale_gene_list_id (optional) Filter by small_scale_gene_list.@id
    #' @param small_scale_gene_list_geneid (optional) Filter by small_scale_gene_list.geneid
    #' @param small_scale_gene_list_name (optional) Filter by small_scale_gene_list.name
    #' @param small_scale_gene_list_symbol (optional) Filter by small_scale_gene_list.symbol
    #' @param small_scale_gene_list_synonyms (optional) Filter by small_scale_gene_list.synonyms
    #' @param small_scale_loci_list (optional) Filter by small_scale_loci_list
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_files_timestamp (optional) Filter by submitted_files_timestamp
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PredictionSetResults) with additional information such as HTTP status code, headers
    #' @export
    PredictionSetsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, award_contact_pi_id = NULL, award_contact_pi_title = NULL, award_title = NULL, collections = NULL, control_for_id = NULL, control_for_accession = NULL, control_for_aliases = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, donors_id = NULL, donors_accession = NULL, donors_aliases = NULL, donors_sex = NULL, donors_status = NULL, donors_taxa = NULL, file_set_type = NULL, files_id = NULL, files_accession = NULL, files_aliases = NULL, files_content_type = NULL, files_creation_timestamp = NULL, files_file_format = NULL, files_file_size = NULL, files_href = NULL, files_s3_uri = NULL, files_sequencing_platform = NULL, files_submitted_file_name = NULL, files_upload_status = NULL, input_file_set_for = NULL, input_file_sets = NULL, lab_id = NULL, lab_title = NULL, large_scale_gene_list_id = NULL, large_scale_gene_list_accession = NULL, large_scale_gene_list_aliases = NULL, large_scale_loci_list_id = NULL, large_scale_loci_list_accession = NULL, large_scale_loci_list_aliases = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, samples_id = NULL, samples_accession = NULL, samples_aliases = NULL, samples_cell_fate_change_treatments = NULL, samples_classifications = NULL, samples_construct_library_sets_id = NULL, samples_construct_library_sets_accession = NULL, samples_construct_library_sets_summary = NULL, samples_disease_terms_id = NULL, samples_disease_terms_term_name = NULL, samples_modifications = NULL, samples_sample_terms_id = NULL, samples_sample_terms_aliases = NULL, samples_sample_terms_status = NULL, samples_sample_terms_summary = NULL, samples_sample_terms_term_name = NULL, samples_status = NULL, samples_summary = NULL, samples_targeted_sample_term_id = NULL, samples_targeted_sample_term_term_name = NULL, samples_taxa = NULL, samples_treatments = NULL, scope = NULL, small_scale_gene_list_id = NULL, small_scale_gene_list_geneid = NULL, small_scale_gene_list_name = NULL, small_scale_gene_list_symbol = NULL, small_scale_gene_list_synonyms = NULL, small_scale_loci_list = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_files_timestamp = NULL, submitter_comment = NULL, summary = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }




























































      if (length(`samples_classifications`) > 2) {
        stop("Invalid length for `samples_classifications` when calling IgvfApi$PredictionSets, number of items must be less than or equal to 2.")
      }






      if (length(`samples_modifications`) > 2) {
        stop("Invalid length for `samples_modifications` when calling IgvfApi$PredictionSets, number of items must be less than or equal to 2.")
      }



























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$PredictionSets. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$PredictionSets. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_id`) {
        query_params[["award.contact_pi.@id"]] <- c(query_params[["award.contact_pi.@id"]], list(`award.contact_pi.@id` = query_item))
      }

      # explore
      for (query_item in `award_contact_pi_title`) {
        query_params[["award.contact_pi.title"]] <- c(query_params[["award.contact_pi.title"]], list(`award.contact_pi.title` = query_item))
      }

      # explore
      for (query_item in `award_title`) {
        query_params[["award.title"]] <- c(query_params[["award.title"]], list(`award.title` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$PredictionSets. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `control_for_id`) {
        query_params[["control_for.@id"]] <- c(query_params[["control_for.@id"]], list(`control_for.@id` = query_item))
      }

      # explore
      for (query_item in `control_for_accession`) {
        query_params[["control_for.accession"]] <- c(query_params[["control_for.accession"]], list(`control_for.accession` = query_item))
      }

      # explore
      for (query_item in `control_for_aliases`) {
        query_params[["control_for.aliases"]] <- c(query_params[["control_for.aliases"]], list(`control_for.aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors_id`) {
        query_params[["donors.@id"]] <- c(query_params[["donors.@id"]], list(`donors.@id` = query_item))
      }

      # explore
      for (query_item in `donors_accession`) {
        query_params[["donors.accession"]] <- c(query_params[["donors.accession"]], list(`donors.accession` = query_item))
      }

      # explore
      for (query_item in `donors_aliases`) {
        query_params[["donors.aliases"]] <- c(query_params[["donors.aliases"]], list(`donors.aliases` = query_item))
      }

      # explore
      for (query_item in `donors_sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for donors_sex when calling IgvfApi$PredictionSets. Must be [male, female, unspecified].")
        }
        query_params[["donors.sex"]] <- c(query_params[["donors.sex"]], list(`donors.sex` = query_item))
      }

      # explore
      for (query_item in `donors_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for donors_status when calling IgvfApi$PredictionSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["donors.status"]] <- c(query_params[["donors.status"]], list(`donors.status` = query_item))
      }

      # explore
      for (query_item in `donors_taxa`) {
        query_params[["donors.taxa"]] <- c(query_params[["donors.taxa"]], list(`donors.taxa` = query_item))
      }

      # explore
      for (query_item in `file_set_type`) {
        # validate enum values
        if (!(query_item %in% c("pathogenicity", "functional effect", "protein stability", "activity level", "binding effect"))) {
          stop("Invalid value for file_set_type when calling IgvfApi$PredictionSets. Must be [pathogenicity, functional effect, protein stability, activity level, binding effect].")
        }
        query_params[["file_set_type"]] <- c(query_params[["file_set_type"]], list(`file_set_type` = query_item))
      }

      # explore
      for (query_item in `files_id`) {
        query_params[["files.@id"]] <- c(query_params[["files.@id"]], list(`files.@id` = query_item))
      }

      # explore
      for (query_item in `files_accession`) {
        query_params[["files.accession"]] <- c(query_params[["files.accession"]], list(`files.accession` = query_item))
      }

      # explore
      for (query_item in `files_aliases`) {
        query_params[["files.aliases"]] <- c(query_params[["files.aliases"]], list(`files.aliases` = query_item))
      }

      # explore
      for (query_item in `files_content_type`) {
        query_params[["files.content_type"]] <- c(query_params[["files.content_type"]], list(`files.content_type` = query_item))
      }

      # explore
      for (query_item in `files_creation_timestamp`) {
        query_params[["files.creation_timestamp"]] <- c(query_params[["files.creation_timestamp"]], list(`files.creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `files_file_format`) {
        query_params[["files.file_format"]] <- c(query_params[["files.file_format"]], list(`files.file_format` = query_item))
      }

      # explore
      for (query_item in `files_file_size`) {
        query_params[["files.file_size"]] <- c(query_params[["files.file_size"]], list(`files.file_size` = query_item))
      }

      # explore
      for (query_item in `files_href`) {
        query_params[["files.href"]] <- c(query_params[["files.href"]], list(`files.href` = query_item))
      }

      # explore
      for (query_item in `files_s3_uri`) {
        query_params[["files.s3_uri"]] <- c(query_params[["files.s3_uri"]], list(`files.s3_uri` = query_item))
      }

      # explore
      for (query_item in `files_sequencing_platform`) {
        query_params[["files.sequencing_platform"]] <- c(query_params[["files.sequencing_platform"]], list(`files.sequencing_platform` = query_item))
      }

      # explore
      for (query_item in `files_submitted_file_name`) {
        query_params[["files.submitted_file_name"]] <- c(query_params[["files.submitted_file_name"]], list(`files.submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `files_upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for files_upload_status when calling IgvfApi$PredictionSets. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["files.upload_status"]] <- c(query_params[["files.upload_status"]], list(`files.upload_status` = query_item))
      }

      # explore
      for (query_item in `input_file_set_for`) {
        query_params[["input_file_set_for"]] <- c(query_params[["input_file_set_for"]], list(`input_file_set_for` = query_item))
      }

      # explore
      for (query_item in `input_file_sets`) {
        query_params[["input_file_sets"]] <- c(query_params[["input_file_sets"]], list(`input_file_sets` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `large_scale_gene_list_id`) {
        query_params[["large_scale_gene_list.@id"]] <- c(query_params[["large_scale_gene_list.@id"]], list(`large_scale_gene_list.@id` = query_item))
      }

      # explore
      for (query_item in `large_scale_gene_list_accession`) {
        query_params[["large_scale_gene_list.accession"]] <- c(query_params[["large_scale_gene_list.accession"]], list(`large_scale_gene_list.accession` = query_item))
      }

      # explore
      for (query_item in `large_scale_gene_list_aliases`) {
        query_params[["large_scale_gene_list.aliases"]] <- c(query_params[["large_scale_gene_list.aliases"]], list(`large_scale_gene_list.aliases` = query_item))
      }

      # explore
      for (query_item in `large_scale_loci_list_id`) {
        query_params[["large_scale_loci_list.@id"]] <- c(query_params[["large_scale_loci_list.@id"]], list(`large_scale_loci_list.@id` = query_item))
      }

      # explore
      for (query_item in `large_scale_loci_list_accession`) {
        query_params[["large_scale_loci_list.accession"]] <- c(query_params[["large_scale_loci_list.accession"]], list(`large_scale_loci_list.accession` = query_item))
      }

      # explore
      for (query_item in `large_scale_loci_list_aliases`) {
        query_params[["large_scale_loci_list.aliases"]] <- c(query_params[["large_scale_loci_list.aliases"]], list(`large_scale_loci_list.aliases` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `samples_id`) {
        query_params[["samples.@id"]] <- c(query_params[["samples.@id"]], list(`samples.@id` = query_item))
      }

      # explore
      for (query_item in `samples_accession`) {
        query_params[["samples.accession"]] <- c(query_params[["samples.accession"]], list(`samples.accession` = query_item))
      }

      # explore
      for (query_item in `samples_aliases`) {
        query_params[["samples.aliases"]] <- c(query_params[["samples.aliases"]], list(`samples.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_cell_fate_change_treatments`) {
        query_params[["samples.cell_fate_change_treatments"]] <- c(query_params[["samples.cell_fate_change_treatments"]], list(`samples.cell_fate_change_treatments` = query_item))
      }

      # explore
      for (query_item in `samples_classifications`) {
        # validate enum values
        if (!(query_item %in% c("organoid", "gastruloid", "embryoid", "cell line", "differentiated cell specimen", "reprogrammed cell specimen", "pooled cell specimen"))) {
          stop("Invalid value for samples_classifications when calling IgvfApi$PredictionSets. Must be [organoid, gastruloid, embryoid, cell line, differentiated cell specimen, reprogrammed cell specimen, pooled cell specimen].")
        }
        query_params[["samples.classifications"]] <- c(query_params[["samples.classifications"]], list(`samples.classifications` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_id`) {
        query_params[["samples.construct_library_sets.@id"]] <- c(query_params[["samples.construct_library_sets.@id"]], list(`samples.construct_library_sets.@id` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_accession`) {
        query_params[["samples.construct_library_sets.accession"]] <- c(query_params[["samples.construct_library_sets.accession"]], list(`samples.construct_library_sets.accession` = query_item))
      }

      # explore
      for (query_item in `samples_construct_library_sets_summary`) {
        query_params[["samples.construct_library_sets.summary"]] <- c(query_params[["samples.construct_library_sets.summary"]], list(`samples.construct_library_sets.summary` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_id`) {
        query_params[["samples.disease_terms.@id"]] <- c(query_params[["samples.disease_terms.@id"]], list(`samples.disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_disease_terms_term_name`) {
        query_params[["samples.disease_terms.term_name"]] <- c(query_params[["samples.disease_terms.term_name"]], list(`samples.disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_modifications`) {
        query_params[["samples.modifications"]] <- c(query_params[["samples.modifications"]], list(`samples.modifications` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_id`) {
        query_params[["samples.sample_terms.@id"]] <- c(query_params[["samples.sample_terms.@id"]], list(`samples.sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_aliases`) {
        query_params[["samples.sample_terms.aliases"]] <- c(query_params[["samples.sample_terms.aliases"]], list(`samples.sample_terms.aliases` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for samples_sample_terms_status when calling IgvfApi$PredictionSets. Must be [archived, deleted, in progress, released].")
        }
        query_params[["samples.sample_terms.status"]] <- c(query_params[["samples.sample_terms.status"]], list(`samples.sample_terms.status` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_summary`) {
        query_params[["samples.sample_terms.summary"]] <- c(query_params[["samples.sample_terms.summary"]], list(`samples.sample_terms.summary` = query_item))
      }

      # explore
      for (query_item in `samples_sample_terms_term_name`) {
        query_params[["samples.sample_terms.term_name"]] <- c(query_params[["samples.sample_terms.term_name"]], list(`samples.sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for samples_status when calling IgvfApi$PredictionSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["samples.status"]] <- c(query_params[["samples.status"]], list(`samples.status` = query_item))
      }

      # explore
      for (query_item in `samples_summary`) {
        query_params[["samples.summary"]] <- c(query_params[["samples.summary"]], list(`samples.summary` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_id`) {
        query_params[["samples.targeted_sample_term.@id"]] <- c(query_params[["samples.targeted_sample_term.@id"]], list(`samples.targeted_sample_term.@id` = query_item))
      }

      # explore
      for (query_item in `samples_targeted_sample_term_term_name`) {
        query_params[["samples.targeted_sample_term.term_name"]] <- c(query_params[["samples.targeted_sample_term.term_name"]], list(`samples.targeted_sample_term.term_name` = query_item))
      }

      # explore
      for (query_item in `samples_taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for samples_taxa when calling IgvfApi$PredictionSets. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["samples.taxa"]] <- c(query_params[["samples.taxa"]], list(`samples.taxa` = query_item))
      }

      # explore
      for (query_item in `samples_treatments`) {
        query_params[["samples.treatments"]] <- c(query_params[["samples.treatments"]], list(`samples.treatments` = query_item))
      }

      # explore
      for (query_item in `scope`) {
        # validate enum values
        if (!(query_item %in% c("genes", "loci", "genome-wide"))) {
          stop("Invalid value for scope when calling IgvfApi$PredictionSets. Must be [genes, loci, genome-wide].")
        }
        query_params[["scope"]] <- c(query_params[["scope"]], list(`scope` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_id`) {
        query_params[["small_scale_gene_list.@id"]] <- c(query_params[["small_scale_gene_list.@id"]], list(`small_scale_gene_list.@id` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_geneid`) {
        query_params[["small_scale_gene_list.geneid"]] <- c(query_params[["small_scale_gene_list.geneid"]], list(`small_scale_gene_list.geneid` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_name`) {
        query_params[["small_scale_gene_list.name"]] <- c(query_params[["small_scale_gene_list.name"]], list(`small_scale_gene_list.name` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_symbol`) {
        query_params[["small_scale_gene_list.symbol"]] <- c(query_params[["small_scale_gene_list.symbol"]], list(`small_scale_gene_list.symbol` = query_item))
      }

      # explore
      for (query_item in `small_scale_gene_list_synonyms`) {
        query_params[["small_scale_gene_list.synonyms"]] <- c(query_params[["small_scale_gene_list.synonyms"]], list(`small_scale_gene_list.synonyms` = query_item))
      }

      # explore
      for (query_item in `small_scale_loci_list`) {
        query_params[["small_scale_loci_list"]] <- c(query_params[["small_scale_loci_list"]], list(`small_scale_loci_list` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$PredictionSets. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_files_timestamp`) {
        query_params[["submitted_files_timestamp"]] <- c(query_params[["submitted_files_timestamp"]], list(`submitted_files_timestamp` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/prediction-sets/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PredictionSetResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the PrimaryCell collection.
    #'
    #' @description
    #' List items in the PrimaryCell collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from (optional) Filter by originated_from
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param passage_number (optional) Filter by passage_number
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PrimaryCellResults
    #' @export
    PrimaryCells = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from = NULL, part_of = NULL, parts = NULL, passage_number = NULL, pooled_from = NULL, pooled_in = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$PrimaryCellsWithHttpInfo(frame, query, limit, sort, id, accession, age, age_units, aliases, alternate_accessions, award_id, award_component, biomarkers, cellular_sub_pool, classifications, collections, construct_library_sets, creation_timestamp, date_obtained, dbxrefs, description, disease_terms_id, disease_terms_term_name, documents, donors, embryonic, file_sets_id, file_sets_accession, file_sets_aliases, file_sets_assay_term_term_name, file_sets_lab_title, file_sets_status, file_sets_summary, institutional_certificates_id, institutional_certificates_certificate_identifier, lab_id, lab_title, lot_id, lower_bound_age, lower_bound_age_in_hours, modifications_id, modifications_status, modifications_summary, moi, multiplexed_in_id, multiplexed_in_accession, notes, nucleic_acid_delivery, origin_of, originated_from, part_of, parts, passage_number, pooled_from, pooled_in, product_id, protocols, publication_identifiers, publications, release_timestamp, revoke_detail, sample_terms_id, sample_terms_term_name, sex, sorted_fractions, sorted_from_id, sorted_from_accession, sorted_from_detail, sources_id, starting_amount, starting_amount_units, status, submitted_by_id, submitted_by_title, submitter_comment, summary, taxa, time_post_library_delivery, time_post_library_delivery_units, treatments_id, treatments_purpose, treatments_status, treatments_summary, treatments_treatment_type, upper_bound_age, upper_bound_age_in_hours, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the PrimaryCell collection.
    #'
    #' @description
    #' List items in the PrimaryCell collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from (optional) Filter by originated_from
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param passage_number (optional) Filter by passage_number
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PrimaryCellResults) with additional information such as HTTP status code, headers
    #' @export
    PrimaryCellsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from = NULL, part_of = NULL, parts = NULL, passage_number = NULL, pooled_from = NULL, pooled_in = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }


























































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$PrimaryCells. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `age`) {
        query_params[["age"]] <- c(query_params[["age"]], list(`age` = query_item))
      }

      # explore
      for (query_item in `age_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month", "year"))) {
          stop("Invalid value for age_units when calling IgvfApi$PrimaryCells. Must be [minute, hour, day, week, month, year].")
        }
        query_params[["age_units"]] <- c(query_params[["age_units"]], list(`age_units` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$PrimaryCells. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biomarkers`) {
        query_params[["biomarkers"]] <- c(query_params[["biomarkers"]], list(`biomarkers` = query_item))
      }

      # explore
      for (query_item in `cellular_sub_pool`) {
        query_params[["cellular_sub_pool"]] <- c(query_params[["cellular_sub_pool"]], list(`cellular_sub_pool` = query_item))
      }

      # explore
      for (query_item in `classifications`) {
        query_params[["classifications"]] <- c(query_params[["classifications"]], list(`classifications` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$PrimaryCells. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets`) {
        query_params[["construct_library_sets"]] <- c(query_params[["construct_library_sets"]], list(`construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_obtained`) {
        query_params[["date_obtained"]] <- c(query_params[["date_obtained"]], list(`date_obtained` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `disease_terms_id`) {
        query_params[["disease_terms.@id"]] <- c(query_params[["disease_terms.@id"]], list(`disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `disease_terms_term_name`) {
        query_params[["disease_terms.term_name"]] <- c(query_params[["disease_terms.term_name"]], list(`disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors`) {
        query_params[["donors"]] <- c(query_params[["donors"]], list(`donors` = query_item))
      }

      # explore
      for (query_item in `embryonic`) {
        query_params[["embryonic"]] <- c(query_params[["embryonic"]], list(`embryonic` = query_item))
      }

      # explore
      for (query_item in `file_sets_id`) {
        query_params[["file_sets.@id"]] <- c(query_params[["file_sets.@id"]], list(`file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `file_sets_accession`) {
        query_params[["file_sets.accession"]] <- c(query_params[["file_sets.accession"]], list(`file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `file_sets_aliases`) {
        query_params[["file_sets.aliases"]] <- c(query_params[["file_sets.aliases"]], list(`file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `file_sets_assay_term_term_name`) {
        query_params[["file_sets.assay_term.term_name"]] <- c(query_params[["file_sets.assay_term.term_name"]], list(`file_sets.assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `file_sets_lab_title`) {
        query_params[["file_sets.lab.title"]] <- c(query_params[["file_sets.lab.title"]], list(`file_sets.lab.title` = query_item))
      }

      # explore
      for (query_item in `file_sets_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for file_sets_status when calling IgvfApi$PrimaryCells. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["file_sets.status"]] <- c(query_params[["file_sets.status"]], list(`file_sets.status` = query_item))
      }

      # explore
      for (query_item in `file_sets_summary`) {
        query_params[["file_sets.summary"]] <- c(query_params[["file_sets.summary"]], list(`file_sets.summary` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_id`) {
        query_params[["institutional_certificates.@id"]] <- c(query_params[["institutional_certificates.@id"]], list(`institutional_certificates.@id` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_certificate_identifier`) {
        query_params[["institutional_certificates.certificate_identifier"]] <- c(query_params[["institutional_certificates.certificate_identifier"]], list(`institutional_certificates.certificate_identifier` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age`) {
        query_params[["lower_bound_age"]] <- c(query_params[["lower_bound_age"]], list(`lower_bound_age` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age_in_hours`) {
        query_params[["lower_bound_age_in_hours"]] <- c(query_params[["lower_bound_age_in_hours"]], list(`lower_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `modifications_id`) {
        query_params[["modifications.@id"]] <- c(query_params[["modifications.@id"]], list(`modifications.@id` = query_item))
      }

      # explore
      for (query_item in `modifications_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for modifications_status when calling IgvfApi$PrimaryCells. Must be [archived, deleted, in progress, released].")
        }
        query_params[["modifications.status"]] <- c(query_params[["modifications.status"]], list(`modifications.status` = query_item))
      }

      # explore
      for (query_item in `modifications_summary`) {
        query_params[["modifications.summary"]] <- c(query_params[["modifications.summary"]], list(`modifications.summary` = query_item))
      }

      # explore
      for (query_item in `moi`) {
        query_params[["moi"]] <- c(query_params[["moi"]], list(`moi` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_id`) {
        query_params[["multiplexed_in.@id"]] <- c(query_params[["multiplexed_in.@id"]], list(`multiplexed_in.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_accession`) {
        query_params[["multiplexed_in.accession"]] <- c(query_params[["multiplexed_in.accession"]], list(`multiplexed_in.accession` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `nucleic_acid_delivery`) {
        # validate enum values
        if (!(query_item %in% c("transfection", "adenoviral transduction", "lentiviral transduction"))) {
          stop("Invalid value for nucleic_acid_delivery when calling IgvfApi$PrimaryCells. Must be [transfection, adenoviral transduction, lentiviral transduction].")
        }
        query_params[["nucleic_acid_delivery"]] <- c(query_params[["nucleic_acid_delivery"]], list(`nucleic_acid_delivery` = query_item))
      }

      # explore
      for (query_item in `origin_of`) {
        query_params[["origin_of"]] <- c(query_params[["origin_of"]], list(`origin_of` = query_item))
      }

      # explore
      for (query_item in `originated_from`) {
        query_params[["originated_from"]] <- c(query_params[["originated_from"]], list(`originated_from` = query_item))
      }

      # explore
      for (query_item in `part_of`) {
        query_params[["part_of"]] <- c(query_params[["part_of"]], list(`part_of` = query_item))
      }

      # explore
      for (query_item in `parts`) {
        query_params[["parts"]] <- c(query_params[["parts"]], list(`parts` = query_item))
      }

      # explore
      for (query_item in `passage_number`) {
        query_params[["passage_number"]] <- c(query_params[["passage_number"]], list(`passage_number` = query_item))
      }

      # explore
      for (query_item in `pooled_from`) {
        query_params[["pooled_from"]] <- c(query_params[["pooled_from"]], list(`pooled_from` = query_item))
      }

      # explore
      for (query_item in `pooled_in`) {
        query_params[["pooled_in"]] <- c(query_params[["pooled_in"]], list(`pooled_in` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sample_terms_id`) {
        query_params[["sample_terms.@id"]] <- c(query_params[["sample_terms.@id"]], list(`sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `sample_terms_term_name`) {
        query_params[["sample_terms.term_name"]] <- c(query_params[["sample_terms.term_name"]], list(`sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `sex`) {
        # validate enum values
        if (!(query_item %in% c("female", "male", "mixed", "unspecified"))) {
          stop("Invalid value for sex when calling IgvfApi$PrimaryCells. Must be [female, male, mixed, unspecified].")
        }
        query_params[["sex"]] <- c(query_params[["sex"]], list(`sex` = query_item))
      }

      # explore
      for (query_item in `sorted_fractions`) {
        query_params[["sorted_fractions"]] <- c(query_params[["sorted_fractions"]], list(`sorted_fractions` = query_item))
      }

      # explore
      for (query_item in `sorted_from_id`) {
        query_params[["sorted_from.@id"]] <- c(query_params[["sorted_from.@id"]], list(`sorted_from.@id` = query_item))
      }

      # explore
      for (query_item in `sorted_from_accession`) {
        query_params[["sorted_from.accession"]] <- c(query_params[["sorted_from.accession"]], list(`sorted_from.accession` = query_item))
      }

      # explore
      for (query_item in `sorted_from_detail`) {
        query_params[["sorted_from_detail"]] <- c(query_params[["sorted_from_detail"]], list(`sorted_from_detail` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `starting_amount`) {
        query_params[["starting_amount"]] <- c(query_params[["starting_amount"]], list(`starting_amount` = query_item))
      }

      # explore
      for (query_item in `starting_amount_units`) {
        # validate enum values
        if (!(query_item %in% c("cells", "cells/ml", "g", "items", "mg", "whole animals", "whole embryos", "μg", "ng"))) {
          stop("Invalid value for starting_amount_units when calling IgvfApi$PrimaryCells. Must be [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng].")
        }
        query_params[["starting_amount_units"]] <- c(query_params[["starting_amount_units"]], list(`starting_amount_units` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$PrimaryCells. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$PrimaryCells. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery`) {
        query_params[["time_post_library_delivery"]] <- c(query_params[["time_post_library_delivery"]], list(`time_post_library_delivery` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_library_delivery_units when calling IgvfApi$PrimaryCells. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_library_delivery_units"]] <- c(query_params[["time_post_library_delivery_units"]], list(`time_post_library_delivery_units` = query_item))
      }

      # explore
      for (query_item in `treatments_id`) {
        query_params[["treatments.@id"]] <- c(query_params[["treatments.@id"]], list(`treatments.@id` = query_item))
      }

      # explore
      for (query_item in `treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for treatments_purpose when calling IgvfApi$PrimaryCells. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["treatments.purpose"]] <- c(query_params[["treatments.purpose"]], list(`treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `treatments_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for treatments_status when calling IgvfApi$PrimaryCells. Must be [archived, deleted, in progress, released].")
        }
        query_params[["treatments.status"]] <- c(query_params[["treatments.status"]], list(`treatments.status` = query_item))
      }

      # explore
      for (query_item in `treatments_summary`) {
        query_params[["treatments.summary"]] <- c(query_params[["treatments.summary"]], list(`treatments.summary` = query_item))
      }

      # explore
      for (query_item in `treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for treatments_treatment_type when calling IgvfApi$PrimaryCells. Must be [chemical, protein, environmental].")
        }
        query_params[["treatments.treatment_type"]] <- c(query_params[["treatments.treatment_type"]], list(`treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age`) {
        query_params[["upper_bound_age"]] <- c(query_params[["upper_bound_age"]], list(`upper_bound_age` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age_in_hours`) {
        query_params[["upper_bound_age_in_hours"]] <- c(query_params[["upper_bound_age_in_hours"]], list(`upper_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/primary-cells/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PrimaryCellResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Publication collection.
    #'
    #' @description
    #' List items in the Publication collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param abstract (optional) Filter by abstract
    #' @param aliases (optional) Filter by aliases
    #' @param authors (optional) Filter by authors
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_published (optional) Filter by date_published
    #' @param date_revised (optional) Filter by date_revised
    #' @param description (optional) Filter by description
    #' @param donors (optional) Filter by donors
    #' @param file_sets (optional) Filter by file_sets
    #' @param issue (optional) Filter by issue
    #' @param journal (optional) Filter by journal
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param page (optional) Filter by page
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publication_year (optional) Filter by publication_year
    #' @param published_by (optional) Filter by published_by
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param samples (optional) Filter by samples
    #' @param software (optional) Filter by software
    #' @param software_versions (optional) Filter by software_versions
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param volume (optional) Filter by volume
    #' @param workflows (optional) Filter by workflows
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return PublicationResults
    #' @export
    Publications = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, abstract = NULL, aliases = NULL, authors = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, date_published = NULL, date_revised = NULL, description = NULL, donors = NULL, file_sets = NULL, issue = NULL, journal = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, page = NULL, publication_identifiers = NULL, publication_year = NULL, published_by = NULL, release_timestamp = NULL, samples = NULL, software = NULL, software_versions = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, volume = NULL, workflows = NULL, data_file = NULL, ...) {
      local_var_response <- self$PublicationsWithHttpInfo(frame, query, limit, sort, id, abstract, aliases, authors, award_id, award_component, creation_timestamp, date_published, date_revised, description, donors, file_sets, issue, journal, lab_id, lab_title, notes, page, publication_identifiers, publication_year, published_by, release_timestamp, samples, software, software_versions, status, submitted_by_id, submitted_by_title, submitter_comment, summary, title, uuid, volume, workflows, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Publication collection.
    #'
    #' @description
    #' List items in the Publication collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param abstract (optional) Filter by abstract
    #' @param aliases (optional) Filter by aliases
    #' @param authors (optional) Filter by authors
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_published (optional) Filter by date_published
    #' @param date_revised (optional) Filter by date_revised
    #' @param description (optional) Filter by description
    #' @param donors (optional) Filter by donors
    #' @param file_sets (optional) Filter by file_sets
    #' @param issue (optional) Filter by issue
    #' @param journal (optional) Filter by journal
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param notes (optional) Filter by notes
    #' @param page (optional) Filter by page
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publication_year (optional) Filter by publication_year
    #' @param published_by (optional) Filter by published_by
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param samples (optional) Filter by samples
    #' @param software (optional) Filter by software
    #' @param software_versions (optional) Filter by software_versions
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param volume (optional) Filter by volume
    #' @param workflows (optional) Filter by workflows
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (PublicationResults) with additional information such as HTTP status code, headers
    #' @export
    PublicationsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, abstract = NULL, aliases = NULL, authors = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, date_published = NULL, date_revised = NULL, description = NULL, donors = NULL, file_sets = NULL, issue = NULL, journal = NULL, lab_id = NULL, lab_title = NULL, notes = NULL, page = NULL, publication_identifiers = NULL, publication_year = NULL, published_by = NULL, release_timestamp = NULL, samples = NULL, software = NULL, software_versions = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, volume = NULL, workflows = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }







































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Publications. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `abstract`) {
        query_params[["abstract"]] <- c(query_params[["abstract"]], list(`abstract` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `authors`) {
        query_params[["authors"]] <- c(query_params[["authors"]], list(`authors` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Publications. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_published`) {
        query_params[["date_published"]] <- c(query_params[["date_published"]], list(`date_published` = query_item))
      }

      # explore
      for (query_item in `date_revised`) {
        query_params[["date_revised"]] <- c(query_params[["date_revised"]], list(`date_revised` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `donors`) {
        query_params[["donors"]] <- c(query_params[["donors"]], list(`donors` = query_item))
      }

      # explore
      for (query_item in `file_sets`) {
        query_params[["file_sets"]] <- c(query_params[["file_sets"]], list(`file_sets` = query_item))
      }

      # explore
      for (query_item in `issue`) {
        query_params[["issue"]] <- c(query_params[["issue"]], list(`issue` = query_item))
      }

      # explore
      for (query_item in `journal`) {
        query_params[["journal"]] <- c(query_params[["journal"]], list(`journal` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `page`) {
        query_params[["page"]] <- c(query_params[["page"]], list(`page` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publication_year`) {
        query_params[["publication_year"]] <- c(query_params[["publication_year"]], list(`publication_year` = query_item))
      }

      # explore
      for (query_item in `published_by`) {
        # validate enum values
        if (!(query_item %in% c("community", "IGVF", "ENCODE"))) {
          stop("Invalid value for published_by when calling IgvfApi$Publications. Must be [community, IGVF, ENCODE].")
        }
        query_params[["published_by"]] <- c(query_params[["published_by"]], list(`published_by` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `samples`) {
        query_params[["samples"]] <- c(query_params[["samples"]], list(`samples` = query_item))
      }

      # explore
      for (query_item in `software`) {
        query_params[["software"]] <- c(query_params[["software"]], list(`software` = query_item))
      }

      # explore
      for (query_item in `software_versions`) {
        query_params[["software_versions"]] <- c(query_params[["software_versions"]], list(`software_versions` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Publications. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `volume`) {
        query_params[["volume"]] <- c(query_params[["volume"]], list(`volume` = query_item))
      }

      # explore
      for (query_item in `workflows`) {
        query_params[["workflows"]] <- c(query_params[["workflows"]], list(`workflows` = query_item))
      }

      local_var_url_path <- "/publications/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "PublicationResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the ReferenceFile collection.
    #'
    #' @description
    #' List items in the ReferenceFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param external (optional) Filter by external
    #' @param external_id (optional) Filter by external_id
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_format_type (optional) Filter by file_format_type
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param source_url (optional) Filter by source_url
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return ReferenceFileResults
    #' @export
    ReferenceFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, external = NULL, external_id = NULL, file_format = NULL, file_format_specifications = NULL, file_format_type = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, source_url = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$ReferenceFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, anvil_url, assembly, award_id, award_component, collections, content_md5sum, content_type, controlled_access, creation_timestamp, dbxrefs, derived_from, description, documents, external, external_id, file_format, file_format_specifications, file_format_type, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, release_timestamp, revoke_detail, s3_uri, source_url, sources, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, transcriptome_annotation, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the ReferenceFile collection.
    #'
    #' @description
    #' List items in the ReferenceFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param external (optional) Filter by external
    #' @param external_id (optional) Filter by external_id
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_format_type (optional) Filter by file_format_type
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param source_url (optional) Filter by source_url
    #' @param sources (optional) Filter by sources
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (ReferenceFileResults) with additional information such as HTTP status code, headers
    #' @export
    ReferenceFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, external = NULL, external_id = NULL, file_format = NULL, file_format_specifications = NULL, file_format_type = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, source_url = NULL, sources = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }






















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$ReferenceFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `anvil_url`) {
        query_params[["anvil_url"]] <- c(query_params[["anvil_url"]], list(`anvil_url` = query_item))
      }

      # explore
      for (query_item in `assembly`) {
        # validate enum values
        if (!(query_item %in% c("GRCh38", "hg19", "GRCm39", "mm10"))) {
          stop("Invalid value for assembly when calling IgvfApi$ReferenceFiles. Must be [GRCh38, hg19, GRCm39, mm10].")
        }
        query_params[["assembly"]] <- c(query_params[["assembly"]], list(`assembly` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$ReferenceFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$ReferenceFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `controlled_access`) {
        query_params[["controlled_access"]] <- c(query_params[["controlled_access"]], list(`controlled_access` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `external`) {
        query_params[["external"]] <- c(query_params[["external"]], list(`external` = query_item))
      }

      # explore
      for (query_item in `external_id`) {
        query_params[["external_id"]] <- c(query_params[["external_id"]], list(`external_id` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("bed", "csv", "dat", "fasta", "gaf", "gds", "gtf", "obo", "owl", "PWM", "tar", "tsv", "txt", "vcf", "xml"))) {
          stop("Invalid value for file_format when calling IgvfApi$ReferenceFiles. Must be [bed, csv, dat, fasta, gaf, gds, gtf, obo, owl, PWM, tar, tsv, txt, vcf, xml].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_format_type`) {
        # validate enum values
        if (!(query_item %in% c("bed12", "bed3", "bed3+", "bed5", "bed6", "bed6+", "bed9", "bed9+", "mpra_starr"))) {
          stop("Invalid value for file_format_type when calling IgvfApi$ReferenceFiles. Must be [bed12, bed3, bed3+, bed5, bed6, bed6+, bed9, bed9+, mpra_starr].")
        }
        query_params[["file_format_type"]] <- c(query_params[["file_format_type"]], list(`file_format_type` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `source_url`) {
        query_params[["source_url"]] <- c(query_params[["source_url"]], list(`source_url` = query_item))
      }

      # explore
      for (query_item in `sources`) {
        query_params[["sources"]] <- c(query_params[["sources"]], list(`sources` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$ReferenceFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotation`) {
        # validate enum values
        if (!(query_item %in% c("GENCODE 40", "GENCODE 41", "GENCODE 42", "GENCODE 43", "GENCODE 44", "GENCODE 45", "GENCODE M30", "GENCODE M31", "GENCODE M32", "GENCODE M33", "GENCODE M34"))) {
          stop("Invalid value for transcriptome_annotation when calling IgvfApi$ReferenceFiles. Must be [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34].")
        }
        query_params[["transcriptome_annotation"]] <- c(query_params[["transcriptome_annotation"]], list(`transcriptome_annotation` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$ReferenceFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/reference-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "ReferenceFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Generate a report based on search query. All results are returned.
    #'
    #' @description
    #' Generate a report based on search query. All results are returned.
    #'
    #' @param query (optional) Query string for searching.
    #' @param type (optional) Filter by item type.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
    #' @param field_filters (optional) Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
    #' @param include_fields (optional) Fields to include in the response. Can be repeated for multiple fields.
    #' @param frame (optional) Object with links, or object with some links embedded.
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return character
    #' @export
    Report = function(query = NULL, type = NULL, limit = NULL, sort = NULL, field_filters = NULL, include_fields = NULL, frame = NULL, data_file = NULL, ...) {
      local_var_response <- self$ReportWithHttpInfo(query, type, limit, sort, field_filters, include_fields, frame, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Generate a report based on search query. All results are returned.
    #'
    #' @description
    #' Generate a report based on search query. All results are returned.
    #'
    #' @param query (optional) Query string for searching.
    #' @param type (optional) Filter by item type.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
    #' @param field_filters (optional) Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
    #' @param include_fields (optional) Fields to include in the response. Can be repeated for multiple fields.
    #' @param frame (optional) Object with links, or object with some links embedded.
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (character) with additional information such as HTTP status code, headers
    #' @export
    ReportWithHttpInfo = function(query = NULL, type = NULL, limit = NULL, sort = NULL, field_filters = NULL, include_fields = NULL, frame = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE








      query_params[["query"]] <- `query`

      # explore
      for (query_item in `type`) {
        query_params[["type"]] <- c(query_params[["type"]], list(`type` = query_item))
      }

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      for (i in seq_along(`field_filters`)) {
          k <- names(`field_filters`)[[i]]
          v <- `field_filters`[[i]]
          query_params[[k]] <- v
      }

      # explore
      for (query_item in `include_fields`) {
        query_params[["field"]] <- c(query_params[["field"]], list(`field` = query_item))
      }

      if (!(`frame` %in% c("object", "embedded"))) {
        stop("Invalid value for frame when calling IgvfApi$Report. Must be [object, embedded].")
      }
      query_params[["frame"]] <- `frame`

      local_var_url_path <- "/multireport.tsv"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("text/tab-separated-values", "application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "character", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the RodentDonor collection.
    #'
    #' @description
    #' List items in the RodentDonor collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param genotype (optional) Filter by genotype
    #' @param individual_rodent (optional) Filter by individual_rodent
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param notes (optional) Filter by notes
    #' @param phenotypic_features_id (optional) Filter by phenotypic_features.@id
    #' @param phenotypic_features_feature_id (optional) Filter by phenotypic_features.feature.@id
    #' @param phenotypic_features_feature_term_id (optional) Filter by phenotypic_features.feature.term_id
    #' @param phenotypic_features_feature_term_name (optional) Filter by phenotypic_features.feature.term_name
    #' @param phenotypic_features_observation_date (optional) Filter by phenotypic_features.observation_date
    #' @param phenotypic_features_quantity_units (optional) Filter by phenotypic_features.quantity_units
    #' @param product_id (optional) Filter by product_id
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param rodent_identifier (optional) Filter by rodent_identifier
    #' @param sex (optional) Filter by sex
    #' @param sources_id (optional) Filter by sources.@id
    #' @param status (optional) Filter by status
    #' @param strain (optional) Filter by strain
    #' @param strain_background (optional) Filter by strain_background
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return RodentDonorResults
    #' @export
    RodentDonors = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, collections = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, genotype = NULL, individual_rodent = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, notes = NULL, phenotypic_features_id = NULL, phenotypic_features_feature_id = NULL, phenotypic_features_feature_term_id = NULL, phenotypic_features_feature_term_name = NULL, phenotypic_features_observation_date = NULL, phenotypic_features_quantity_units = NULL, product_id = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, rodent_identifier = NULL, sex = NULL, sources_id = NULL, status = NULL, strain = NULL, strain_background = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$RodentDonorsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, award_id, award_component, collections, creation_timestamp, dbxrefs, description, documents, genotype, individual_rodent, lab_id, lab_title, lot_id, notes, phenotypic_features_id, phenotypic_features_feature_id, phenotypic_features_feature_term_id, phenotypic_features_feature_term_name, phenotypic_features_observation_date, phenotypic_features_quantity_units, product_id, publication_identifiers, publications, release_timestamp, revoke_detail, rodent_identifier, sex, sources_id, status, strain, strain_background, submitted_by_id, submitted_by_title, submitter_comment, summary, taxa, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the RodentDonor collection.
    #'
    #' @description
    #' List items in the RodentDonor collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param genotype (optional) Filter by genotype
    #' @param individual_rodent (optional) Filter by individual_rodent
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param notes (optional) Filter by notes
    #' @param phenotypic_features_id (optional) Filter by phenotypic_features.@id
    #' @param phenotypic_features_feature_id (optional) Filter by phenotypic_features.feature.@id
    #' @param phenotypic_features_feature_term_id (optional) Filter by phenotypic_features.feature.term_id
    #' @param phenotypic_features_feature_term_name (optional) Filter by phenotypic_features.feature.term_name
    #' @param phenotypic_features_observation_date (optional) Filter by phenotypic_features.observation_date
    #' @param phenotypic_features_quantity_units (optional) Filter by phenotypic_features.quantity_units
    #' @param product_id (optional) Filter by product_id
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param rodent_identifier (optional) Filter by rodent_identifier
    #' @param sex (optional) Filter by sex
    #' @param sources_id (optional) Filter by sources.@id
    #' @param status (optional) Filter by status
    #' @param strain (optional) Filter by strain
    #' @param strain_background (optional) Filter by strain_background
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (RodentDonorResults) with additional information such as HTTP status code, headers
    #' @export
    RodentDonorsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, collections = NULL, creation_timestamp = NULL, dbxrefs = NULL, description = NULL, documents = NULL, genotype = NULL, individual_rodent = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, notes = NULL, phenotypic_features_id = NULL, phenotypic_features_feature_id = NULL, phenotypic_features_feature_term_id = NULL, phenotypic_features_feature_term_name = NULL, phenotypic_features_observation_date = NULL, phenotypic_features_quantity_units = NULL, product_id = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, rodent_identifier = NULL, sex = NULL, sources_id = NULL, status = NULL, strain = NULL, strain_background = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$RodentDonors. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$RodentDonors. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$RodentDonors. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `genotype`) {
        query_params[["genotype"]] <- c(query_params[["genotype"]], list(`genotype` = query_item))
      }

      # explore
      for (query_item in `individual_rodent`) {
        query_params[["individual_rodent"]] <- c(query_params[["individual_rodent"]], list(`individual_rodent` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_id`) {
        query_params[["phenotypic_features.@id"]] <- c(query_params[["phenotypic_features.@id"]], list(`phenotypic_features.@id` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_feature_id`) {
        query_params[["phenotypic_features.feature.@id"]] <- c(query_params[["phenotypic_features.feature.@id"]], list(`phenotypic_features.feature.@id` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_feature_term_id`) {
        query_params[["phenotypic_features.feature.term_id"]] <- c(query_params[["phenotypic_features.feature.term_id"]], list(`phenotypic_features.feature.term_id` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_feature_term_name`) {
        query_params[["phenotypic_features.feature.term_name"]] <- c(query_params[["phenotypic_features.feature.term_name"]], list(`phenotypic_features.feature.term_name` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_observation_date`) {
        query_params[["phenotypic_features.observation_date"]] <- c(query_params[["phenotypic_features.observation_date"]], list(`phenotypic_features.observation_date` = query_item))
      }

      # explore
      for (query_item in `phenotypic_features_quantity_units`) {
        # validate enum values
        if (!(query_item %in% c("meter", "micromole", "nanogram", "microgram", "milligram", "gram", "kilogram", "milli-International Unit per milliliter", "picogram per milliliter", "nanogram per milliliter", "milligram per deciliter"))) {
          stop("Invalid value for phenotypic_features_quantity_units when calling IgvfApi$RodentDonors. Must be [meter, micromole, nanogram, microgram, milligram, gram, kilogram, milli-International Unit per milliliter, picogram per milliliter, nanogram per milliliter, milligram per deciliter].")
        }
        query_params[["phenotypic_features.quantity_units"]] <- c(query_params[["phenotypic_features.quantity_units"]], list(`phenotypic_features.quantity_units` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `rodent_identifier`) {
        query_params[["rodent_identifier"]] <- c(query_params[["rodent_identifier"]], list(`rodent_identifier` = query_item))
      }

      # explore
      for (query_item in `sex`) {
        # validate enum values
        if (!(query_item %in% c("male", "female", "unspecified"))) {
          stop("Invalid value for sex when calling IgvfApi$RodentDonors. Must be [male, female, unspecified].")
        }
        query_params[["sex"]] <- c(query_params[["sex"]], list(`sex` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$RodentDonors. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `strain`) {
        query_params[["strain"]] <- c(query_params[["strain"]], list(`strain` = query_item))
      }

      # explore
      for (query_item in `strain_background`) {
        # validate enum values
        if (!(query_item %in% c("A/J (AJ)", "C57BL/6J (B6)", "129S1/SvImJ (129)", "NOD/ShiLtJ (NOD)", "NZO/H1LtJ (NZO)", "CAST/EiJ (CAST)", "PWK/PhJ (PWK)", "WSB/EiJ (WSB)", "CAST (M. m. castaneus)", "WSB (M. m. domesticus)", "PWK (M. m. musculus)"))) {
          stop("Invalid value for strain_background when calling IgvfApi$RodentDonors. Must be [A/J (AJ), C57BL/6J (B6), 129S1/SvImJ (129), NOD/ShiLtJ (NOD), NZO/H1LtJ (NZO), CAST/EiJ (CAST), PWK/PhJ (PWK), WSB/EiJ (WSB), CAST (M. m. castaneus), WSB (M. m. domesticus), PWK (M. m. musculus)].")
        }
        query_params[["strain_background"]] <- c(query_params[["strain_background"]], list(`strain_background` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$RodentDonors. Must be [Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/rodent-donors/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "RodentDonorResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the SampleTerm collection.
    #'
    #' @description
    #' List items in the SampleTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param cell_slims (optional) Filter by cell_slims
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param developmental_slims (optional) Filter by developmental_slims
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param ontology (optional) Filter by ontology
    #' @param organ_slims (optional) Filter by organ_slims
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param system_slims (optional) Filter by system_slims
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SampleTermResults
    #' @export
    SampleTerms = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, cell_slims = NULL, creation_timestamp = NULL, dbxrefs = NULL, deprecated_ntr_terms = NULL, description = NULL, developmental_slims = NULL, is_a = NULL, name = NULL, notes = NULL, ontology = NULL, organ_slims = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, system_slims = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$SampleTermsWithHttpInfo(frame, query, limit, sort, id, aliases, ancestors, cell_slims, creation_timestamp, dbxrefs, deprecated_ntr_terms, description, developmental_slims, is_a, name, notes, ontology, organ_slims, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, synonyms, system_slims, term_id, term_name, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the SampleTerm collection.
    #'
    #' @description
    #' List items in the SampleTerm collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param ancestors (optional) Filter by ancestors
    #' @param cell_slims (optional) Filter by cell_slims
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param deprecated_ntr_terms (optional) Filter by deprecated_ntr_terms
    #' @param description (optional) Filter by description
    #' @param developmental_slims (optional) Filter by developmental_slims
    #' @param is_a (optional) Filter by is_a
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param ontology (optional) Filter by ontology
    #' @param organ_slims (optional) Filter by organ_slims
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param synonyms (optional) Filter by synonyms
    #' @param system_slims (optional) Filter by system_slims
    #' @param term_id (optional) Filter by term_id
    #' @param term_name (optional) Filter by term_name
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SampleTermResults) with additional information such as HTTP status code, headers
    #' @export
    SampleTermsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, ancestors = NULL, cell_slims = NULL, creation_timestamp = NULL, dbxrefs = NULL, deprecated_ntr_terms = NULL, description = NULL, developmental_slims = NULL, is_a = NULL, name = NULL, notes = NULL, ontology = NULL, organ_slims = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, synonyms = NULL, system_slims = NULL, term_id = NULL, term_name = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }






























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$SampleTerms. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `ancestors`) {
        query_params[["ancestors"]] <- c(query_params[["ancestors"]], list(`ancestors` = query_item))
      }

      # explore
      for (query_item in `cell_slims`) {
        query_params[["cell_slims"]] <- c(query_params[["cell_slims"]], list(`cell_slims` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `deprecated_ntr_terms`) {
        query_params[["deprecated_ntr_terms"]] <- c(query_params[["deprecated_ntr_terms"]], list(`deprecated_ntr_terms` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `developmental_slims`) {
        query_params[["developmental_slims"]] <- c(query_params[["developmental_slims"]], list(`developmental_slims` = query_item))
      }

      # explore
      for (query_item in `is_a`) {
        query_params[["is_a"]] <- c(query_params[["is_a"]], list(`is_a` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `ontology`) {
        query_params[["ontology"]] <- c(query_params[["ontology"]], list(`ontology` = query_item))
      }

      # explore
      for (query_item in `organ_slims`) {
        query_params[["organ_slims"]] <- c(query_params[["organ_slims"]], list(`organ_slims` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$SampleTerms. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `synonyms`) {
        query_params[["synonyms"]] <- c(query_params[["synonyms"]], list(`synonyms` = query_item))
      }

      # explore
      for (query_item in `system_slims`) {
        query_params[["system_slims"]] <- c(query_params[["system_slims"]], list(`system_slims` = query_item))
      }

      # explore
      for (query_item in `term_id`) {
        query_params[["term_id"]] <- c(query_params[["term_id"]], list(`term_id` = query_item))
      }

      # explore
      for (query_item in `term_name`) {
        query_params[["term_name"]] <- c(query_params[["term_name"]], list(`term_name` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/sample-terms/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SampleTermResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve JSON schema for item type
    #'
    #' @description
    #' Retrieve JSON schema for item type
    #'
    #' @param item_type The name of the item type
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return object
    #' @export
    SchemaForItemType = function(item_type, data_file = NULL, ...) {
      local_var_response <- self$SchemaForItemTypeWithHttpInfo(item_type, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve JSON schema for item type
    #'
    #' @description
    #' Retrieve JSON schema for item type
    #'
    #' @param item_type The name of the item type
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (object) with additional information such as HTTP status code, headers
    #' @export
    SchemaForItemTypeWithHttpInfo = function(item_type, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`item_type`)) {
        stop("Missing required parameter `item_type`.")
      }


      local_var_url_path <- "/profiles/{item_type}"
      if (!missing(`item_type`)) {
        local_var_url_path <- gsub("\\{item_type\\}", URLencode(as.character(`item_type`), reserved = TRUE), local_var_url_path)
      }

      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "object", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Retrieve JSON schemas for all item types
    #'
    #' @description
    #' Retrieve JSON schemas for all item types
    #'
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return object
    #' @export
    Schemas = function(data_file = NULL, ...) {
      local_var_response <- self$SchemasWithHttpInfo(data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Retrieve JSON schemas for all item types
    #'
    #' @description
    #' Retrieve JSON schemas for all item types
    #'
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (object) with additional information such as HTTP status code, headers
    #' @export
    SchemasWithHttpInfo = function(data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      local_var_url_path <- "/profiles"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "object", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' Search for items in the IGVF Project.
    #'
    #' @description
    #' Search for items in the IGVF Project.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param type (optional) Filter by item type.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
    #' @param field_filters (optional) Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SearchResults
    #' @export
    Search = function(frame, query = NULL, type = NULL, limit = NULL, sort = NULL, field_filters = NULL, data_file = NULL, ...) {
      local_var_response <- self$SearchWithHttpInfo(frame, query, type, limit, sort, field_filters, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' Search for items in the IGVF Project.
    #'
    #' @description
    #' Search for items in the IGVF Project.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param type (optional) Filter by item type.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Does not work with limit=all.
    #' @param field_filters (optional) Any field from any item type can be used as a filter. Use '!' at end of field name for negation and 'lt:', 'lte:', 'gt:', 'gte:' with value for range queries on numeric fields. Examples: {'status!': 'in progress', 'file_size': 'gte:30000'}
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SearchResults) with additional information such as HTTP status code, headers
    #' @export
    SearchWithHttpInfo = function(frame, query = NULL, type = NULL, limit = NULL, sort = NULL, field_filters = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }







      query_params[["query"]] <- `query`

      # explore
      for (query_item in `type`) {
        query_params[["type"]] <- c(query_params[["type"]], list(`type` = query_item))
      }

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      for (i in seq_along(`field_filters`)) {
          k <- names(`field_filters`)[[i]]
          v <- `field_filters`[[i]]
          query_params[[k]] <- v
      }

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Search. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      local_var_url_path <- "/search"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SearchResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the SequenceFile collection.
    #'
    #' @description
    #' List items in the SequenceFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param flowcell_id (optional) Filter by flowcell_id
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param illumina_read_type (optional) Filter by illumina_read_type
    #' @param index (optional) Filter by index
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lane (optional) Filter by lane
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param maximum_read_length (optional) Filter by maximum_read_length
    #' @param md5sum (optional) Filter by md5sum
    #' @param mean_read_length (optional) Filter by mean_read_length
    #' @param minimum_read_length (optional) Filter by minimum_read_length
    #' @param notes (optional) Filter by notes
    #' @param read_count (optional) Filter by read_count
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param seqspecs (optional) Filter by seqspecs
    #' @param sequencing_kit (optional) Filter by sequencing_kit
    #' @param sequencing_platform (optional) Filter by sequencing_platform
    #' @param sequencing_run (optional) Filter by sequencing_run
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SequenceFileResults
    #' @export
    SequenceFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, flowcell_id = NULL, gene_list_for = NULL, href = NULL, illumina_read_type = NULL, index = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, lane = NULL, loci_list_for = NULL, maximum_read_length = NULL, md5sum = NULL, mean_read_length = NULL, minimum_read_length = NULL, notes = NULL, read_count = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, seqspecs = NULL, sequencing_kit = NULL, sequencing_platform = NULL, sequencing_run = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$SequenceFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, anvil_url, award_id, award_component, collections, content_md5sum, content_type, controlled_access, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_set, file_size, flowcell_id, gene_list_for, href, illumina_read_type, index, input_file_for, integrated_in, lab_id, lab_title, lane, loci_list_for, maximum_read_length, md5sum, mean_read_length, minimum_read_length, notes, read_count, release_timestamp, revoke_detail, s3_uri, seqspecs, sequencing_kit, sequencing_platform, sequencing_run, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the SequenceFile collection.
    #'
    #' @description
    #' List items in the SequenceFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param flowcell_id (optional) Filter by flowcell_id
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param illumina_read_type (optional) Filter by illumina_read_type
    #' @param index (optional) Filter by index
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lane (optional) Filter by lane
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param maximum_read_length (optional) Filter by maximum_read_length
    #' @param md5sum (optional) Filter by md5sum
    #' @param mean_read_length (optional) Filter by mean_read_length
    #' @param minimum_read_length (optional) Filter by minimum_read_length
    #' @param notes (optional) Filter by notes
    #' @param read_count (optional) Filter by read_count
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param seqspecs (optional) Filter by seqspecs
    #' @param sequencing_kit (optional) Filter by sequencing_kit
    #' @param sequencing_platform (optional) Filter by sequencing_platform
    #' @param sequencing_run (optional) Filter by sequencing_run
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SequenceFileResults) with additional information such as HTTP status code, headers
    #' @export
    SequenceFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, flowcell_id = NULL, gene_list_for = NULL, href = NULL, illumina_read_type = NULL, index = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, lane = NULL, loci_list_for = NULL, maximum_read_length = NULL, md5sum = NULL, mean_read_length = NULL, minimum_read_length = NULL, notes = NULL, read_count = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, seqspecs = NULL, sequencing_kit = NULL, sequencing_platform = NULL, sequencing_run = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



























































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$SequenceFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `anvil_url`) {
        query_params[["anvil_url"]] <- c(query_params[["anvil_url"]], list(`anvil_url` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$SequenceFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$SequenceFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `controlled_access`) {
        query_params[["controlled_access"]] <- c(query_params[["controlled_access"]], list(`controlled_access` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("bam", "fastq", "pod5"))) {
          stop("Invalid value for file_format when calling IgvfApi$SequenceFiles. Must be [bam, fastq, pod5].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `flowcell_id`) {
        query_params[["flowcell_id"]] <- c(query_params[["flowcell_id"]], list(`flowcell_id` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `illumina_read_type`) {
        # validate enum values
        if (!(query_item %in% c("R1", "R2", "R3", "I1", "I2"))) {
          stop("Invalid value for illumina_read_type when calling IgvfApi$SequenceFiles. Must be [R1, R2, R3, I1, I2].")
        }
        query_params[["illumina_read_type"]] <- c(query_params[["illumina_read_type"]], list(`illumina_read_type` = query_item))
      }

      # explore
      for (query_item in `index`) {
        query_params[["index"]] <- c(query_params[["index"]], list(`index` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lane`) {
        query_params[["lane"]] <- c(query_params[["lane"]], list(`lane` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `maximum_read_length`) {
        query_params[["maximum_read_length"]] <- c(query_params[["maximum_read_length"]], list(`maximum_read_length` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `mean_read_length`) {
        query_params[["mean_read_length"]] <- c(query_params[["mean_read_length"]], list(`mean_read_length` = query_item))
      }

      # explore
      for (query_item in `minimum_read_length`) {
        query_params[["minimum_read_length"]] <- c(query_params[["minimum_read_length"]], list(`minimum_read_length` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `read_count`) {
        query_params[["read_count"]] <- c(query_params[["read_count"]], list(`read_count` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `seqspecs`) {
        query_params[["seqspecs"]] <- c(query_params[["seqspecs"]], list(`seqspecs` = query_item))
      }

      # explore
      for (query_item in `sequencing_kit`) {
        # validate enum values
        if (!(query_item %in% c("HiSeq SBS Kit v4", "HiSeq SR Cluster Kit v4-cBot-HS", "HiSeq PE Cluster Kit v4-cBot-HS", "HiSeq SR Rapid Cluster Kit v2", "HiSeq PE Rapid Cluster Kit v2", "HiSeq Rapid SBS Kit v2", "HiSeq 3000/4000 SBS Kit", "HiSeq 3000/4000 SR Cluster Kit", "HiSeq 3000/4000 PE Cluster Kit", "MiSeq Reagent Kit v2", "NextSeq 500 Mid Output Kit", "NextSeq 500 High Output Kit", "NextSeq 500 Mid Output v2 Kit", "NextSeq 500 High Output v2 Kit", "NextSeq 500/550 Mid-Output v2.5 Kit", "NextSeq 500/550 High-Output v2.5 Kit", "TG NextSeq 500/550 Mid-Output Kit v2.5", "TG NextSeq 500/550 High-Output Kit v2.5", "NextSeq 1000/2000 P1 Reagent Kit", "NextSeq 1000/2000 P2 Reagent Kit", "NextSeq 1000/2000 P3 Reagent Kit", "NextSeq 1000/2000 P1 XLEAP-SBS Reagent Kit", "NextSeq 1000/2000 P2 XLEAP-SBS Reagent Kit", "NextSeq 2000 P3 XLEAP-SBS Reagent Kit", "NextSeq 2000 P4 XLEAP-SBS Reagent Kit", "NovaSeq 6000 SP Reagent Kit v1.5", "NovaSeq 6000 S1 Reagent Kit v1.5", "NovaSeq 6000 S2 Reagent Kit v1.5", "NovaSeq 6000 S4 Reagent Kit v1.5", "NovaSeq X Series 10B Reagent Kit", "ONT Ligation Sequencing Kit V14", "Sequel sequencing kit 3.0", "Sequel II sequencing kit 2.0"))) {
          stop("Invalid value for sequencing_kit when calling IgvfApi$SequenceFiles. Must be [HiSeq SBS Kit v4, HiSeq SR Cluster Kit v4-cBot-HS, HiSeq PE Cluster Kit v4-cBot-HS, HiSeq SR Rapid Cluster Kit v2, HiSeq PE Rapid Cluster Kit v2, HiSeq Rapid SBS Kit v2, HiSeq 3000/4000 SBS Kit, HiSeq 3000/4000 SR Cluster Kit, HiSeq 3000/4000 PE Cluster Kit, MiSeq Reagent Kit v2, NextSeq 500 Mid Output Kit, NextSeq 500 High Output Kit, NextSeq 500 Mid Output v2 Kit, NextSeq 500 High Output v2 Kit, NextSeq 500/550 Mid-Output v2.5 Kit, NextSeq 500/550 High-Output v2.5 Kit, TG NextSeq 500/550 Mid-Output Kit v2.5, TG NextSeq 500/550 High-Output Kit v2.5, NextSeq 1000/2000 P1 Reagent Kit, NextSeq 1000/2000 P2 Reagent Kit, NextSeq 1000/2000 P3 Reagent Kit, NextSeq 1000/2000 P1 XLEAP-SBS Reagent Kit, NextSeq 1000/2000 P2 XLEAP-SBS Reagent Kit, NextSeq 2000 P3 XLEAP-SBS Reagent Kit, NextSeq 2000 P4 XLEAP-SBS Reagent Kit, NovaSeq 6000 SP Reagent Kit v1.5, NovaSeq 6000 S1 Reagent Kit v1.5, NovaSeq 6000 S2 Reagent Kit v1.5, NovaSeq 6000 S4 Reagent Kit v1.5, NovaSeq X Series 10B Reagent Kit, ONT Ligation Sequencing Kit V14, Sequel sequencing kit 3.0, Sequel II sequencing kit 2.0].")
        }
        query_params[["sequencing_kit"]] <- c(query_params[["sequencing_kit"]], list(`sequencing_kit` = query_item))
      }

      # explore
      for (query_item in `sequencing_platform`) {
        query_params[["sequencing_platform"]] <- c(query_params[["sequencing_platform"]], list(`sequencing_platform` = query_item))
      }

      # explore
      for (query_item in `sequencing_run`) {
        query_params[["sequencing_run"]] <- c(query_params[["sequencing_run"]], list(`sequencing_run` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$SequenceFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$SequenceFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/sequence-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SequenceFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the SignalFile collection.
    #'
    #' @description
    #' List items in the SignalFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_summary (optional) Filter by content_summary
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param filtered (optional) Filter by filtered
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param normalized (optional) Filter by normalized
    #' @param notes (optional) Filter by notes
    #' @param reference_files (optional) Filter by reference_files
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param start_view_position (optional) Filter by start_view_position
    #' @param status (optional) Filter by status
    #' @param strand_specificity (optional) Filter by strand_specificity
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SignalFileResults
    #' @export
    SignalFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_summary = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, filtered = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, normalized = NULL, notes = NULL, reference_files = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, start_view_position = NULL, status = NULL, strand_specificity = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$SignalFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, assembly, award_id, award_component, collections, content_md5sum, content_summary, content_type, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_set, file_size, filtered, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, normalized, notes, reference_files, release_timestamp, revoke_detail, s3_uri, start_view_position, status, strand_specificity, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, transcriptome_annotation, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the SignalFile collection.
    #'
    #' @description
    #' List items in the SignalFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_summary (optional) Filter by content_summary
    #' @param content_type (optional) Filter by content_type
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param filtered (optional) Filter by filtered
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param normalized (optional) Filter by normalized
    #' @param notes (optional) Filter by notes
    #' @param reference_files (optional) Filter by reference_files
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param start_view_position (optional) Filter by start_view_position
    #' @param status (optional) Filter by status
    #' @param strand_specificity (optional) Filter by strand_specificity
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SignalFileResults) with additional information such as HTTP status code, headers
    #' @export
    SignalFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_summary = NULL, content_type = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_set = NULL, file_size = NULL, filtered = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, normalized = NULL, notes = NULL, reference_files = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, start_view_position = NULL, status = NULL, strand_specificity = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$SignalFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `assembly`) {
        # validate enum values
        if (!(query_item %in% c("GRCh38", "GRCm39"))) {
          stop("Invalid value for assembly when calling IgvfApi$SignalFiles. Must be [GRCh38, GRCm39].")
        }
        query_params[["assembly"]] <- c(query_params[["assembly"]], list(`assembly` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$SignalFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$SignalFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_summary`) {
        query_params[["content_summary"]] <- c(query_params[["content_summary"]], list(`content_summary` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("bigWig"))) {
          stop("Invalid value for file_format when calling IgvfApi$SignalFiles. Must be [bigWig].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `filtered`) {
        query_params[["filtered"]] <- c(query_params[["filtered"]], list(`filtered` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `normalized`) {
        query_params[["normalized"]] <- c(query_params[["normalized"]], list(`normalized` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `reference_files`) {
        query_params[["reference_files"]] <- c(query_params[["reference_files"]], list(`reference_files` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `start_view_position`) {
        query_params[["start_view_position"]] <- c(query_params[["start_view_position"]], list(`start_view_position` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$SignalFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `strand_specificity`) {
        # validate enum values
        if (!(query_item %in% c("plus", "minus", "unstranded"))) {
          stop("Invalid value for strand_specificity when calling IgvfApi$SignalFiles. Must be [plus, minus, unstranded].")
        }
        query_params[["strand_specificity"]] <- c(query_params[["strand_specificity"]], list(`strand_specificity` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotation`) {
        # validate enum values
        if (!(query_item %in% c("GENCODE 40", "GENCODE 41", "GENCODE 42", "GENCODE 43", "GENCODE 44", "GENCODE 45", "GENCODE M30", "GENCODE M31", "GENCODE M32", "GENCODE M33", "GENCODE M34"))) {
          stop("Invalid value for transcriptome_annotation when calling IgvfApi$SignalFiles. Must be [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34].")
        }
        query_params[["transcriptome_annotation"]] <- c(query_params[["transcriptome_annotation"]], list(`transcriptome_annotation` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$SignalFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/signal-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SignalFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Software collection.
    #'
    #' @description
    #' List items in the Software collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param source_url (optional) Filter by source_url
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param used_by (optional) Filter by used_by
    #' @param uuid (optional) Filter by uuid
    #' @param versions (optional) Filter by versions
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SoftwareResults
    #' @export
    Software = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, source_url = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, used_by = NULL, uuid = NULL, versions = NULL, data_file = NULL, ...) {
      local_var_response <- self$SoftwareWithHttpInfo(frame, query, limit, sort, id, aliases, award_id, award_component, creation_timestamp, description, lab_id, lab_title, name, notes, publication_identifiers, publications, release_timestamp, source_url, status, submitted_by_id, submitted_by_title, submitter_comment, summary, title, used_by, uuid, versions, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Software collection.
    #'
    #' @description
    #' List items in the Software collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param source_url (optional) Filter by source_url
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param used_by (optional) Filter by used_by
    #' @param uuid (optional) Filter by uuid
    #' @param versions (optional) Filter by versions
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SoftwareResults) with additional information such as HTTP status code, headers
    #' @export
    SoftwareWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, source_url = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, used_by = NULL, uuid = NULL, versions = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }




























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Software. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Software. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `source_url`) {
        query_params[["source_url"]] <- c(query_params[["source_url"]], list(`source_url` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Software. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `used_by`) {
        # validate enum values
        if (!(query_item %in% c("consortium analysis", "DACC", "processing pipelines"))) {
          stop("Invalid value for used_by when calling IgvfApi$Software. Must be [consortium analysis, DACC, processing pipelines].")
        }
        query_params[["used_by"]] <- c(query_params[["used_by"]], list(`used_by` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `versions`) {
        query_params[["versions"]] <- c(query_params[["versions"]], list(`versions` = query_item))
      }

      local_var_url_path <- "/software/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SoftwareResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the SoftwareVersion collection.
    #'
    #' @description
    #' List items in the SoftwareVersion collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param download_id (optional) Filter by download_id
    #' @param downloaded_url (optional) Filter by downloaded_url
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param software_id (optional) Filter by software.@id
    #' @param software_title (optional) Filter by software.title
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param version (optional) Filter by version
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SoftwareVersionResults
    #' @export
    SoftwareVersions = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, download_id = NULL, downloaded_url = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, software_id = NULL, software_title = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, version = NULL, data_file = NULL, ...) {
      local_var_response <- self$SoftwareVersionsWithHttpInfo(frame, query, limit, sort, id, aliases, award_id, award_component, creation_timestamp, description, download_id, downloaded_url, lab_id, lab_title, name, notes, publication_identifiers, publications, release_timestamp, software_id, software_title, status, submitted_by_id, submitted_by_title, submitter_comment, summary, uuid, version, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the SoftwareVersion collection.
    #'
    #' @description
    #' List items in the SoftwareVersion collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param download_id (optional) Filter by download_id
    #' @param downloaded_url (optional) Filter by downloaded_url
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param software_id (optional) Filter by software.@id
    #' @param software_title (optional) Filter by software.title
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param version (optional) Filter by version
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SoftwareVersionResults) with additional information such as HTTP status code, headers
    #' @export
    SoftwareVersionsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, award_id = NULL, award_component = NULL, creation_timestamp = NULL, description = NULL, download_id = NULL, downloaded_url = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, software_id = NULL, software_title = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, version = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$SoftwareVersions. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$SoftwareVersions. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `download_id`) {
        query_params[["download_id"]] <- c(query_params[["download_id"]], list(`download_id` = query_item))
      }

      # explore
      for (query_item in `downloaded_url`) {
        query_params[["downloaded_url"]] <- c(query_params[["downloaded_url"]], list(`downloaded_url` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `software_id`) {
        query_params[["software.@id"]] <- c(query_params[["software.@id"]], list(`software.@id` = query_item))
      }

      # explore
      for (query_item in `software_title`) {
        query_params[["software.title"]] <- c(query_params[["software.title"]], list(`software.title` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$SoftwareVersions. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `version`) {
        query_params[["version"]] <- c(query_params[["version"]], list(`version` = query_item))
      }

      local_var_url_path <- "/software-versions/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SoftwareVersionResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Source collection.
    #'
    #' @description
    #' List items in the Source collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return SourceResults
    #' @export
    Sources = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, creation_timestamp = NULL, description = NULL, name = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$SourcesWithHttpInfo(frame, query, limit, sort, id, aliases, creation_timestamp, description, name, notes, release_timestamp, status, submitted_by_id, submitted_by_title, submitter_comment, summary, title, url, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Source collection.
    #'
    #' @description
    #' List items in the Source collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (SourceResults) with additional information such as HTTP status code, headers
    #' @export
    SourcesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, creation_timestamp = NULL, description = NULL, name = NULL, notes = NULL, release_timestamp = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, url = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }




















      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Sources. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Sources. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/sources/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "SourceResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the TabularFile collection.
    #'
    #' @description
    #' List items in the TabularFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_format_type (optional) Filter by file_format_type
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return TabularFileResults
    #' @export
    TabularFiles = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_format_type = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      local_var_response <- self$TabularFilesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_step_version, anvil_url, assembly, award_id, award_component, collections, content_md5sum, content_type, controlled_access, creation_timestamp, dbxrefs, derived_from, description, documents, file_format, file_format_specifications, file_format_type, file_set, file_size, gene_list_for, href, input_file_for, integrated_in, lab_id, lab_title, loci_list_for, md5sum, notes, release_timestamp, revoke_detail, s3_uri, status, submitted_by_id, submitted_by_title, submitted_file_name, submitter_comment, summary, transcriptome_annotation, upload_status, uuid, validation_error_detail, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the TabularFile collection.
    #'
    #' @description
    #' List items in the TabularFile collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_step_version (optional) Filter by analysis_step_version
    #' @param anvil_url (optional) Filter by anvil_url
    #' @param assembly (optional) Filter by assembly
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param content_md5sum (optional) Filter by content_md5sum
    #' @param content_type (optional) Filter by content_type
    #' @param controlled_access (optional) Filter by controlled_access
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param derived_from (optional) Filter by derived_from
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_format (optional) Filter by file_format
    #' @param file_format_specifications (optional) Filter by file_format_specifications
    #' @param file_format_type (optional) Filter by file_format_type
    #' @param file_set (optional) Filter by file_set
    #' @param file_size (optional) Filter by file_size
    #' @param gene_list_for (optional) Filter by gene_list_for
    #' @param href (optional) Filter by href
    #' @param input_file_for (optional) Filter by input_file_for
    #' @param integrated_in (optional) Filter by integrated_in
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param loci_list_for (optional) Filter by loci_list_for
    #' @param md5sum (optional) Filter by md5sum
    #' @param notes (optional) Filter by notes
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param s3_uri (optional) Filter by s3_uri
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitted_file_name (optional) Filter by submitted_file_name
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param transcriptome_annotation (optional) Filter by transcriptome_annotation
    #' @param upload_status (optional) Filter by upload_status
    #' @param uuid (optional) Filter by uuid
    #' @param validation_error_detail (optional) Filter by validation_error_detail
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (TabularFileResults) with additional information such as HTTP status code, headers
    #' @export
    TabularFilesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_step_version = NULL, anvil_url = NULL, assembly = NULL, award_id = NULL, award_component = NULL, collections = NULL, content_md5sum = NULL, content_type = NULL, controlled_access = NULL, creation_timestamp = NULL, dbxrefs = NULL, derived_from = NULL, description = NULL, documents = NULL, file_format = NULL, file_format_specifications = NULL, file_format_type = NULL, file_set = NULL, file_size = NULL, gene_list_for = NULL, href = NULL, input_file_for = NULL, integrated_in = NULL, lab_id = NULL, lab_title = NULL, loci_list_for = NULL, md5sum = NULL, notes = NULL, release_timestamp = NULL, revoke_detail = NULL, s3_uri = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitted_file_name = NULL, submitter_comment = NULL, summary = NULL, transcriptome_annotation = NULL, upload_status = NULL, uuid = NULL, validation_error_detail = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }


















































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$TabularFiles. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_step_version`) {
        query_params[["analysis_step_version"]] <- c(query_params[["analysis_step_version"]], list(`analysis_step_version` = query_item))
      }

      # explore
      for (query_item in `anvil_url`) {
        query_params[["anvil_url"]] <- c(query_params[["anvil_url"]], list(`anvil_url` = query_item))
      }

      # explore
      for (query_item in `assembly`) {
        # validate enum values
        if (!(query_item %in% c("GRCh38", "GRCm39"))) {
          stop("Invalid value for assembly when calling IgvfApi$TabularFiles. Must be [GRCh38, GRCm39].")
        }
        query_params[["assembly"]] <- c(query_params[["assembly"]], list(`assembly` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$TabularFiles. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$TabularFiles. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `content_md5sum`) {
        query_params[["content_md5sum"]] <- c(query_params[["content_md5sum"]], list(`content_md5sum` = query_item))
      }

      # explore
      for (query_item in `content_type`) {
        query_params[["content_type"]] <- c(query_params[["content_type"]], list(`content_type` = query_item))
      }

      # explore
      for (query_item in `controlled_access`) {
        query_params[["controlled_access"]] <- c(query_params[["controlled_access"]], list(`controlled_access` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `derived_from`) {
        query_params[["derived_from"]] <- c(query_params[["derived_from"]], list(`derived_from` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_format`) {
        # validate enum values
        if (!(query_item %in% c("bed", "csv", "gtf", "tsv", "txt", "vcf"))) {
          stop("Invalid value for file_format when calling IgvfApi$TabularFiles. Must be [bed, csv, gtf, tsv, txt, vcf].")
        }
        query_params[["file_format"]] <- c(query_params[["file_format"]], list(`file_format` = query_item))
      }

      # explore
      for (query_item in `file_format_specifications`) {
        query_params[["file_format_specifications"]] <- c(query_params[["file_format_specifications"]], list(`file_format_specifications` = query_item))
      }

      # explore
      for (query_item in `file_format_type`) {
        # validate enum values
        if (!(query_item %in% c("bed12", "bed3", "bed3+", "bed5", "bed6", "bed6+", "bed9", "bed9+", "mpra_starr"))) {
          stop("Invalid value for file_format_type when calling IgvfApi$TabularFiles. Must be [bed12, bed3, bed3+, bed5, bed6, bed6+, bed9, bed9+, mpra_starr].")
        }
        query_params[["file_format_type"]] <- c(query_params[["file_format_type"]], list(`file_format_type` = query_item))
      }

      # explore
      for (query_item in `file_set`) {
        query_params[["file_set"]] <- c(query_params[["file_set"]], list(`file_set` = query_item))
      }

      # explore
      for (query_item in `file_size`) {
        query_params[["file_size"]] <- c(query_params[["file_size"]], list(`file_size` = query_item))
      }

      # explore
      for (query_item in `gene_list_for`) {
        query_params[["gene_list_for"]] <- c(query_params[["gene_list_for"]], list(`gene_list_for` = query_item))
      }

      # explore
      for (query_item in `href`) {
        query_params[["href"]] <- c(query_params[["href"]], list(`href` = query_item))
      }

      # explore
      for (query_item in `input_file_for`) {
        query_params[["input_file_for"]] <- c(query_params[["input_file_for"]], list(`input_file_for` = query_item))
      }

      # explore
      for (query_item in `integrated_in`) {
        query_params[["integrated_in"]] <- c(query_params[["integrated_in"]], list(`integrated_in` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `loci_list_for`) {
        query_params[["loci_list_for"]] <- c(query_params[["loci_list_for"]], list(`loci_list_for` = query_item))
      }

      # explore
      for (query_item in `md5sum`) {
        query_params[["md5sum"]] <- c(query_params[["md5sum"]], list(`md5sum` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `s3_uri`) {
        query_params[["s3_uri"]] <- c(query_params[["s3_uri"]], list(`s3_uri` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$TabularFiles. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitted_file_name`) {
        query_params[["submitted_file_name"]] <- c(query_params[["submitted_file_name"]], list(`submitted_file_name` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `transcriptome_annotation`) {
        # validate enum values
        if (!(query_item %in% c("GENCODE 40", "GENCODE 41", "GENCODE 42", "GENCODE 43", "GENCODE 44", "GENCODE 45", "GENCODE M30", "GENCODE M31", "GENCODE M32", "GENCODE M33", "GENCODE M34"))) {
          stop("Invalid value for transcriptome_annotation when calling IgvfApi$TabularFiles. Must be [GENCODE 40, GENCODE 41, GENCODE 42, GENCODE 43, GENCODE 44, GENCODE 45, GENCODE M30, GENCODE M31, GENCODE M32, GENCODE M33, GENCODE M34].")
        }
        query_params[["transcriptome_annotation"]] <- c(query_params[["transcriptome_annotation"]], list(`transcriptome_annotation` = query_item))
      }

      # explore
      for (query_item in `upload_status`) {
        # validate enum values
        if (!(query_item %in% c("pending", "file not found", "invalidated", "validated"))) {
          stop("Invalid value for upload_status when calling IgvfApi$TabularFiles. Must be [pending, file not found, invalidated, validated].")
        }
        query_params[["upload_status"]] <- c(query_params[["upload_status"]], list(`upload_status` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `validation_error_detail`) {
        query_params[["validation_error_detail"]] <- c(query_params[["validation_error_detail"]], list(`validation_error_detail` = query_item))
      }

      local_var_url_path <- "/tabular-files/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "TabularFileResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the TechnicalSample collection.
    #'
    #' @description
    #' List items in the TechnicalSample collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates (optional) Filter by institutional_certificates
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_material (optional) Filter by sample_material
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return TechnicalSampleResults
    #' @export
    TechnicalSamples = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, documents = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_material = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$TechnicalSamplesWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, award_id, award_component, classifications, collections, construct_library_sets, creation_timestamp, date_obtained, dbxrefs, description, documents, file_sets_id, file_sets_accession, file_sets_aliases, file_sets_assay_term_term_name, file_sets_lab_title, file_sets_status, file_sets_summary, institutional_certificates, lab_id, lab_title, lot_id, moi, multiplexed_in_id, multiplexed_in_accession, notes, nucleic_acid_delivery, origin_of, product_id, protocols, publication_identifiers, publications, release_timestamp, revoke_detail, sample_material, sample_terms_id, sample_terms_term_name, sorted_fractions, sorted_from_id, sorted_from_accession, sorted_from_detail, sources_id, starting_amount, starting_amount_units, status, submitted_by_id, submitted_by_title, submitter_comment, summary, taxa, time_post_library_delivery, time_post_library_delivery_units, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the TechnicalSample collection.
    #'
    #' @description
    #' List items in the TechnicalSample collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates (optional) Filter by institutional_certificates
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_material (optional) Filter by sample_material
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (TechnicalSampleResults) with additional information such as HTTP status code, headers
    #' @export
    TechnicalSamplesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, documents = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_material = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$TechnicalSamples. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$TechnicalSamples. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `classifications`) {
        query_params[["classifications"]] <- c(query_params[["classifications"]], list(`classifications` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$TechnicalSamples. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets`) {
        query_params[["construct_library_sets"]] <- c(query_params[["construct_library_sets"]], list(`construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_obtained`) {
        query_params[["date_obtained"]] <- c(query_params[["date_obtained"]], list(`date_obtained` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `file_sets_id`) {
        query_params[["file_sets.@id"]] <- c(query_params[["file_sets.@id"]], list(`file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `file_sets_accession`) {
        query_params[["file_sets.accession"]] <- c(query_params[["file_sets.accession"]], list(`file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `file_sets_aliases`) {
        query_params[["file_sets.aliases"]] <- c(query_params[["file_sets.aliases"]], list(`file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `file_sets_assay_term_term_name`) {
        query_params[["file_sets.assay_term.term_name"]] <- c(query_params[["file_sets.assay_term.term_name"]], list(`file_sets.assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `file_sets_lab_title`) {
        query_params[["file_sets.lab.title"]] <- c(query_params[["file_sets.lab.title"]], list(`file_sets.lab.title` = query_item))
      }

      # explore
      for (query_item in `file_sets_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for file_sets_status when calling IgvfApi$TechnicalSamples. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["file_sets.status"]] <- c(query_params[["file_sets.status"]], list(`file_sets.status` = query_item))
      }

      # explore
      for (query_item in `file_sets_summary`) {
        query_params[["file_sets.summary"]] <- c(query_params[["file_sets.summary"]], list(`file_sets.summary` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates`) {
        query_params[["institutional_certificates"]] <- c(query_params[["institutional_certificates"]], list(`institutional_certificates` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `moi`) {
        query_params[["moi"]] <- c(query_params[["moi"]], list(`moi` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_id`) {
        query_params[["multiplexed_in.@id"]] <- c(query_params[["multiplexed_in.@id"]], list(`multiplexed_in.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_accession`) {
        query_params[["multiplexed_in.accession"]] <- c(query_params[["multiplexed_in.accession"]], list(`multiplexed_in.accession` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `nucleic_acid_delivery`) {
        # validate enum values
        if (!(query_item %in% c("transfection", "adenoviral transduction", "lentiviral transduction"))) {
          stop("Invalid value for nucleic_acid_delivery when calling IgvfApi$TechnicalSamples. Must be [transfection, adenoviral transduction, lentiviral transduction].")
        }
        query_params[["nucleic_acid_delivery"]] <- c(query_params[["nucleic_acid_delivery"]], list(`nucleic_acid_delivery` = query_item))
      }

      # explore
      for (query_item in `origin_of`) {
        query_params[["origin_of"]] <- c(query_params[["origin_of"]], list(`origin_of` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sample_material`) {
        # validate enum values
        if (!(query_item %in% c("undefined", "inorganic", "synthetic", "organic"))) {
          stop("Invalid value for sample_material when calling IgvfApi$TechnicalSamples. Must be [undefined, inorganic, synthetic, organic].")
        }
        query_params[["sample_material"]] <- c(query_params[["sample_material"]], list(`sample_material` = query_item))
      }

      # explore
      for (query_item in `sample_terms_id`) {
        query_params[["sample_terms.@id"]] <- c(query_params[["sample_terms.@id"]], list(`sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `sample_terms_term_name`) {
        query_params[["sample_terms.term_name"]] <- c(query_params[["sample_terms.term_name"]], list(`sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `sorted_fractions`) {
        query_params[["sorted_fractions"]] <- c(query_params[["sorted_fractions"]], list(`sorted_fractions` = query_item))
      }

      # explore
      for (query_item in `sorted_from_id`) {
        query_params[["sorted_from.@id"]] <- c(query_params[["sorted_from.@id"]], list(`sorted_from.@id` = query_item))
      }

      # explore
      for (query_item in `sorted_from_accession`) {
        query_params[["sorted_from.accession"]] <- c(query_params[["sorted_from.accession"]], list(`sorted_from.accession` = query_item))
      }

      # explore
      for (query_item in `sorted_from_detail`) {
        query_params[["sorted_from_detail"]] <- c(query_params[["sorted_from_detail"]], list(`sorted_from_detail` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `starting_amount`) {
        query_params[["starting_amount"]] <- c(query_params[["starting_amount"]], list(`starting_amount` = query_item))
      }

      # explore
      for (query_item in `starting_amount_units`) {
        # validate enum values
        if (!(query_item %in% c("cells", "cells/ml", "g", "items", "mg", "whole animals", "whole embryos", "μg", "ng"))) {
          stop("Invalid value for starting_amount_units when calling IgvfApi$TechnicalSamples. Must be [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng].")
        }
        query_params[["starting_amount_units"]] <- c(query_params[["starting_amount_units"]], list(`starting_amount_units` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$TechnicalSamples. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Saccharomyces cerevisiae", "Homo sapiens"))) {
          stop("Invalid value for taxa when calling IgvfApi$TechnicalSamples. Must be [Saccharomyces cerevisiae, Homo sapiens].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery`) {
        query_params[["time_post_library_delivery"]] <- c(query_params[["time_post_library_delivery"]], list(`time_post_library_delivery` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_library_delivery_units when calling IgvfApi$TechnicalSamples. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_library_delivery_units"]] <- c(query_params[["time_post_library_delivery_units"]], list(`time_post_library_delivery_units` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/technical-samples/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "TechnicalSampleResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Tissue collection.
    #'
    #' @description
    #' List items in the Tissue collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param ccf_id (optional) Filter by ccf_id
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from (optional) Filter by originated_from
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param pmi (optional) Filter by pmi
    #' @param pmi_units (optional) Filter by pmi_units
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param preservation_method (optional) Filter by preservation_method
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return TissueResults
    #' @export
    Tissues = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, ccf_id = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from = NULL, part_of = NULL, parts = NULL, pmi = NULL, pmi_units = NULL, pooled_from = NULL, pooled_in = NULL, preservation_method = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$TissuesWithHttpInfo(frame, query, limit, sort, id, accession, age, age_units, aliases, alternate_accessions, award_id, award_component, biomarkers, ccf_id, cellular_sub_pool, classifications, collections, construct_library_sets, creation_timestamp, date_obtained, dbxrefs, description, disease_terms_id, disease_terms_term_name, documents, donors, embryonic, file_sets_id, file_sets_accession, file_sets_aliases, file_sets_assay_term_term_name, file_sets_lab_title, file_sets_status, file_sets_summary, institutional_certificates_id, institutional_certificates_certificate_identifier, lab_id, lab_title, lot_id, lower_bound_age, lower_bound_age_in_hours, modifications_id, modifications_status, modifications_summary, moi, multiplexed_in_id, multiplexed_in_accession, notes, nucleic_acid_delivery, origin_of, originated_from, part_of, parts, pmi, pmi_units, pooled_from, pooled_in, preservation_method, product_id, protocols, publication_identifiers, publications, release_timestamp, revoke_detail, sample_terms_id, sample_terms_term_name, sex, sorted_fractions, sorted_from_id, sorted_from_accession, sorted_from_detail, sources_id, starting_amount, starting_amount_units, status, submitted_by_id, submitted_by_title, submitter_comment, summary, taxa, time_post_library_delivery, time_post_library_delivery_units, treatments_id, treatments_purpose, treatments_status, treatments_summary, treatments_treatment_type, upper_bound_age, upper_bound_age_in_hours, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Tissue collection.
    #'
    #' @description
    #' List items in the Tissue collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param ccf_id (optional) Filter by ccf_id
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from (optional) Filter by originated_from
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param pmi (optional) Filter by pmi
    #' @param pmi_units (optional) Filter by pmi_units
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param preservation_method (optional) Filter by preservation_method
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (TissueResults) with additional information such as HTTP status code, headers
    #' @export
    TissuesWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, ccf_id = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from = NULL, part_of = NULL, parts = NULL, pmi = NULL, pmi_units = NULL, pooled_from = NULL, pooled_in = NULL, preservation_method = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }





























































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Tissues. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `age`) {
        query_params[["age"]] <- c(query_params[["age"]], list(`age` = query_item))
      }

      # explore
      for (query_item in `age_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month", "year"))) {
          stop("Invalid value for age_units when calling IgvfApi$Tissues. Must be [minute, hour, day, week, month, year].")
        }
        query_params[["age_units"]] <- c(query_params[["age_units"]], list(`age_units` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Tissues. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biomarkers`) {
        query_params[["biomarkers"]] <- c(query_params[["biomarkers"]], list(`biomarkers` = query_item))
      }

      # explore
      for (query_item in `ccf_id`) {
        query_params[["ccf_id"]] <- c(query_params[["ccf_id"]], list(`ccf_id` = query_item))
      }

      # explore
      for (query_item in `cellular_sub_pool`) {
        query_params[["cellular_sub_pool"]] <- c(query_params[["cellular_sub_pool"]], list(`cellular_sub_pool` = query_item))
      }

      # explore
      for (query_item in `classifications`) {
        query_params[["classifications"]] <- c(query_params[["classifications"]], list(`classifications` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$Tissues. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets`) {
        query_params[["construct_library_sets"]] <- c(query_params[["construct_library_sets"]], list(`construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_obtained`) {
        query_params[["date_obtained"]] <- c(query_params[["date_obtained"]], list(`date_obtained` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `disease_terms_id`) {
        query_params[["disease_terms.@id"]] <- c(query_params[["disease_terms.@id"]], list(`disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `disease_terms_term_name`) {
        query_params[["disease_terms.term_name"]] <- c(query_params[["disease_terms.term_name"]], list(`disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors`) {
        query_params[["donors"]] <- c(query_params[["donors"]], list(`donors` = query_item))
      }

      # explore
      for (query_item in `embryonic`) {
        query_params[["embryonic"]] <- c(query_params[["embryonic"]], list(`embryonic` = query_item))
      }

      # explore
      for (query_item in `file_sets_id`) {
        query_params[["file_sets.@id"]] <- c(query_params[["file_sets.@id"]], list(`file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `file_sets_accession`) {
        query_params[["file_sets.accession"]] <- c(query_params[["file_sets.accession"]], list(`file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `file_sets_aliases`) {
        query_params[["file_sets.aliases"]] <- c(query_params[["file_sets.aliases"]], list(`file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `file_sets_assay_term_term_name`) {
        query_params[["file_sets.assay_term.term_name"]] <- c(query_params[["file_sets.assay_term.term_name"]], list(`file_sets.assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `file_sets_lab_title`) {
        query_params[["file_sets.lab.title"]] <- c(query_params[["file_sets.lab.title"]], list(`file_sets.lab.title` = query_item))
      }

      # explore
      for (query_item in `file_sets_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for file_sets_status when calling IgvfApi$Tissues. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["file_sets.status"]] <- c(query_params[["file_sets.status"]], list(`file_sets.status` = query_item))
      }

      # explore
      for (query_item in `file_sets_summary`) {
        query_params[["file_sets.summary"]] <- c(query_params[["file_sets.summary"]], list(`file_sets.summary` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_id`) {
        query_params[["institutional_certificates.@id"]] <- c(query_params[["institutional_certificates.@id"]], list(`institutional_certificates.@id` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_certificate_identifier`) {
        query_params[["institutional_certificates.certificate_identifier"]] <- c(query_params[["institutional_certificates.certificate_identifier"]], list(`institutional_certificates.certificate_identifier` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age`) {
        query_params[["lower_bound_age"]] <- c(query_params[["lower_bound_age"]], list(`lower_bound_age` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age_in_hours`) {
        query_params[["lower_bound_age_in_hours"]] <- c(query_params[["lower_bound_age_in_hours"]], list(`lower_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `modifications_id`) {
        query_params[["modifications.@id"]] <- c(query_params[["modifications.@id"]], list(`modifications.@id` = query_item))
      }

      # explore
      for (query_item in `modifications_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for modifications_status when calling IgvfApi$Tissues. Must be [archived, deleted, in progress, released].")
        }
        query_params[["modifications.status"]] <- c(query_params[["modifications.status"]], list(`modifications.status` = query_item))
      }

      # explore
      for (query_item in `modifications_summary`) {
        query_params[["modifications.summary"]] <- c(query_params[["modifications.summary"]], list(`modifications.summary` = query_item))
      }

      # explore
      for (query_item in `moi`) {
        query_params[["moi"]] <- c(query_params[["moi"]], list(`moi` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_id`) {
        query_params[["multiplexed_in.@id"]] <- c(query_params[["multiplexed_in.@id"]], list(`multiplexed_in.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_accession`) {
        query_params[["multiplexed_in.accession"]] <- c(query_params[["multiplexed_in.accession"]], list(`multiplexed_in.accession` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `nucleic_acid_delivery`) {
        # validate enum values
        if (!(query_item %in% c("transfection", "adenoviral transduction", "lentiviral transduction"))) {
          stop("Invalid value for nucleic_acid_delivery when calling IgvfApi$Tissues. Must be [transfection, adenoviral transduction, lentiviral transduction].")
        }
        query_params[["nucleic_acid_delivery"]] <- c(query_params[["nucleic_acid_delivery"]], list(`nucleic_acid_delivery` = query_item))
      }

      # explore
      for (query_item in `origin_of`) {
        query_params[["origin_of"]] <- c(query_params[["origin_of"]], list(`origin_of` = query_item))
      }

      # explore
      for (query_item in `originated_from`) {
        query_params[["originated_from"]] <- c(query_params[["originated_from"]], list(`originated_from` = query_item))
      }

      # explore
      for (query_item in `part_of`) {
        query_params[["part_of"]] <- c(query_params[["part_of"]], list(`part_of` = query_item))
      }

      # explore
      for (query_item in `parts`) {
        query_params[["parts"]] <- c(query_params[["parts"]], list(`parts` = query_item))
      }

      # explore
      for (query_item in `pmi`) {
        query_params[["pmi"]] <- c(query_params[["pmi"]], list(`pmi` = query_item))
      }

      # explore
      for (query_item in `pmi_units`) {
        # validate enum values
        if (!(query_item %in% c("second", "minute", "hour", "day", "week"))) {
          stop("Invalid value for pmi_units when calling IgvfApi$Tissues. Must be [second, minute, hour, day, week].")
        }
        query_params[["pmi_units"]] <- c(query_params[["pmi_units"]], list(`pmi_units` = query_item))
      }

      # explore
      for (query_item in `pooled_from`) {
        query_params[["pooled_from"]] <- c(query_params[["pooled_from"]], list(`pooled_from` = query_item))
      }

      # explore
      for (query_item in `pooled_in`) {
        query_params[["pooled_in"]] <- c(query_params[["pooled_in"]], list(`pooled_in` = query_item))
      }

      # explore
      for (query_item in `preservation_method`) {
        # validate enum values
        if (!(query_item %in% c("cryopreservation", "flash-freezing"))) {
          stop("Invalid value for preservation_method when calling IgvfApi$Tissues. Must be [cryopreservation, flash-freezing].")
        }
        query_params[["preservation_method"]] <- c(query_params[["preservation_method"]], list(`preservation_method` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sample_terms_id`) {
        query_params[["sample_terms.@id"]] <- c(query_params[["sample_terms.@id"]], list(`sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `sample_terms_term_name`) {
        query_params[["sample_terms.term_name"]] <- c(query_params[["sample_terms.term_name"]], list(`sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `sex`) {
        # validate enum values
        if (!(query_item %in% c("female", "male", "mixed", "unspecified"))) {
          stop("Invalid value for sex when calling IgvfApi$Tissues. Must be [female, male, mixed, unspecified].")
        }
        query_params[["sex"]] <- c(query_params[["sex"]], list(`sex` = query_item))
      }

      # explore
      for (query_item in `sorted_fractions`) {
        query_params[["sorted_fractions"]] <- c(query_params[["sorted_fractions"]], list(`sorted_fractions` = query_item))
      }

      # explore
      for (query_item in `sorted_from_id`) {
        query_params[["sorted_from.@id"]] <- c(query_params[["sorted_from.@id"]], list(`sorted_from.@id` = query_item))
      }

      # explore
      for (query_item in `sorted_from_accession`) {
        query_params[["sorted_from.accession"]] <- c(query_params[["sorted_from.accession"]], list(`sorted_from.accession` = query_item))
      }

      # explore
      for (query_item in `sorted_from_detail`) {
        query_params[["sorted_from_detail"]] <- c(query_params[["sorted_from_detail"]], list(`sorted_from_detail` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `starting_amount`) {
        query_params[["starting_amount"]] <- c(query_params[["starting_amount"]], list(`starting_amount` = query_item))
      }

      # explore
      for (query_item in `starting_amount_units`) {
        # validate enum values
        if (!(query_item %in% c("cells", "cells/ml", "g", "items", "mg", "whole animals", "whole embryos", "μg", "ng"))) {
          stop("Invalid value for starting_amount_units when calling IgvfApi$Tissues. Must be [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng].")
        }
        query_params[["starting_amount_units"]] <- c(query_params[["starting_amount_units"]], list(`starting_amount_units` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$Tissues. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$Tissues. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery`) {
        query_params[["time_post_library_delivery"]] <- c(query_params[["time_post_library_delivery"]], list(`time_post_library_delivery` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_library_delivery_units when calling IgvfApi$Tissues. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_library_delivery_units"]] <- c(query_params[["time_post_library_delivery_units"]], list(`time_post_library_delivery_units` = query_item))
      }

      # explore
      for (query_item in `treatments_id`) {
        query_params[["treatments.@id"]] <- c(query_params[["treatments.@id"]], list(`treatments.@id` = query_item))
      }

      # explore
      for (query_item in `treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for treatments_purpose when calling IgvfApi$Tissues. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["treatments.purpose"]] <- c(query_params[["treatments.purpose"]], list(`treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `treatments_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for treatments_status when calling IgvfApi$Tissues. Must be [archived, deleted, in progress, released].")
        }
        query_params[["treatments.status"]] <- c(query_params[["treatments.status"]], list(`treatments.status` = query_item))
      }

      # explore
      for (query_item in `treatments_summary`) {
        query_params[["treatments.summary"]] <- c(query_params[["treatments.summary"]], list(`treatments.summary` = query_item))
      }

      # explore
      for (query_item in `treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for treatments_treatment_type when calling IgvfApi$Tissues. Must be [chemical, protein, environmental].")
        }
        query_params[["treatments.treatment_type"]] <- c(query_params[["treatments.treatment_type"]], list(`treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age`) {
        query_params[["upper_bound_age"]] <- c(query_params[["upper_bound_age"]], list(`upper_bound_age` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age_in_hours`) {
        query_params[["upper_bound_age_in_hours"]] <- c(query_params[["upper_bound_age_in_hours"]], list(`upper_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/tissues/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "TissueResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Treatment collection.
    #'
    #' @description
    #' List items in the Treatment collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param amount (optional) Filter by amount
    #' @param amount_units (optional) Filter by amount_units
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biosamples_treated (optional) Filter by biosamples_treated
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param depletion (optional) Filter by depletion
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param duration (optional) Filter by duration
    #' @param duration_units (optional) Filter by duration_units
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param notes (optional) Filter by notes
    #' @param p_h (optional) Filter by pH
    #' @param post_treatment_time (optional) Filter by post_treatment_time
    #' @param post_treatment_time_units (optional) Filter by post_treatment_time_units
    #' @param product_id (optional) Filter by product_id
    #' @param purpose (optional) Filter by purpose
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sources_id (optional) Filter by sources.@id
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param temperature (optional) Filter by temperature
    #' @param temperature_units (optional) Filter by temperature_units
    #' @param treatment_term_id (optional) Filter by treatment_term_id
    #' @param treatment_term_name (optional) Filter by treatment_term_name
    #' @param treatment_type (optional) Filter by treatment_type
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return TreatmentResults
    #' @export
    Treatments = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, amount = NULL, amount_units = NULL, award_id = NULL, award_component = NULL, biosamples_treated = NULL, creation_timestamp = NULL, depletion = NULL, description = NULL, documents = NULL, duration = NULL, duration_units = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, notes = NULL, p_h = NULL, post_treatment_time = NULL, post_treatment_time_units = NULL, product_id = NULL, purpose = NULL, release_timestamp = NULL, sources_id = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, temperature = NULL, temperature_units = NULL, treatment_term_id = NULL, treatment_term_name = NULL, treatment_type = NULL, uuid = NULL, data_file = NULL, ...) {
      local_var_response <- self$TreatmentsWithHttpInfo(frame, query, limit, sort, id, aliases, amount, amount_units, award_id, award_component, biosamples_treated, creation_timestamp, depletion, description, documents, duration, duration_units, lab_id, lab_title, lot_id, notes, p_h, post_treatment_time, post_treatment_time_units, product_id, purpose, release_timestamp, sources_id, status, submitted_by_id, submitted_by_title, submitter_comment, summary, temperature, temperature_units, treatment_term_id, treatment_term_name, treatment_type, uuid, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Treatment collection.
    #'
    #' @description
    #' List items in the Treatment collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param amount (optional) Filter by amount
    #' @param amount_units (optional) Filter by amount_units
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biosamples_treated (optional) Filter by biosamples_treated
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param depletion (optional) Filter by depletion
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param duration (optional) Filter by duration
    #' @param duration_units (optional) Filter by duration_units
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param notes (optional) Filter by notes
    #' @param p_h (optional) Filter by pH
    #' @param post_treatment_time (optional) Filter by post_treatment_time
    #' @param post_treatment_time_units (optional) Filter by post_treatment_time_units
    #' @param product_id (optional) Filter by product_id
    #' @param purpose (optional) Filter by purpose
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param sources_id (optional) Filter by sources.@id
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param temperature (optional) Filter by temperature
    #' @param temperature_units (optional) Filter by temperature_units
    #' @param treatment_term_id (optional) Filter by treatment_term_id
    #' @param treatment_term_name (optional) Filter by treatment_term_name
    #' @param treatment_type (optional) Filter by treatment_type
    #' @param uuid (optional) Filter by uuid
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (TreatmentResults) with additional information such as HTTP status code, headers
    #' @export
    TreatmentsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, amount = NULL, amount_units = NULL, award_id = NULL, award_component = NULL, biosamples_treated = NULL, creation_timestamp = NULL, depletion = NULL, description = NULL, documents = NULL, duration = NULL, duration_units = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, notes = NULL, p_h = NULL, post_treatment_time = NULL, post_treatment_time_units = NULL, product_id = NULL, purpose = NULL, release_timestamp = NULL, sources_id = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, temperature = NULL, temperature_units = NULL, treatment_term_id = NULL, treatment_term_name = NULL, treatment_type = NULL, uuid = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }








































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Treatments. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `amount`) {
        query_params[["amount"]] <- c(query_params[["amount"]], list(`amount` = query_item))
      }

      # explore
      for (query_item in `amount_units`) {
        # validate enum values
        if (!(query_item %in% c("mg/kg", "mg/mL", "mM", "ng/mL", "nM", "percent", "μg/kg", "μg/kg", "μg/mL", "μM", "kpa"))) {
          stop("Invalid value for amount_units when calling IgvfApi$Treatments. Must be [mg/kg, mg/mL, mM, ng/mL, nM, percent, μg/kg, μg/kg, μg/mL, μM, kpa].")
        }
        query_params[["amount_units"]] <- c(query_params[["amount_units"]], list(`amount_units` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Treatments. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biosamples_treated`) {
        query_params[["biosamples_treated"]] <- c(query_params[["biosamples_treated"]], list(`biosamples_treated` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `depletion`) {
        query_params[["depletion"]] <- c(query_params[["depletion"]], list(`depletion` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `duration`) {
        query_params[["duration"]] <- c(query_params[["duration"]], list(`duration` = query_item))
      }

      # explore
      for (query_item in `duration_units`) {
        # validate enum values
        if (!(query_item %in% c("second", "minute", "hour", "day"))) {
          stop("Invalid value for duration_units when calling IgvfApi$Treatments. Must be [second, minute, hour, day].")
        }
        query_params[["duration_units"]] <- c(query_params[["duration_units"]], list(`duration_units` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `p_h`) {
        query_params[["pH"]] <- c(query_params[["pH"]], list(`pH` = query_item))
      }

      # explore
      for (query_item in `post_treatment_time`) {
        query_params[["post_treatment_time"]] <- c(query_params[["post_treatment_time"]], list(`post_treatment_time` = query_item))
      }

      # explore
      for (query_item in `post_treatment_time_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for post_treatment_time_units when calling IgvfApi$Treatments. Must be [minute, hour, day, week, month].")
        }
        query_params[["post_treatment_time_units"]] <- c(query_params[["post_treatment_time_units"]], list(`post_treatment_time_units` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for purpose when calling IgvfApi$Treatments. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["purpose"]] <- c(query_params[["purpose"]], list(`purpose` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for status when calling IgvfApi$Treatments. Must be [archived, deleted, in progress, released].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `temperature`) {
        query_params[["temperature"]] <- c(query_params[["temperature"]], list(`temperature` = query_item))
      }

      # explore
      for (query_item in `temperature_units`) {
        # validate enum values
        if (!(query_item %in% c("Celsius"))) {
          stop("Invalid value for temperature_units when calling IgvfApi$Treatments. Must be [Celsius].")
        }
        query_params[["temperature_units"]] <- c(query_params[["temperature_units"]], list(`temperature_units` = query_item))
      }

      # explore
      for (query_item in `treatment_term_id`) {
        query_params[["treatment_term_id"]] <- c(query_params[["treatment_term_id"]], list(`treatment_term_id` = query_item))
      }

      # explore
      for (query_item in `treatment_term_name`) {
        query_params[["treatment_term_name"]] <- c(query_params[["treatment_term_name"]], list(`treatment_term_name` = query_item))
      }

      # explore
      for (query_item in `treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for treatment_type when calling IgvfApi$Treatments. Must be [chemical, protein, environmental].")
        }
        query_params[["treatment_type"]] <- c(query_params[["treatment_type"]], list(`treatment_type` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      local_var_url_path <- "/treatments/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "TreatmentResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the User collection.
    #'
    #' @description
    #' List items in the User collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param email (optional) Filter by email
    #' @param first_name (optional) Filter by first_name
    #' @param groups (optional) Filter by groups
    #' @param job_title (optional) Filter by job_title
    #' @param lab (optional) Filter by lab
    #' @param last_name (optional) Filter by last_name
    #' @param notes (optional) Filter by notes
    #' @param status (optional) Filter by status
    #' @param submits_for (optional) Filter by submits_for
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param viewing_groups (optional) Filter by viewing_groups
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return UserResults
    #' @export
    Users = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, creation_timestamp = NULL, description = NULL, email = NULL, first_name = NULL, groups = NULL, job_title = NULL, lab = NULL, last_name = NULL, notes = NULL, status = NULL, submits_for = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, viewing_groups = NULL, data_file = NULL, ...) {
      local_var_response <- self$UsersWithHttpInfo(frame, query, limit, sort, id, aliases, creation_timestamp, description, email, first_name, groups, job_title, lab, last_name, notes, status, submits_for, submitted_by_id, submitted_by_title, submitter_comment, summary, title, uuid, viewing_groups, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the User collection.
    #'
    #' @description
    #' List items in the User collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param aliases (optional) Filter by aliases
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param email (optional) Filter by email
    #' @param first_name (optional) Filter by first_name
    #' @param groups (optional) Filter by groups
    #' @param job_title (optional) Filter by job_title
    #' @param lab (optional) Filter by lab
    #' @param last_name (optional) Filter by last_name
    #' @param notes (optional) Filter by notes
    #' @param status (optional) Filter by status
    #' @param submits_for (optional) Filter by submits_for
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param title (optional) Filter by title
    #' @param uuid (optional) Filter by uuid
    #' @param viewing_groups (optional) Filter by viewing_groups
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (UserResults) with additional information such as HTTP status code, headers
    #' @export
    UsersWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, aliases = NULL, creation_timestamp = NULL, description = NULL, email = NULL, first_name = NULL, groups = NULL, job_title = NULL, lab = NULL, last_name = NULL, notes = NULL, status = NULL, submits_for = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, title = NULL, uuid = NULL, viewing_groups = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }

























      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Users. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `email`) {
        query_params[["email"]] <- c(query_params[["email"]], list(`email` = query_item))
      }

      # explore
      for (query_item in `first_name`) {
        query_params[["first_name"]] <- c(query_params[["first_name"]], list(`first_name` = query_item))
      }

      # explore
      for (query_item in `groups`) {
        # validate enum values
        if (!(query_item %in% c("admin", "read-only-admin", "verified"))) {
          stop("Invalid value for groups when calling IgvfApi$Users. Must be [admin, read-only-admin, verified].")
        }
        query_params[["groups"]] <- c(query_params[["groups"]], list(`groups` = query_item))
      }

      # explore
      for (query_item in `job_title`) {
        # validate enum values
        if (!(query_item %in% c("Principal Investigator", "Co-Investigator", "Project Manager", "Submitter", "Post Doc", "Data Wrangler", "Scientist", "Computational Scientist", "Software Developer", "NHGRI staff member", "Other"))) {
          stop("Invalid value for job_title when calling IgvfApi$Users. Must be [Principal Investigator, Co-Investigator, Project Manager, Submitter, Post Doc, Data Wrangler, Scientist, Computational Scientist, Software Developer, NHGRI staff member, Other].")
        }
        query_params[["job_title"]] <- c(query_params[["job_title"]], list(`job_title` = query_item))
      }

      # explore
      for (query_item in `lab`) {
        query_params[["lab"]] <- c(query_params[["lab"]], list(`lab` = query_item))
      }

      # explore
      for (query_item in `last_name`) {
        query_params[["last_name"]] <- c(query_params[["last_name"]], list(`last_name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("current", "deleted", "disabled"))) {
          stop("Invalid value for status when calling IgvfApi$Users. Must be [current, deleted, disabled].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submits_for`) {
        query_params[["submits_for"]] <- c(query_params[["submits_for"]], list(`submits_for` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `title`) {
        query_params[["title"]] <- c(query_params[["title"]], list(`title` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `viewing_groups`) {
        # validate enum values
        if (!(query_item %in% c("community", "IGVF"))) {
          stop("Invalid value for viewing_groups when calling IgvfApi$Users. Must be [community, IGVF].")
        }
        query_params[["viewing_groups"]] <- c(query_params[["viewing_groups"]], list(`viewing_groups` = query_item))
      }

      local_var_url_path <- "/users/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "UserResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the WholeOrganism collection.
    #'
    #' @description
    #' List items in the WholeOrganism collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from (optional) Filter by originated_from
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return WholeOrganismResults
    #' @export
    WholeOrganisms = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from = NULL, part_of = NULL, parts = NULL, pooled_from = NULL, pooled_in = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      local_var_response <- self$WholeOrganismsWithHttpInfo(frame, query, limit, sort, id, accession, age, age_units, aliases, alternate_accessions, award_id, award_component, biomarkers, cellular_sub_pool, classifications, collections, construct_library_sets, creation_timestamp, date_obtained, dbxrefs, description, disease_terms_id, disease_terms_term_name, documents, donors, embryonic, file_sets_id, file_sets_accession, file_sets_aliases, file_sets_assay_term_term_name, file_sets_lab_title, file_sets_status, file_sets_summary, institutional_certificates_id, institutional_certificates_certificate_identifier, lab_id, lab_title, lot_id, lower_bound_age, lower_bound_age_in_hours, modifications_id, modifications_status, modifications_summary, moi, multiplexed_in_id, multiplexed_in_accession, notes, nucleic_acid_delivery, origin_of, originated_from, part_of, parts, pooled_from, pooled_in, product_id, protocols, publication_identifiers, publications, release_timestamp, revoke_detail, sample_terms_id, sample_terms_term_name, sex, sorted_fractions, sorted_from_id, sorted_from_accession, sorted_from_detail, sources_id, starting_amount, starting_amount_units, status, submitted_by_id, submitted_by_title, submitter_comment, summary, taxa, time_post_library_delivery, time_post_library_delivery_units, treatments_id, treatments_purpose, treatments_status, treatments_summary, treatments_treatment_type, upper_bound_age, upper_bound_age_in_hours, url, uuid, virtual, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the WholeOrganism collection.
    #'
    #' @description
    #' List items in the WholeOrganism collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param age (optional) Filter by age
    #' @param age_units (optional) Filter by age_units
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param biomarkers (optional) Filter by biomarkers
    #' @param cellular_sub_pool (optional) Filter by cellular_sub_pool
    #' @param classifications (optional) Filter by classifications
    #' @param collections (optional) Filter by collections
    #' @param construct_library_sets (optional) Filter by construct_library_sets
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param date_obtained (optional) Filter by date_obtained
    #' @param dbxrefs (optional) Filter by dbxrefs
    #' @param description (optional) Filter by description
    #' @param disease_terms_id (optional) Filter by disease_terms.@id
    #' @param disease_terms_term_name (optional) Filter by disease_terms.term_name
    #' @param documents (optional) Filter by documents
    #' @param donors (optional) Filter by donors
    #' @param embryonic (optional) Filter by embryonic
    #' @param file_sets_id (optional) Filter by file_sets.@id
    #' @param file_sets_accession (optional) Filter by file_sets.accession
    #' @param file_sets_aliases (optional) Filter by file_sets.aliases
    #' @param file_sets_assay_term_term_name (optional) Filter by file_sets.assay_term.term_name
    #' @param file_sets_lab_title (optional) Filter by file_sets.lab.title
    #' @param file_sets_status (optional) Filter by file_sets.status
    #' @param file_sets_summary (optional) Filter by file_sets.summary
    #' @param institutional_certificates_id (optional) Filter by institutional_certificates.@id
    #' @param institutional_certificates_certificate_identifier (optional) Filter by institutional_certificates.certificate_identifier
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param lot_id (optional) Filter by lot_id
    #' @param lower_bound_age (optional) Filter by lower_bound_age
    #' @param lower_bound_age_in_hours (optional) Filter by lower_bound_age_in_hours
    #' @param modifications_id (optional) Filter by modifications.@id
    #' @param modifications_status (optional) Filter by modifications.status
    #' @param modifications_summary (optional) Filter by modifications.summary
    #' @param moi (optional) Filter by moi
    #' @param multiplexed_in_id (optional) Filter by multiplexed_in.@id
    #' @param multiplexed_in_accession (optional) Filter by multiplexed_in.accession
    #' @param notes (optional) Filter by notes
    #' @param nucleic_acid_delivery (optional) Filter by nucleic_acid_delivery
    #' @param origin_of (optional) Filter by origin_of
    #' @param originated_from (optional) Filter by originated_from
    #' @param part_of (optional) Filter by part_of
    #' @param parts (optional) Filter by parts
    #' @param pooled_from (optional) Filter by pooled_from
    #' @param pooled_in (optional) Filter by pooled_in
    #' @param product_id (optional) Filter by product_id
    #' @param protocols (optional) Filter by protocols
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param sample_terms_id (optional) Filter by sample_terms.@id
    #' @param sample_terms_term_name (optional) Filter by sample_terms.term_name
    #' @param sex (optional) Filter by sex
    #' @param sorted_fractions (optional) Filter by sorted_fractions
    #' @param sorted_from_id (optional) Filter by sorted_from.@id
    #' @param sorted_from_accession (optional) Filter by sorted_from.accession
    #' @param sorted_from_detail (optional) Filter by sorted_from_detail
    #' @param sources_id (optional) Filter by sources.@id
    #' @param starting_amount (optional) Filter by starting_amount
    #' @param starting_amount_units (optional) Filter by starting_amount_units
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param taxa (optional) Filter by taxa
    #' @param time_post_library_delivery (optional) Filter by time_post_library_delivery
    #' @param time_post_library_delivery_units (optional) Filter by time_post_library_delivery_units
    #' @param treatments_id (optional) Filter by treatments.@id
    #' @param treatments_purpose (optional) Filter by treatments.purpose
    #' @param treatments_status (optional) Filter by treatments.status
    #' @param treatments_summary (optional) Filter by treatments.summary
    #' @param treatments_treatment_type (optional) Filter by treatments.treatment_type
    #' @param upper_bound_age (optional) Filter by upper_bound_age
    #' @param upper_bound_age_in_hours (optional) Filter by upper_bound_age_in_hours
    #' @param url (optional) Filter by url
    #' @param uuid (optional) Filter by uuid
    #' @param virtual (optional) Filter by virtual
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (WholeOrganismResults) with additional information such as HTTP status code, headers
    #' @export
    WholeOrganismsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, age = NULL, age_units = NULL, aliases = NULL, alternate_accessions = NULL, award_id = NULL, award_component = NULL, biomarkers = NULL, cellular_sub_pool = NULL, classifications = NULL, collections = NULL, construct_library_sets = NULL, creation_timestamp = NULL, date_obtained = NULL, dbxrefs = NULL, description = NULL, disease_terms_id = NULL, disease_terms_term_name = NULL, documents = NULL, donors = NULL, embryonic = NULL, file_sets_id = NULL, file_sets_accession = NULL, file_sets_aliases = NULL, file_sets_assay_term_term_name = NULL, file_sets_lab_title = NULL, file_sets_status = NULL, file_sets_summary = NULL, institutional_certificates_id = NULL, institutional_certificates_certificate_identifier = NULL, lab_id = NULL, lab_title = NULL, lot_id = NULL, lower_bound_age = NULL, lower_bound_age_in_hours = NULL, modifications_id = NULL, modifications_status = NULL, modifications_summary = NULL, moi = NULL, multiplexed_in_id = NULL, multiplexed_in_accession = NULL, notes = NULL, nucleic_acid_delivery = NULL, origin_of = NULL, originated_from = NULL, part_of = NULL, parts = NULL, pooled_from = NULL, pooled_in = NULL, product_id = NULL, protocols = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, sample_terms_id = NULL, sample_terms_term_name = NULL, sex = NULL, sorted_fractions = NULL, sorted_from_id = NULL, sorted_from_accession = NULL, sorted_from_detail = NULL, sources_id = NULL, starting_amount = NULL, starting_amount_units = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, taxa = NULL, time_post_library_delivery = NULL, time_post_library_delivery_units = NULL, treatments_id = NULL, treatments_purpose = NULL, treatments_status = NULL, treatments_summary = NULL, treatments_treatment_type = NULL, upper_bound_age = NULL, upper_bound_age_in_hours = NULL, url = NULL, uuid = NULL, virtual = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }

























































































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$WholeOrganisms. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `age`) {
        query_params[["age"]] <- c(query_params[["age"]], list(`age` = query_item))
      }

      # explore
      for (query_item in `age_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month", "year"))) {
          stop("Invalid value for age_units when calling IgvfApi$WholeOrganisms. Must be [minute, hour, day, week, month, year].")
        }
        query_params[["age_units"]] <- c(query_params[["age_units"]], list(`age_units` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$WholeOrganisms. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `biomarkers`) {
        query_params[["biomarkers"]] <- c(query_params[["biomarkers"]], list(`biomarkers` = query_item))
      }

      # explore
      for (query_item in `cellular_sub_pool`) {
        query_params[["cellular_sub_pool"]] <- c(query_params[["cellular_sub_pool"]], list(`cellular_sub_pool` = query_item))
      }

      # explore
      for (query_item in `classifications`) {
        query_params[["classifications"]] <- c(query_params[["classifications"]], list(`classifications` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$WholeOrganisms. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `construct_library_sets`) {
        query_params[["construct_library_sets"]] <- c(query_params[["construct_library_sets"]], list(`construct_library_sets` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `date_obtained`) {
        query_params[["date_obtained"]] <- c(query_params[["date_obtained"]], list(`date_obtained` = query_item))
      }

      # explore
      for (query_item in `dbxrefs`) {
        query_params[["dbxrefs"]] <- c(query_params[["dbxrefs"]], list(`dbxrefs` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `disease_terms_id`) {
        query_params[["disease_terms.@id"]] <- c(query_params[["disease_terms.@id"]], list(`disease_terms.@id` = query_item))
      }

      # explore
      for (query_item in `disease_terms_term_name`) {
        query_params[["disease_terms.term_name"]] <- c(query_params[["disease_terms.term_name"]], list(`disease_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `donors`) {
        query_params[["donors"]] <- c(query_params[["donors"]], list(`donors` = query_item))
      }

      # explore
      for (query_item in `embryonic`) {
        query_params[["embryonic"]] <- c(query_params[["embryonic"]], list(`embryonic` = query_item))
      }

      # explore
      for (query_item in `file_sets_id`) {
        query_params[["file_sets.@id"]] <- c(query_params[["file_sets.@id"]], list(`file_sets.@id` = query_item))
      }

      # explore
      for (query_item in `file_sets_accession`) {
        query_params[["file_sets.accession"]] <- c(query_params[["file_sets.accession"]], list(`file_sets.accession` = query_item))
      }

      # explore
      for (query_item in `file_sets_aliases`) {
        query_params[["file_sets.aliases"]] <- c(query_params[["file_sets.aliases"]], list(`file_sets.aliases` = query_item))
      }

      # explore
      for (query_item in `file_sets_assay_term_term_name`) {
        query_params[["file_sets.assay_term.term_name"]] <- c(query_params[["file_sets.assay_term.term_name"]], list(`file_sets.assay_term.term_name` = query_item))
      }

      # explore
      for (query_item in `file_sets_lab_title`) {
        query_params[["file_sets.lab.title"]] <- c(query_params[["file_sets.lab.title"]], list(`file_sets.lab.title` = query_item))
      }

      # explore
      for (query_item in `file_sets_status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for file_sets_status when calling IgvfApi$WholeOrganisms. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["file_sets.status"]] <- c(query_params[["file_sets.status"]], list(`file_sets.status` = query_item))
      }

      # explore
      for (query_item in `file_sets_summary`) {
        query_params[["file_sets.summary"]] <- c(query_params[["file_sets.summary"]], list(`file_sets.summary` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_id`) {
        query_params[["institutional_certificates.@id"]] <- c(query_params[["institutional_certificates.@id"]], list(`institutional_certificates.@id` = query_item))
      }

      # explore
      for (query_item in `institutional_certificates_certificate_identifier`) {
        query_params[["institutional_certificates.certificate_identifier"]] <- c(query_params[["institutional_certificates.certificate_identifier"]], list(`institutional_certificates.certificate_identifier` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `lot_id`) {
        query_params[["lot_id"]] <- c(query_params[["lot_id"]], list(`lot_id` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age`) {
        query_params[["lower_bound_age"]] <- c(query_params[["lower_bound_age"]], list(`lower_bound_age` = query_item))
      }

      # explore
      for (query_item in `lower_bound_age_in_hours`) {
        query_params[["lower_bound_age_in_hours"]] <- c(query_params[["lower_bound_age_in_hours"]], list(`lower_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `modifications_id`) {
        query_params[["modifications.@id"]] <- c(query_params[["modifications.@id"]], list(`modifications.@id` = query_item))
      }

      # explore
      for (query_item in `modifications_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for modifications_status when calling IgvfApi$WholeOrganisms. Must be [archived, deleted, in progress, released].")
        }
        query_params[["modifications.status"]] <- c(query_params[["modifications.status"]], list(`modifications.status` = query_item))
      }

      # explore
      for (query_item in `modifications_summary`) {
        query_params[["modifications.summary"]] <- c(query_params[["modifications.summary"]], list(`modifications.summary` = query_item))
      }

      # explore
      for (query_item in `moi`) {
        query_params[["moi"]] <- c(query_params[["moi"]], list(`moi` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_id`) {
        query_params[["multiplexed_in.@id"]] <- c(query_params[["multiplexed_in.@id"]], list(`multiplexed_in.@id` = query_item))
      }

      # explore
      for (query_item in `multiplexed_in_accession`) {
        query_params[["multiplexed_in.accession"]] <- c(query_params[["multiplexed_in.accession"]], list(`multiplexed_in.accession` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `nucleic_acid_delivery`) {
        # validate enum values
        if (!(query_item %in% c("transfection", "adenoviral transduction", "lentiviral transduction"))) {
          stop("Invalid value for nucleic_acid_delivery when calling IgvfApi$WholeOrganisms. Must be [transfection, adenoviral transduction, lentiviral transduction].")
        }
        query_params[["nucleic_acid_delivery"]] <- c(query_params[["nucleic_acid_delivery"]], list(`nucleic_acid_delivery` = query_item))
      }

      # explore
      for (query_item in `origin_of`) {
        query_params[["origin_of"]] <- c(query_params[["origin_of"]], list(`origin_of` = query_item))
      }

      # explore
      for (query_item in `originated_from`) {
        query_params[["originated_from"]] <- c(query_params[["originated_from"]], list(`originated_from` = query_item))
      }

      # explore
      for (query_item in `part_of`) {
        query_params[["part_of"]] <- c(query_params[["part_of"]], list(`part_of` = query_item))
      }

      # explore
      for (query_item in `parts`) {
        query_params[["parts"]] <- c(query_params[["parts"]], list(`parts` = query_item))
      }

      # explore
      for (query_item in `pooled_from`) {
        query_params[["pooled_from"]] <- c(query_params[["pooled_from"]], list(`pooled_from` = query_item))
      }

      # explore
      for (query_item in `pooled_in`) {
        query_params[["pooled_in"]] <- c(query_params[["pooled_in"]], list(`pooled_in` = query_item))
      }

      # explore
      for (query_item in `product_id`) {
        query_params[["product_id"]] <- c(query_params[["product_id"]], list(`product_id` = query_item))
      }

      # explore
      for (query_item in `protocols`) {
        query_params[["protocols"]] <- c(query_params[["protocols"]], list(`protocols` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `sample_terms_id`) {
        query_params[["sample_terms.@id"]] <- c(query_params[["sample_terms.@id"]], list(`sample_terms.@id` = query_item))
      }

      # explore
      for (query_item in `sample_terms_term_name`) {
        query_params[["sample_terms.term_name"]] <- c(query_params[["sample_terms.term_name"]], list(`sample_terms.term_name` = query_item))
      }

      # explore
      for (query_item in `sex`) {
        # validate enum values
        if (!(query_item %in% c("female", "male", "mixed", "unspecified"))) {
          stop("Invalid value for sex when calling IgvfApi$WholeOrganisms. Must be [female, male, mixed, unspecified].")
        }
        query_params[["sex"]] <- c(query_params[["sex"]], list(`sex` = query_item))
      }

      # explore
      for (query_item in `sorted_fractions`) {
        query_params[["sorted_fractions"]] <- c(query_params[["sorted_fractions"]], list(`sorted_fractions` = query_item))
      }

      # explore
      for (query_item in `sorted_from_id`) {
        query_params[["sorted_from.@id"]] <- c(query_params[["sorted_from.@id"]], list(`sorted_from.@id` = query_item))
      }

      # explore
      for (query_item in `sorted_from_accession`) {
        query_params[["sorted_from.accession"]] <- c(query_params[["sorted_from.accession"]], list(`sorted_from.accession` = query_item))
      }

      # explore
      for (query_item in `sorted_from_detail`) {
        query_params[["sorted_from_detail"]] <- c(query_params[["sorted_from_detail"]], list(`sorted_from_detail` = query_item))
      }

      # explore
      for (query_item in `sources_id`) {
        query_params[["sources.@id"]] <- c(query_params[["sources.@id"]], list(`sources.@id` = query_item))
      }

      # explore
      for (query_item in `starting_amount`) {
        query_params[["starting_amount"]] <- c(query_params[["starting_amount"]], list(`starting_amount` = query_item))
      }

      # explore
      for (query_item in `starting_amount_units`) {
        # validate enum values
        if (!(query_item %in% c("cells", "cells/ml", "g", "items", "mg", "whole animals", "whole embryos", "μg", "ng"))) {
          stop("Invalid value for starting_amount_units when calling IgvfApi$WholeOrganisms. Must be [cells, cells/ml, g, items, mg, whole animals, whole embryos, μg, ng].")
        }
        query_params[["starting_amount_units"]] <- c(query_params[["starting_amount_units"]], list(`starting_amount_units` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$WholeOrganisms. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `taxa`) {
        # validate enum values
        if (!(query_item %in% c("Homo sapiens", "Mus musculus"))) {
          stop("Invalid value for taxa when calling IgvfApi$WholeOrganisms. Must be [Homo sapiens, Mus musculus].")
        }
        query_params[["taxa"]] <- c(query_params[["taxa"]], list(`taxa` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery`) {
        query_params[["time_post_library_delivery"]] <- c(query_params[["time_post_library_delivery"]], list(`time_post_library_delivery` = query_item))
      }

      # explore
      for (query_item in `time_post_library_delivery_units`) {
        # validate enum values
        if (!(query_item %in% c("minute", "hour", "day", "week", "month"))) {
          stop("Invalid value for time_post_library_delivery_units when calling IgvfApi$WholeOrganisms. Must be [minute, hour, day, week, month].")
        }
        query_params[["time_post_library_delivery_units"]] <- c(query_params[["time_post_library_delivery_units"]], list(`time_post_library_delivery_units` = query_item))
      }

      # explore
      for (query_item in `treatments_id`) {
        query_params[["treatments.@id"]] <- c(query_params[["treatments.@id"]], list(`treatments.@id` = query_item))
      }

      # explore
      for (query_item in `treatments_purpose`) {
        # validate enum values
        if (!(query_item %in% c("activation", "agonist", "antagonist", "control", "differentiation", "de-differentiation", "perturbation", "selection", "stimulation"))) {
          stop("Invalid value for treatments_purpose when calling IgvfApi$WholeOrganisms. Must be [activation, agonist, antagonist, control, differentiation, de-differentiation, perturbation, selection, stimulation].")
        }
        query_params[["treatments.purpose"]] <- c(query_params[["treatments.purpose"]], list(`treatments.purpose` = query_item))
      }

      # explore
      for (query_item in `treatments_status`) {
        # validate enum values
        if (!(query_item %in% c("archived", "deleted", "in progress", "released"))) {
          stop("Invalid value for treatments_status when calling IgvfApi$WholeOrganisms. Must be [archived, deleted, in progress, released].")
        }
        query_params[["treatments.status"]] <- c(query_params[["treatments.status"]], list(`treatments.status` = query_item))
      }

      # explore
      for (query_item in `treatments_summary`) {
        query_params[["treatments.summary"]] <- c(query_params[["treatments.summary"]], list(`treatments.summary` = query_item))
      }

      # explore
      for (query_item in `treatments_treatment_type`) {
        # validate enum values
        if (!(query_item %in% c("chemical", "protein", "environmental"))) {
          stop("Invalid value for treatments_treatment_type when calling IgvfApi$WholeOrganisms. Must be [chemical, protein, environmental].")
        }
        query_params[["treatments.treatment_type"]] <- c(query_params[["treatments.treatment_type"]], list(`treatments.treatment_type` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age`) {
        query_params[["upper_bound_age"]] <- c(query_params[["upper_bound_age"]], list(`upper_bound_age` = query_item))
      }

      # explore
      for (query_item in `upper_bound_age_in_hours`) {
        query_params[["upper_bound_age_in_hours"]] <- c(query_params[["upper_bound_age_in_hours"]], list(`upper_bound_age_in_hours` = query_item))
      }

      # explore
      for (query_item in `url`) {
        query_params[["url"]] <- c(query_params[["url"]], list(`url` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `virtual`) {
        query_params[["virtual"]] <- c(query_params[["virtual"]], list(`virtual` = query_item))
      }

      local_var_url_path <- "/whole-organisms/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "WholeOrganismResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    },
    #' List items in the Workflow collection.
    #'
    #' @description
    #' List items in the Workflow collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_steps (optional) Filter by analysis_steps
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param source_url (optional) Filter by source_url
    #' @param standards_page_id (optional) Filter by standards_page.@id
    #' @param standards_page_title (optional) Filter by standards_page.title
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param workflow_repositories (optional) Filter by workflow_repositories
    #' @param workflow_version (optional) Filter by workflow_version
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return WorkflowResults
    #' @export
    Workflows = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_steps = NULL, award_id = NULL, award_component = NULL, collections = NULL, creation_timestamp = NULL, description = NULL, documents = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, source_url = NULL, standards_page_id = NULL, standards_page_title = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, workflow_repositories = NULL, workflow_version = NULL, data_file = NULL, ...) {
      local_var_response <- self$WorkflowsWithHttpInfo(frame, query, limit, sort, id, accession, aliases, alternate_accessions, analysis_steps, award_id, award_component, collections, creation_timestamp, description, documents, lab_id, lab_title, name, notes, publication_identifiers, publications, release_timestamp, revoke_detail, source_url, standards_page_id, standards_page_title, status, submitted_by_id, submitted_by_title, submitter_comment, summary, uuid, workflow_repositories, workflow_version, data_file = data_file, ...)
      if (local_var_response$status_code >= 200 && local_var_response$status_code <= 299) {
        local_var_response$content
      } else if (local_var_response$status_code >= 300 && local_var_response$status_code <= 399) {
        local_var_response
      } else if (local_var_response$status_code >= 400 && local_var_response$status_code <= 499) {
        local_var_response
      } else if (local_var_response$status_code >= 500 && local_var_response$status_code <= 599) {
        local_var_response
      }
    },
    #' List items in the Workflow collection.
    #'
    #' @description
    #' List items in the Workflow collection.
    #'
    #' @param frame Constant value. Do not set.
    #' @param query (optional) Query string for searching.
    #' @param limit (optional) Maximum number of results to return. Default is 25. Use 'all' for all results.
    #' @param sort (optional) Fields to sort results by. Prefix with '-' for descending order. Can be repeated for multiple sort fields. Does not work with limit=all.
    #' @param id (optional) Filter by @id
    #' @param accession (optional) Filter by accession
    #' @param aliases (optional) Filter by aliases
    #' @param alternate_accessions (optional) Filter by alternate_accessions
    #' @param analysis_steps (optional) Filter by analysis_steps
    #' @param award_id (optional) Filter by award.@id
    #' @param award_component (optional) Filter by award.component
    #' @param collections (optional) Filter by collections
    #' @param creation_timestamp (optional) Filter by creation_timestamp
    #' @param description (optional) Filter by description
    #' @param documents (optional) Filter by documents
    #' @param lab_id (optional) Filter by lab.@id
    #' @param lab_title (optional) Filter by lab.title
    #' @param name (optional) Filter by name
    #' @param notes (optional) Filter by notes
    #' @param publication_identifiers (optional) Filter by publication_identifiers
    #' @param publications (optional) Filter by publications
    #' @param release_timestamp (optional) Filter by release_timestamp
    #' @param revoke_detail (optional) Filter by revoke_detail
    #' @param source_url (optional) Filter by source_url
    #' @param standards_page_id (optional) Filter by standards_page.@id
    #' @param standards_page_title (optional) Filter by standards_page.title
    #' @param status (optional) Filter by status
    #' @param submitted_by_id (optional) Filter by submitted_by.@id
    #' @param submitted_by_title (optional) Filter by submitted_by.title
    #' @param submitter_comment (optional) Filter by submitter_comment
    #' @param summary (optional) Filter by summary
    #' @param uuid (optional) Filter by uuid
    #' @param workflow_repositories (optional) Filter by workflow_repositories
    #' @param workflow_version (optional) Filter by workflow_version
    #' @param data_file (optional) name of the data file to save the result
    #' @param ... Other optional arguments
    #' @return API response (WorkflowResults) with additional information such as HTTP status code, headers
    #' @export
    WorkflowsWithHttpInfo = function(frame, query = NULL, limit = NULL, sort = NULL, id = NULL, accession = NULL, aliases = NULL, alternate_accessions = NULL, analysis_steps = NULL, award_id = NULL, award_component = NULL, collections = NULL, creation_timestamp = NULL, description = NULL, documents = NULL, lab_id = NULL, lab_title = NULL, name = NULL, notes = NULL, publication_identifiers = NULL, publications = NULL, release_timestamp = NULL, revoke_detail = NULL, source_url = NULL, standards_page_id = NULL, standards_page_title = NULL, status = NULL, submitted_by_id = NULL, submitted_by_title = NULL, submitter_comment = NULL, summary = NULL, uuid = NULL, workflow_repositories = NULL, workflow_version = NULL, data_file = NULL, ...) {
      args <- list(...)
      query_params <- list()
      header_params <- c()
      form_params <- list()
      file_params <- list()
      local_var_body <- NULL
      oauth_scopes <- NULL
      is_oauth <- FALSE

      if (missing(`frame`)) {
        stop("Missing required parameter `frame`.")
      }



































      query_params[["query"]] <- `query`

      if (!(`frame` %in% c("object"))) {
        stop("Invalid value for frame when calling IgvfApi$Workflows. Must be [object].")
      }
      query_params[["frame"]] <- `frame`

      query_params[["limit"]] <- `limit`

      # explore
      for (query_item in `sort`) {
        query_params[["sort"]] <- c(query_params[["sort"]], list(`sort` = query_item))
      }

      # explore
      for (query_item in `id`) {
        query_params[["@id"]] <- c(query_params[["@id"]], list(`@id` = query_item))
      }

      # explore
      for (query_item in `accession`) {
        query_params[["accession"]] <- c(query_params[["accession"]], list(`accession` = query_item))
      }

      # explore
      for (query_item in `aliases`) {
        query_params[["aliases"]] <- c(query_params[["aliases"]], list(`aliases` = query_item))
      }

      # explore
      for (query_item in `alternate_accessions`) {
        query_params[["alternate_accessions"]] <- c(query_params[["alternate_accessions"]], list(`alternate_accessions` = query_item))
      }

      # explore
      for (query_item in `analysis_steps`) {
        query_params[["analysis_steps"]] <- c(query_params[["analysis_steps"]], list(`analysis_steps` = query_item))
      }

      # explore
      for (query_item in `award_id`) {
        query_params[["award.@id"]] <- c(query_params[["award.@id"]], list(`award.@id` = query_item))
      }

      # explore
      for (query_item in `award_component`) {
        # validate enum values
        if (!(query_item %in% c("affiliate", "data analysis", "data coordination", "functional characterization", "mapping", "networks", "predictive modeling"))) {
          stop("Invalid value for award_component when calling IgvfApi$Workflows. Must be [affiliate, data analysis, data coordination, functional characterization, mapping, networks, predictive modeling].")
        }
        query_params[["award.component"]] <- c(query_params[["award.component"]], list(`award.component` = query_item))
      }

      # explore
      for (query_item in `collections`) {
        # validate enum values
        if (!(query_item %in% c("ClinGen", "ENCODE", "GREGoR", "IGVF_catalog_beta_v0.1", "IGVF_catalog_beta_v0.2", "IGVF_catalog_beta_v0.3", "IGVF_catalog_beta_v0.4", "MaveDB", "MPRAbase", "Vista"))) {
          stop("Invalid value for collections when calling IgvfApi$Workflows. Must be [ClinGen, ENCODE, GREGoR, IGVF_catalog_beta_v0.1, IGVF_catalog_beta_v0.2, IGVF_catalog_beta_v0.3, IGVF_catalog_beta_v0.4, MaveDB, MPRAbase, Vista].")
        }
        query_params[["collections"]] <- c(query_params[["collections"]], list(`collections` = query_item))
      }

      # explore
      for (query_item in `creation_timestamp`) {
        query_params[["creation_timestamp"]] <- c(query_params[["creation_timestamp"]], list(`creation_timestamp` = query_item))
      }

      # explore
      for (query_item in `description`) {
        query_params[["description"]] <- c(query_params[["description"]], list(`description` = query_item))
      }

      # explore
      for (query_item in `documents`) {
        query_params[["documents"]] <- c(query_params[["documents"]], list(`documents` = query_item))
      }

      # explore
      for (query_item in `lab_id`) {
        query_params[["lab.@id"]] <- c(query_params[["lab.@id"]], list(`lab.@id` = query_item))
      }

      # explore
      for (query_item in `lab_title`) {
        query_params[["lab.title"]] <- c(query_params[["lab.title"]], list(`lab.title` = query_item))
      }

      # explore
      for (query_item in `name`) {
        query_params[["name"]] <- c(query_params[["name"]], list(`name` = query_item))
      }

      # explore
      for (query_item in `notes`) {
        query_params[["notes"]] <- c(query_params[["notes"]], list(`notes` = query_item))
      }

      # explore
      for (query_item in `publication_identifiers`) {
        query_params[["publication_identifiers"]] <- c(query_params[["publication_identifiers"]], list(`publication_identifiers` = query_item))
      }

      # explore
      for (query_item in `publications`) {
        query_params[["publications"]] <- c(query_params[["publications"]], list(`publications` = query_item))
      }

      # explore
      for (query_item in `release_timestamp`) {
        query_params[["release_timestamp"]] <- c(query_params[["release_timestamp"]], list(`release_timestamp` = query_item))
      }

      # explore
      for (query_item in `revoke_detail`) {
        query_params[["revoke_detail"]] <- c(query_params[["revoke_detail"]], list(`revoke_detail` = query_item))
      }

      # explore
      for (query_item in `source_url`) {
        query_params[["source_url"]] <- c(query_params[["source_url"]], list(`source_url` = query_item))
      }

      # explore
      for (query_item in `standards_page_id`) {
        query_params[["standards_page.@id"]] <- c(query_params[["standards_page.@id"]], list(`standards_page.@id` = query_item))
      }

      # explore
      for (query_item in `standards_page_title`) {
        query_params[["standards_page.title"]] <- c(query_params[["standards_page.title"]], list(`standards_page.title` = query_item))
      }

      # explore
      for (query_item in `status`) {
        # validate enum values
        if (!(query_item %in% c("in progress", "released", "deleted", "replaced", "revoked", "archived"))) {
          stop("Invalid value for status when calling IgvfApi$Workflows. Must be [in progress, released, deleted, replaced, revoked, archived].")
        }
        query_params[["status"]] <- c(query_params[["status"]], list(`status` = query_item))
      }

      # explore
      for (query_item in `submitted_by_id`) {
        query_params[["submitted_by.@id"]] <- c(query_params[["submitted_by.@id"]], list(`submitted_by.@id` = query_item))
      }

      # explore
      for (query_item in `submitted_by_title`) {
        query_params[["submitted_by.title"]] <- c(query_params[["submitted_by.title"]], list(`submitted_by.title` = query_item))
      }

      # explore
      for (query_item in `submitter_comment`) {
        query_params[["submitter_comment"]] <- c(query_params[["submitter_comment"]], list(`submitter_comment` = query_item))
      }

      # explore
      for (query_item in `summary`) {
        query_params[["summary"]] <- c(query_params[["summary"]], list(`summary` = query_item))
      }

      # explore
      for (query_item in `uuid`) {
        query_params[["uuid"]] <- c(query_params[["uuid"]], list(`uuid` = query_item))
      }

      # explore
      for (query_item in `workflow_repositories`) {
        query_params[["workflow_repositories"]] <- c(query_params[["workflow_repositories"]], list(`workflow_repositories` = query_item))
      }

      # explore
      for (query_item in `workflow_version`) {
        query_params[["workflow_version"]] <- c(query_params[["workflow_version"]], list(`workflow_version` = query_item))
      }

      local_var_url_path <- "/workflows/@@listing"
      # HTTP basic auth
      if (!is.null(self$api_client$username) || !is.null(self$api_client$password)) {
        header_params["Authorization"] <- paste("Basic", base64enc::base64encode(charToRaw(paste(self$api_client$username, self$api_client$password, sep = ":"))))
      }

      # The Accept request HTTP header
      local_var_accepts <- list("application/json")

      # The Content-Type representation header
      local_var_content_types <- list()

      local_var_resp <- self$api_client$CallApi(url = paste0(self$api_client$base_path, local_var_url_path),
                                 method = "GET",
                                 query_params = query_params,
                                 header_params = header_params,
                                 form_params = form_params,
                                 file_params = file_params,
                                 accepts = local_var_accepts,
                                 content_types = local_var_content_types,
                                 body = local_var_body,
                                 is_oauth = is_oauth,
                                 oauth_scopes = oauth_scopes,
                                 ...)

      if ((local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) || local_var_resp$status_code == 404) {
        # save response in a file
        if (!is.null(data_file)) {
          write(local_var_resp$response, data_file)
        }

        deserialized_resp_obj <- tryCatch(
          self$api_client$deserialize(local_var_resp$response_as_text(), "WorkflowResults", loadNamespace("igvfclient")),
          error = function(e) {
            print("Error deserializing, returning raw")
	    return(local_var_resp$response)
          }
        )
        if (!is.null(deserialized_resp_obj)) {
            local_var_resp$content <- deserialized_resp_obj
        }
        local_var_resp
      } else if (local_var_resp$status_code >= 300 && local_var_resp$status_code <= 399) {
        ApiResponse$new(paste("Server returned ", local_var_resp$status_code, " response status code."), local_var_resp)
      } else if (local_var_resp$status_code >= 400 && local_var_resp$status_code <= 499) {
        ApiResponse$new("API client error", local_var_resp)
      } else if (local_var_resp$status_code >= 500 && local_var_resp$status_code <= 599) {
        if (is.null(local_var_resp$response) || local_var_resp$response == "") {
          local_var_resp$response <- "API server error"
        }
        local_var_resp
      }
    }
  )
)
